<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Gyan Prakash Mishra">

<title>Methylation independent gene regulation by methyl-CpG binding protein 2</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="MECP2_project_files/libs/clipboard/clipboard.min.js"></script>
<script src="MECP2_project_files/libs/quarto-html/quarto.js"></script>
<script src="MECP2_project_files/libs/quarto-html/popper.min.js"></script>
<script src="MECP2_project_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="MECP2_project_files/libs/quarto-html/anchor.min.js"></script>
<link href="MECP2_project_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="MECP2_project_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="MECP2_project_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="MECP2_project_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="MECP2_project_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body>

<div id="quarto-content" class="page-columns page-rows-contents page-layout-full toc-left">
<div id="quarto-sidebar-toc-left" class="sidebar toc-left">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction">Introduction</a></li>
  <li><a href="#load-libraries" id="toc-load-libraries" class="nav-link" data-scroll-target="#load-libraries">Load Libraries</a></li>
  <li><a href="#data-processing" id="toc-data-processing" class="nav-link" data-scroll-target="#data-processing">Data processing</a></li>
  <li><a href="#figure-1-cutrun-uncovers-mecp2-binding-hotspots-mbhs-across-the-neural-genome." id="toc-figure-1-cutrun-uncovers-mecp2-binding-hotspots-mbhs-across-the-neural-genome." class="nav-link" data-scroll-target="#figure-1-cutrun-uncovers-mecp2-binding-hotspots-mbhs-across-the-neural-genome."><strong>Figure 1: CUT&amp;RUN uncovers MECP2 Binding Hotspots (MBHs) across the neural genome.</strong></a></li>
  <li><a href="#figure-2-mecp2-binding-to-mbh-sites-is-largely-methylation-independent." id="toc-figure-2-mecp2-binding-to-mbh-sites-is-largely-methylation-independent." class="nav-link" data-scroll-target="#figure-2-mecp2-binding-to-mbh-sites-is-largely-methylation-independent."><strong>Figure 2: MECP2 binding to MBH sites is largely methylation-independent.</strong></a></li>
  <li><a href="#figure-3-mecp2-modulates-enhancer-activity-at-mbh-sites." id="toc-figure-3-mecp2-modulates-enhancer-activity-at-mbh-sites." class="nav-link" data-scroll-target="#figure-3-mecp2-modulates-enhancer-activity-at-mbh-sites."><strong>Figure 3: MECP2 modulates enhancer activity at MBH sites.</strong></a></li>
  <li><a href="#figure-4-mecp2-preferentially-represses-genes-harboring-multiple-mbh-sites." id="toc-figure-4-mecp2-preferentially-represses-genes-harboring-multiple-mbh-sites." class="nav-link" data-scroll-target="#figure-4-mecp2-preferentially-represses-genes-harboring-multiple-mbh-sites."><strong>Figure 4: MECP2 preferentially represses genes harboring multiple MBH sites.</strong></a></li>
  <li><a href="#figure-5-mbhs-repress-genes-independently-of-the-mca-pathway." id="toc-figure-5-mbhs-repress-genes-independently-of-the-mca-pathway." class="nav-link" data-scroll-target="#figure-5-mbhs-repress-genes-independently-of-the-mca-pathway."><strong>Figure 5: MBHs repress genes independently of the mCA pathway.</strong></a></li>
  <li><a href="#section" id="toc-section" class="nav-link" data-scroll-target="#section"></a></li>
  </ul>
</nav>
</div>
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
</div>
<main class="content column-page-right" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Methylation independent gene regulation by methyl-CpG binding protein 2</h1>
</div>


<div class="quarto-title-meta-author">
  <div class="quarto-title-meta-heading">Author</div>
  <div class="quarto-title-meta-heading">Affiliations</div>
  
    <div class="quarto-title-meta-contents">
    <p class="author">Gyan Prakash Mishra </p>
  </div>
    <div class="quarto-title-meta-contents">
        <p class="affiliation">
            Department of Neuroscience, Peter O’Donnell Jr.&nbsp;Brain Institute,
          </p>
        <p class="affiliation">
            UT Southwestern Medical Center
          </p>
        <p class="affiliation">
            
          </p>
        <p class="affiliation">
            
          </p>
        <p class="affiliation">
            
          </p>
        <p class="affiliation">
            
          </p>
      </div>
    </div>

<div class="quarto-title-meta">

      
  
    
  </div>
  

</header>

<section id="introduction" class="level3">
<h3 class="anchored" data-anchor-id="introduction">Introduction</h3>
<p><br>
This document encapsulates the R code employed for the comprehensive quantitative, integrative, and statistical analysis of the datasets outlined in the study.</p>
<table class="table">
<tbody>
<tr class="odd">
<td><strong>SI. No</strong></td>
<td><strong>Sequencing data type</strong></td>
<td><strong>Source</strong></td>
</tr>
<tr class="even">
<td>1.</td>
<td>MECP2 CUT&amp;RUN (Adult Brain Cortex of <em>Mecp2</em> WT and KO mice)</td>
<td><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE150538">GSE150538</a></td>
</tr>
<tr class="odd">
<td>2.</td>
<td>MECP2 CUT&amp;RUN&nbsp;(Adult hippocampus of <em>MECP2</em>-TG1, <em>Mecp2</em> KO, and WT)</td>
<td><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE213752">GSE213752</a></td>
</tr>
<tr class="even">
<td>3.&nbsp;</td>
<td>MECP2 ChIP-seq&nbsp;</td>
<td><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE67293">GSE67293</a></td>
</tr>
<tr class="odd">
<td>4.&nbsp;</td>
<td>MECP2 ChIP-seq&nbsp;</td>
<td><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE139509">GSE139509</a></td>
</tr>
<tr class="even">
<td>5.</td>
<td>BS-seq&nbsp;(Brain Cortex of 10 week old DNMT3a cKO and WT mice)</td>
<td><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE103214">GSE103214</a></td>
</tr>
<tr class="odd">
<td>6.</td>
<td>MECP2 CUT&amp;RUN ( WT and G118E )</td>
<td><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi">GSE243009</a></td>
</tr>
<tr class="even">
<td>7.</td>
<td>RNA-seq (Supplementary Table S1)</td>
<td><a href="https://www.sciencedirect.com/science/article/pii/S109727651930810X?via%3Dihub#app2">Boxer et. al, 2020</a></td>
</tr>
<tr class="odd">
<td>8.</td>
<td>snRNA-seq (Supplementary Table S7)</td>
<td><a href="https://www.nature.com/articles/s41593-018-0270-6#citeas">Renthal et. al, 2018</a></td>
</tr>
</tbody>
</table>
</section>
<section id="load-libraries" class="level3">
<h3 class="anchored" data-anchor-id="load-libraries">Load Libraries</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_knit<span class="sc">$</span><span class="fu">set</span>(<span class="at">root.dir =</span> <span class="st">"/work/OBI/Neuroinformatics_Core/s225347/Mecp2_project/"</span>)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressPackageStartupMessages</span>({</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ChIPseeker)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(openxlsx)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(reshape2)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggExtra)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(MASS)</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scales)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(regioneR)</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(plyranges)</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(AnnotationHub)</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Rsamtools)</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Rsubread)</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(edgeR)</span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpointdensity)</span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(viridis)</span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(methylKit)</span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(EnrichedHeatmap)</span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(circlize)</span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer)</span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(org.Mm.eg.db)</span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(BSgenome)</span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(TxDb.Mmusculus.UCSC.mm10.knownGene)</span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(BSgenome.Mmusculus.UCSC.mm10)</span>
<span id="cb1-29"><a href="#cb1-29" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Repitools)</span>
<span id="cb1-30"><a href="#cb1-30" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(gridExtra)</span>
<span id="cb1-31"><a href="#cb1-31" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(seqLogo)</span>
<span id="cb1-32"><a href="#cb1-32" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(universalmotif) </span>
<span id="cb1-33"><a href="#cb1-33" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(MotifDb) </span>
<span id="cb1-34"><a href="#cb1-34" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(memes)</span>
<span id="cb1-35"><a href="#cb1-35" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(BRGenomics)</span>
<span id="cb1-36"><a href="#cb1-36" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(rtracklayer)</span>
<span id="cb1-37"><a href="#cb1-37" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GenometriCorr)</span>
<span id="cb1-38"><a href="#cb1-38" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(parallel)</span>
<span id="cb1-39"><a href="#cb1-39" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-40"><a href="#cb1-40" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GeneOverlap)</span>
<span id="cb1-41"><a href="#cb1-41" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggvenn)</span>
<span id="cb1-42"><a href="#cb1-42" aria-hidden="true" tabindex="-1"></a>txdb <span class="ot">=</span> TxDb.Mmusculus.UCSC.mm10.knownGene</span>
<span id="cb1-43"><a href="#cb1-43" aria-hidden="true" tabindex="-1"></a>})</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-evl="false">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>path <span class="ot">=</span> <span class="fu">getwd</span>()</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dir.create</span>(<span class="fu">file.path</span>(<span class="fu">paste0</span>(path,<span class="st">"/results"</span>)), <span class="at">showWarnings =</span> <span class="cn">FALSE</span>)</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dir.create</span>(<span class="fu">file.path</span>(<span class="fu">paste0</span>(path,<span class="st">"/results/RData"</span>)), <span class="at">showWarnings =</span> <span class="cn">FALSE</span>)</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="fu">dir.create</span>(<span class="fu">file.path</span>(<span class="fu">paste0</span>(path,<span class="st">"/Plots"</span>)), <span class="at">showWarnings =</span> <span class="cn">FALSE</span>)</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="co">#dir.create(file.path(paste0(path,"/Plots")), showWarnings = FALSE)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="data-processing" class="level3">
<h3 class="anchored" data-anchor-id="data-processing">Data processing</h3>
<ol type="1">
<li><p><strong>MECP2 peak processing</strong></p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># read tag count output from annotatepeaks.pl </span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>MECP2_tagCount.bed <span class="ot">=</span> <span class="fu">read.csv</span>(</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>  <span class="st">"results/MECP2/MECP2WT.vs.KO_overlap_MECP2WT.vs.IgG.noBlacklist_tagCount.tsv"</span>,</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>   <span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>,<span class="at">header=</span>T) </span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="co">#head(MECP2_tagCount.bed)</span></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(MECP2_tagCount.bed)[<span class="dv">1</span>] <span class="ot">=</span> <span class="st">'PeakID'</span></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>MECP2_tagCount <span class="ot">=</span> MECP2_tagCount.bed <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">1</span>,<span class="dv">20</span><span class="sc">:</span><span class="dv">23</span>) </span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(MECP2_tagCount)[<span class="dv">1</span>] <span class="ot">=</span> <span class="st">"PeakID"</span></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(MECP2_tagCount) <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">'……CUT_and_RUN.results.|_CNR_.*'</span>,<span class="st">''</span>,</span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">colnames</span>(MECP2_tagCount))</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a><span class="co"># After looking at the peak height in IGV, </span></span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a><span class="co"># we decided to filter the peaks whose average tag count is below 20 across the </span></span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a><span class="co"># wild type samples</span></span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>MECP2_tagCount.filtered <span class="ot">=</span> MECP2_tagCount <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">rowMeans</span>(.[,<span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">5</span>)]) <span class="sc">&gt;</span><span class="dv">20</span>) </span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a>MECP2_tagCount.filtered.bed  <span class="ot">=</span> MECP2_tagCount.bed <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(PeakID <span class="sc">%in%</span> MECP2_tagCount.filtered<span class="sc">$</span>PeakID) </span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-20"><a href="#cb3-20" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Total number of filtered MECP2 peaks = "</span>,<span class="fu">dim</span>(MECP2_tagCount.filtered.bed)[<span class="dv">1</span>]))</span>
<span id="cb3-21"><a href="#cb3-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-22"><a href="#cb3-22" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(MECP2_tagCount.filtered.bed,</span>
<span id="cb3-23"><a href="#cb3-23" aria-hidden="true" tabindex="-1"></a>              <span class="at">file=</span><span class="st">"results/MECP2/MECP2WT.vs.KO_overlap_MECP2WT.vs.IgG.noBlacklist_filtered.bed"</span>,</span>
<span id="cb3-24"><a href="#cb3-24" aria-hidden="true" tabindex="-1"></a>              <span class="at">sep=</span><span class="st">'</span><span class="sc">\t</span><span class="st">'</span>,</span>
<span id="cb3-25"><a href="#cb3-25" aria-hidden="true" tabindex="-1"></a>              <span class="at">quote =</span> F,</span>
<span id="cb3-26"><a href="#cb3-26" aria-hidden="true" tabindex="-1"></a>              <span class="at">col.names =</span> F,</span>
<span id="cb3-27"><a href="#cb3-27" aria-hidden="true" tabindex="-1"></a>              <span class="at">row.names =</span> F)</span>
<span id="cb3-28"><a href="#cb3-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-29"><a href="#cb3-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-30"><a href="#cb3-30" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(MECP2_tagCount.filtered.bed,</span>
<span id="cb3-31"><a href="#cb3-31" aria-hidden="true" tabindex="-1"></a>        <span class="at">file=</span><span class="st">"results/MECP2/MECP2WT.vs.KO_overlap_MECP2WT.vs.IgG.noBlacklist_filtered.rds"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></li>
<li><p><strong>Peak annotation of MECP2</strong></p>
<p><strong>Figure 3A</strong></p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Read MECP2 Binding peaks</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>mecp2  <span class="ot">=</span> <span class="fu">paste0</span>(<span class="st">"results/MECP2/MECP2WT.vs.KO_overlap_MECP2WT.vs.IgG.noBlacklist_filtered.bed"</span>)</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>peak <span class="ot">=</span> <span class="fu">readPeakFile</span>(mecp2)</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>peakAnno.mecp2 <span class="ot">=</span> <span class="fu">annotatePeak</span>(mecp2, </span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>                               <span class="at">tssRegion=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">1000</span>, <span class="dv">1000</span>),</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>                               <span class="at">TxDb=</span>txdb, <span class="at">annoDb=</span><span class="st">"org.Mm.eg.db"</span>)</span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a><span class="co"># plot peak distribution </span></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a><span class="fu">plotAnnoPie</span>(peakAnno.mecp2)</span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>mecp2.binding <span class="ot">=</span> <span class="fu">as.data.frame</span>(peakAnno.mecp2)</span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a><span class="co">#save RData of MECP2 annotation</span></span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(mecp2.binding,</span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a>     <span class="at">file=</span> <span class="st">"results/RData/MECP2_peakAnnotation.RData"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></li>
<li><p><strong>Read bisulfite sequencing (BS-seq) data from 8week old mice cortex</strong></p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"scripts/R/readBismarkFiles.R"</span>)</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Read CG methylation BSmap file</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>CG_methFiles <span class="ot">=</span> <span class="st">"data/8wk_CTX_Stroud2017_mm10_BSmap_CG.txt.gz"</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>CTX_CG <span class="ot">=</span> <span class="fu">readBismarkCoverage</span>(CG_methFiles,<span class="at">sample.id =</span> <span class="st">"CG"</span>,<span class="at">assembly=</span><span class="st">"unknown"</span>,</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>                             <span class="at">context=</span><span class="st">"CG"</span>,<span class="at">min.cov=</span><span class="dv">5</span>)</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(CTX_CG, <span class="at">file=</span><span class="st">'results/BSMap/8wk_CTX_Stroud2017_mm10_BSmap_CG.RData'</span>)</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>CTX_CG.df <span class="ot">=</span> <span class="fu">getData</span>(CTX_CG)</span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>CTX_CG.df<span class="sc">$</span>meth <span class="ot">=</span> CTX_CG.df<span class="sc">$</span>numCs<span class="sc">/</span>CTX_CG.df<span class="sc">$</span>coverage</span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>CTX_CG.df <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">8</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write.table</span>(<span class="st">'data/8wk_CTX_Stroud2017_mm10_BSmap_CG.cov5.bed'</span>,</span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a>              <span class="at">sep=</span><span class="st">'</span><span class="sc">\t</span><span class="st">'</span>,<span class="at">col.names =</span> <span class="cn">FALSE</span>,<span class="at">row.names =</span> <span class="cn">FALSE</span>,<span class="at">quote =</span> <span class="cn">FALSE</span>)</span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Read CA methylation BSmap file</span></span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a>CA_methFiles <span class="ot">=</span> <span class="fu">paste0</span>(path,<span class="st">"data/8wk_CTX_Stroud2017_mm10_BSmap_CA.txt.gz"</span>)</span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a>CTX_CA <span class="ot">=</span> <span class="fu">readBismarkCoverage</span>(CA_methFiles,<span class="at">sample.id =</span> <span class="st">"CA"</span>,<span class="at">assembly=</span><span class="st">"unknown"</span>,</span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a>                             <span class="at">context=</span><span class="st">"CA"</span>,<span class="at">min.cov=</span><span class="dv">5</span>)</span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(CTX_CA, <span class="at">file=</span><span class="st">'results/BSMap/8wk_CTX_Stroud2017_mm10_BSmap_CA.RData'</span>)</span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true" tabindex="-1"></a>CTX_CA.df <span class="ot">=</span> <span class="fu">getData</span>(CTX_CA)</span>
<span id="cb5-22"><a href="#cb5-22" aria-hidden="true" tabindex="-1"></a>CTX_CA.df<span class="sc">$</span>meth <span class="ot">=</span> CTX_CA.df<span class="sc">$</span>numCs<span class="sc">/</span>CTX_CA.df<span class="sc">$</span>coverage</span>
<span id="cb5-23"><a href="#cb5-23" aria-hidden="true" tabindex="-1"></a>CTX_CA.df <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">8</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb5-24"><a href="#cb5-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write.table</span>(<span class="st">'data/8wk_CTX_Stroud2017_mm10_BSmap_CA.cov5.bed'</span>,</span>
<span id="cb5-25"><a href="#cb5-25" aria-hidden="true" tabindex="-1"></a>              <span class="at">sep=</span><span class="st">'</span><span class="sc">\t</span><span class="st">'</span>,<span class="at">col.names =</span> <span class="cn">FALSE</span>,<span class="at">row.names =</span> <span class="cn">FALSE</span>,<span class="at">quote =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></li>
<li><p><strong>Read excel having list of Differentially expressed genes between following conditions.</strong></p>
<ul>
<li>MECP2 KO vs MECP2 WT (whole-cell, nuclear and Chromatin associated RNA-seq)</li>
<li>DNMT3a KO vs DNMT3A WT</li>
<li>R306C vs MECP2 WT (whole-cell, nuclear and Chromatin associated RNA-seq)</li>
</ul>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Read RNA-seq data to extract DNMT3A KOvsWT analysis</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>path <span class="ot">=</span> <span class="st">"data/Boxer_RNAseq_data.xlsx"</span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>sheets <span class="ot">=</span> openxlsx<span class="sc">::</span><span class="fu">getSheetNames</span>(path)</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a><span class="co"># print name of all the sheet</span></span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>sheets</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a><span class="co"># open each sheet by name ans make a list </span></span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>RNAseq.list <span class="ot">=</span> <span class="fu">lapply</span>(sheets, openxlsx<span class="sc">::</span>read.xlsx, <span class="at">xlsxFile=</span>path,<span class="at">startRow=</span><span class="dv">2</span>)</span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a><span class="co"># name each element of the list</span></span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(RNAseq.list) <span class="ot">=</span> sheets</span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(RNAseq.list)</span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert Gene Symbol to ENSEMBL id </span></span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a>RNAseq.list <span class="ot">=</span> <span class="fu">lapply</span>(RNAseq.list, </span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a>                     <span class="cf">function</span>(x) { <span class="fu">mutate</span>(x,<span class="at">ENSEMBL =</span> <span class="fu">mapIds</span>(org.Mm.eg.db,</span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a>                                                             <span class="at">keys=</span>x<span class="sc">$</span>Gene, </span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a>                                                             <span class="at">column=</span><span class="st">"ENSEMBL"</span>,</span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true" tabindex="-1"></a>                                                             <span class="at">keytype=</span><span class="st">"SYMBOL"</span>,</span>
<span id="cb6-20"><a href="#cb6-20" aria-hidden="true" tabindex="-1"></a>                                                             <span class="at">multiVals=</span><span class="st">"first"</span> ))})</span>
<span id="cb6-21"><a href="#cb6-21" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(<span class="at">file =</span> <span class="st">'results/RData/Boxer_RNAseq_data.RData'</span>,RNAseq.list)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></li>
<li><p><strong>Calculate gene body CA methylation</strong></p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dir.create</span>(<span class="fu">file.path</span>(<span class="fu">paste0</span>(path,<span class="st">"/results/BSMap"</span>)), <span class="at">showWarnings =</span> <span class="cn">FALSE</span>)</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Use BioMart to extract start and end location of ensembl genes </span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="co"># the below code has been taken from https://davetang.org/muse/2012/04/27/learning-to-use-biomart/</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a><span class="fu">listMarts</span>()</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>ensembl <span class="ot">=</span> <span class="fu">useMart</span>(<span class="st">"ensembl"</span>,</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>                  <span class="at">dataset=</span><span class="st">"mmusculus_gene_ensembl"</span>,</span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>                  <span class="at">host=</span><span class="st">'https://nov2020.archive.ensembl.org/'</span>)</span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a><span class="fu">listDatasets</span>(ensembl)</span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a><span class="co"># building a query, requires filters, attributes and values</span></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a><span class="co"># listFilters shows all firslte</span></span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a>filters <span class="ot">=</span> <span class="fu">listFilters</span>(ensembl)</span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a>attributes <span class="ot">=</span> <span class="fu">listAttributes</span>(ensembl)</span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(attributes)</span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a><span class="co"># the gene length for MECP2 unbound genes need to be extracted to filter gene &lt; 4.5kb</span></span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a>ensenbl_location <span class="ot">=</span> <span class="fu">getBM</span>(<span class="at">attributes=</span><span class="fu">c</span>(<span class="st">'ensembl_gene_id'</span>,</span>
<span id="cb7-19"><a href="#cb7-19" aria-hidden="true" tabindex="-1"></a>                                      <span class="st">'chromosome_name'</span>,</span>
<span id="cb7-20"><a href="#cb7-20" aria-hidden="true" tabindex="-1"></a>                                      <span class="st">'start_position'</span>,</span>
<span id="cb7-21"><a href="#cb7-21" aria-hidden="true" tabindex="-1"></a>                                      <span class="st">'end_position'</span>,<span class="st">'strand'</span>),<span class="at">mart =</span> ensembl) </span>
<span id="cb7-22"><a href="#cb7-22" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(<span class="at">file =</span> <span class="st">'results/RData/ENSEMBL.IDs_with_start_end.RData'</span>,ensenbl_location)</span>
<span id="cb7-23"><a href="#cb7-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-24"><a href="#cb7-24" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(ensenbl_location)</span>
<span id="cb7-25"><a href="#cb7-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-26"><a href="#cb7-26" aria-hidden="true" tabindex="-1"></a><span class="co">#load("results/MECP2/MECP2.Rdata",verbose =TRUE)</span></span>
<span id="cb7-27"><a href="#cb7-27" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(ensenbl_location) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">'ENSEMBL'</span>,<span class="st">'chr'</span>,<span class="st">'start'</span>,<span class="st">'end'</span>,<span class="st">'strand'</span>)</span>
<span id="cb7-28"><a href="#cb7-28" aria-hidden="true" tabindex="-1"></a>ensenbl_location <span class="ot">=</span> ensenbl_location <span class="sc">%&gt;%</span> <span class="fu">distinct</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb7-29"><a href="#cb7-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(end<span class="sc">-</span>start <span class="sc">&gt;</span><span class="dv">4500</span>)</span>
<span id="cb7-30"><a href="#cb7-30" aria-hidden="true" tabindex="-1"></a>ensenbl_location.chr1.<span class="fl">19.</span>XY <span class="ot">=</span> ensenbl_location <span class="sc">%&gt;%</span> </span>
<span id="cb7-31"><a href="#cb7-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(chr <span class="sc">%in%</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">19</span>,<span class="st">'X'</span>,<span class="st">'Y'</span>))</span>
<span id="cb7-32"><a href="#cb7-32" aria-hidden="true" tabindex="-1"></a>ensenbl_location.chr1.<span class="fl">19.</span>XY<span class="sc">$</span>chr <span class="ot">=</span> <span class="fu">paste0</span>(<span class="st">'chr'</span>,ensenbl_location.chr1.<span class="fl">19.</span>XY<span class="sc">$</span>chr)</span>
<span id="cb7-33"><a href="#cb7-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-34"><a href="#cb7-34" aria-hidden="true" tabindex="-1"></a><span class="co">#Remove first 3kb of gene </span></span>
<span id="cb7-35"><a href="#cb7-35" aria-hidden="true" tabindex="-1"></a>ensenbl_location.chr1.<span class="fl">19.</span>XY<span class="sc">$</span>start <span class="ot">=</span> <span class="fu">ifelse</span>(ensenbl_location.chr1.<span class="fl">19.</span>XY<span class="sc">$</span>strand <span class="sc">==</span> <span class="dv">1</span>,</span>
<span id="cb7-36"><a href="#cb7-36" aria-hidden="true" tabindex="-1"></a>                                           ensenbl_location.chr1.<span class="fl">19.</span>XY<span class="sc">$</span>start<span class="sc">+</span><span class="dv">3000</span>, </span>
<span id="cb7-37"><a href="#cb7-37" aria-hidden="true" tabindex="-1"></a>                                           ensenbl_location.chr1.<span class="fl">19.</span>XY<span class="sc">$</span>end<span class="dv">-3000</span>)</span>
<span id="cb7-38"><a href="#cb7-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-39"><a href="#cb7-39" aria-hidden="true" tabindex="-1"></a>ensenbl_location.GR <span class="ot">=</span> <span class="fu">GRanges</span>(ensenbl_location.chr1.<span class="fl">19.</span>XY[,<span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">1</span>)])</span>
<span id="cb7-40"><a href="#cb7-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-41"><a href="#cb7-41" aria-hidden="true" tabindex="-1"></a><span class="co"># load CA methylation data</span></span>
<span id="cb7-42"><a href="#cb7-42" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">'results/BSMap/8wk_CTX_Stroud2017_mm10_BSmap_CA.RData'</span>,<span class="at">verbose =</span> <span class="cn">TRUE</span>)</span>
<span id="cb7-43"><a href="#cb7-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-44"><a href="#cb7-44" aria-hidden="true" tabindex="-1"></a>ensenbl_location.GR_CA_meth <span class="ot">=</span> <span class="fu">regionCounts</span>(CTX_CA,ensenbl_location.GR)</span>
<span id="cb7-45"><a href="#cb7-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-46"><a href="#cb7-46" aria-hidden="true" tabindex="-1"></a>ensenbl_location.df_CA_meth <span class="ot">=</span> <span class="fu">getData</span>(ensenbl_location.GR_CA_meth)</span>
<span id="cb7-47"><a href="#cb7-47" aria-hidden="true" tabindex="-1"></a>ensenbl_location.df_CA_meth<span class="sc">$</span>meth <span class="ot">=</span> ensenbl_location.df_CA_meth<span class="sc">$</span>numCs<span class="sc">/</span>ensenbl_location.df_CA_meth<span class="sc">$</span>coverage</span>
<span id="cb7-48"><a href="#cb7-48" aria-hidden="true" tabindex="-1"></a>ensenbl_location.df_CA_meth<span class="sc">$</span>ID <span class="ot">=</span> <span class="fu">paste0</span>(ensenbl_location.df_CA_meth<span class="sc">$</span>chr,<span class="st">'_'</span>,</span>
<span id="cb7-49"><a href="#cb7-49" aria-hidden="true" tabindex="-1"></a>                                        ensenbl_location.df_CA_meth<span class="sc">$</span>start,<span class="st">'_'</span>,</span>
<span id="cb7-50"><a href="#cb7-50" aria-hidden="true" tabindex="-1"></a>                                        ensenbl_location.df_CA_meth<span class="sc">$</span>end)</span>
<span id="cb7-51"><a href="#cb7-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-52"><a href="#cb7-52" aria-hidden="true" tabindex="-1"></a>ensenbl_location.chr1.<span class="fl">19.</span>XY<span class="sc">$</span>ID <span class="ot">=</span> <span class="fu">paste0</span>(ensenbl_location.chr1.<span class="fl">19.</span>XY<span class="sc">$</span>chr,<span class="st">'_'</span>,</span>
<span id="cb7-53"><a href="#cb7-53" aria-hidden="true" tabindex="-1"></a>                                        ensenbl_location.chr1.<span class="fl">19.</span>XY<span class="sc">$</span>start,<span class="st">'_'</span>,</span>
<span id="cb7-54"><a href="#cb7-54" aria-hidden="true" tabindex="-1"></a>                                        ensenbl_location.chr1.<span class="fl">19.</span>XY<span class="sc">$</span>end)</span>
<span id="cb7-55"><a href="#cb7-55" aria-hidden="true" tabindex="-1"></a>ensenbl_location.df_CA_meth <span class="ot">=</span> <span class="fu">merge</span>(ensenbl_location.df_CA_meth,</span>
<span id="cb7-56"><a href="#cb7-56" aria-hidden="true" tabindex="-1"></a>                                    ensenbl_location.chr1.<span class="fl">19.</span>XY,<span class="at">by=</span><span class="st">'ID'</span>,<span class="at">all.x=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb7-57"><a href="#cb7-57" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">distinct</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb7-58"><a href="#cb7-58" aria-hidden="true" tabindex="-1"></a>                                    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">10</span>,<span class="dv">11</span>,<span class="dv">12</span>,<span class="dv">13</span>,<span class="dv">14</span>,<span class="dv">6</span>,<span class="dv">7</span>,<span class="dv">8</span>,<span class="dv">9</span>)</span>
<span id="cb7-59"><a href="#cb7-59" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(ensenbl_location.df_CA_meth) <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">'</span><span class="sc">\\</span><span class="st">.y'</span>,<span class="st">''</span>,</span>
<span id="cb7-60"><a href="#cb7-60" aria-hidden="true" tabindex="-1"></a>                                             <span class="fu">colnames</span>(ensenbl_location.df_CA_meth))</span>
<span id="cb7-61"><a href="#cb7-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-62"><a href="#cb7-62" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(ensenbl_location.df_CA_meth,</span>
<span id="cb7-63"><a href="#cb7-63" aria-hidden="true" tabindex="-1"></a>     <span class="at">file=</span><span class="st">'results/BSMap/8wk_CTX_Stroud2017_mm10_BSmap_CA_within_mm10_ensembl_genes.First3kb_excluded.RData'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></li>
<li><p><strong>Integrate MECP2 peaks, DEGs (MECP2 KO vs WT), and gene body CA and CG methylation</strong></p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">'results/RData/Boxer_RNAseq_data.RData'</span>,<span class="at">verbose =</span> <span class="cn">TRUE</span>)</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>RNAseq_Mecp2KO <span class="ot">=</span> RNAseq.list[<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>)]</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>geneSet1 <span class="ot">=</span> <span class="fu">lapply</span>(RNAseq_Mecp2KO, </span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>                  <span class="cf">function</span>(x) x <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(FDR <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> logFC <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">pull</span>(Gene))</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>geneSet2 <span class="ot">=</span> <span class="fu">lapply</span>(RNAseq_Mecp2KO, </span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>                  <span class="cf">function</span>(x) x <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(FDR <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> logFC <span class="sc">&lt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">pull</span>(Gene))</span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>geneSet3 <span class="ot">=</span> <span class="fu">lapply</span>(RNAseq_Mecp2KO, </span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>                  <span class="cf">function</span>(x) x <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(FDR <span class="sc">&gt;</span> <span class="fl">0.05</span> <span class="sc">|</span> <span class="fu">is.na</span>(FDR)) <span class="sc">%&gt;%</span></span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">pull</span>(Gene))</span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(geneSet1) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"MECP2.KO_vs_Control.WholeCell.Repressed"</span>,</span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a>                    <span class="st">"MECP2.KO_vs_Control.Nuclear.Repressed"</span>,</span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a>                    <span class="st">"MECP2.KO_vs_Control.Chromatin.Repressed"</span>)</span>
<span id="cb8-16"><a href="#cb8-16" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(geneSet2) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"MECP2.KO_vs_Control.WholeCell.Activated"</span>,</span>
<span id="cb8-17"><a href="#cb8-17" aria-hidden="true" tabindex="-1"></a>                    <span class="st">"MECP2.KO_vs_Control.Nuclear.Activated"</span>,</span>
<span id="cb8-18"><a href="#cb8-18" aria-hidden="true" tabindex="-1"></a>                    <span class="st">"MECP2.KO_vs_Control.Chromatin.Activated"</span>)</span>
<span id="cb8-19"><a href="#cb8-19" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(geneSet3) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"MECP2.KO_vs_Control.WholeCell.Unchanged"</span>,</span>
<span id="cb8-20"><a href="#cb8-20" aria-hidden="true" tabindex="-1"></a>                    <span class="st">"MECP2.KO_vs_Control.Nuclear.Unchanged"</span>,</span>
<span id="cb8-21"><a href="#cb8-21" aria-hidden="true" tabindex="-1"></a>                    <span class="st">"MECP2.KO_vs_Control.Chromatin.Unchanged"</span>)</span>
<span id="cb8-22"><a href="#cb8-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-23"><a href="#cb8-23" aria-hidden="true" tabindex="-1"></a>RNAseq_Mecp2KO_GeneSets <span class="ot">=</span> <span class="fu">c</span>(geneSet1,geneSet2,geneSet3)</span>
<span id="cb8-24"><a href="#cb8-24" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(RNAseq_Mecp2KO_GeneSets)</span>
<span id="cb8-25"><a href="#cb8-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-26"><a href="#cb8-26" aria-hidden="true" tabindex="-1"></a><span class="do">##############################################################################################</span></span>
<span id="cb8-27"><a href="#cb8-27" aria-hidden="true" tabindex="-1"></a><span class="co"># Read Methylation Data</span></span>
<span id="cb8-28"><a href="#cb8-28" aria-hidden="true" tabindex="-1"></a><span class="co"># CG methylation </span></span>
<span id="cb8-29"><a href="#cb8-29" aria-hidden="true" tabindex="-1"></a>CG <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">"data/8wk_CTX_Stroud2017_mm10_BSmap_CG_within_mm10_ensembl_genes.txt"</span>,</span>
<span id="cb8-30"><a href="#cb8-30" aria-hidden="true" tabindex="-1"></a>              <span class="at">sep=</span><span class="st">'</span><span class="sc">\t</span><span class="st">'</span>,<span class="at">header =</span> F)</span>
<span id="cb8-31"><a href="#cb8-31" aria-hidden="true" tabindex="-1"></a>CG<span class="sc">$</span>CG_meth <span class="ot">=</span> CG<span class="sc">$</span>V2<span class="sc">/</span>CG<span class="sc">$</span>V3</span>
<span id="cb8-32"><a href="#cb8-32" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(CG)</span>
<span id="cb8-33"><a href="#cb8-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-34"><a href="#cb8-34" aria-hidden="true" tabindex="-1"></a><span class="co"># Read Methylation Data</span></span>
<span id="cb8-35"><a href="#cb8-35" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"results/BSMap/8wk_CTX_Stroud2017_mm10_BSmap_CA_within_mm10_ensembl_genes.First3kb_excluded.RData"</span>)</span>
<span id="cb8-36"><a href="#cb8-36" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(ensenbl_location.df_CA_meth)</span>
<span id="cb8-37"><a href="#cb8-37" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(ensenbl_location.df_CA_meth)[<span class="dv">9</span>] <span class="ot">=</span><span class="st">"CA_meth"</span></span>
<span id="cb8-38"><a href="#cb8-38" aria-hidden="true" tabindex="-1"></a>CA <span class="ot">=</span> ensenbl_location.df_CA_meth </span>
<span id="cb8-39"><a href="#cb8-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-40"><a href="#cb8-40" aria-hidden="true" tabindex="-1"></a>meth <span class="ot">=</span> <span class="fu">merge</span>(CG,CA,<span class="at">by.x=</span><span class="st">'V1'</span>,<span class="at">by.y=</span><span class="st">'ENSEMBL'</span>,<span class="at">all=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">1</span>,<span class="dv">4</span>,<span class="dv">12</span>)</span>
<span id="cb8-41"><a href="#cb8-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-42"><a href="#cb8-42" aria-hidden="true" tabindex="-1"></a><span class="do">##############################################################################################</span></span>
<span id="cb8-43"><a href="#cb8-43" aria-hidden="true" tabindex="-1"></a><span class="co"># merge binding and methylation and RNAseq</span></span>
<span id="cb8-44"><a href="#cb8-44" aria-hidden="true" tabindex="-1"></a>mecp2.binding.meth <span class="ot">=</span> <span class="fu">merge</span>(mecp2.binding,meth,<span class="at">by.x=</span><span class="st">'ENSEMBL'</span>,<span class="at">by.y=</span><span class="st">'V1'</span>,<span class="at">all=</span><span class="cn">TRUE</span>) </span>
<span id="cb8-45"><a href="#cb8-45" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(mecp2.binding.meth)</span>
<span id="cb8-46"><a href="#cb8-46" aria-hidden="true" tabindex="-1"></a><span class="do">##############################################################################################</span></span>
<span id="cb8-47"><a href="#cb8-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-48"><a href="#cb8-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-49"><a href="#cb8-49" aria-hidden="true" tabindex="-1"></a><span class="co"># Whole Cell</span></span>
<span id="cb8-50"><a href="#cb8-50" aria-hidden="true" tabindex="-1"></a>mecp2.binding.meth.RNAseqKO_wholeCell <span class="ot">=</span> <span class="fu">merge</span>(mecp2.binding.meth, RNAseq_Mecp2KO[[<span class="dv">1</span>]], </span>
<span id="cb8-51"><a href="#cb8-51" aria-hidden="true" tabindex="-1"></a>      <span class="at">by =</span><span class="st">'ENSEMBL'</span>, <span class="at">all=</span>T) <span class="sc">%&gt;%</span></span>
<span id="cb8-52"><a href="#cb8-52" aria-hidden="true" tabindex="-1"></a>      <span class="fu">mutate</span>(<span class="at">Binding =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(SYMBOL), <span class="st">'Unbound'</span>,<span class="st">'Bound'</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb8-53"><a href="#cb8-53" aria-hidden="true" tabindex="-1"></a>      <span class="fu">mutate</span>(<span class="at">Meth_CA =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(CA_meth), <span class="st">'No_CA'</span>,<span class="st">'CA'</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb8-54"><a href="#cb8-54" aria-hidden="true" tabindex="-1"></a>      <span class="fu">mutate</span>(<span class="at">Meth_CG =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(CG_meth), <span class="st">'No_CG'</span>,<span class="st">'CG'</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb8-55"><a href="#cb8-55" aria-hidden="true" tabindex="-1"></a>      <span class="fu">mutate</span>(<span class="at">Expression =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(logFC), <span class="st">'Not_Expressed'</span>,<span class="st">'Expressed'</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb8-56"><a href="#cb8-56" aria-hidden="true" tabindex="-1"></a>      <span class="fu">mutate</span>(<span class="at">DEG =</span> <span class="fu">case_when</span>(logFC <span class="sc">&gt;</span><span class="dv">0</span> <span class="sc">&amp;</span> FDR <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">~</span> <span class="st">"Repressed"</span>,</span>
<span id="cb8-57"><a href="#cb8-57" aria-hidden="true" tabindex="-1"></a>                       logFC <span class="sc">&lt;</span><span class="dv">0</span> <span class="sc">&amp;</span> FDR <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">~</span> <span class="st">"Activated"</span>,</span>
<span id="cb8-58"><a href="#cb8-58" aria-hidden="true" tabindex="-1"></a>                       FDR <span class="sc">&gt;=</span><span class="fl">0.05</span> <span class="sc">|</span> <span class="fu">is.na</span>(FDR) <span class="sc">~</span> <span class="st">"Unchanged"</span>))</span>
<span id="cb8-59"><a href="#cb8-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-60"><a href="#cb8-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-61"><a href="#cb8-61" aria-hidden="true" tabindex="-1"></a><span class="co"># Whole Cell Bound</span></span>
<span id="cb8-62"><a href="#cb8-62" aria-hidden="true" tabindex="-1"></a>Mecp2.Bound.meth.expressed.wholeCell <span class="ot">=</span> mecp2.binding.meth.RNAseqKO_wholeCell <span class="sc">%&gt;%</span> </span>
<span id="cb8-63"><a href="#cb8-63" aria-hidden="true" tabindex="-1"></a>      <span class="fu">filter</span>(Expression <span class="sc">==</span><span class="st">'Expressed'</span> <span class="sc">&amp;</span> Binding <span class="sc">==</span> <span class="st">'Bound'</span> <span class="sc">&amp;</span> Meth_CA <span class="sc">==</span> <span class="st">'CA'</span> <span class="sc">&amp;</span> Meth_CG<span class="sc">==</span><span class="st">'CG'</span>)</span>
<span id="cb8-64"><a href="#cb8-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-65"><a href="#cb8-65" aria-hidden="true" tabindex="-1"></a><span class="co"># wholeCell Unbound</span></span>
<span id="cb8-66"><a href="#cb8-66" aria-hidden="true" tabindex="-1"></a>Mecp2.UnBound.meth.expressed.wholeCell <span class="ot">=</span> mecp2.binding.meth.RNAseqKO_wholeCell <span class="sc">%&gt;%</span> </span>
<span id="cb8-67"><a href="#cb8-67" aria-hidden="true" tabindex="-1"></a>      <span class="fu">filter</span>(Expression <span class="sc">==</span><span class="st">'Expressed'</span> <span class="sc">&amp;</span> Binding <span class="sc">==</span> <span class="st">'Unbound'</span> <span class="sc">&amp;</span> Meth_CA <span class="sc">==</span> <span class="st">'CA'</span> <span class="sc">&amp;</span> Meth_CG<span class="sc">==</span><span class="st">'CG'</span>)</span>
<span id="cb8-68"><a href="#cb8-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-69"><a href="#cb8-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-70"><a href="#cb8-70" aria-hidden="true" tabindex="-1"></a><span class="co"># Write table </span></span>
<span id="cb8-71"><a href="#cb8-71" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(Mecp2.Bound.meth.expressed.wholeCell,</span>
<span id="cb8-72"><a href="#cb8-72" aria-hidden="true" tabindex="-1"></a>            <span class="at">file =</span> <span class="st">"results/MECP2/Mecp2.Bound.meth.expressed.wholeCell.tsv"</span>,</span>
<span id="cb8-73"><a href="#cb8-73" aria-hidden="true" tabindex="-1"></a>            <span class="at">sep=</span><span class="st">'</span><span class="sc">\t</span><span class="st">'</span>,<span class="at">quote =</span> F,</span>
<span id="cb8-74"><a href="#cb8-74" aria-hidden="true" tabindex="-1"></a>            <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> F)</span>
<span id="cb8-75"><a href="#cb8-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-76"><a href="#cb8-76" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(Mecp2.UnBound.meth.expressed.wholeCell,</span>
<span id="cb8-77"><a href="#cb8-77" aria-hidden="true" tabindex="-1"></a>            <span class="at">file =</span> <span class="st">"results/MECP2/Mecp2.UnBound.meth.expressed.wholeCell.tsv"</span>,</span>
<span id="cb8-78"><a href="#cb8-78" aria-hidden="true" tabindex="-1"></a>            <span class="at">sep=</span><span class="st">'</span><span class="sc">\t</span><span class="st">'</span>,<span class="at">quote =</span> F,</span>
<span id="cb8-79"><a href="#cb8-79" aria-hidden="true" tabindex="-1"></a>            <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> F)</span>
<span id="cb8-80"><a href="#cb8-80" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-81"><a href="#cb8-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-82"><a href="#cb8-82" aria-hidden="true" tabindex="-1"></a><span class="do">##############################################################################################</span></span>
<span id="cb8-83"><a href="#cb8-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-84"><a href="#cb8-84" aria-hidden="true" tabindex="-1"></a><span class="co"># nuclear</span></span>
<span id="cb8-85"><a href="#cb8-85" aria-hidden="true" tabindex="-1"></a>mecp2.binding.meth.RNAseqKO_nuclear <span class="ot">=</span> <span class="fu">merge</span>(mecp2.binding.meth, RNAseq_Mecp2KO[[<span class="dv">2</span>]], </span>
<span id="cb8-86"><a href="#cb8-86" aria-hidden="true" tabindex="-1"></a>  <span class="at">by =</span><span class="st">'ENSEMBL'</span>, <span class="at">all=</span>T) <span class="sc">%&gt;%</span></span>
<span id="cb8-87"><a href="#cb8-87" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Binding =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(SYMBOL), <span class="st">'Unbound'</span>,<span class="st">'Bound'</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb8-88"><a href="#cb8-88" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Meth_CA =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(CA_meth), <span class="st">'No_CA'</span>,<span class="st">'CA'</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb8-89"><a href="#cb8-89" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Meth_CG =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(CG_meth), <span class="st">'No_CG'</span>,<span class="st">'CG'</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb8-90"><a href="#cb8-90" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Expression =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(logFC), <span class="st">'Not_Expressed'</span>,<span class="st">'Expressed'</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb8-91"><a href="#cb8-91" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">DEG =</span> <span class="fu">case_when</span>(logFC <span class="sc">&gt;</span><span class="dv">0</span> <span class="sc">&amp;</span> FDR <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">~</span> <span class="st">"Repressed"</span>,</span>
<span id="cb8-92"><a href="#cb8-92" aria-hidden="true" tabindex="-1"></a>                         logFC <span class="sc">&lt;</span><span class="dv">0</span> <span class="sc">&amp;</span> FDR <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">~</span> <span class="st">"Activated"</span>,</span>
<span id="cb8-93"><a href="#cb8-93" aria-hidden="true" tabindex="-1"></a>                         FDR <span class="sc">&gt;=</span><span class="fl">0.05</span> <span class="sc">|</span> <span class="fu">is.na</span>(FDR) <span class="sc">~</span> <span class="st">"Unchanged"</span>))</span>
<span id="cb8-94"><a href="#cb8-94" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-95"><a href="#cb8-95" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-96"><a href="#cb8-96" aria-hidden="true" tabindex="-1"></a><span class="co"># nuclear Bound</span></span>
<span id="cb8-97"><a href="#cb8-97" aria-hidden="true" tabindex="-1"></a>Mecp2.Bound.meth.expressed.nuclear <span class="ot">=</span> mecp2.binding.meth.RNAseqKO_nuclear <span class="sc">%&gt;%</span> </span>
<span id="cb8-98"><a href="#cb8-98" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Expression <span class="sc">==</span><span class="st">'Expressed'</span> <span class="sc">&amp;</span> Binding <span class="sc">==</span> <span class="st">'Bound'</span> <span class="sc">&amp;</span> Meth_CA <span class="sc">==</span> <span class="st">'CA'</span> <span class="sc">&amp;</span> Meth_CG<span class="sc">==</span><span class="st">'CG'</span>)</span>
<span id="cb8-99"><a href="#cb8-99" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-100"><a href="#cb8-100" aria-hidden="true" tabindex="-1"></a><span class="co"># nuclear Unbound</span></span>
<span id="cb8-101"><a href="#cb8-101" aria-hidden="true" tabindex="-1"></a>Mecp2.UnBound.meth.expressed.nuclear <span class="ot">=</span> mecp2.binding.meth.RNAseqKO_nuclear <span class="sc">%&gt;%</span> </span>
<span id="cb8-102"><a href="#cb8-102" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Expression <span class="sc">==</span><span class="st">'Expressed'</span> <span class="sc">&amp;</span> Binding <span class="sc">==</span> <span class="st">'Unbound'</span> <span class="sc">&amp;</span> Meth_CA <span class="sc">==</span> <span class="st">'CA'</span> <span class="sc">&amp;</span> Meth_CG<span class="sc">==</span><span class="st">'CG'</span>)</span>
<span id="cb8-103"><a href="#cb8-103" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-104"><a href="#cb8-104" aria-hidden="true" tabindex="-1"></a><span class="co"># write table</span></span>
<span id="cb8-105"><a href="#cb8-105" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(Mecp2.UnBound.meth.expressed.nuclear,</span>
<span id="cb8-106"><a href="#cb8-106" aria-hidden="true" tabindex="-1"></a>            <span class="at">file =</span> <span class="st">"results/MECP2/Mecp2.UnBound.meth.expressed.nuclear.tsv"</span>,</span>
<span id="cb8-107"><a href="#cb8-107" aria-hidden="true" tabindex="-1"></a>            <span class="at">sep=</span><span class="st">'</span><span class="sc">\t</span><span class="st">'</span>,<span class="at">quote =</span> F,</span>
<span id="cb8-108"><a href="#cb8-108" aria-hidden="true" tabindex="-1"></a>            <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> F)</span>
<span id="cb8-109"><a href="#cb8-109" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(Mecp2.Bound.meth.expressed.nuclear,</span>
<span id="cb8-110"><a href="#cb8-110" aria-hidden="true" tabindex="-1"></a>            <span class="at">file =</span> <span class="st">"results/MECP2/Mecp2.Bound.meth.expressed.nuclear.tsv"</span>,</span>
<span id="cb8-111"><a href="#cb8-111" aria-hidden="true" tabindex="-1"></a>            <span class="at">sep=</span><span class="st">'</span><span class="sc">\t</span><span class="st">'</span>,<span class="at">quote =</span> F,</span>
<span id="cb8-112"><a href="#cb8-112" aria-hidden="true" tabindex="-1"></a>            <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> F)</span>
<span id="cb8-113"><a href="#cb8-113" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-114"><a href="#cb8-114" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-115"><a href="#cb8-115" aria-hidden="true" tabindex="-1"></a><span class="do">##############################################################################################</span></span>
<span id="cb8-116"><a href="#cb8-116" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-117"><a href="#cb8-117" aria-hidden="true" tabindex="-1"></a><span class="co"># chromatin</span></span>
<span id="cb8-118"><a href="#cb8-118" aria-hidden="true" tabindex="-1"></a>mecp2.binding.meth.RNAseqKO_chromatin <span class="ot">=</span> <span class="fu">merge</span>(mecp2.binding.meth, RNAseq_Mecp2KO[[<span class="dv">3</span>]],</span>
<span id="cb8-119"><a href="#cb8-119" aria-hidden="true" tabindex="-1"></a>  <span class="at">by =</span><span class="st">'ENSEMBL'</span>, <span class="at">all=</span>T) <span class="sc">%&gt;%</span></span>
<span id="cb8-120"><a href="#cb8-120" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Binding =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(SYMBOL), <span class="st">'Unbound'</span>,<span class="st">'Bound'</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb8-121"><a href="#cb8-121" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Meth_CA =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(CA_meth), <span class="st">'No_CA'</span>,<span class="st">'CA'</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb8-122"><a href="#cb8-122" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Meth_CG =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(CG_meth), <span class="st">'No_CG'</span>,<span class="st">'CG'</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb8-123"><a href="#cb8-123" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Expression =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(logFC), <span class="st">'Not_Expressed'</span>,<span class="st">'Expressed'</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb8-124"><a href="#cb8-124" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">DEG =</span> <span class="fu">case_when</span>(logFC <span class="sc">&gt;</span><span class="dv">0</span> <span class="sc">&amp;</span> FDR <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">~</span> <span class="st">"Repressed"</span>,</span>
<span id="cb8-125"><a href="#cb8-125" aria-hidden="true" tabindex="-1"></a>                         logFC <span class="sc">&lt;</span><span class="dv">0</span> <span class="sc">&amp;</span> FDR <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">~</span> <span class="st">"Activated"</span>,</span>
<span id="cb8-126"><a href="#cb8-126" aria-hidden="true" tabindex="-1"></a>                         FDR <span class="sc">&gt;=</span><span class="fl">0.05</span> <span class="sc">|</span> <span class="fu">is.na</span>(FDR) <span class="sc">~</span> <span class="st">"Unchanged"</span>))</span>
<span id="cb8-127"><a href="#cb8-127" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-128"><a href="#cb8-128" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-129"><a href="#cb8-129" aria-hidden="true" tabindex="-1"></a><span class="co"># chromatin Bound</span></span>
<span id="cb8-130"><a href="#cb8-130" aria-hidden="true" tabindex="-1"></a>Mecp2.Bound.meth.expressed.chromatin <span class="ot">=</span> mecp2.binding.meth.RNAseqKO_chromatin <span class="sc">%&gt;%</span> </span>
<span id="cb8-131"><a href="#cb8-131" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Expression <span class="sc">==</span><span class="st">'Expressed'</span> <span class="sc">&amp;</span> Binding <span class="sc">==</span> <span class="st">'Bound'</span> <span class="sc">&amp;</span> Meth_CA <span class="sc">==</span> <span class="st">'CA'</span> <span class="sc">&amp;</span> Meth_CG<span class="sc">==</span><span class="st">'CG'</span>)</span>
<span id="cb8-132"><a href="#cb8-132" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-133"><a href="#cb8-133" aria-hidden="true" tabindex="-1"></a><span class="co"># chromatin Unbound</span></span>
<span id="cb8-134"><a href="#cb8-134" aria-hidden="true" tabindex="-1"></a>Mecp2.UnBound.meth.expressed.chromatin <span class="ot">=</span> mecp2.binding.meth.RNAseqKO_chromatin <span class="sc">%&gt;%</span> </span>
<span id="cb8-135"><a href="#cb8-135" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Expression <span class="sc">==</span><span class="st">'Expressed'</span> <span class="sc">&amp;</span> Binding <span class="sc">==</span> <span class="st">'Unbound'</span> <span class="sc">&amp;</span> Meth_CA <span class="sc">==</span> <span class="st">'CA'</span> <span class="sc">&amp;</span> Meth_CG<span class="sc">==</span><span class="st">'CG'</span>)</span>
<span id="cb8-136"><a href="#cb8-136" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-137"><a href="#cb8-137" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-138"><a href="#cb8-138" aria-hidden="true" tabindex="-1"></a><span class="co"># write table </span></span>
<span id="cb8-139"><a href="#cb8-139" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(Mecp2.Bound.meth.expressed.chromatin,</span>
<span id="cb8-140"><a href="#cb8-140" aria-hidden="true" tabindex="-1"></a>            <span class="at">file =</span> <span class="st">"results/MECP2/Mecp2.Bound.meth.expressed.chromatin.tsv"</span>,</span>
<span id="cb8-141"><a href="#cb8-141" aria-hidden="true" tabindex="-1"></a>            <span class="at">sep=</span><span class="st">'</span><span class="sc">\t</span><span class="st">'</span>,<span class="at">quote =</span> F,</span>
<span id="cb8-142"><a href="#cb8-142" aria-hidden="true" tabindex="-1"></a>            <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> F)</span>
<span id="cb8-143"><a href="#cb8-143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-144"><a href="#cb8-144" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(Mecp2.UnBound.meth.expressed.chromatin,</span>
<span id="cb8-145"><a href="#cb8-145" aria-hidden="true" tabindex="-1"></a>            <span class="at">file =</span> <span class="st">"results/MECP2/Mecp2.UnBound.meth.expressed.chromatin.tsv"</span>,</span>
<span id="cb8-146"><a href="#cb8-146" aria-hidden="true" tabindex="-1"></a>            <span class="at">sep=</span><span class="st">'</span><span class="sc">\t</span><span class="st">'</span>,<span class="at">quote =</span> F,</span>
<span id="cb8-147"><a href="#cb8-147" aria-hidden="true" tabindex="-1"></a>            <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> F)</span>
<span id="cb8-148"><a href="#cb8-148" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-149"><a href="#cb8-149" aria-hidden="true" tabindex="-1"></a><span class="do">##############################################################################################</span></span>
<span id="cb8-150"><a href="#cb8-150" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(Mecp2.Bound.meth.expressed.wholeCell,</span>
<span id="cb8-151"><a href="#cb8-151" aria-hidden="true" tabindex="-1"></a>     Mecp2.UnBound.meth.expressed.wholeCell,</span>
<span id="cb8-152"><a href="#cb8-152" aria-hidden="true" tabindex="-1"></a>     mecp2.binding.meth.RNAseqKO_wholeCell,</span>
<span id="cb8-153"><a href="#cb8-153" aria-hidden="true" tabindex="-1"></a>     Mecp2.Bound.meth.expressed.nuclear,</span>
<span id="cb8-154"><a href="#cb8-154" aria-hidden="true" tabindex="-1"></a>     Mecp2.UnBound.meth.expressed.nuclear,</span>
<span id="cb8-155"><a href="#cb8-155" aria-hidden="true" tabindex="-1"></a>     mecp2.binding.meth.RNAseqKO_nuclear,</span>
<span id="cb8-156"><a href="#cb8-156" aria-hidden="true" tabindex="-1"></a>     Mecp2.Bound.meth.expressed.chromatin,</span>
<span id="cb8-157"><a href="#cb8-157" aria-hidden="true" tabindex="-1"></a>     Mecp2.UnBound.meth.expressed.chromatin,</span>
<span id="cb8-158"><a href="#cb8-158" aria-hidden="true" tabindex="-1"></a>     mecp2.binding.meth.RNAseqKO_chromatin,</span>
<span id="cb8-159"><a href="#cb8-159" aria-hidden="true" tabindex="-1"></a>     <span class="at">file =</span> <span class="st">'results/RData/MECP2_regulatedGenes.RData'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></li>
<li><p><strong>Download BS-seq data (</strong>GSE103214)</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GEOquery)</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"scripts/R/readBismarkFiles.R"</span>)</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a><span class="co"># GSE103214</span></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a><span class="fu">dir.create</span>(<span class="fu">file.path</span>(<span class="st">'data/'</span>, <span class="st">'GSE103214'</span>), <span class="at">showWarnings =</span> <span class="cn">FALSE</span>)</span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>BSdata <span class="ot">=</span> <span class="fu">paste0</span>(<span class="st">'GSM'</span>,<span class="st">''</span>, <span class="dv">2757476</span><span class="sc">:</span><span class="dv">2757479</span>)</span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a><span class="fu">lapply</span>(BSdata, <span class="cf">function</span>(x) { <span class="fu">getGEOSuppFiles</span>(<span class="at">GEO =</span> x,<span class="at">fetch_files =</span> <span class="cn">TRUE</span>,</span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a>                                             <span class="at">makeDirectory =</span> <span class="cn">FALSE</span>,</span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a>                                             <span class="at">baseDir =</span> <span class="st">'data/GSE103214'</span>)})</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Run these shell command to extract CG and CA sites</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="ex">module</span> load parallel/20150122</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span> data/GSE103214/<span class="pp">*</span>.gz <span class="kw">|</span> <span class="ex">parallel</span> <span class="at">--verbose</span> <span class="st">'zcat {} | grep "CG" &gt;{.}_CG.txt'</span></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span> data/GSE103214/<span class="pp">*</span>.gz <span class="kw">|</span> <span class="ex">parallel</span> <span class="at">--verbose</span> <span class="st">'zcat {} | grep "CA" &gt;{.}_CA.txt'</span></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span> data/GSE103214/<span class="pp">*</span>.txt <span class="kw">|</span><span class="ex">parallel</span> <span class="at">--verbose</span> <span class="st">'gzip {}'</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>chain <span class="ot">=</span> <span class="fu">import.chain</span>(<span class="st">"data/mm10/mm9ToMm10.over.chain"</span>)</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>D3ABSseq <span class="ot">=</span> <span class="fu">list.files</span>(<span class="at">path =</span> <span class="st">'data/GSE103214/'</span>,<span class="at">full.names =</span> T,<span class="at">pattern =</span> <span class="st">'C[A-Z].txt.gz'</span>)</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>D3ABSeq.list <span class="ot">=</span> <span class="fu">list</span>()</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">seq_along</span>(D3ABSseq)){</span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>  list.index <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">'.txt.gz'</span>,<span class="st">''</span>,<span class="fu">basename</span>(D3ABSseq[i]))</span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>  D3ACG <span class="ot">=</span> <span class="fu">readBismarkCoverage_col45</span>(D3ABSseq[i],</span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">sample.id =</span> <span class="fu">gsub</span>(<span class="st">'.txt.gz'</span>,<span class="st">''</span>,<span class="fu">basename</span>(D3ABSseq[i])),</span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">assembly=</span><span class="st">"mm9"</span>,<span class="at">min.cov=</span><span class="dv">5</span>)</span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a>  D3ACG.GR <span class="ot">=</span> <span class="fu">getData</span>(D3ACG) <span class="sc">%&gt;%</span> <span class="fu">GRanges</span>()</span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a>  D3ACG.GR.mm10 <span class="ot">=</span> <span class="fu">liftOver</span>(D3ACG.GR, chain)</span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a>  D3ABSeq.list[[list.index]] <span class="ot">=</span> <span class="fu">unlist</span>(<span class="fu">as</span>(D3ACG.GR.mm10,<span class="st">"GRangesList"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">meth =</span> numCs<span class="sc">/</span>coverage)</span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write.table</span>(D3ABSeq.list[[i]] <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">9</span>),</span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true" tabindex="-1"></a>              <span class="at">file =</span> <span class="fu">paste0</span>(<span class="st">'data/GSE103214/'</span>,<span class="fu">names</span>(D3ABSeq.list)[i],<span class="st">'.bedGraph'</span>),</span>
<span id="cb11-17"><a href="#cb11-17" aria-hidden="true" tabindex="-1"></a>              <span class="at">row.names =</span> <span class="cn">FALSE</span>,<span class="at">col.names =</span> <span class="cn">FALSE</span>,<span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>,</span>
<span id="cb11-18"><a href="#cb11-18" aria-hidden="true" tabindex="-1"></a>              <span class="at">quote =</span> <span class="cn">FALSE</span>)</span>
<span id="cb11-19"><a href="#cb11-19" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb11-20"><a href="#cb11-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-21"><a href="#cb11-21" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(D3ABSeq.list,<span class="at">file=</span><span class="st">'results/RData/GSE103214_Cortex_10wk_BSmap.RDS'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></li>
<li><p>Download G118E MECP2 CUT&amp;RUN data</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co"># GSE243009</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>CNRdata <span class="ot">=</span> <span class="fu">paste0</span>(<span class="st">'GSM'</span>,<span class="dv">7777160</span><span class="sc">:</span><span class="dv">7777171</span>)</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a><span class="fu">dir.create</span>(<span class="fu">file.path</span>(<span class="st">'data/'</span>, <span class="st">'GSE243009'</span>), <span class="at">showWarnings =</span> <span class="cn">FALSE</span>)</span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a><span class="fu">lapply</span>(CNRdata, <span class="cf">function</span>(x) { <span class="fu">getGEOSuppFiles</span>(<span class="at">GEO =</span> x,<span class="at">fetch_files =</span> <span class="cn">TRUE</span>,</span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>                                             <span class="at">makeDirectory =</span> <span class="cn">FALSE</span>,</span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>                                             <span class="at">baseDir =</span> <span class="st">'data/GSE243009'</span>)})</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></li>
<li><p>Download</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co"># GSE213752</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dir.create</span>(<span class="fu">file.path</span>(<span class="st">'data/'</span>, <span class="st">'GSE213752'</span>), <span class="at">showWarnings =</span> <span class="cn">FALSE</span>)</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a><span class="fu">getGEOSuppFiles</span>(<span class="at">GEO =</span> <span class="st">"GSE213752"</span>,<span class="at">fetch_files =</span> <span class="cn">TRUE</span>,</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>                                             <span class="at">makeDirectory =</span> <span class="cn">FALSE</span>,</span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>                                             <span class="at">baseDir =</span> <span class="st">'data/GSE213752'</span>)</span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a><span class="co"># GSE67293</span></span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a><span class="fu">dir.create</span>(<span class="fu">file.path</span>(<span class="st">'data/'</span>, <span class="st">'GSE67293'</span>), <span class="at">showWarnings =</span> <span class="cn">FALSE</span>)</span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a><span class="fu">getGEOSuppFiles</span>(<span class="at">GEO =</span> <span class="st">"GSE67293"</span>,<span class="at">fetch_files =</span> <span class="cn">TRUE</span>,</span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a>                <span class="at">makeDirectory =</span> <span class="cn">FALSE</span>,</span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a>                <span class="at">baseDir =</span> <span class="st">'data/GSE67293'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></li>
</ol>
</section>
<section id="figure-1-cutrun-uncovers-mecp2-binding-hotspots-mbhs-across-the-neural-genome." class="level3">
<h3 class="anchored" data-anchor-id="figure-1-cutrun-uncovers-mecp2-binding-hotspots-mbhs-across-the-neural-genome."><strong>Figure 1: CUT&amp;RUN uncovers MECP2 Binding Hotspots (MBHs) across the neural genome.</strong></h3>
<ol type="1">
<li><p>Figure 1A.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to average column value across bins and step size</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>summarize_logFC_mCA <span class="ot">=</span> <span class="cf">function</span>(x,<span class="at">binSize =</span> <span class="dv">19</span>,<span class="at">stepSize =</span><span class="dv">4</span>){</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>  df.bin <span class="ot">=</span><span class="fu">list</span>()</span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">seq</span>(<span class="at">from=</span><span class="dv">1</span>, <span class="at">to=</span><span class="fu">nrow</span>(x), <span class="at">by=</span>stepSize)){</span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(i)</span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>    start <span class="ot">=</span> i</span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>    stop  <span class="ot">=</span> i<span class="sc">+</span>binSize</span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(stop <span class="sc">&lt;=</span> <span class="fu">nrow</span>(x)){</span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>      df <span class="ot">=</span> x[start<span class="sc">:</span>stop,]</span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a>      df.bin[[i]] <span class="ot">=</span> df  <span class="sc">%&gt;%</span> <span class="fu">summarize</span>(<span class="at">cov.mean=</span><span class="fu">mean</span>(.[,<span class="dv">1</span>]),<span class="at">cov.mean.sd =</span> <span class="fu">sd</span>(.[,<span class="dv">1</span>]),<span class="at">mCA.mean=</span><span class="fu">mean</span>(.[,<span class="dv">2</span>]))</span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>{</span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a>      <span class="fu">return</span>(df.bin)</span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb14-16"><a href="#cb14-16" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb14-17"><a href="#cb14-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-18"><a href="#cb14-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Load genome wide CA methylation BSMap data</span></span>
<span id="cb14-19"><a href="#cb14-19" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">'results/BSMap/8wk_CTX_Stroud2017_mm10_BSmap_CA.RData'</span>,<span class="at">verbose =</span> <span class="cn">TRUE</span>)</span>
<span id="cb14-20"><a href="#cb14-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-21"><a href="#cb14-21" aria-hidden="true" tabindex="-1"></a><span class="co"># Read mean coverage of MECP2 7wk_MECP2 sample</span></span>
<span id="cb14-22"><a href="#cb14-22" aria-hidden="true" tabindex="-1"></a>MECP2_7wk_cov <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">"results/MECP2/7wk_CTX_MECP2_gene_Cov.bed"</span>,<span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>,<span class="at">header =</span> F)</span>
<span id="cb14-23"><a href="#cb14-23" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(MECP2_7wk_cov) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">'chr'</span>,<span class="st">'start'</span>,<span class="st">'end'</span>,<span class="st">'7wk_MECP2'</span>)</span>
<span id="cb14-24"><a href="#cb14-24" aria-hidden="true" tabindex="-1"></a>MECP2_7wk_cov<span class="sc">$</span><span class="st">`</span><span class="at">7wk_MECP2</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">ifelse</span>(MECP2_7wk_cov<span class="sc">$</span><span class="st">`</span><span class="at">7wk_MECP2</span><span class="st">`</span> <span class="sc">==</span> <span class="st">'.'</span>, <span class="dv">0</span>, MECP2_7wk_cov<span class="sc">$</span><span class="st">`</span><span class="at">7wk_MECP2</span><span class="st">`</span>)</span>
<span id="cb14-25"><a href="#cb14-25" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(MECP2_7wk_cov)</span>
<span id="cb14-26"><a href="#cb14-26" aria-hidden="true" tabindex="-1"></a>MECP2_7wk_cov <span class="ot">=</span> MECP2_7wk_cov <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="st">`</span><span class="at">7wk_MECP2</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">as.numeric</span>(<span class="st">`</span><span class="at">7wk_MECP2</span><span class="st">`</span>), </span>
<span id="cb14-27"><a href="#cb14-27" aria-hidden="true" tabindex="-1"></a>                                         <span class="at">geneLength =</span> (end<span class="sc">-</span>start)<span class="sc">/</span><span class="dv">1000</span>,</span>
<span id="cb14-28"><a href="#cb14-28" aria-hidden="true" tabindex="-1"></a>                                         <span class="at">norm_7wk_MECP2 =</span> <span class="st">`</span><span class="at">7wk_MECP2</span><span class="st">`</span><span class="sc">/</span>geneLength) <span class="co">#%&gt;% </span></span>
<span id="cb14-29"><a href="#cb14-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(norm_7wk_MECP2 <span class="sc">!=</span> <span class="fu">max</span>(norm_7wk_MECP2))</span>
<span id="cb14-30"><a href="#cb14-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-31"><a href="#cb14-31" aria-hidden="true" tabindex="-1"></a><span class="co"># Read mean coverage of IgG 7wk_IgG sample</span></span>
<span id="cb14-32"><a href="#cb14-32" aria-hidden="true" tabindex="-1"></a>IgG_7wk_cov <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">"results/MECP2/7wk_CTX_IgG_gene_Cov.bed"</span>,<span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>,<span class="at">header =</span> F)</span>
<span id="cb14-33"><a href="#cb14-33" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(IgG_7wk_cov) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">'chr'</span>,<span class="st">'start'</span>,<span class="st">'end'</span>,<span class="st">'7wk_IgG'</span>)</span>
<span id="cb14-34"><a href="#cb14-34" aria-hidden="true" tabindex="-1"></a>IgG_7wk_cov<span class="sc">$</span><span class="st">`</span><span class="at">7wk_IgG</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">ifelse</span>(IgG_7wk_cov<span class="sc">$</span><span class="st">`</span><span class="at">7wk_IgG</span><span class="st">`</span> <span class="sc">==</span> <span class="st">'.'</span>, <span class="dv">0</span>, IgG_7wk_cov<span class="sc">$</span><span class="st">`</span><span class="at">7wk_IgG</span><span class="st">`</span>)</span>
<span id="cb14-35"><a href="#cb14-35" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(IgG_7wk_cov)</span>
<span id="cb14-36"><a href="#cb14-36" aria-hidden="true" tabindex="-1"></a>IgG_7wk_cov <span class="ot">=</span> IgG_7wk_cov <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="st">`</span><span class="at">7wk_IgG</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">as.numeric</span>(<span class="st">`</span><span class="at">7wk_IgG</span><span class="st">`</span>), </span>
<span id="cb14-37"><a href="#cb14-37" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">geneLength =</span> (end<span class="sc">-</span>start)<span class="sc">/</span><span class="dv">1000</span>,</span>
<span id="cb14-38"><a href="#cb14-38" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">norm_IgG_7wk_cov =</span> <span class="st">`</span><span class="at">7wk_IgG</span><span class="st">`</span><span class="sc">/</span>geneLength) <span class="co">#%&gt;% </span></span>
<span id="cb14-39"><a href="#cb14-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(norm_IgG_7wk_cov <span class="sc">!=</span> <span class="fu">max</span>(norm_IgG_7wk_cov))</span>
<span id="cb14-40"><a href="#cb14-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-41"><a href="#cb14-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-42"><a href="#cb14-42" aria-hidden="true" tabindex="-1"></a><span class="co"># combine all the above data frame</span></span>
<span id="cb14-43"><a href="#cb14-43" aria-hidden="true" tabindex="-1"></a>MECP2_sample_coverage <span class="ot">=</span> <span class="fu">cbind</span>(MECP2_7wk_cov,IgG_7wk_cov) <span class="sc">%&gt;%</span> </span>
<span id="cb14-44"><a href="#cb14-44" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">6</span>,<span class="dv">12</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb14-45"><a href="#cb14-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">ID =</span> <span class="fu">paste0</span>(chr,<span class="st">'_'</span>,start,<span class="st">'_'</span>,end)) <span class="co">#%&gt;% </span></span>
<span id="cb14-46"><a href="#cb14-46" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(MECP2_sample_coverage)</span>
<span id="cb14-47"><a href="#cb14-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-48"><a href="#cb14-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-49"><a href="#cb14-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-50"><a href="#cb14-50" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert MECP2 coordinated to GRanges</span></span>
<span id="cb14-51"><a href="#cb14-51" aria-hidden="true" tabindex="-1"></a>MECP2.bed <span class="ot">=</span> MECP2_7wk_cov[,<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>)]</span>
<span id="cb14-52"><a href="#cb14-52" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(MECP2.bed) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">'chr'</span>,<span class="st">'start'</span>,<span class="st">'end'</span>)</span>
<span id="cb14-53"><a href="#cb14-53" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(MECP2.bed)</span>
<span id="cb14-54"><a href="#cb14-54" aria-hidden="true" tabindex="-1"></a>MECP2.bed.GR <span class="ot">=</span> <span class="fu">GRanges</span>(MECP2.bed)</span>
<span id="cb14-55"><a href="#cb14-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-56"><a href="#cb14-56" aria-hidden="true" tabindex="-1"></a><span class="co"># calculate mCA on MECP2 peaks</span></span>
<span id="cb14-57"><a href="#cb14-57" aria-hidden="true" tabindex="-1"></a>MECP2_peak_cov_meth <span class="ot">=</span> <span class="fu">regionCounts</span>(CTX_CA,MECP2.bed.GR)</span>
<span id="cb14-58"><a href="#cb14-58" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(MECP2_peak_cov_meth)</span>
<span id="cb14-59"><a href="#cb14-59" aria-hidden="true" tabindex="-1"></a>MECP2_peak_cov_meth.df <span class="ot">=</span> <span class="fu">getData</span>(MECP2_peak_cov_meth)</span>
<span id="cb14-60"><a href="#cb14-60" aria-hidden="true" tabindex="-1"></a>MECP2_peak_cov_meth.df<span class="sc">$</span>mCA <span class="ot">=</span> MECP2_peak_cov_meth.df<span class="sc">$</span>numCs<span class="sc">/</span>MECP2_peak_cov_meth.df<span class="sc">$</span>coverage</span>
<span id="cb14-61"><a href="#cb14-61" aria-hidden="true" tabindex="-1"></a>MECP2_peak_cov_meth.df<span class="sc">$</span>ID <span class="ot">=</span> <span class="fu">paste0</span>(MECP2_peak_cov_meth.df<span class="sc">$</span>chr,<span class="st">'_'</span>,MECP2_peak_cov_meth.df<span class="sc">$</span>start,<span class="st">"_"</span>,MECP2_peak_cov_meth.df<span class="sc">$</span>end)</span>
<span id="cb14-62"><a href="#cb14-62" aria-hidden="true" tabindex="-1"></a><span class="co">#MECP2_peak_cov_meth.df = MECP2_peak_cov_meth.df %&gt;% column_to_rownames('ID')</span></span>
<span id="cb14-63"><a href="#cb14-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-64"><a href="#cb14-64" aria-hidden="true" tabindex="-1"></a><span class="co"># Merge MECP2 coverage and mCA level on MECP2 peaks</span></span>
<span id="cb14-65"><a href="#cb14-65" aria-hidden="true" tabindex="-1"></a>MECP2_peak_cov_meth.merged <span class="ot">=</span> <span class="fu">merge</span>(MECP2_sample_coverage,MECP2_peak_cov_meth.df,<span class="at">by=</span><span class="st">'ID'</span>)</span>
<span id="cb14-66"><a href="#cb14-66" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(MECP2_peak_cov_meth.merged)</span>
<span id="cb14-67"><a href="#cb14-67" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(MECP2_peak_cov_meth.merged)</span>
<span id="cb14-68"><a href="#cb14-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-69"><a href="#cb14-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-70"><a href="#cb14-70" aria-hidden="true" tabindex="-1"></a>MECP2_peak_cov_meth.merged.list <span class="ot">=</span> <span class="fu">list</span>(<span class="st">`</span><span class="at">7wk_MECP2</span><span class="st">`</span> <span class="ot">=</span> MECP2_peak_cov_meth.merged[,<span class="fu">c</span>(<span class="dv">5</span>,<span class="dv">14</span>)],</span>
<span id="cb14-71"><a href="#cb14-71" aria-hidden="true" tabindex="-1"></a>                                       <span class="st">`</span><span class="at">7wk_IgG</span><span class="st">`</span> <span class="ot">=</span> MECP2_peak_cov_meth.merged[,<span class="fu">c</span>(<span class="dv">6</span>,<span class="dv">14</span>)])</span>
<span id="cb14-72"><a href="#cb14-72" aria-hidden="true" tabindex="-1"></a>MECP2_peak_cov_meth.merged.list <span class="ot">=</span> <span class="fu">lapply</span>(MECP2_peak_cov_meth.merged.list, <span class="cf">function</span>(x)</span>
<span id="cb14-73"><a href="#cb14-73" aria-hidden="true" tabindex="-1"></a>{ x <span class="sc">%&gt;%</span> <span class="fu">mutate_if</span>(is.character, as.numeric)})</span>
<span id="cb14-74"><a href="#cb14-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-75"><a href="#cb14-75" aria-hidden="true" tabindex="-1"></a>MECP2_peak_cov_meth.merged.list <span class="ot">=</span> <span class="fu">lapply</span>(MECP2_peak_cov_meth.merged.list, <span class="cf">function</span>(x) x <span class="sc">%&gt;%</span> </span>
<span id="cb14-76"><a href="#cb14-76" aria-hidden="true" tabindex="-1"></a>                                           <span class="fu">arrange</span>(<span class="fu">desc</span>(mCA)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(.[,<span class="dv">1</span>] <span class="sc">&gt;</span><span class="dv">0</span>))</span>
<span id="cb14-77"><a href="#cb14-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-78"><a href="#cb14-78" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-79"><a href="#cb14-79" aria-hidden="true" tabindex="-1"></a>tmp <span class="ot">=</span> <span class="fu">lapply</span>(MECP2_peak_cov_meth.merged.list, <span class="cf">function</span>(x) </span>
<span id="cb14-80"><a href="#cb14-80" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize_logFC_mCA</span>(x <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>(),<span class="at">binSize =</span> <span class="dv">800</span>,<span class="at">stepSize =</span> <span class="dv">80</span>))</span>
<span id="cb14-81"><a href="#cb14-81" aria-hidden="true" tabindex="-1"></a>tmp1 <span class="ot">=</span> <span class="fu">lapply</span>(tmp, <span class="cf">function</span>(x) <span class="fu">do.call</span>(rbind,x))</span>
<span id="cb14-82"><a href="#cb14-82" aria-hidden="true" tabindex="-1"></a>tmp2 <span class="ot">=</span> plyr<span class="sc">::</span><span class="fu">ldply</span>(tmp1,data.frame)</span>
<span id="cb14-83"><a href="#cb14-83" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(tmp2)[<span class="dv">1</span>] <span class="ot">=</span> <span class="st">'MECP2'</span></span>
<span id="cb14-84"><a href="#cb14-84" aria-hidden="true" tabindex="-1"></a>MECP2_peak_cov_meth.merged.avg <span class="ot">=</span> tmp2[,<span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">1</span>)]</span>
<span id="cb14-85"><a href="#cb14-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-86"><a href="#cb14-86" aria-hidden="true" tabindex="-1"></a><span class="co"># calculate pearson correlation between mCA and MECP2 on geneBody</span></span>
<span id="cb14-87"><a href="#cb14-87" aria-hidden="true" tabindex="-1"></a>mCA_MECP2.corr.df <span class="ot">=</span> <span class="fu">rbindlist</span>(<span class="fu">lapply</span>(</span>
<span id="cb14-88"><a href="#cb14-88" aria-hidden="true" tabindex="-1"></a>  <span class="fu">lapply</span>(MECP2_peak_cov_meth.merged.list, <span class="cf">function</span>(x) <span class="fu">cor.test</span>(x[,<span class="dv">1</span>],x[,<span class="dv">2</span>])), tidy),</span>
<span id="cb14-89"><a href="#cb14-89" aria-hidden="true" tabindex="-1"></a>  <span class="at">use.names =</span> <span class="cn">TRUE</span>, <span class="at">idcol =</span> <span class="st">'Sample'</span>)</span>
<span id="cb14-90"><a href="#cb14-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-91"><a href="#cb14-91" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-92"><a href="#cb14-92" aria-hidden="true" tabindex="-1"></a>cor.plot <span class="ot">=</span> MECP2_peak_cov_meth.merged.avg  <span class="sc">%&gt;%</span> <span class="co">#filter(cov.mean &lt; 2) %&gt;% #head()</span></span>
<span id="cb14-93"><a href="#cb14-93" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>mCA.mean,<span class="at">y=</span>cov.mean,<span class="at">color=</span>MECP2))<span class="sc">+</span></span>
<span id="cb14-94"><a href="#cb14-94" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size=</span><span class="fl">0.5</span>)<span class="sc">+</span></span>
<span id="cb14-95"><a href="#cb14-95" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>()<span class="sc">+</span></span>
<span id="cb14-96"><a href="#cb14-96" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_ribbon</span>(<span class="fu">aes</span>(<span class="at">x=</span>mCA.mean,<span class="at">y=</span>cov.mean,</span>
<span id="cb14-97"><a href="#cb14-97" aria-hidden="true" tabindex="-1"></a>                 <span class="at">ymin=</span>cov.mean<span class="sc">-</span>cov.mean.sd,</span>
<span id="cb14-98"><a href="#cb14-98" aria-hidden="true" tabindex="-1"></a>                <span class="at">ymax=</span>cov.mean<span class="sc">+</span>cov.mean.sd,<span class="at">fill=</span>MECP2),</span>
<span id="cb14-99"><a href="#cb14-99" aria-hidden="true" tabindex="-1"></a>           <span class="at">alpha=</span><span class="fl">0.2</span>,<span class="at">color=</span><span class="st">'NA'</span>) <span class="sc">+</span></span>
<span id="cb14-100"><a href="#cb14-100" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">name =</span> <span class="st">"MECP2"</span>,<span class="at">values =</span> <span class="fu">c</span>(<span class="st">'black'</span>,<span class="st">'red'</span>),<span class="at">aesthetics =</span> <span class="fu">c</span>(<span class="st">"colour"</span>, <span class="st">"fill"</span>))<span class="sc">+</span></span>
<span id="cb14-101"><a href="#cb14-101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-102"><a href="#cb14-102" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_test</span>(<span class="dv">20</span>)<span class="sc">+</span></span>
<span id="cb14-103"><a href="#cb14-103" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">'mCA/CA'</span>)<span class="sc">+</span></span>
<span id="cb14-104"><a href="#cb14-104" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">'Gene Body MECP2 coverage'</span>)</span>
<span id="cb14-105"><a href="#cb14-105" aria-hidden="true" tabindex="-1"></a>cor.plot</span>
<span id="cb14-106"><a href="#cb14-106" aria-hidden="true" tabindex="-1"></a>cor.plot<span class="sc">+</span><span class="fu">annotate</span>(<span class="at">geom=</span><span class="st">"text"</span>, <span class="at">x=</span><span class="fl">0.015</span>, <span class="at">y=</span><span class="dv">65</span>, </span>
<span id="cb14-107"><a href="#cb14-107" aria-hidden="true" tabindex="-1"></a>                  <span class="at">label=</span><span class="fu">paste0</span>(<span class="st">'R = '</span>,<span class="fu">round</span>(mCA_MECP2.corr.df<span class="sc">$</span>estimate[<span class="dv">1</span>],<span class="dv">2</span>),</span>
<span id="cb14-108"><a href="#cb14-108" aria-hidden="true" tabindex="-1"></a>                               <span class="fu">ifelse</span>(mCA_MECP2.corr.df<span class="sc">$</span>p.value[<span class="dv">1</span>] <span class="sc">==</span> <span class="dv">0</span>,</span>
<span id="cb14-109"><a href="#cb14-109" aria-hidden="true" tabindex="-1"></a>                               <span class="st">', p &lt; 2.2e-16'</span>,mCA_MECP2.corr.df<span class="sc">$</span>p.value[<span class="dv">1</span>])),</span>
<span id="cb14-110"><a href="#cb14-110" aria-hidden="true" tabindex="-1"></a>                  <span class="at">color=</span><span class="st">"red"</span>)<span class="sc">+</span></span>
<span id="cb14-111"><a href="#cb14-111" aria-hidden="true" tabindex="-1"></a>         <span class="fu">annotate</span>(<span class="at">geom=</span><span class="st">"text"</span>, <span class="at">x=</span><span class="fl">0.015</span>, <span class="at">y=</span><span class="dv">70</span>, </span>
<span id="cb14-112"><a href="#cb14-112" aria-hidden="true" tabindex="-1"></a>                  <span class="at">label=</span><span class="fu">paste0</span>(<span class="st">'R = '</span>,<span class="fu">round</span>(mCA_MECP2.corr.df<span class="sc">$</span>estimate[<span class="dv">2</span>],<span class="dv">2</span>),</span>
<span id="cb14-113"><a href="#cb14-113" aria-hidden="true" tabindex="-1"></a>                               <span class="fu">ifelse</span>(mCA_MECP2.corr.df<span class="sc">$</span>p.value[<span class="dv">2</span>] <span class="sc">==</span> <span class="dv">0</span>,</span>
<span id="cb14-114"><a href="#cb14-114" aria-hidden="true" tabindex="-1"></a>                                <span class="st">', p &lt; 2.2e-16'</span>,<span class="fu">paste0</span>(<span class="st">', p = '</span>,<span class="fu">round</span>(mCA_MECP2.corr.df<span class="sc">$</span>p.value[<span class="dv">2</span>],<span class="dv">2</span>)))),</span>
<span id="cb14-115"><a href="#cb14-115" aria-hidden="true" tabindex="-1"></a>           <span class="at">color=</span><span class="st">"black"</span>)</span>
<span id="cb14-116"><a href="#cb14-116" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">file=</span><span class="st">'Plots/MECP2_mCA_corr_geneBody.pdf'</span>,<span class="at">height =</span> <span class="dv">12</span>,<span class="at">width =</span> <span class="dv">16</span>,<span class="at">units =</span> <span class="st">'cm'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></li>
<li><p>Figure 1D.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co"># compare MECP2 CUT&amp;RUN in MBHs, high mCA aand H3K9me3.</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="at">file =</span> <span class="st">"results/BSMap/8wk_CTX_Stroud2017_mm10_BSmap_CA.RData"</span>,<span class="at">verbose =</span> T)</span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>tiles <span class="ot">=</span> <span class="fu">tileGenome</span>(<span class="fu">seqinfo</span>(BSgenome.Mmusculus.UCSC.mm10), <span class="at">tilewidth=</span><span class="dv">5000</span>,</span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">cut.last.tile.in.chrom=</span><span class="cn">TRUE</span>)</span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>CTX_CA_meth_tiles <span class="ot">=</span><span class="fu">regionCounts</span>(CTX_CA,tiles)</span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a>CTX_CA_meth_tiles.df <span class="ot">=</span> <span class="fu">getData</span>(CTX_CA_meth_tiles)</span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a>CTX_CA_meth_tiles.df<span class="sc">$</span>meth <span class="ot">=</span> CTX_CA_meth_tiles.df<span class="sc">$</span>numCs<span class="sc">/</span>CTX_CA_meth_tiles.df<span class="sc">$</span>coverage</span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a>CTX_CA_meth_tiles.df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(coverage <span class="sc">&gt;</span> <span class="dv">100</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(meth)) <span class="sc">%&gt;%</span> </span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">top_n</span>(<span class="dv">1000</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb15-14"><a href="#cb15-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write.table</span>(<span class="st">"results/BSMap/8wk_CTX_Stroud2017_mm10_BSmap_top1000_CA_5kbbin.bed"</span>,<span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>,</span>
<span id="cb15-15"><a href="#cb15-15" aria-hidden="true" tabindex="-1"></a>              <span class="at">col.names =</span> F,<span class="at">row.names =</span> F,<span class="at">quote =</span> <span class="cn">FALSE</span>)</span>
<span id="cb15-16"><a href="#cb15-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-17"><a href="#cb15-17" aria-hidden="true" tabindex="-1"></a>mecp2_cov <span class="ot">=</span> <span class="fu">c</span>(<span class="fu">list.files</span>(<span class="st">"results/MECP2"</span>,<span class="at">pattern =</span> <span class="st">".*_MECP2WT.*_top1000_CA.bed"</span>,</span>
<span id="cb15-18"><a href="#cb15-18" aria-hidden="true" tabindex="-1"></a>                        <span class="at">full.names =</span> <span class="cn">TRUE</span>),</span>
<span id="cb15-19"><a href="#cb15-19" aria-hidden="true" tabindex="-1"></a>             <span class="fu">list.files</span>(<span class="st">"results/MECP2"</span>,<span class="at">pattern =</span> <span class="st">".*_MECP2WT.*_H3K9me3_noBlacklist.bed"</span>,</span>
<span id="cb15-20"><a href="#cb15-20" aria-hidden="true" tabindex="-1"></a>                        <span class="at">full.names =</span> <span class="cn">TRUE</span>),</span>
<span id="cb15-21"><a href="#cb15-21" aria-hidden="true" tabindex="-1"></a>             <span class="fu">list.files</span>(<span class="st">"results/MECP2"</span>,<span class="at">pattern =</span> <span class="st">".*_MECP2WT.*peak_Coverage.bed"</span>,</span>
<span id="cb15-22"><a href="#cb15-22" aria-hidden="true" tabindex="-1"></a>                        <span class="at">full.names =</span> <span class="cn">TRUE</span>))</span>
<span id="cb15-23"><a href="#cb15-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-24"><a href="#cb15-24" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(mecp2_cov) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"IgG_WT_Rep1_high.mCA"</span>, <span class="st">"MECP2_WT_Rep1_high.mCA"</span>, <span class="st">"IgG_WT_Rep2_high.mCA"</span>, <span class="st">"MECP2_WT_Rep2_high.mCA"</span>,</span>
<span id="cb15-25"><a href="#cb15-25" aria-hidden="true" tabindex="-1"></a>                     <span class="st">"IgG_WT_Rep1_H3K9me3"</span>, <span class="st">"MECP2_WT_Rep1_H3K9me3"</span>, <span class="st">"IgG_WT_Rep2_H3K9me3"</span>, <span class="st">"MECP2_WT_Rep2_H3K9me3"</span>, </span>
<span id="cb15-26"><a href="#cb15-26" aria-hidden="true" tabindex="-1"></a>                     <span class="st">"IgG_WT_Rep1_MBHs"</span>, <span class="st">"MECP2_WT_Rep1_MBHs"</span>, <span class="st">"IgG_WT_Rep2_MBHs"</span>, <span class="st">"MECP2_WT_Rep2_MBHs"</span>)</span>
<span id="cb15-27"><a href="#cb15-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-28"><a href="#cb15-28" aria-hidden="true" tabindex="-1"></a>mecp2_cov <span class="ot">=</span> <span class="fu">lapply</span>(mecp2_cov, <span class="cf">function</span>(x)</span>
<span id="cb15-29"><a href="#cb15-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">read.csv</span>(x,<span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>,<span class="at">header =</span> F) <span class="sc">%&gt;%</span> </span>
<span id="cb15-30"><a href="#cb15-30" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">V4 =</span> <span class="fu">as.numeric</span>(V4), <span class="at">len=</span>(V3<span class="sc">-</span>V2)<span class="sc">/</span><span class="dv">1000</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb15-31"><a href="#cb15-31" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(V4)) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">norm_cov =</span> V4<span class="sc">/</span>len))</span>
<span id="cb15-32"><a href="#cb15-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-33"><a href="#cb15-33" aria-hidden="true" tabindex="-1"></a><span class="fu">rbindlist</span>(mecp2_cov,<span class="at">idcol =</span> <span class="st">'Id'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb15-34"><a href="#cb15-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Group =</span> <span class="fu">gsub</span>(<span class="st">'MECP2_WT_Rep._|IgG_WT_Rep._'</span>,<span class="st">''</span>,Id)) <span class="sc">%&gt;%</span> </span>
<span id="cb15-35"><a href="#cb15-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sample =</span> <span class="fu">gsub</span>(<span class="st">'_WT_Rep[0-9]_H3K9me3|_WT_Rep[0-9]_high.mCA|_WT_Rep[0-9]_MBHs'</span>,<span class="st">''</span>,Id)) <span class="sc">%&gt;%</span> </span>
<span id="cb15-36"><a href="#cb15-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Id =</span> <span class="fu">paste0</span>(V1,<span class="st">'_'</span>,V2,<span class="st">'_'</span>,V3)) <span class="sc">%&gt;%</span> </span>
<span id="cb15-37"><a href="#cb15-37" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">1</span>,<span class="dv">8</span>,<span class="dv">9</span>,<span class="dv">7</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb15-38"><a href="#cb15-38" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(Id,Group,Sample) <span class="sc">%&gt;%</span> </span>
<span id="cb15-39"><a href="#cb15-39" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">avg_norm_covv =</span> <span class="fu">mean</span>(norm_cov, <span class="at">na.rm=</span><span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb15-40"><a href="#cb15-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>Group,<span class="at">y=</span><span class="fu">log2</span>(avg_norm_covv<span class="sc">+</span><span class="dv">1</span>),<span class="at">color=</span>Sample))<span class="sc">+</span></span>
<span id="cb15-41"><a href="#cb15-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb15-42"><a href="#cb15-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_test</span>(<span class="dv">20</span>) <span class="sc">+</span></span>
<span id="cb15-43"><a href="#cb15-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="st">"CUT &amp; RUN"</span>,<span class="at">values =</span> <span class="fu">c</span>(<span class="st">'grey'</span>,<span class="st">'red'</span>))<span class="sc">+</span></span>
<span id="cb15-44"><a href="#cb15-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>(<span class="at">comparisons =</span> <span class="fu">list</span>(</span>
<span id="cb15-45"><a href="#cb15-45" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">c</span>(<span class="st">"H3K9me3"</span>,<span class="st">'MBHs'</span>),</span>
<span id="cb15-46"><a href="#cb15-46" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">c</span>(<span class="st">'high.mCA'</span>,<span class="st">'MBHs'</span>),</span>
<span id="cb15-47"><a href="#cb15-47" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">c</span>(<span class="st">'high.mCA'</span>,<span class="st">'H3K9me3'</span>))) <span class="sc">+</span></span>
<span id="cb15-48"><a href="#cb15-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"log2 (MECP2 Coverage +1)"</span>)</span>
<span id="cb15-49"><a href="#cb15-49" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">'Plots/MECP2_on_H3K9me3_highmCA_MBHs.pdf'</span>,<span class="at">height =</span> <span class="dv">6</span>,<span class="at">width =</span> <span class="dv">8</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></li>
<li><p>Figure 1F</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co"># import MECP2 peaks</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>regions <span class="ot">=</span> <span class="fu">import</span>(<span class="st">"results/MECP2/MECP2WT.vs.KO_overlap_MECP2WT.vs.IgG.noBlacklist_filtered.bed"</span>)</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a><span class="co"># generate random genomic regions</span></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>randomRegions <span class="ot">=</span> <span class="fu">createRandomRegions</span>(<span class="at">nregions=</span><span class="dv">23193</span>,</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">length.mean=</span><span class="dv">500</span>, <span class="at">length.sd=</span><span class="dv">20</span>, <span class="at">genome=</span><span class="st">"mm10"</span>, </span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">mask=</span><span class="cn">NULL</span>, <span class="at">non.overlapping=</span><span class="cn">TRUE</span>)</span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a>regions_GC <span class="ot">=</span> <span class="fu">gcContentCalc</span>(regions , <span class="at">organism=</span>Mmusculus, <span class="at">verbose=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">MECP2 =</span> <span class="dv">1</span>) </span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a>randomRegions_GC <span class="ot">=</span> <span class="fu">gcContentCalc</span>(randomRegions , <span class="at">organism=</span>Mmusculus, <span class="at">verbose=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb16-14"><a href="#cb16-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb16-15"><a href="#cb16-15" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">Random =</span> <span class="dv">1</span>) </span>
<span id="cb16-16"><a href="#cb16-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-17"><a href="#cb16-17" aria-hidden="true" tabindex="-1"></a>p <span class="ot">=</span> <span class="fu">cbind</span>(randomRegions_GC,regions_GC) <span class="sc">%&gt;%</span> </span>
<span id="cb16-18"><a href="#cb16-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">melt</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb16-19"><a href="#cb16-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggboxplot</span>(<span class="at">x =</span> <span class="st">"variable"</span>, <span class="at">y =</span> <span class="st">"value"</span>,</span>
<span id="cb16-20"><a href="#cb16-20" aria-hidden="true" tabindex="-1"></a>                 <span class="at">color =</span> <span class="st">"variable"</span>, <span class="at">palette =</span> <span class="fu">c</span>(<span class="st">"gray"</span>,<span class="st">"blue"</span>))<span class="sc">+</span></span>
<span id="cb16-21"><a href="#cb16-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylab</span>(<span class="st">"GC%"</span>)</span>
<span id="cb16-22"><a href="#cb16-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-23"><a href="#cb16-23" aria-hidden="true" tabindex="-1"></a><span class="co">#  Add p-value</span></span>
<span id="cb16-24"><a href="#cb16-24" aria-hidden="true" tabindex="-1"></a>p <span class="sc">+</span> <span class="fu">stat_compare_means</span>()</span>
<span id="cb16-25"><a href="#cb16-25" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">'Plots/GC_percent_MECP2_peaks.pdf'</span>,<span class="at">height =</span> <span class="dv">4</span>,<span class="at">width =</span> <span class="dv">3</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></li>
<li><p>Figure 1G.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co"># GGGTGT, GGGTTT, TTTGGG frequency. on MBH sites</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>x <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">'results/CAsequence/GT_motif_freq_on_MECP2.txt'</span>,<span class="at">sep =</span> <span class="st">'</span><span class="sc">\t</span><span class="st">'</span>,<span class="at">header =</span> T) <span class="sc">%&gt;%</span> </span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">1</span>,<span class="dv">5</span>,<span class="dv">2</span>,<span class="dv">14</span>)</span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(x)</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(x) <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">'X[0-9]</span><span class="sc">\\</span><span class="st">.|</span><span class="sc">\\</span><span class="st">..*'</span>,<span class="st">''</span>,<span class="fu">colnames</span>(x))  </span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>x <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">abs</span>(.[,<span class="dv">1</span>]) <span class="sc">&lt;</span> <span class="dv">1990</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>  <span class="co">#mutate(GGTGT = GGTGT-0.09) %&gt;% </span></span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a>  <span class="co">#mutate(AKTCATTT = AKTCATTT-0.01) %&gt;% </span></span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">melt</span>(<span class="at">id.vars =</span> <span class="st">'Distance'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>Distance,<span class="at">y=</span>value,<span class="at">color=</span>variable))<span class="sc">+</span> </span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>()<span class="sc">+</span></span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">'red'</span>,<span class="st">'black'</span>,<span class="st">'grey'</span>))<span class="sc">+</span></span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_test</span>(<span class="dv">20</span>)<span class="sc">+</span></span>
<span id="cb17-15"><a href="#cb17-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">color=</span><span class="cn">NULL</span>)<span class="sc">+</span></span>
<span id="cb17-16"><a href="#cb17-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">'Motifs per base per peak'</span>)</span>
<span id="cb17-17"><a href="#cb17-17" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">'Plots/GT_motif_on_MECP2_peaks.pdf'</span>,<span class="at">height =</span> <span class="dv">6</span>,<span class="at">width =</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></li>
<li><p>Supplementary Figure 1A.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="co"># This code has been taken from https://davidzeleny.net/wiki/doku.php/recol:monte-carlo-pearson</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to test correlation using permutation</span></span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>test.cor.perm <span class="ot">=</span> <span class="cf">function</span> (x, y, <span class="at">nperm =</span> <span class="dv">499</span>)</span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a>  r.obs <span class="ot">=</span> <span class="fu">cor.test</span>(<span class="at">x =</span> x, <span class="at">y =</span> y,<span class="at">method =</span> <span class="st">"spearman"</span>,<span class="at">exact =</span> <span class="cn">FALSE</span>)<span class="sc">$</span>estimate</span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>  P.par <span class="ot">=</span> <span class="fu">cor.test</span> (<span class="at">x =</span> x, <span class="at">y =</span> y,<span class="at">method =</span> <span class="st">"spearman"</span>,<span class="at">exact =</span> <span class="cn">FALSE</span>)<span class="sc">$</span>p.value</span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a>  <span class="co">#  r.per = replicate (nperm, expr = cor (x = x, y = sample (y)))</span></span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a>  r.per <span class="ot">=</span> <span class="fu">sapply</span> (<span class="dv">1</span><span class="sc">:</span>nperm, <span class="at">FUN =</span> <span class="cf">function</span> (i) <span class="fu">cor.test</span>(<span class="at">x =</span> x, <span class="at">y =</span> <span class="fu">sample</span> (y),<span class="at">method =</span> <span class="st">"spearman"</span>,<span class="at">exact =</span> <span class="cn">FALSE</span>)<span class="sc">$</span>estimate)</span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a>  r.per <span class="ot">=</span> <span class="fu">c</span>(r.per, r.obs)</span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">hist</span> (r.per, <span class="at">xlim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">1</span>))</span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">abline</span> (<span class="at">v =</span> r.obs, <span class="at">col =</span> <span class="st">'red'</span>)</span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a>  P.per <span class="ot">=</span> <span class="fu">sum</span> (<span class="fu">abs</span> (r.per) <span class="sc">&gt;=</span> <span class="fu">abs</span> (r.obs))<span class="sc">/</span>(nperm <span class="sc">+</span> <span class="dv">1</span>) </span>
<span id="cb18-14"><a href="#cb18-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span> (<span class="fu">list</span> (<span class="at">r.obs =</span> r.obs, <span class="at">P.par =</span> P.par, <span class="at">P.per =</span> P.per))</span>
<span id="cb18-15"><a href="#cb18-15" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb18-16"><a href="#cb18-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-17"><a href="#cb18-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-18"><a href="#cb18-18" aria-hidden="true" tabindex="-1"></a><span class="co"># read the multibigwigout file has been generated using deeptools for MECP2 CUT&amp;RUN and MECP2 ChIPSeq</span></span>
<span id="cb18-19"><a href="#cb18-19" aria-hidden="true" tabindex="-1"></a>mecp2_cnr_chip_corr <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">"results/MECP2_CNR_ChIPseq_comparison/MECP2_CNR_ChIPseq_multiBigWigOut.tab"</span>,<span class="at">header =</span> T,<span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>,<span class="at">check.names =</span> <span class="cn">FALSE</span>) <span class="co">#%&gt;% head()</span></span>
<span id="cb18-20"><a href="#cb18-20" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(mecp2_cnr_chip_corr) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"chr"</span>,<span class="st">'start'</span>,<span class="st">'end'</span>,</span>
<span id="cb18-21"><a href="#cb18-21" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">'MECP2_CNR_R1'</span>,<span class="st">'MECP2_CNR_R2'</span>,</span>
<span id="cb18-22"><a href="#cb18-22" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">'MECP2_ChIPseq_Ab1'</span>,<span class="st">'MECP2_ChIPseq_Ab2'</span>)</span>
<span id="cb18-23"><a href="#cb18-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-24"><a href="#cb18-24" aria-hidden="true" tabindex="-1"></a><span class="co"># read the multibigwigout file has been generated using deeptools for MECP2 CUT&amp;RUN and MECP2 CUT&amp;TAG</span></span>
<span id="cb18-25"><a href="#cb18-25" aria-hidden="true" tabindex="-1"></a>mecp2_cnr_cnt_corr <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">"results/MECP2_CNR_ChIPseq_comparison/MECP2_CNR_CNT_multiBigWigOut.tab"</span>,<span class="at">header =</span> T,<span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>,<span class="at">check.names =</span> <span class="cn">FALSE</span>) <span class="co">#%&gt;% head()</span></span>
<span id="cb18-26"><a href="#cb18-26" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(mecp2_cnr_cnt_corr) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"chr"</span>,<span class="st">'start'</span>,<span class="st">'end'</span>,</span>
<span id="cb18-27"><a href="#cb18-27" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">'MECP2_CNR_R1'</span>,<span class="st">'MECP2_CNR_R2'</span>,</span>
<span id="cb18-28"><a href="#cb18-28" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">'MECP2_CNT_Ab1'</span>,<span class="st">'MECP2_CNT_Ab2'</span>)</span>
<span id="cb18-29"><a href="#cb18-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-30"><a href="#cb18-30" aria-hidden="true" tabindex="-1"></a>cor.perm.list <span class="ot">=</span> <span class="fu">list</span>()</span>
<span id="cb18-31"><a href="#cb18-31" aria-hidden="true" tabindex="-1"></a>cor.perm.list[[<span class="dv">1</span>]] <span class="ot">=</span> <span class="fu">test.cor.perm</span>(mecp2_cnr_chip_corr<span class="sc">$</span>MECP2_CNR_R1,</span>
<span id="cb18-32"><a href="#cb18-32" aria-hidden="true" tabindex="-1"></a>                                   mecp2_cnr_chip_corr<span class="sc">$</span>MECP2_CNR_R2,<span class="at">nperm =</span> <span class="dv">999</span>)</span>
<span id="cb18-33"><a href="#cb18-33" aria-hidden="true" tabindex="-1"></a>cor.perm.list[[<span class="dv">2</span>]] <span class="ot">=</span> <span class="fu">test.cor.perm</span>(mecp2_cnr_chip_corr<span class="sc">$</span>MECP2_CNR_R1,</span>
<span id="cb18-34"><a href="#cb18-34" aria-hidden="true" tabindex="-1"></a>                                   mecp2_cnr_chip_corr<span class="sc">$</span>MECP2_ChIPseq_Ab1,<span class="at">nperm =</span> <span class="dv">999</span>)</span>
<span id="cb18-35"><a href="#cb18-35" aria-hidden="true" tabindex="-1"></a>cor.perm.list[[<span class="dv">3</span>]] <span class="ot">=</span> <span class="fu">test.cor.perm</span>(mecp2_cnr_chip_corr<span class="sc">$</span>MECP2_CNR_R1,</span>
<span id="cb18-36"><a href="#cb18-36" aria-hidden="true" tabindex="-1"></a>                                   mecp2_cnr_chip_corr<span class="sc">$</span>MECP2_ChIPseq_Ab2,<span class="at">nperm =</span> <span class="dv">999</span>)</span>
<span id="cb18-37"><a href="#cb18-37" aria-hidden="true" tabindex="-1"></a>cor.perm.list[[<span class="dv">4</span>]] <span class="ot">=</span> <span class="fu">test.cor.perm</span>(mecp2_cnr_chip_corr<span class="sc">$</span>MECP2_CNR_R2,</span>
<span id="cb18-38"><a href="#cb18-38" aria-hidden="true" tabindex="-1"></a>                                   mecp2_cnr_chip_corr<span class="sc">$</span>MECP2_ChIPseq_Ab1,<span class="at">nperm =</span> <span class="dv">999</span>)</span>
<span id="cb18-39"><a href="#cb18-39" aria-hidden="true" tabindex="-1"></a>cor.perm.list[[<span class="dv">5</span>]] <span class="ot">=</span> <span class="fu">test.cor.perm</span>(mecp2_cnr_chip_corr<span class="sc">$</span>MECP2_CNR_R2,</span>
<span id="cb18-40"><a href="#cb18-40" aria-hidden="true" tabindex="-1"></a>                                   mecp2_cnr_chip_corr<span class="sc">$</span>MECP2_ChIPseq_Ab2,<span class="at">nperm =</span> <span class="dv">999</span>)</span>
<span id="cb18-41"><a href="#cb18-41" aria-hidden="true" tabindex="-1"></a>cor.perm.list[[<span class="dv">6</span>]] <span class="ot">=</span> <span class="fu">test.cor.perm</span>(mecp2_cnr_chip_corr<span class="sc">$</span>MECP2_ChIPseq_Ab1,</span>
<span id="cb18-42"><a href="#cb18-42" aria-hidden="true" tabindex="-1"></a>                                   mecp2_cnr_chip_corr<span class="sc">$</span>MECP2_ChIPseq_Ab2,<span class="at">nperm =</span> <span class="dv">999</span>)</span>
<span id="cb18-43"><a href="#cb18-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-44"><a href="#cb18-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-45"><a href="#cb18-45" aria-hidden="true" tabindex="-1"></a>cor.perm.list[[<span class="dv">7</span>]]  <span class="ot">=</span> <span class="fu">test.cor.perm</span>(mecp2_cnr_cnt_corr<span class="sc">$</span>MECP2_CNR_R1,</span>
<span id="cb18-46"><a href="#cb18-46" aria-hidden="true" tabindex="-1"></a>                                    mecp2_cnr_cnt_corr<span class="sc">$</span>MECP2_CNR_R2,<span class="at">nperm =</span> <span class="dv">999</span>)</span>
<span id="cb18-47"><a href="#cb18-47" aria-hidden="true" tabindex="-1"></a>cor.perm.list[[<span class="dv">8</span>]]  <span class="ot">=</span> <span class="fu">test.cor.perm</span>(mecp2_cnr_cnt_corr<span class="sc">$</span>MECP2_CNR_R1,</span>
<span id="cb18-48"><a href="#cb18-48" aria-hidden="true" tabindex="-1"></a>                                    mecp2_cnr_cnt_corr<span class="sc">$</span>MECP2_CNT_Ab1,<span class="at">nperm =</span> <span class="dv">999</span>)</span>
<span id="cb18-49"><a href="#cb18-49" aria-hidden="true" tabindex="-1"></a>cor.perm.list[[<span class="dv">9</span>]]  <span class="ot">=</span> <span class="fu">test.cor.perm</span>(mecp2_cnr_cnt_corr<span class="sc">$</span>MECP2_CNR_R1,</span>
<span id="cb18-50"><a href="#cb18-50" aria-hidden="true" tabindex="-1"></a>                                    mecp2_cnr_cnt_corr<span class="sc">$</span>MECP2_CNT_Ab2,<span class="at">nperm =</span> <span class="dv">999</span>)</span>
<span id="cb18-51"><a href="#cb18-51" aria-hidden="true" tabindex="-1"></a>cor.perm.list[[<span class="dv">10</span>]]  <span class="ot">=</span> <span class="fu">test.cor.perm</span>(mecp2_cnr_cnt_corr<span class="sc">$</span>MECP2_CNR_R2,</span>
<span id="cb18-52"><a href="#cb18-52" aria-hidden="true" tabindex="-1"></a>                                     mecp2_cnr_cnt_corr<span class="sc">$</span>MECP2_CNT_Ab1,<span class="at">nperm =</span> <span class="dv">999</span>)</span>
<span id="cb18-53"><a href="#cb18-53" aria-hidden="true" tabindex="-1"></a>cor.perm.list[[<span class="dv">11</span>]]  <span class="ot">=</span> <span class="fu">test.cor.perm</span>(mecp2_cnr_cnt_corr<span class="sc">$</span>MECP2_CNR_R2,</span>
<span id="cb18-54"><a href="#cb18-54" aria-hidden="true" tabindex="-1"></a>                                     mecp2_cnr_cnt_corr<span class="sc">$</span>MECP2_CNT_Ab2,<span class="at">nperm =</span> <span class="dv">999</span>)</span>
<span id="cb18-55"><a href="#cb18-55" aria-hidden="true" tabindex="-1"></a>cor.perm.list[[<span class="dv">12</span>]]  <span class="ot">=</span> <span class="fu">test.cor.perm</span>(mecp2_cnr_cnt_corr<span class="sc">$</span>MECP2_CNT_Ab1,</span>
<span id="cb18-56"><a href="#cb18-56" aria-hidden="true" tabindex="-1"></a>                                     mecp2_cnr_cnt_corr<span class="sc">$</span>MECP2_CNT_Ab2,<span class="at">nperm =</span> <span class="dv">999</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></li>
<li><p>Supplementary Figure 1G.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to average column value across bins and step size</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>summarize_logFC_mCA <span class="ot">=</span> <span class="cf">function</span>(x,<span class="at">binSize =</span> <span class="dv">19</span>,<span class="at">stepSize =</span><span class="dv">4</span>){</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>  df.bin <span class="ot">=</span><span class="fu">list</span>()</span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">seq</span>(<span class="at">from=</span><span class="dv">1</span>, <span class="at">to=</span><span class="fu">nrow</span>(x), <span class="at">by=</span>stepSize)){</span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(i)</span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a>    start <span class="ot">=</span> i</span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a>    stop  <span class="ot">=</span> i<span class="sc">+</span>binSize</span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(stop <span class="sc">&lt;=</span> <span class="fu">nrow</span>(x)){</span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a>      df <span class="ot">=</span> x[start<span class="sc">:</span>stop,]</span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a>      df.bin[[i]] <span class="ot">=</span> df  <span class="sc">%&gt;%</span> <span class="fu">summarize</span>(<span class="at">cov.mean=</span><span class="fu">mean</span>(.[,<span class="dv">1</span>]),<span class="at">cov.mean.sd =</span> <span class="fu">sd</span>(.[,<span class="dv">1</span>]),<span class="at">mCA.mean=</span><span class="fu">mean</span>(.[,<span class="dv">2</span>]))</span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>{</span>
<span id="cb19-13"><a href="#cb19-13" aria-hidden="true" tabindex="-1"></a>      <span class="fu">return</span>(df.bin)</span>
<span id="cb19-14"><a href="#cb19-14" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb19-15"><a href="#cb19-15" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb19-16"><a href="#cb19-16" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb19-17"><a href="#cb19-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-18"><a href="#cb19-18" aria-hidden="true" tabindex="-1"></a>CNR_files <span class="ot">=</span> <span class="fu">list.files</span>(<span class="st">'results/CNR_Coverage/'</span>,<span class="at">pattern =</span> <span class="st">"G118E_vs_IgG_Rep*"</span>,<span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb19-19"><a href="#cb19-19" aria-hidden="true" tabindex="-1"></a>CNR_files.list <span class="ot">=</span> <span class="fu">lapply</span>(CNR_files, read.csv, <span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>,<span class="at">header=</span>F)</span>
<span id="cb19-20"><a href="#cb19-20" aria-hidden="true" tabindex="-1"></a>CNR_files.names <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"G118E_vs_WT_Rep1"</span>,<span class="st">"G118E_vs_WT_Rep2"</span>,<span class="st">"G118E_vs_WT_Rep3"</span>)</span>
<span id="cb19-21"><a href="#cb19-21" aria-hidden="true" tabindex="-1"></a>CNR_files.list <span class="ot">=</span> <span class="fu">lapply</span>(<span class="fu">seq_along</span>(CNR_files.list),<span class="cf">function</span>(x) {</span>
<span id="cb19-22"><a href="#cb19-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">setNames</span>(CNR_files.list[[x]],<span class="at">nm =</span> <span class="fu">c</span>(<span class="st">'chr'</span>,<span class="st">'start'</span>,<span class="st">'end'</span>,CNR_files.names[x])) })</span>
<span id="cb19-23"><a href="#cb19-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-24"><a href="#cb19-24" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(CNR_files.list)) {</span>
<span id="cb19-25"><a href="#cb19-25" aria-hidden="true" tabindex="-1"></a>  CNR_files.list[[i]]<span class="sc">$</span>Cov <span class="ot">=</span> <span class="fu">ifelse</span>(CNR_files.list[[i]][,<span class="dv">4</span>] <span class="sc">==</span> <span class="st">'.'</span>, <span class="dv">0</span> , CNR_files.list[[i]][,<span class="dv">4</span>])</span>
<span id="cb19-26"><a href="#cb19-26" aria-hidden="true" tabindex="-1"></a>  CNR_files.list[[i]]<span class="sc">$</span>Cov <span class="ot">=</span> <span class="fu">as.numeric</span>(CNR_files.list[[i]]<span class="sc">$</span>Cov)</span>
<span id="cb19-27"><a href="#cb19-27" aria-hidden="true" tabindex="-1"></a>  CNR_files.list[[i]]<span class="sc">$</span>geneLength <span class="ot">=</span> (CNR_files.list[[i]]<span class="sc">$</span>end <span class="sc">-</span> CNR_files.list[[i]]<span class="sc">$</span>start)<span class="sc">/</span><span class="dv">1000</span></span>
<span id="cb19-28"><a href="#cb19-28" aria-hidden="true" tabindex="-1"></a>  CNR_files.list[[i]]<span class="sc">$</span>norm <span class="ot">=</span> CNR_files.list[[i]]<span class="sc">$</span>Cov<span class="sc">/</span>CNR_files.list[[i]]<span class="sc">$</span>geneLength</span>
<span id="cb19-29"><a href="#cb19-29" aria-hidden="true" tabindex="-1"></a>  CNR_files.list[[i]] <span class="ot">=</span> CNR_files.list[[i]][,<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">7</span>)] <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(norm)) </span>
<span id="cb19-30"><a href="#cb19-30" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb19-31"><a href="#cb19-31" aria-hidden="true" tabindex="-1"></a>CNR_files.list <span class="ot">=</span> <span class="fu">lapply</span>(<span class="fu">seq_along</span>(CNR_files.list),<span class="cf">function</span>(x) {</span>
<span id="cb19-32"><a href="#cb19-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">setNames</span>(CNR_files.list[[x]],<span class="at">nm =</span> <span class="fu">c</span>(<span class="st">'chr'</span>,<span class="st">'start'</span>,<span class="st">'end'</span>,CNR_files.names[x])) })</span>
<span id="cb19-33"><a href="#cb19-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-34"><a href="#cb19-34" aria-hidden="true" tabindex="-1"></a>CNR_files.df <span class="ot">=</span> <span class="fu">Reduce</span>(full_join,CNR_files.list)</span>
<span id="cb19-35"><a href="#cb19-35" aria-hidden="true" tabindex="-1"></a>CNR_files.df<span class="sc">$</span>ID <span class="ot">=</span> <span class="fu">paste0</span>(CNR_files.df<span class="sc">$</span>chr,<span class="st">'_'</span>,CNR_files.df<span class="sc">$</span>start,<span class="st">'_'</span>,CNR_files.df<span class="sc">$</span>end)</span>
<span id="cb19-36"><a href="#cb19-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-37"><a href="#cb19-37" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert GeneBody coordinated to GRanges</span></span>
<span id="cb19-38"><a href="#cb19-38" aria-hidden="true" tabindex="-1"></a>geneBody.bed <span class="ot">=</span> CNR_files.df[,<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>)]</span>
<span id="cb19-39"><a href="#cb19-39" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(geneBody.bed) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">'chr'</span>,<span class="st">'start'</span>,<span class="st">'end'</span>)</span>
<span id="cb19-40"><a href="#cb19-40" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(geneBody.bed)</span>
<span id="cb19-41"><a href="#cb19-41" aria-hidden="true" tabindex="-1"></a>geneBody.bed.GR <span class="ot">=</span> <span class="fu">GRanges</span>(geneBody.bed)</span>
<span id="cb19-42"><a href="#cb19-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-43"><a href="#cb19-43" aria-hidden="true" tabindex="-1"></a>geneBody.bed.GR</span>
<span id="cb19-44"><a href="#cb19-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-45"><a href="#cb19-45" aria-hidden="true" tabindex="-1"></a><span class="co"># calculate mCA on MECP2 peaks</span></span>
<span id="cb19-46"><a href="#cb19-46" aria-hidden="true" tabindex="-1"></a><span class="co"># Load genome wide CA methylation BSMap data</span></span>
<span id="cb19-47"><a href="#cb19-47" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">'results/BSMap/8wk_CTX_Stroud2017_mm10_BSmap_CA.RData'</span>,<span class="at">verbose =</span> <span class="cn">TRUE</span>)</span>
<span id="cb19-48"><a href="#cb19-48" aria-hidden="true" tabindex="-1"></a>geneBody_cov_meth <span class="ot">=</span> <span class="fu">regionCounts</span>(CTX_CA,geneBody.bed.GR)</span>
<span id="cb19-49"><a href="#cb19-49" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(geneBody_cov_meth)</span>
<span id="cb19-50"><a href="#cb19-50" aria-hidden="true" tabindex="-1"></a><span class="co">#save(file = 'results/RData/MECP2_G118E_CNR_geneBody_Coverage.RData',geneBody_cov_meth)</span></span>
<span id="cb19-51"><a href="#cb19-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-52"><a href="#cb19-52" aria-hidden="true" tabindex="-1"></a>geneBody_cov_meth.df <span class="ot">=</span> <span class="fu">getData</span>(geneBody_cov_meth)</span>
<span id="cb19-53"><a href="#cb19-53" aria-hidden="true" tabindex="-1"></a>geneBody_cov_meth.df<span class="sc">$</span>mCA <span class="ot">=</span> geneBody_cov_meth.df<span class="sc">$</span>numCs<span class="sc">/</span>geneBody_cov_meth.df<span class="sc">$</span>coverage</span>
<span id="cb19-54"><a href="#cb19-54" aria-hidden="true" tabindex="-1"></a>geneBody_cov_meth.df<span class="sc">$</span>ID <span class="ot">=</span> <span class="fu">paste0</span>(geneBody_cov_meth.df<span class="sc">$</span>chr,<span class="st">'_'</span>,geneBody_cov_meth.df<span class="sc">$</span>start,<span class="st">"_"</span>,geneBody_cov_meth.df<span class="sc">$</span>end)</span>
<span id="cb19-55"><a href="#cb19-55" aria-hidden="true" tabindex="-1"></a><span class="co">#MECP2_peak_cov_meth.df = MECP2_peak_cov_meth.df %&gt;% column_to_rownames('ID')</span></span>
<span id="cb19-56"><a href="#cb19-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-57"><a href="#cb19-57" aria-hidden="true" tabindex="-1"></a><span class="co"># Merge MECP2 coverage and mCA level on MECP2 peaks</span></span>
<span id="cb19-58"><a href="#cb19-58" aria-hidden="true" tabindex="-1"></a>geneBody_cov_meth<span class="ot">=</span> <span class="fu">merge</span>(CNR_files.df,geneBody_cov_meth.df,<span class="at">by=</span><span class="st">'ID'</span>)</span>
<span id="cb19-59"><a href="#cb19-59" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(geneBody_cov_meth)</span>
<span id="cb19-60"><a href="#cb19-60" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(geneBody_cov_meth)</span>
<span id="cb19-61"><a href="#cb19-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-62"><a href="#cb19-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-63"><a href="#cb19-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-64"><a href="#cb19-64" aria-hidden="true" tabindex="-1"></a>geneBody_cov_meth.list <span class="ot">=</span> <span class="fu">list</span>(geneBody_cov_meth[,<span class="fu">c</span>(<span class="dv">5</span>,<span class="dv">15</span>)],</span>
<span id="cb19-65"><a href="#cb19-65" aria-hidden="true" tabindex="-1"></a>                              geneBody_cov_meth[,<span class="fu">c</span>(<span class="dv">6</span>,<span class="dv">15</span>)],</span>
<span id="cb19-66"><a href="#cb19-66" aria-hidden="true" tabindex="-1"></a>                              geneBody_cov_meth[,<span class="fu">c</span>(<span class="dv">7</span>,<span class="dv">15</span>)])</span>
<span id="cb19-67"><a href="#cb19-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-68"><a href="#cb19-68" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(geneBody_cov_meth.list) <span class="ot">=</span> CNR_files.names</span>
<span id="cb19-69"><a href="#cb19-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-70"><a href="#cb19-70" aria-hidden="true" tabindex="-1"></a>geneBody_cov_meth.list <span class="ot">=</span> <span class="fu">lapply</span>(geneBody_cov_meth.list, <span class="cf">function</span>(x) x <span class="sc">%&gt;%</span> <span class="fu">arrange</span>((mCA)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>( <span class="sc">!</span>(<span class="fu">is.na</span>(.[,<span class="dv">1</span>])))) </span>
<span id="cb19-71"><a href="#cb19-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-72"><a href="#cb19-72" aria-hidden="true" tabindex="-1"></a><span class="co">#geneBody_cov_meth.list[[1]]$G118E_vs_WT_Rep1 = log2(geneBody_cov_meth.list[[1]]$G118E_vs_WT_Rep1)</span></span>
<span id="cb19-73"><a href="#cb19-73" aria-hidden="true" tabindex="-1"></a>tmp <span class="ot">=</span> <span class="fu">lapply</span>(geneBody_cov_meth.list, <span class="cf">function</span>(x) <span class="fu">summarize_logFC_mCA</span>(x,<span class="at">binSize =</span> <span class="dv">800</span>,<span class="at">stepSize =</span> <span class="dv">80</span>))</span>
<span id="cb19-74"><a href="#cb19-74" aria-hidden="true" tabindex="-1"></a>tmp1 <span class="ot">=</span> <span class="fu">lapply</span>(tmp, <span class="cf">function</span>(x) <span class="fu">do.call</span>(rbind,x))</span>
<span id="cb19-75"><a href="#cb19-75" aria-hidden="true" tabindex="-1"></a>tmp2 <span class="ot">=</span> plyr<span class="sc">::</span><span class="fu">ldply</span>(tmp1,data.frame)</span>
<span id="cb19-76"><a href="#cb19-76" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(tmp2)[<span class="dv">1</span>] <span class="ot">=</span> <span class="st">'Samples'</span></span>
<span id="cb19-77"><a href="#cb19-77" aria-hidden="true" tabindex="-1"></a>geneBody_cov_meth.merged.avg <span class="ot">=</span> tmp2[,<span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">1</span>)]</span>
<span id="cb19-78"><a href="#cb19-78" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-79"><a href="#cb19-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-80"><a href="#cb19-80" aria-hidden="true" tabindex="-1"></a><span class="co">#geneBody_cov_meth.merged.avg$Samples = CNR_files.names</span></span>
<span id="cb19-81"><a href="#cb19-81" aria-hidden="true" tabindex="-1"></a>geneBody_cov_meth.merged.avg<span class="sc">$</span>Rep <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">"MECP2.*_"</span>,<span class="st">""</span>,geneBody_cov_meth.merged.avg<span class="sc">$</span>Samples)</span>
<span id="cb19-82"><a href="#cb19-82" aria-hidden="true" tabindex="-1"></a>geneBody_cov_meth.merged.avg<span class="sc">$</span>Samples <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">"_Rep."</span>,<span class="st">""</span>,geneBody_cov_meth.merged.avg<span class="sc">$</span>Samples)</span>
<span id="cb19-83"><a href="#cb19-83" aria-hidden="true" tabindex="-1"></a>geneBody_cov_meth.merged.avg<span class="sc">$</span>Samples <span class="ot">=</span> <span class="fu">factor</span>(geneBody_cov_meth.merged.avg<span class="sc">$</span>Samples,</span>
<span id="cb19-84"><a href="#cb19-84" aria-hidden="true" tabindex="-1"></a>                                              <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"MECP2_WT"</span>,<span class="st">"MECP2_G118E"</span>))</span>
<span id="cb19-85"><a href="#cb19-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-86"><a href="#cb19-86" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-87"><a href="#cb19-87" aria-hidden="true" tabindex="-1"></a>geneBody_cov_meth.merged.avg  <span class="sc">%&gt;%</span> <span class="co">#filter(!(grepl('pre',Samples))) %&gt;% #head()</span></span>
<span id="cb19-88"><a href="#cb19-88" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>mCA.mean,<span class="at">y=</span>cov.mean,))<span class="sc">+</span></span>
<span id="cb19-89"><a href="#cb19-89" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size=</span><span class="fl">0.5</span>)<span class="sc">+</span></span>
<span id="cb19-90"><a href="#cb19-90" aria-hidden="true" tabindex="-1"></a>  <span class="co">#scale_color_manual(values = c('red','red','red','grey','grey','grey'))+</span></span>
<span id="cb19-91"><a href="#cb19-91" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_line()+</span></span>
<span id="cb19-92"><a href="#cb19-92" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_cor</span>( <span class="at">method =</span> <span class="st">"pearson"</span>)<span class="sc">+</span></span>
<span id="cb19-93"><a href="#cb19-93" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_test</span>(<span class="dv">20</span>)<span class="sc">+</span></span>
<span id="cb19-94"><a href="#cb19-94" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">'mCA/CA'</span>)<span class="sc">+</span></span>
<span id="cb19-95"><a href="#cb19-95" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"log2 (G118E/IgG) - log2 (WT/IgG)"</span>)<span class="sc">+</span></span>
<span id="cb19-96"><a href="#cb19-96" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">color =</span> <span class="fu">guide_legend</span>(<span class="at">override.aes =</span> <span class="fu">list</span>(<span class="at">size =</span> <span class="dv">2</span>)))<span class="sc">+</span></span>
<span id="cb19-97"><a href="#cb19-97" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>Rep,<span class="at">scales =</span> <span class="st">"free_y"</span>)<span class="sc">+</span></span>
<span id="cb19-98"><a href="#cb19-98" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_ribbon</span>(<span class="fu">aes</span>(<span class="at">x=</span>mCA.mean,<span class="at">y=</span>cov.mean,</span>
<span id="cb19-99"><a href="#cb19-99" aria-hidden="true" tabindex="-1"></a>                  <span class="at">ymin=</span>cov.mean<span class="sc">-</span>cov.mean.sd,</span>
<span id="cb19-100"><a href="#cb19-100" aria-hidden="true" tabindex="-1"></a>                  <span class="at">ymax=</span>cov.mean<span class="sc">+</span>cov.mean.sd),</span>
<span id="cb19-101"><a href="#cb19-101" aria-hidden="true" tabindex="-1"></a>              <span class="at">alpha=</span><span class="fl">0.2</span>,<span class="at">color=</span><span class="st">'NA'</span>) <span class="sc">+</span></span>
<span id="cb19-102"><a href="#cb19-102" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">name =</span> <span class="st">"MECP2"</span>,<span class="at">values =</span> <span class="fu">c</span>(<span class="st">'red'</span>,<span class="st">'black'</span>),<span class="at">aesthetics =</span> <span class="fu">c</span>(<span class="st">"colour"</span>, <span class="st">"fill"</span>))</span>
<span id="cb19-103"><a href="#cb19-103" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-104"><a href="#cb19-104" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">'Plots/GSE243009_CNR_vs_mCA_G118E_vs_WT_corr.pdf'</span>,<span class="at">height =</span> <span class="dv">7</span>,<span class="at">width =</span> <span class="dv">18</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></li>
<li><p>Supplementary Figure 1I.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dir.create</span>(<span class="fu">file.path</span>(<span class="fu">paste0</span>(path,<span class="st">"/results/MECP2_motifs"</span>)), <span class="at">showWarnings =</span> <span class="cn">FALSE</span>)</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">meme_bin =</span> <span class="st">"/cm/shared/apps/meme/5.3.3/bin/"</span>)</span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a><span class="fu">check_meme_install</span>()</span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a>mecp2  <span class="ot">=</span> <span class="st">"results/MECP2/MECP2WT.vs.KO_overlap_MECP2WT.vs.IgG.noBlacklist_filtered.bed"</span></span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a>MECP2_peaks.GR <span class="ot">=</span> <span class="fu">readPeakFile</span>(mecp2)</span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a>MECP2_peaks.GR<span class="fl">.100</span>bp <span class="ot">=</span> <span class="fu">resize</span>(MECP2_peaks.GR,<span class="at">width =</span> <span class="dv">100</span>,<span class="at">fix =</span> <span class="st">'center'</span>)</span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a><span class="fu">as.data.frame</span>(MECP2_peaks.GR<span class="fl">.100</span>bp) <span class="sc">%&gt;%</span> </span>
<span id="cb20-10"><a href="#cb20-10" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">4</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb20-11"><a href="#cb20-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write.table</span>(<span class="st">'results/MECP2/MECP2WT.vs.KO_overlap_MECP2WT.vs.IgG.noBlacklist_filtered_pm50bp.bed'</span>,</span>
<span id="cb20-12"><a href="#cb20-12" aria-hidden="true" tabindex="-1"></a>              <span class="at">col.names =</span> F,<span class="at">row.names =</span> F,<span class="at">sep=</span><span class="st">'</span><span class="sc">\t</span><span class="st">'</span>,<span class="at">quote =</span> F)</span>
<span id="cb20-13"><a href="#cb20-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-14"><a href="#cb20-14" aria-hidden="true" tabindex="-1"></a>MECP2_peaks.GR<span class="fl">.100</span>bp.seq <span class="ot">=</span> <span class="fu">get_sequence</span>(MECP2_peaks.GR<span class="fl">.100</span>bp,Mmusculus)</span>
<span id="cb20-15"><a href="#cb20-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-16"><a href="#cb20-16" aria-hidden="true" tabindex="-1"></a>MECP2_peaks.GR<span class="fl">.100</span>bp.DREME <span class="ot">=</span> <span class="fu">runDreme</span>(MECP2_peaks.GR<span class="fl">.100</span>bp.seq,</span>
<span id="cb20-17"><a href="#cb20-17" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">control=</span><span class="st">"shuffle"</span>,</span>
<span id="cb20-18"><a href="#cb20-18" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">maxk =</span><span class="dv">12</span>,</span>
<span id="cb20-19"><a href="#cb20-19" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">outdir =</span> <span class="st">"results/MECP2_motifs/MECP2_all_peaks_DREME"</span>,<span class="at">dna=</span><span class="cn">TRUE</span>)</span>
<span id="cb20-20"><a href="#cb20-20" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(MECP2_peaks.GR<span class="fl">.100</span>bp.DREME,</span>
<span id="cb20-21"><a href="#cb20-21" aria-hidden="true" tabindex="-1"></a>     <span class="at">file=</span><span class="st">'results/MECP2_motifs/MECP2_all_peaks_DREME/MECP2_all_peaks_DREME.RData'</span>)</span>
<span id="cb20-22"><a href="#cb20-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-23"><a href="#cb20-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-24"><a href="#cb20-24" aria-hidden="true" tabindex="-1"></a>MECP2_peaks.GR<span class="fl">.100</span>bp.Streme <span class="ot">=</span> <span class="fu">runStreme</span>(MECP2_peaks.GR<span class="fl">.100</span>bp.seq,</span>
<span id="cb20-25"><a href="#cb20-25" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">control=</span><span class="st">"shuffle"</span>,</span>
<span id="cb20-26"><a href="#cb20-26" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">outdir =</span> <span class="st">"results/MECP2_motifs/MECP2_all_peaks_STREME"</span>,</span>
<span id="cb20-27"><a href="#cb20-27" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">dna=</span><span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></li>
<li><p>Supplementary Figure 1J</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="co"># CCAC and A/T repeats motif frequency on MBHs sites</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>x <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">'results/MECP2_motifs/MECP2_all_peaks_STREME/Motif1-4_density_on_MECP2_peaks.txt'</span>,</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>             <span class="at">sep =</span> <span class="st">'</span><span class="sc">\t</span><span class="st">'</span>,<span class="at">header =</span> T) <span class="sc">%&gt;%</span> </span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">5</span>)</span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(x)</span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(x) <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">'X[0-9]</span><span class="sc">\\</span><span class="st">.|</span><span class="sc">\\</span><span class="st">..*'</span>,<span class="st">''</span>,<span class="fu">colnames</span>(x))  </span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a>x <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">abs</span>(.[,<span class="dv">1</span>]) <span class="sc">&lt;</span> <span class="dv">990</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a>  <span class="co">#mutate(CAC = CAC-0.024) %&gt;% </span></span>
<span id="cb21-11"><a href="#cb21-11" aria-hidden="true" tabindex="-1"></a>  <span class="co">#mutate(AKTCATTT = AKTCATTT-0.01) %&gt;% </span></span>
<span id="cb21-12"><a href="#cb21-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">melt</span>(<span class="at">id.vars =</span> <span class="st">'Distance'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb21-13"><a href="#cb21-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>Distance,<span class="at">y=</span>value,<span class="at">color=</span>variable))<span class="sc">+</span> </span>
<span id="cb21-14"><a href="#cb21-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>()<span class="sc">+</span></span>
<span id="cb21-15"><a href="#cb21-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">'red'</span>,<span class="st">'blue'</span>))<span class="sc">+</span></span>
<span id="cb21-16"><a href="#cb21-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_test</span>(<span class="dv">20</span>)<span class="sc">+</span></span>
<span id="cb21-17"><a href="#cb21-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">color=</span><span class="cn">NULL</span>)<span class="sc">+</span></span>
<span id="cb21-18"><a href="#cb21-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">'Motifs per base per peak'</span>)</span>
<span id="cb21-19"><a href="#cb21-19" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">'Plots/motif24_on_MECP2_peaks.pdf'</span>,<span class="at">height =</span> <span class="dv">6</span>,<span class="at">width =</span> <span class="dv">14</span>,)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></li>
<li><p>Supplementary Figure 1K.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Run known motif enrichment analyses using AME</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>ame_vs_shuffle <span class="ot">=</span> <span class="fu">runAme</span>(MECP2_peaks.GR<span class="fl">.100</span>bp.seq,</span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>                         <span class="at">outdir =</span> <span class="st">"results/MECP2_motifs/MECP2_all_peaks_AME"</span>,</span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>                         <span class="at">database =</span> motifs,<span class="at">control =</span> )</span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a>AME <span class="ot">=</span> <span class="fu">importAme</span>(<span class="st">"results/MECP2_motifs/MECP2_all_peaks_AME/ame.tsv"</span>)</span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a>motif.list <span class="ot">=</span> <span class="fu">list</span>()</span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a>motif.list.rc <span class="ot">=</span> <span class="fu">list</span>()</span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">4</span>,<span class="dv">5</span>,<span class="dv">9</span>,<span class="dv">12</span><span class="sc">:</span><span class="dv">20</span>)){</span>
<span id="cb22-12"><a href="#cb22-12" aria-hidden="true" tabindex="-1"></a>  con <span class="ot">=</span> AME[i,<span class="dv">5</span>] <span class="sc">%&gt;%</span> <span class="fu">as.vector</span>() <span class="sc">%&gt;%</span> <span class="fu">unlist</span>() <span class="sc">%&gt;%</span> <span class="fu">setNames</span>(<span class="cn">NULL</span>)</span>
<span id="cb22-13"><a href="#cb22-13" aria-hidden="true" tabindex="-1"></a>  con_name <span class="ot">=</span> AME[i,<span class="dv">3</span>] <span class="sc">%&gt;%</span> <span class="fu">as.vector</span>() <span class="sc">%&gt;%</span> <span class="fu">unlist</span>() <span class="sc">%&gt;%</span> <span class="fu">setNames</span>(<span class="cn">NULL</span>)</span>
<span id="cb22-14"><a href="#cb22-14" aria-hidden="true" tabindex="-1"></a>  motif.list.rc[[i]] <span class="ot">=</span> <span class="fu">motif_rc</span>(<span class="fu">create_motif</span>(con,<span class="at">name=</span>con_name))</span>
<span id="cb22-15"><a href="#cb22-15" aria-hidden="true" tabindex="-1"></a>  motif.list[[i]] <span class="ot">=</span> <span class="fu">create_motif</span>(con,<span class="at">name=</span>con_name)</span>
<span id="cb22-16"><a href="#cb22-16" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb22-17"><a href="#cb22-17" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">"results/MECP2_motifs/MECP2_all_peaks_AME/SelectedEnriched_motifLogo.pdf"</span>,<span class="at">height =</span> <span class="dv">10</span>)</span>
<span id="cb22-18"><a href="#cb22-18" aria-hidden="true" tabindex="-1"></a><span class="fu">view_motifs</span>(motif.list, <span class="at">show.positions.once =</span> <span class="cn">FALSE</span>, <span class="at">names.pos =</span> <span class="st">"right"</span>)</span>
<span id="cb22-19"><a href="#cb22-19" aria-hidden="true" tabindex="-1"></a><span class="fu">view_motifs</span>(motif.list.rc, <span class="at">show.positions.once =</span> <span class="cn">FALSE</span>, <span class="at">names.pos =</span> <span class="st">"right"</span>)</span>
<span id="cb22-20"><a href="#cb22-20" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></li>
</ol>
</section>
<section id="figure-2-mecp2-binding-to-mbh-sites-is-largely-methylation-independent." class="level3">
<h3 class="anchored" data-anchor-id="figure-2-mecp2-binding-to-mbh-sites-is-largely-methylation-independent."><strong>Figure 2: MECP2 binding to MBH sites is largely methylation-independent.</strong></h3>
<ol type="1">
<li><p>Figure 2A.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="at">file =</span> <span class="st">"results/BSMap/8wk_CTX_Stroud2017_mm10_BSmap_CA.RData"</span>,<span class="at">verbose =</span> T)</span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>MECP2_peaks <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">"results/MECP2/MECP2WT.vs.KO_overlap_MECP2WT.vs.IgG.noBlacklist_peakID_filtered.bed"</span>,<span class="at">sep=</span><span class="st">'</span><span class="sc">\t</span><span class="st">'</span>,<span class="at">header =</span> F) </span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(MECP2_peaks) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">'Chr'</span>,<span class="st">'Start'</span>,<span class="st">'End'</span>,<span class="st">'PeakID'</span>)</span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a>MECP2_peaks.GR <span class="ot">=</span> MECP2_peaks <span class="sc">%&gt;%</span> <span class="fu">GRanges</span>() </span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a>tiles <span class="ot">=</span> <span class="fu">tileMethylCounts</span>(CTX_CA,<span class="at">win.size=</span><span class="dv">10000</span>,<span class="at">step.size=</span><span class="dv">10000</span>,<span class="at">cov.bases =</span> <span class="dv">5</span>)</span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a>tiles.df <span class="ot">=</span> <span class="fu">getData</span>(tiles) <span class="sc">|&gt;</span> </span>
<span id="cb23-9"><a href="#cb23-9" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">meth =</span> numCs<span class="sc">/</span>coverage) <span class="sc">|&gt;</span> </span>
<span id="cb23-10"><a href="#cb23-10" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">8</span>)) <span class="sc">%&gt;%</span> <span class="fu">GRanges</span>()</span>
<span id="cb23-11"><a href="#cb23-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-12"><a href="#cb23-12" aria-hidden="true" tabindex="-1"></a>CTX_CA_meth <span class="ot">=</span><span class="fu">regionCounts</span>(CTX_CA,MECP2_peaks.GR)</span>
<span id="cb23-13"><a href="#cb23-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-14"><a href="#cb23-14" aria-hidden="true" tabindex="-1"></a><span class="fu">getMethylationStats</span>(CTX_CA, <span class="at">plot =</span> <span class="cn">TRUE</span>)</span>
<span id="cb23-15"><a href="#cb23-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-16"><a href="#cb23-16" aria-hidden="true" tabindex="-1"></a>CTX_CA_meth <span class="ot">=</span><span class="fu">regionCounts</span>(CTX_CA,MECP2_peaks.GR)</span>
<span id="cb23-17"><a href="#cb23-17" aria-hidden="true" tabindex="-1"></a>CTX_CA_meth.df <span class="ot">=</span> <span class="fu">getData</span>(CTX_CA_meth)</span>
<span id="cb23-18"><a href="#cb23-18" aria-hidden="true" tabindex="-1"></a>CTX_CA_meth.df<span class="sc">$</span>meth <span class="ot">=</span> CTX_CA_meth.df<span class="sc">$</span>numCs<span class="sc">/</span>CTX_CA_meth.df<span class="sc">$</span>coverage</span>
<span id="cb23-19"><a href="#cb23-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-20"><a href="#cb23-20" aria-hidden="true" tabindex="-1"></a>CTX_CA.df <span class="ot">=</span> <span class="fu">getData</span>(CTX_CA) <span class="sc">|&gt;</span> dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">meth =</span> numCs<span class="sc">/</span>coverage) <span class="sc">|&gt;</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">8</span>))</span>
<span id="cb23-21"><a href="#cb23-21" aria-hidden="true" tabindex="-1"></a>CTX_CA.GR <span class="ot">=</span> CTX_CA.df <span class="sc">|&gt;</span> <span class="fu">GRanges</span>()</span>
<span id="cb23-22"><a href="#cb23-22" aria-hidden="true" tabindex="-1"></a>mat2 <span class="ot">=</span> <span class="fu">normalizeToMatrix</span>(CTX_CA.GR, <span class="fu">resize</span>(MECP2_peaks.GR,<span class="at">width =</span> <span class="dv">1</span>,<span class="at">fix=</span><span class="st">'center'</span>), <span class="at">value_column =</span> <span class="st">"meth"</span>, <span class="at">mean_mode =</span> <span class="st">"absolute"</span>,</span>
<span id="cb23-23"><a href="#cb23-23" aria-hidden="true" tabindex="-1"></a>                         <span class="at">extend =</span> <span class="dv">4000</span>, <span class="at">w =</span> <span class="dv">50</span>, <span class="at">background =</span> <span class="cn">NA</span>, <span class="at">smooth =</span> <span class="cn">TRUE</span>,<span class="at">target_ratio =</span> <span class="fl">0.3</span>)</span>
<span id="cb23-24"><a href="#cb23-24" aria-hidden="true" tabindex="-1"></a>meth_col_fun <span class="ot">=</span> <span class="fu">colorRamp2</span>(<span class="fu">c</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="fu">c</span>(<span class="st">"white"</span>, <span class="st">"red"</span>))</span>
<span id="cb23-25"><a href="#cb23-25" aria-hidden="true" tabindex="-1"></a>ht_list <span class="ot">=</span> <span class="fu">EnrichedHeatmap</span>(mat2, <span class="at">col =</span> meth_col_fun, <span class="at">name =</span> <span class="st">"methylation"</span>, <span class="at">column_title =</span> <span class="st">"CA methylation near TSS"</span>,</span>
<span id="cb23-26"><a href="#cb23-26" aria-hidden="true" tabindex="-1"></a>                          <span class="at">top_annotation =</span> <span class="fu">HeatmapAnnotation</span>(<span class="at">enriched =</span> <span class="fu">anno_enriched</span>(<span class="at">ylim=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">0.03</span>))))</span>
<span id="cb23-27"><a href="#cb23-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-28"><a href="#cb23-28" aria-hidden="true" tabindex="-1"></a>density_map <span class="ot">=</span> <span class="fu">extract_anno_enriched</span>(<span class="fu">draw</span>(ht_list))</span>
<span id="cb23-29"><a href="#cb23-29" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="at">file=</span><span class="st">'Plots/mCA_density_on_MECP2_peaks.pdf'</span>,<span class="at">height =</span> <span class="dv">8</span>,<span class="at">width =</span> <span class="dv">10</span>)</span>
<span id="cb23-30"><a href="#cb23-30" aria-hidden="true" tabindex="-1"></a><span class="fu">extract_anno_enriched</span>(<span class="fu">draw</span>(ht_list))</span>
<span id="cb23-31"><a href="#cb23-31" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb23-32"><a href="#cb23-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-33"><a href="#cb23-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-34"><a href="#cb23-34" aria-hidden="true" tabindex="-1"></a><span class="do">#######################################################################################################</span></span>
<span id="cb23-35"><a href="#cb23-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-36"><a href="#cb23-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-37"><a href="#cb23-37" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="at">file =</span> <span class="st">"results/BSMap/8wk_CTX_Stroud2017_mm10_BSmap_CG.RData"</span>,<span class="at">verbose=</span><span class="cn">TRUE</span>)</span>
<span id="cb23-38"><a href="#cb23-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-39"><a href="#cb23-39" aria-hidden="true" tabindex="-1"></a>CTX_CG_meth <span class="ot">=</span><span class="fu">regionCounts</span>(CTX_CG,MECP2_peaks.GR)</span>
<span id="cb23-40"><a href="#cb23-40" aria-hidden="true" tabindex="-1"></a>CTX_CG_meth.df <span class="ot">=</span> <span class="fu">getData</span>(CTX_CG_meth)</span>
<span id="cb23-41"><a href="#cb23-41" aria-hidden="true" tabindex="-1"></a>CTX_CG_meth.df<span class="sc">$</span>meth <span class="ot">=</span> CTX_CG_meth.df<span class="sc">$</span>numCs<span class="sc">/</span>CTX_CG_meth.df<span class="sc">$</span>coverage</span>
<span id="cb23-42"><a href="#cb23-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-43"><a href="#cb23-43" aria-hidden="true" tabindex="-1"></a>CTX_CG.df <span class="ot">=</span> <span class="fu">getData</span>(CTX_CG) <span class="sc">|&gt;</span> dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">meth =</span> numCs<span class="sc">/</span>coverage) <span class="sc">|&gt;</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">8</span>))</span>
<span id="cb23-44"><a href="#cb23-44" aria-hidden="true" tabindex="-1"></a>CTX_CG.GR <span class="ot">=</span> CTX_CG.df <span class="sc">|&gt;</span> <span class="fu">GRanges</span>()</span>
<span id="cb23-45"><a href="#cb23-45" aria-hidden="true" tabindex="-1"></a>mat2 <span class="ot">=</span> <span class="fu">normalizeToMatrix</span>(CTX_CG.GR, <span class="fu">resize</span>(MECP2_peaks.GR,<span class="at">width =</span> <span class="dv">1</span>,<span class="at">fix=</span><span class="st">'center'</span>), </span>
<span id="cb23-46"><a href="#cb23-46" aria-hidden="true" tabindex="-1"></a>                         <span class="at">value_column =</span> <span class="st">"meth"</span>, <span class="at">mean_mode =</span> <span class="st">"absolute"</span>,</span>
<span id="cb23-47"><a href="#cb23-47" aria-hidden="true" tabindex="-1"></a>                         <span class="at">extend =</span> <span class="dv">4000</span>, <span class="at">w =</span> <span class="dv">100</span>, <span class="at">background =</span> <span class="cn">NA</span>, <span class="at">smooth =</span> <span class="cn">TRUE</span>,<span class="at">target_ratio =</span> <span class="fl">0.3</span>)</span>
<span id="cb23-48"><a href="#cb23-48" aria-hidden="true" tabindex="-1"></a>meth_col_fun <span class="ot">=</span> <span class="fu">colorRamp2</span>(<span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>), <span class="fu">c</span>(<span class="st">"white"</span>, <span class="st">"red"</span>))</span>
<span id="cb23-49"><a href="#cb23-49" aria-hidden="true" tabindex="-1"></a>top_annotation <span class="ot">=</span>  <span class="fu">HeatmapAnnotation</span>(<span class="at">enriched =</span> <span class="fu">anno_enriched</span>(<span class="at">ylim=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>)))</span>
<span id="cb23-50"><a href="#cb23-50" aria-hidden="true" tabindex="-1"></a>ht_list  <span class="ot">=</span><span class="fu">EnrichedHeatmap</span>(mat2, <span class="at">col =</span> meth_col_fun, <span class="at">name =</span> <span class="st">"methylation"</span>, </span>
<span id="cb23-51"><a href="#cb23-51" aria-hidden="true" tabindex="-1"></a>                          <span class="at">column_title =</span> <span class="st">"CG methylation near MECP2"</span>,</span>
<span id="cb23-52"><a href="#cb23-52" aria-hidden="true" tabindex="-1"></a>                          <span class="at">top_annotation =</span> <span class="fu">HeatmapAnnotation</span>(<span class="at">enriched =</span> <span class="fu">anno_enriched</span>(<span class="at">ylim=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">0.5</span>))))</span>
<span id="cb23-53"><a href="#cb23-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-54"><a href="#cb23-54" aria-hidden="true" tabindex="-1"></a>density_map <span class="ot">=</span> <span class="fu">extract_anno_enriched</span>(<span class="fu">draw</span>(ht_list))</span>
<span id="cb23-55"><a href="#cb23-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-56"><a href="#cb23-56" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="at">file=</span><span class="st">'Plots/mCG_density_on_MECP2_peaks.pdf'</span>,<span class="at">height =</span> <span class="dv">8</span>,<span class="at">width =</span> <span class="dv">10</span>)</span>
<span id="cb23-57"><a href="#cb23-57" aria-hidden="true" tabindex="-1"></a><span class="fu">extract_anno_enriched</span>(<span class="fu">draw</span>(ht_list))</span>
<span id="cb23-58"><a href="#cb23-58" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></li>
<li><p>Figure 2B and Supplementary Figure 6.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>Diif_MECP2_D3AKO_vs_WT <span class="ot">=</span> <span class="fu">list.files</span>(<span class="at">path=</span><span class="st">'results/MECP2/'</span>,</span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">pattern =</span> <span class="st">'MECP2.*_DNMT3A.*.noBlacklist.bed'</span>,</span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">full.names =</span> T)[<span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>)]</span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>Diif_MECP2_D3AKO_vs_WT</span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>file.name <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">'.noBlacklist.bed'</span>,<span class="st">''</span>,<span class="fu">basename</span>(Diif_MECP2_D3AKO_vs_WT))</span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(Diif_MECP2_D3AKO_vs_WT) <span class="ot">=</span> file.name</span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a>Diif_MECP2_D3AKO_vs_WT.bed<span class="ot">=</span> <span class="fu">lapply</span>(Diif_MECP2_D3AKO_vs_WT, readPeakFile)</span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a>MBHs_counts <span class="ot">=</span> <span class="fu">lapply</span>(Diif_MECP2_D3AKO_vs_WT.bed, as.data.frame) <span class="sc">%&gt;%</span> </span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">lapply</span>(.,dim) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> <span class="fu">t</span>() <span class="sc">%&gt;%</span> .[,<span class="dv">1</span>] <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>()</span>
<span id="cb24-11"><a href="#cb24-11" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(MBHs_counts) <span class="ot">=</span> <span class="st">'Count'</span></span>
<span id="cb24-12"><a href="#cb24-12" aria-hidden="true" tabindex="-1"></a>MBHs_counts<span class="sc">$</span>Group <span class="ot">=</span> <span class="fu">c</span>(<span class="st">'No Change'</span>,<span class="st">'Gain'</span>,<span class="st">'Loss'</span>)</span>
<span id="cb24-13"><a href="#cb24-13" aria-hidden="true" tabindex="-1"></a>MBHs_counts<span class="sc">$</span>Group2 <span class="ot">=</span> <span class="fu">c</span>(<span class="st">'MECP2'</span>,<span class="st">'MECP2'</span>,<span class="st">'MECP2'</span>)</span>
<span id="cb24-14"><a href="#cb24-14" aria-hidden="true" tabindex="-1"></a>MBHs_counts[<span class="sc">-</span><span class="dv">5</span>,] <span class="sc">%&gt;%</span> </span>
<span id="cb24-15"><a href="#cb24-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Group=</span> <span class="fu">factor</span>(Group, <span class="at">levels=</span><span class="fu">c</span>(<span class="st">'Loss'</span>,<span class="st">'Gain'</span>,<span class="st">'No Change'</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb24-16"><a href="#cb24-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-17"><a href="#cb24-17" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>Group,<span class="at">y=</span>Count,<span class="at">fill=</span>Group2))<span class="sc">+</span></span>
<span id="cb24-18"><a href="#cb24-18" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">'identity'</span>)<span class="sc">+</span></span>
<span id="cb24-19"><a href="#cb24-19" aria-hidden="true" tabindex="-1"></a><span class="fu">coord_flip</span>()<span class="sc">+</span></span>
<span id="cb24-20"><a href="#cb24-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">'Dnmt3a cKO vs WT'</span>)<span class="sc">+</span></span>
<span id="cb24-21"><a href="#cb24-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_test</span>(<span class="dv">20</span>)<span class="sc">+</span></span>
<span id="cb24-22"><a href="#cb24-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">'grey'</span>))</span>
<span id="cb24-23"><a href="#cb24-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-24"><a href="#cb24-24" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"results/RData/Boxer_RNAseq_data.RData"</span>)</span>
<span id="cb24-25"><a href="#cb24-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-26"><a href="#cb24-26" aria-hidden="true" tabindex="-1"></a>Diif_MECP2_D3AKO_vs_WT.anno.logFC <span class="ot">=</span> <span class="fu">lapply</span>(Diif_MECP2_D3AKO_vs_WT.anno, <span class="cf">function</span>(x) {</span>
<span id="cb24-27"><a href="#cb24-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">merge</span>(x <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(ENSEMBL) <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(ENSEMBL) <span class="sc">%&gt;%</span> </span>
<span id="cb24-28"><a href="#cb24-28" aria-hidden="true" tabindex="-1"></a>          <span class="fu">summarise</span>(<span class="at">Count=</span>dplyr<span class="sc">::</span><span class="fu">n</span>()) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Count <span class="sc">&gt;=</span><span class="dv">2</span>),</span>
<span id="cb24-29"><a href="#cb24-29" aria-hidden="true" tabindex="-1"></a>        RNAseq.list[[<span class="dv">7</span>]]) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(FDR <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> <span class="sc">!</span>(<span class="fu">is.na</span>(ENSEMBL)))  </span>
<span id="cb24-30"><a href="#cb24-30" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb24-31"><a href="#cb24-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-32"><a href="#cb24-32" aria-hidden="true" tabindex="-1"></a>Diif_MECP2_D3AKO_vs_WT.anno.logFC.df <span class="ot">=</span> <span class="fu">rbindlist</span>(Diif_MECP2_D3AKO_vs_WT.anno.logFC,<span class="at">idcol =</span> <span class="st">'Group'</span>) <span class="sc">%&gt;%</span>   <span class="fu">mutate</span>(<span class="at">MBHs =</span> <span class="fu">case_when</span>(<span class="fu">grepl</span>(<span class="st">'unchaged'</span>,Group)<span class="sc">~</span><span class="st">'No Change'</span>,</span>
<span id="cb24-33"><a href="#cb24-33" aria-hidden="true" tabindex="-1"></a>                         <span class="fu">grepl</span>(<span class="st">'*_up.*WT_vs_KO'</span>,Group)<span class="sc">~</span><span class="st">'Loss'</span>,</span>
<span id="cb24-34"><a href="#cb24-34" aria-hidden="true" tabindex="-1"></a>                         <span class="fu">grepl</span>(<span class="st">'*_up.*KO_vs_WT'</span>,Group)<span class="sc">~</span><span class="st">'Gain'</span>),</span>
<span id="cb24-35"><a href="#cb24-35" aria-hidden="true" tabindex="-1"></a>         <span class="at">MBHs=</span> <span class="fu">factor</span>(MBHs,<span class="at">levels=</span><span class="fu">c</span>(<span class="st">'Gain'</span>,<span class="st">'No Change'</span>,<span class="st">'Loss'</span>))) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">5</span>,<span class="dv">9</span>) <span class="sc">%&gt;%</span> <span class="fu">distinct</span>() </span>
<span id="cb24-36"><a href="#cb24-36" aria-hidden="true" tabindex="-1"></a>MBHs_number <span class="ot">=</span> <span class="fu">table</span>(Diif_MECP2_D3AKO_vs_WT.anno.logFC.df<span class="sc">$</span>MBHs)</span>
<span id="cb24-37"><a href="#cb24-37" aria-hidden="true" tabindex="-1"></a>MBHs_number <span class="ot">=</span> <span class="fu">paste0</span>(<span class="fu">names</span>(MBHs_number), <span class="st">' (n='</span>,MBHs_number,<span class="st">')'</span>)</span>
<span id="cb24-38"><a href="#cb24-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-39"><a href="#cb24-39" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(Diif_MECP2_D3AKO_vs_WT.anno.logFC.df, <span class="fu">aes</span>(<span class="at">x=</span>MBHs,<span class="at">y=</span>logFC))<span class="sc">+</span></span>
<span id="cb24-40"><a href="#cb24-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>()<span class="sc">+</span></span>
<span id="cb24-41"><a href="#cb24-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">'logFC (DNMT3A KO vs WT RNA-seq)'</span>)<span class="sc">+</span></span>
<span id="cb24-42"><a href="#cb24-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_test</span>(<span class="dv">20</span>)<span class="sc">+</span></span>
<span id="cb24-43"><a href="#cb24-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="dv">45</span>,<span class="at">hjust =</span> <span class="dv">1</span>,<span class="at">vjust =</span> <span class="dv">1</span>))<span class="sc">+</span></span>
<span id="cb24-44"><a href="#cb24-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>,<span class="at">linetype=</span><span class="st">'dashed'</span>)<span class="sc">+</span></span>
<span id="cb24-45"><a href="#cb24-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>(<span class="at">comparisons =</span> <span class="fu">list</span>(<span class="fu">c</span>(<span class="st">'Gain'</span>,<span class="st">'Loss'</span>),</span>
<span id="cb24-46"><a href="#cb24-46" aria-hidden="true" tabindex="-1"></a>                                        <span class="fu">c</span>(<span class="st">'Gain'</span>,<span class="st">'No Change'</span>),</span>
<span id="cb24-47"><a href="#cb24-47" aria-hidden="true" tabindex="-1"></a>                                        <span class="fu">c</span>(<span class="st">'Loss'</span>,<span class="st">'No Change'</span>)))<span class="sc">+</span></span>
<span id="cb24-48"><a href="#cb24-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">labels=</span><span class="fu">c</span>(<span class="st">"Gain"</span> <span class="ot">=</span> MBHs_number[<span class="dv">1</span>], </span>
<span id="cb24-49"><a href="#cb24-49" aria-hidden="true" tabindex="-1"></a>                            <span class="st">"No Change"</span> <span class="ot">=</span> MBHs_number[<span class="dv">2</span>],</span>
<span id="cb24-50"><a href="#cb24-50" aria-hidden="true" tabindex="-1"></a>                            <span class="st">"Loss"</span> <span class="ot">=</span> MBHs_number[<span class="dv">3</span>]))<span class="sc">+</span></span>
<span id="cb24-51"><a href="#cb24-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">'MBHs'</span>)</span>
<span id="cb24-52"><a href="#cb24-52" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">'Plots/MBHs_gene_DNMT3a_KOvs_WT.pdf'</span>,<span class="at">height =</span> <span class="dv">7</span>,<span class="at">width =</span> <span class="dv">4</span>)</span>
<span id="cb24-53"><a href="#cb24-53" aria-hidden="true" tabindex="-1"></a><span class="co">#ggsave("Plots/Diff_MBHs_DNMT3AKO_vs_WT.pdf",height = 5,width = 8)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></li>
<li><p>Figure 2C.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>MECP2_diff.dnmt3awt <span class="ot">=</span> <span class="fu">list.files</span>(<span class="st">"results/MECP2/"</span>,<span class="at">pattern =</span> <span class="st">".sort_multiBigWigOut.tab"</span>,</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(MECP2_diff.dnmt3awt) <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">"MECP2WT.vs.KO_overlap_MECP2WT.vs.IgG.noBlacklist_filtered_|.sort_multiBigWigOut.tab"</span>, <span class="st">''</span>, <span class="fu">basename</span>(MECP2_diff.dnmt3awt))</span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>MECP2_diff.dnmt3awt <span class="ot">=</span> <span class="fu">lapply</span>(MECP2_diff.dnmt3awt, read.csv,  <span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>,<span class="at">header=</span>T,<span class="at">check.names=</span>F)</span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a><span class="fu">rbindlist</span>(MECP2_diff.dnmt3awt,<span class="at">idcol =</span> <span class="st">"Group"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">1</span>,<span class="dv">5</span><span class="sc">:</span><span class="dv">8</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">melt</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">variable =</span> <span class="fu">gsub</span>(<span class="st">'</span><span class="sc">\'</span><span class="st">'</span>,<span class="st">''</span>,variable)) <span class="sc">%&gt;%</span></span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Rep =</span> <span class="fu">gsub</span>(<span class="st">'Dnmt3a_.*_'</span>,<span class="st">''</span>,variable)) <span class="sc">%&gt;%</span> </span>
<span id="cb25-12"><a href="#cb25-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sample =</span> <span class="fu">gsub</span>(<span class="st">'_rep[0-9]'</span>,<span class="st">''</span>,variable)) <span class="sc">%&gt;%</span> </span>
<span id="cb25-13"><a href="#cb25-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sample =</span> <span class="fu">factor</span>(Sample,<span class="at">levels=</span><span class="fu">c</span>(<span class="st">"Dnmt3a_WT"</span>,<span class="st">"Dnmt3a_KO"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb25-14"><a href="#cb25-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>Sample,<span class="at">y=</span><span class="fu">log2</span>(value),<span class="at">color=</span>Sample))<span class="sc">+</span></span>
<span id="cb25-15"><a href="#cb25-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>()<span class="sc">+</span></span>
<span id="cb25-16"><a href="#cb25-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_wrap</span>(<span class="sc">~</span>Group)<span class="sc">+</span></span>
<span id="cb25-17"><a href="#cb25-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">stat_compare_means</span>() <span class="sc">+</span></span>
<span id="cb25-18"><a href="#cb25-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_test</span>(<span class="dv">20</span>)<span class="sc">+</span></span>
<span id="cb25-19"><a href="#cb25-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"grey"</span>,<span class="st">"red"</span>))<span class="sc">+</span></span>
<span id="cb25-20"><a href="#cb25-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">""</span>)<span class="sc">+</span></span>
<span id="cb25-21"><a href="#cb25-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"log2(MECP2 Coverage"</span>)</span>
<span id="cb25-22"><a href="#cb25-22" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"Plots/Diff_MECP2_coverage_Dnmt3aWT.KO.multiBigWigOut.pdf"</span>,<span class="at">height =</span> <span class="dv">5</span>,<span class="at">width =</span> <span class="dv">6</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></li>
<li><p>Supplementary Figure 2B.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to average column value across bins and step size</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>summarize_logFC_mCA <span class="ot">=</span> <span class="cf">function</span>(x,<span class="at">binSize =</span> <span class="dv">19</span>,<span class="at">stepSize =</span><span class="dv">4</span>){</span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>  df.bin <span class="ot">=</span><span class="fu">list</span>()</span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">seq</span>(<span class="at">from=</span><span class="dv">1</span>, <span class="at">to=</span><span class="fu">nrow</span>(x), <span class="at">by=</span>stepSize)){</span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(i)</span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a>    start <span class="ot">=</span> i</span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a>    stop  <span class="ot">=</span> i<span class="sc">+</span>binSize</span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(stop <span class="sc">&lt;=</span> <span class="fu">nrow</span>(x)){</span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a>      df <span class="ot">=</span> x[start<span class="sc">:</span>stop,]</span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a>      df.bin[[i]] <span class="ot">=</span> df  <span class="sc">%&gt;%</span> <span class="fu">summarize</span>(<span class="at">cov.mean=</span><span class="fu">mean</span>(.[,<span class="dv">1</span>]),<span class="at">cov.mean.sd =</span> <span class="fu">sd</span>(.[,<span class="dv">1</span>]),<span class="at">mCA.mean=</span><span class="fu">mean</span>(.[,<span class="dv">2</span>]))</span>
<span id="cb26-11"><a href="#cb26-11" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb26-12"><a href="#cb26-12" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>{</span>
<span id="cb26-13"><a href="#cb26-13" aria-hidden="true" tabindex="-1"></a>      <span class="fu">return</span>(df.bin)</span>
<span id="cb26-14"><a href="#cb26-14" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb26-15"><a href="#cb26-15" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb26-16"><a href="#cb26-16" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb26-17"><a href="#cb26-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-18"><a href="#cb26-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-19"><a href="#cb26-19" aria-hidden="true" tabindex="-1"></a>CNR_files <span class="ot">=</span> <span class="fu">list.files</span>(<span class="st">'results/CNR_Coverage/'</span>,<span class="at">full.names =</span> <span class="cn">TRUE</span>,<span class="at">pattern =</span> <span class="st">"*_Mecp2_*.*_gene.cov.bed"</span>)</span>
<span id="cb26-20"><a href="#cb26-20" aria-hidden="true" tabindex="-1"></a>CNR_files.list <span class="ot">=</span> <span class="fu">lapply</span>(CNR_files, read.csv, <span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>,<span class="at">header=</span>F)</span>
<span id="cb26-21"><a href="#cb26-21" aria-hidden="true" tabindex="-1"></a>CNR_files.names <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">".bed"</span>,<span class="st">""</span>,<span class="fu">basename</span>(CNR_files))</span>
<span id="cb26-22"><a href="#cb26-22" aria-hidden="true" tabindex="-1"></a>CNR_files.list <span class="ot">=</span> <span class="fu">lapply</span>(<span class="fu">seq_along</span>(CNR_files.list),<span class="cf">function</span>(x) {</span>
<span id="cb26-23"><a href="#cb26-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">setNames</span>(CNR_files.list[[x]],<span class="at">nm =</span> <span class="fu">c</span>(<span class="st">'chr'</span>,<span class="st">'start'</span>,<span class="st">'end'</span>,CNR_files.names[x])) })</span>
<span id="cb26-24"><a href="#cb26-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-25"><a href="#cb26-25" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(CNR_files.list)) {</span>
<span id="cb26-26"><a href="#cb26-26" aria-hidden="true" tabindex="-1"></a>  CNR_files.list[[i]]<span class="sc">$</span>Cov <span class="ot">=</span> <span class="fu">ifelse</span>(CNR_files.list[[i]][,<span class="dv">4</span>] <span class="sc">==</span> <span class="st">'.'</span>, <span class="dv">0</span> , CNR_files.list[[i]][,<span class="dv">4</span>])</span>
<span id="cb26-27"><a href="#cb26-27" aria-hidden="true" tabindex="-1"></a>  CNR_files.list[[i]]<span class="sc">$</span>Cov <span class="ot">=</span> <span class="fu">as.numeric</span>(CNR_files.list[[i]]<span class="sc">$</span>Cov)</span>
<span id="cb26-28"><a href="#cb26-28" aria-hidden="true" tabindex="-1"></a>  CNR_files.list[[i]]<span class="sc">$</span>geneLength <span class="ot">=</span> (CNR_files.list[[i]]<span class="sc">$</span>end <span class="sc">-</span> CNR_files.list[[i]]<span class="sc">$</span>start)<span class="sc">/</span><span class="dv">1000</span></span>
<span id="cb26-29"><a href="#cb26-29" aria-hidden="true" tabindex="-1"></a>  CNR_files.list[[i]]<span class="sc">$</span>norm <span class="ot">=</span> CNR_files.list[[i]]<span class="sc">$</span>Cov<span class="sc">/</span>CNR_files.list[[i]]<span class="sc">$</span>geneLength</span>
<span id="cb26-30"><a href="#cb26-30" aria-hidden="true" tabindex="-1"></a>  CNR_files.list[[i]] <span class="ot">=</span> CNR_files.list[[i]][,<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">7</span>)] <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(norm)) <span class="co">#%&gt;% dplyr::slice(-1:-5)</span></span>
<span id="cb26-31"><a href="#cb26-31" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb26-32"><a href="#cb26-32" aria-hidden="true" tabindex="-1"></a>CNR_files.list <span class="ot">=</span> <span class="fu">lapply</span>(<span class="fu">seq_along</span>(CNR_files.list),<span class="cf">function</span>(x) {</span>
<span id="cb26-33"><a href="#cb26-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">setNames</span>(CNR_files.list[[x]],<span class="at">nm =</span> <span class="fu">c</span>(<span class="st">'chr'</span>,<span class="st">'start'</span>,<span class="st">'end'</span>,CNR_files.names[x])) })</span>
<span id="cb26-34"><a href="#cb26-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-35"><a href="#cb26-35" aria-hidden="true" tabindex="-1"></a>CNR_files.df <span class="ot">=</span> <span class="fu">Reduce</span>(full_join,CNR_files.list)</span>
<span id="cb26-36"><a href="#cb26-36" aria-hidden="true" tabindex="-1"></a>CNR_files.df<span class="sc">$</span>ID <span class="ot">=</span> <span class="fu">paste0</span>(CNR_files.df<span class="sc">$</span>chr,<span class="st">'_'</span>,CNR_files.df<span class="sc">$</span>start,<span class="st">'_'</span>,CNR_files.df<span class="sc">$</span>end)</span>
<span id="cb26-37"><a href="#cb26-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-38"><a href="#cb26-38" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert GeneBody coordinated to GRanges</span></span>
<span id="cb26-39"><a href="#cb26-39" aria-hidden="true" tabindex="-1"></a>geneBody.bed <span class="ot">=</span> CNR_files.df[,<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>)]</span>
<span id="cb26-40"><a href="#cb26-40" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(geneBody.bed) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">'chr'</span>,<span class="st">'start'</span>,<span class="st">'end'</span>)</span>
<span id="cb26-41"><a href="#cb26-41" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(geneBody.bed)</span>
<span id="cb26-42"><a href="#cb26-42" aria-hidden="true" tabindex="-1"></a>geneBody.bed.GR <span class="ot">=</span> <span class="fu">GRanges</span>(geneBody.bed)</span>
<span id="cb26-43"><a href="#cb26-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-44"><a href="#cb26-44" aria-hidden="true" tabindex="-1"></a><span class="co"># calculate mCA on MECP2 peaks</span></span>
<span id="cb26-45"><a href="#cb26-45" aria-hidden="true" tabindex="-1"></a><span class="co"># Load genome wide CA methylation BSMap data</span></span>
<span id="cb26-46"><a href="#cb26-46" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">'results/BSMap/8wk_CTX_Stroud2017_mm10_BSmap_CA.RData'</span>,<span class="at">verbose =</span> <span class="cn">TRUE</span>)</span>
<span id="cb26-47"><a href="#cb26-47" aria-hidden="true" tabindex="-1"></a>geneBody_cov_meth <span class="ot">=</span> <span class="fu">regionCounts</span>(CTX_CA,geneBody.bed.GR)</span>
<span id="cb26-48"><a href="#cb26-48" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(geneBody_cov_meth)</span>
<span id="cb26-49"><a href="#cb26-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-50"><a href="#cb26-50" aria-hidden="true" tabindex="-1"></a>geneBody_cov_meth.df <span class="ot">=</span> <span class="fu">getData</span>(geneBody_cov_meth)</span>
<span id="cb26-51"><a href="#cb26-51" aria-hidden="true" tabindex="-1"></a>geneBody_cov_meth.df<span class="sc">$</span>mCA <span class="ot">=</span> geneBody_cov_meth.df<span class="sc">$</span>numCs<span class="sc">/</span>geneBody_cov_meth.df<span class="sc">$</span>coverage</span>
<span id="cb26-52"><a href="#cb26-52" aria-hidden="true" tabindex="-1"></a>geneBody_cov_meth.df<span class="sc">$</span>ID <span class="ot">=</span> <span class="fu">paste0</span>(geneBody_cov_meth.df<span class="sc">$</span>chr,<span class="st">'_'</span>,geneBody_cov_meth.df<span class="sc">$</span>start,<span class="st">"_"</span>,geneBody_cov_meth.df<span class="sc">$</span>end)</span>
<span id="cb26-53"><a href="#cb26-53" aria-hidden="true" tabindex="-1"></a><span class="co">#MECP2_peak_cov_meth.df = MECP2_peak_cov_meth.df %&gt;% column_to_rownames('ID')</span></span>
<span id="cb26-54"><a href="#cb26-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-55"><a href="#cb26-55" aria-hidden="true" tabindex="-1"></a><span class="co"># Merge MECP2 coverage and mCA level on MECP2 peaks</span></span>
<span id="cb26-56"><a href="#cb26-56" aria-hidden="true" tabindex="-1"></a>geneBody_cov_meth<span class="ot">=</span> <span class="fu">merge</span>(CNR_files.df,geneBody_cov_meth.df,<span class="at">by=</span><span class="st">'ID'</span>)</span>
<span id="cb26-57"><a href="#cb26-57" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(geneBody_cov_meth)</span>
<span id="cb26-58"><a href="#cb26-58" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(geneBody_cov_meth)</span>
<span id="cb26-59"><a href="#cb26-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-60"><a href="#cb26-60" aria-hidden="true" tabindex="-1"></a>geneBody_cov_meth.list <span class="ot">=</span> <span class="fu">list</span>(geneBody_cov_meth[,<span class="fu">c</span>(<span class="dv">5</span>,<span class="dv">16</span>)],</span>
<span id="cb26-61"><a href="#cb26-61" aria-hidden="true" tabindex="-1"></a>                              geneBody_cov_meth[,<span class="fu">c</span>(<span class="dv">6</span>,<span class="dv">16</span>)],</span>
<span id="cb26-62"><a href="#cb26-62" aria-hidden="true" tabindex="-1"></a>                              geneBody_cov_meth[,<span class="fu">c</span>(<span class="dv">7</span>,<span class="dv">16</span>)],</span>
<span id="cb26-63"><a href="#cb26-63" aria-hidden="true" tabindex="-1"></a>                              geneBody_cov_meth[,<span class="fu">c</span>(<span class="dv">8</span>,<span class="dv">16</span>)])</span>
<span id="cb26-64"><a href="#cb26-64" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(geneBody_cov_meth.list) <span class="ot">=</span>CNR_files.names</span>
<span id="cb26-65"><a href="#cb26-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-66"><a href="#cb26-66" aria-hidden="true" tabindex="-1"></a>geneBody_cov_meth.list <span class="ot">=</span> <span class="fu">lapply</span>(geneBody_cov_meth.list, </span>
<span id="cb26-67"><a href="#cb26-67" aria-hidden="true" tabindex="-1"></a>                                <span class="cf">function</span>(x) x <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(mCA)) <span class="sc">%&gt;%</span> </span>
<span id="cb26-68"><a href="#cb26-68" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">filter</span>(.[,<span class="dv">1</span>] <span class="sc">&gt;</span><span class="dv">0</span> <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(.[,<span class="dv">1</span>])))</span>
<span id="cb26-69"><a href="#cb26-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-70"><a href="#cb26-70" aria-hidden="true" tabindex="-1"></a>tmp  <span class="ot">=</span> <span class="fu">lapply</span>(geneBody_cov_meth.list, <span class="cf">function</span>(x) <span class="fu">summarize_logFC_mCA</span>(x,<span class="at">binSize =</span> <span class="dv">1200</span>,<span class="at">stepSize =</span> <span class="dv">200</span>))</span>
<span id="cb26-71"><a href="#cb26-71" aria-hidden="true" tabindex="-1"></a>tmp1 <span class="ot">=</span> <span class="fu">lapply</span>(tmp, <span class="cf">function</span>(x) <span class="fu">do.call</span>(rbind,x))</span>
<span id="cb26-72"><a href="#cb26-72" aria-hidden="true" tabindex="-1"></a>tmp2 <span class="ot">=</span> plyr<span class="sc">::</span><span class="fu">ldply</span>(tmp1,data.frame)</span>
<span id="cb26-73"><a href="#cb26-73" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(tmp2)[<span class="dv">1</span>] <span class="ot">=</span> <span class="st">'Samples'</span></span>
<span id="cb26-74"><a href="#cb26-74" aria-hidden="true" tabindex="-1"></a>geneBody_cov_meth.merged.avg <span class="ot">=</span> tmp2[,<span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">1</span>)]</span>
<span id="cb26-75"><a href="#cb26-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-76"><a href="#cb26-76" aria-hidden="true" tabindex="-1"></a>plt <span class="ot">=</span> geneBody_cov_meth.merged.avg <span class="sc">%&gt;%</span> </span>
<span id="cb26-77"><a href="#cb26-77" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>mCA.mean,<span class="at">y=</span>cov.mean,<span class="at">color=</span>Samples))<span class="sc">+</span></span>
<span id="cb26-78"><a href="#cb26-78" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">size=</span><span class="fl">0.5</span>)<span class="sc">+</span></span>
<span id="cb26-79"><a href="#cb26-79" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_line</span>()<span class="sc">+</span></span>
<span id="cb26-80"><a href="#cb26-80" aria-hidden="true" tabindex="-1"></a>    <span class="co">#facet_grid(Brain_region~variable)+</span></span>
<span id="cb26-81"><a href="#cb26-81" aria-hidden="true" tabindex="-1"></a>    <span class="co">#geom_ribbon(aes(x=mCA.mean,y=cov.mean,</span></span>
<span id="cb26-82"><a href="#cb26-82" aria-hidden="true" tabindex="-1"></a>    <span class="co">#               ymin=cov.mean-cov.mean.sd,</span></span>
<span id="cb26-83"><a href="#cb26-83" aria-hidden="true" tabindex="-1"></a>    <span class="co">#              ymax=cov.mean+cov.mean.sd,fill=Samples),</span></span>
<span id="cb26-84"><a href="#cb26-84" aria-hidden="true" tabindex="-1"></a>    <span class="co">#         alpha=0.1,color='NA') +</span></span>
<span id="cb26-85"><a href="#cb26-85" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">name =</span> <span class="st">"DEGs (MECP2 KO vs WT)"</span>,</span>
<span id="cb26-86"><a href="#cb26-86" aria-hidden="true" tabindex="-1"></a>                      <span class="at">values =</span> <span class="fu">c</span>(<span class="st">'grey'</span>,<span class="st">'red'</span>,<span class="st">"black"</span>,<span class="st">"darkred"</span>),</span>
<span id="cb26-87"><a href="#cb26-87" aria-hidden="true" tabindex="-1"></a>                      <span class="at">aesthetics =</span> <span class="fu">c</span>(<span class="st">"colour"</span>, <span class="st">"fill"</span>))<span class="sc">+</span></span>
<span id="cb26-88"><a href="#cb26-88" aria-hidden="true" tabindex="-1"></a>    <span class="co">#stat_cor()+</span></span>
<span id="cb26-89"><a href="#cb26-89" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_test</span>(<span class="dv">20</span>)<span class="sc">+</span></span>
<span id="cb26-90"><a href="#cb26-90" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">'mCA/CA'</span>)<span class="sc">+</span></span>
<span id="cb26-91"><a href="#cb26-91" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylab</span>(<span class="st">'Gene Body coverage'</span>)<span class="sc">+</span></span>
<span id="cb26-92"><a href="#cb26-92" aria-hidden="true" tabindex="-1"></a>    <span class="co">#ggtitle(names(df.list)[x])+ </span></span>
<span id="cb26-93"><a href="#cb26-93" aria-hidden="true" tabindex="-1"></a>    <span class="co">#ylim(0,3)+</span></span>
<span id="cb26-94"><a href="#cb26-94" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">'top'</span>,<span class="at">legend.direction =</span> <span class="st">"vertical"</span>)<span class="sc">+</span></span>
<span id="cb26-95"><a href="#cb26-95" aria-hidden="true" tabindex="-1"></a>    <span class="fu">guides</span>(<span class="at">color =</span> <span class="fu">guide_legend</span>(<span class="at">override.aes =</span> <span class="fu">list</span>(<span class="at">size =</span> <span class="dv">2</span>)))</span>
<span id="cb26-96"><a href="#cb26-96" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-97"><a href="#cb26-97" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">'Plots/MECP2_CNR_DNMT3A_KOandWT_mCA_corr.pdf'</span>,<span class="at">height =</span> <span class="dv">10</span>,<span class="at">width =</span> <span class="dv">12</span>)</span>
<span id="cb26-98"><a href="#cb26-98" aria-hidden="true" tabindex="-1"></a>plt</span>
<span id="cb26-99"><a href="#cb26-99" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></li>
<li><p>Supplementary Figure 2C.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>D3A_BSseq <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="at">file =</span> <span class="st">'results/RData/GSE103214_Cortex_10wk_BSmap.RDS'</span>)</span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>D3A_BSseq_methylRaw <span class="ot">=</span> <span class="fu">list</span>()</span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="fu">seq_along</span>(D3A_BSseq)) {</span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a>     x <span class="ot">=</span> D3A_BSseq[[i]]</span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a>     D3A_BSseq_methylRaw[[<span class="fu">names</span>(D3A_BSseq)[i]]] <span class="ot">=</span> <span class="fu">new</span>(<span class="st">"methylRaw"</span>,</span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a>                                                      <span class="fu">data.frame</span>(<span class="at">chr=</span>x[,<span class="dv">1</span>],<span class="at">start=</span>x[,<span class="dv">2</span>],<span class="at">end=</span>x[,<span class="dv">3</span>],</span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a>                           <span class="at">strand=</span>x[,<span class="dv">5</span>],<span class="at">coverage=</span>(x[,<span class="dv">6</span>]),</span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a>                           <span class="at">numCs=</span>x[,<span class="dv">7</span>],<span class="at">numTs=</span>x[,<span class="dv">8</span>]),</span>
<span id="cb27-9"><a href="#cb27-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">sample.id=</span><span class="fu">names</span>(D3A_BSseq)[i],</span>
<span id="cb27-10"><a href="#cb27-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">assembly=</span><span class="st">'mm10'</span>,</span>
<span id="cb27-11"><a href="#cb27-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">context=</span><span class="st">'context'</span>,</span>
<span id="cb27-12"><a href="#cb27-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">resolution=</span><span class="st">"base"</span>)</span>
<span id="cb27-13"><a href="#cb27-13" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb27-14"><a href="#cb27-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-15"><a href="#cb27-15" aria-hidden="true" tabindex="-1"></a><span class="co"># merge replicates </span></span>
<span id="cb27-16"><a href="#cb27-16" aria-hidden="true" tabindex="-1"></a>merge_BSrep <span class="ot">=</span> <span class="cf">function</span>(x,y){</span>
<span id="cb27-17"><a href="#cb27-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rbind</span>(<span class="fu">getData</span>(x),<span class="fu">getData</span>(y)) <span class="sc">%&gt;%</span> </span>
<span id="cb27-18"><a href="#cb27-18" aria-hidden="true" tabindex="-1"></a>          <span class="fu">group_by</span>(chr,start,end,strand) <span class="sc">%&gt;%</span></span>
<span id="cb27-19"><a href="#cb27-19" aria-hidden="true" tabindex="-1"></a>          <span class="fu">summarise_all</span>(., sum, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) </span>
<span id="cb27-20"><a href="#cb27-20" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb27-21"><a href="#cb27-21" aria-hidden="true" tabindex="-1"></a>D3A_BSseq_methylRaw_repMerged <span class="ot">=</span> <span class="fu">list</span>()</span>
<span id="cb27-22"><a href="#cb27-22" aria-hidden="true" tabindex="-1"></a>D3A_BSseq_methylRaw_repMerged <span class="ot">=</span> <span class="fu">mclapply</span>(<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">5</span>,<span class="dv">6</span>),<span class="cf">function</span>(x) </span>
<span id="cb27-23"><a href="#cb27-23" aria-hidden="true" tabindex="-1"></a>                                      { <span class="fu">merge_BSrep</span>(D3A_BSseq_methylRaw[[x]],</span>
<span id="cb27-24"><a href="#cb27-24" aria-hidden="true" tabindex="-1"></a>                                                    D3A_BSseq_methylRaw[[x<span class="sc">+</span><span class="dv">2</span>]])})</span>
<span id="cb27-25"><a href="#cb27-25" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(D3A_BSseq_methylRaw_repMerged) <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">'rep1</span><span class="sc">\\</span><span class="st">.'</span>,<span class="st">''</span>,<span class="fu">names</span>(D3A_BSseq_methylRaw)[<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">5</span>,<span class="dv">6</span>)])</span>
<span id="cb27-26"><a href="#cb27-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-27"><a href="#cb27-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-28"><a href="#cb27-28" aria-hidden="true" tabindex="-1"></a>D3A_BSseq_methylRaw_repMerged.list <span class="ot">=</span> <span class="fu">list</span>()</span>
<span id="cb27-29"><a href="#cb27-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-30"><a href="#cb27-30" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="fu">seq_along</span>(D3A_BSseq_methylRaw_repMerged)) {</span>
<span id="cb27-31"><a href="#cb27-31" aria-hidden="true" tabindex="-1"></a>  x <span class="ot">=</span> D3A_BSseq_methylRaw_repMerged[[i]] <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>()</span>
<span id="cb27-32"><a href="#cb27-32" aria-hidden="true" tabindex="-1"></a>  D3A_BSseq_methylRaw_repMerged.list[[<span class="fu">names</span>(D3A_BSseq_methylRaw_repMerged)[i]]] <span class="ot">=</span> <span class="fu">new</span>(<span class="st">"methylRaw"</span>,</span>
<span id="cb27-33"><a href="#cb27-33" aria-hidden="true" tabindex="-1"></a>                                                       <span class="fu">data.frame</span>(<span class="at">chr=</span>x[,<span class="dv">1</span>],<span class="at">start=</span>x[,<span class="dv">2</span>],<span class="at">end=</span>x[,<span class="dv">3</span>],</span>
<span id="cb27-34"><a href="#cb27-34" aria-hidden="true" tabindex="-1"></a>                                                                  <span class="at">strand=</span>x[,<span class="dv">4</span>],<span class="at">coverage=</span>(x[,<span class="dv">5</span>]),</span>
<span id="cb27-35"><a href="#cb27-35" aria-hidden="true" tabindex="-1"></a>                                                                  <span class="at">numCs=</span>x[,<span class="dv">6</span>],<span class="at">numTs=</span>x[,<span class="dv">7</span>]),</span>
<span id="cb27-36"><a href="#cb27-36" aria-hidden="true" tabindex="-1"></a>                                                   <span class="at">sample.id=</span><span class="fu">names</span>(D3A_BSseq_methylRaw_repMerged)[i],</span>
<span id="cb27-37"><a href="#cb27-37" aria-hidden="true" tabindex="-1"></a>                                                   <span class="at">assembly=</span><span class="st">'mm10'</span>,</span>
<span id="cb27-38"><a href="#cb27-38" aria-hidden="true" tabindex="-1"></a>                                                   <span class="at">context=</span><span class="st">'context'</span>,</span>
<span id="cb27-39"><a href="#cb27-39" aria-hidden="true" tabindex="-1"></a>                                                   <span class="at">resolution=</span><span class="st">"base"</span>)</span>
<span id="cb27-40"><a href="#cb27-40" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb27-41"><a href="#cb27-41" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(D3A_BSseq_methylRaw_repMerged.list,</span>
<span id="cb27-42"><a href="#cb27-42" aria-hidden="true" tabindex="-1"></a>        <span class="at">file =</span> <span class="st">"results/RData/GSE103214_BSseq_Cortex_10wk_repMerged_mm10.RDS"</span>)</span>
<span id="cb27-43"><a href="#cb27-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-44"><a href="#cb27-44" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>( i <span class="cf">in</span> <span class="fu">seq_along</span>(D3A_BSseq_methylRaw_repMerged)){</span>
<span id="cb27-45"><a href="#cb27-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write.table</span>(D3A_BSseq_methylRaw_repMerged[[i]],</span>
<span id="cb27-46"><a href="#cb27-46" aria-hidden="true" tabindex="-1"></a>              <span class="at">file =</span> <span class="fu">paste0</span>(<span class="st">'data/GSE103214/'</span>,</span>
<span id="cb27-47"><a href="#cb27-47" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">names</span>(D3A_BSseq_methylRaw_repMerged)[i],<span class="st">'_repMerged_mm10.txt'</span>))</span>
<span id="cb27-48"><a href="#cb27-48" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb27-49"><a href="#cb27-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-50"><a href="#cb27-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-51"><a href="#cb27-51" aria-hidden="true" tabindex="-1"></a><span class="co"># Classify MECP2 peaks based on decreased mCG after DNMT3A KO</span></span>
<span id="cb27-52"><a href="#cb27-52" aria-hidden="true" tabindex="-1"></a><span class="co"># Read MECP2 final peaks</span></span>
<span id="cb27-53"><a href="#cb27-53" aria-hidden="true" tabindex="-1"></a>MECP2_peaks <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">"results/MECP2/MECP2WT.vs.KO_overlap_MECP2WT.vs.IgG.noBlacklist_peakID_filtered.bed"</span>,<span class="at">sep=</span><span class="st">'</span><span class="sc">\t</span><span class="st">'</span>,<span class="at">header =</span> F) </span>
<span id="cb27-54"><a href="#cb27-54" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(MECP2_peaks) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">'Chr'</span>,<span class="st">'Start'</span>,<span class="st">'End'</span>,<span class="st">'PeakID'</span>)</span>
<span id="cb27-55"><a href="#cb27-55" aria-hidden="true" tabindex="-1"></a>MECP2_peaks.GR <span class="ot">=</span> MECP2_peaks <span class="sc">|&gt;</span> <span class="fu">GRanges</span>() </span>
<span id="cb27-56"><a href="#cb27-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-57"><a href="#cb27-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-58"><a href="#cb27-58" aria-hidden="true" tabindex="-1"></a>MECP2_mCG_D3A <span class="ot">=</span> <span class="fu">list</span>()</span>
<span id="cb27-59"><a href="#cb27-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-60"><a href="#cb27-60" aria-hidden="true" tabindex="-1"></a>MECP2_mCG_D3A <span class="ot">=</span> <span class="fu">lapply</span>(D3A_BSseq_methylRaw_repMerged.list,</span>
<span id="cb27-61"><a href="#cb27-61" aria-hidden="true" tabindex="-1"></a>                       <span class="cf">function</span>(x) <span class="fu">regionCounts</span>(x,MECP2_peaks.GR))</span>
<span id="cb27-62"><a href="#cb27-62" aria-hidden="true" tabindex="-1"></a>MECP2_mCG_D3A.df <span class="ot">=</span> <span class="fu">lapply</span>(MECP2_mCG_D3A,getData)</span>
<span id="cb27-63"><a href="#cb27-63" aria-hidden="true" tabindex="-1"></a>MECP2_mCG_D3A.df <span class="ot">=</span> <span class="fu">lapply</span>(MECP2_mCG_D3A.df, </span>
<span id="cb27-64"><a href="#cb27-64" aria-hidden="true" tabindex="-1"></a>                          <span class="cf">function</span>(x) { x <span class="sc">%&gt;%</span> </span>
<span id="cb27-65"><a href="#cb27-65" aria-hidden="true" tabindex="-1"></a>                                        <span class="fu">mutate</span>(<span class="at">meth=</span>numCs<span class="sc">/</span>coverage,</span>
<span id="cb27-66"><a href="#cb27-66" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">ID =</span> <span class="fu">paste0</span>(chr,<span class="st">'_'</span>,start,<span class="st">'_'</span>,end))})</span>
<span id="cb27-67"><a href="#cb27-67" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="fu">seq_along</span>(MECP2_mCG_D3A.df)){</span>
<span id="cb27-68"><a href="#cb27-68" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(MECP2_mCG_D3A.df[[i]])[<span class="dv">8</span>] <span class="ot">=</span> <span class="fu">names</span>(MECP2_mCG_D3A.df)[i]</span>
<span id="cb27-69"><a href="#cb27-69" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb27-70"><a href="#cb27-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-71"><a href="#cb27-71" aria-hidden="true" tabindex="-1"></a><span class="fu">merge</span>(MECP2_mCG_D3A.df[[<span class="dv">2</span>]],MECP2_mCG_D3A.df[[<span class="dv">4</span>]],<span class="at">by=</span><span class="st">'ID'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb27-72"><a href="#cb27-72" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">1</span>,<span class="dv">9</span>,<span class="dv">17</span>) <span class="sc">%&gt;%</span> <span class="fu">melt</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb27-73"><a href="#cb27-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">variable =</span> <span class="fu">factor</span>(variable, </span>
<span id="cb27-74"><a href="#cb27-74" aria-hidden="true" tabindex="-1"></a>                           <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">'DNMT3a_WT'</span>,<span class="st">'DNMT3a_cKO'</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb27-75"><a href="#cb27-75" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>variable,<span class="at">y=</span>value))<span class="sc">+</span></span>
<span id="cb27-76"><a href="#cb27-76" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>()<span class="sc">+</span></span>
<span id="cb27-77"><a href="#cb27-77" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>(<span class="at">comparisons =</span> <span class="fu">list</span>(<span class="fu">c</span>(<span class="st">'DNMT3a_WT'</span>,<span class="st">'DNMT3a_cKO'</span>)),</span>
<span id="cb27-78"><a href="#cb27-78" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">aes</span>(<span class="at">label =</span> ..p.signif..))<span class="sc">+</span></span>
<span id="cb27-79"><a href="#cb27-79" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_test</span>(<span class="dv">20</span>)<span class="sc">+</span></span>
<span id="cb27-80"><a href="#cb27-80" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"mCG/CG (MBHs)"</span>)<span class="sc">+</span></span>
<span id="cb27-81"><a href="#cb27-81" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>,<span class="at">hjust=</span><span class="dv">1</span>,<span class="at">vjust=</span><span class="dv">1</span>))</span>
<span id="cb27-82"><a href="#cb27-82" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">'Plots/mCG_on_MBHs_D3AWT_KO.pdf'</span>,<span class="at">height =</span> <span class="dv">6</span>,<span class="at">width =</span> <span class="dv">3</span>)</span>
<span id="cb27-83"><a href="#cb27-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-84"><a href="#cb27-84" aria-hidden="true" tabindex="-1"></a><span class="fu">merge</span>(MECP2_mCG_D3A.df[[<span class="dv">1</span>]],MECP2_mCG_D3A.df[[<span class="dv">3</span>]],<span class="at">by=</span><span class="st">'ID'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb27-85"><a href="#cb27-85" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">1</span>,<span class="dv">9</span>,<span class="dv">17</span>) <span class="sc">%&gt;%</span> <span class="fu">melt</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb27-86"><a href="#cb27-86" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">variable =</span> <span class="fu">factor</span>(variable, </span>
<span id="cb27-87"><a href="#cb27-87" aria-hidden="true" tabindex="-1"></a>                           <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">'DNMT3a_WT'</span>,<span class="st">'DNMT3a_cKO'</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb27-88"><a href="#cb27-88" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>variable,<span class="at">y=</span>value))<span class="sc">+</span></span>
<span id="cb27-89"><a href="#cb27-89" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>()<span class="sc">+</span></span>
<span id="cb27-90"><a href="#cb27-90" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>(<span class="at">comparisons =</span> <span class="fu">list</span>(<span class="fu">c</span>(<span class="st">'DNMT3a_WT'</span>,<span class="st">'DNMT3a_cKO'</span>)),</span>
<span id="cb27-91"><a href="#cb27-91" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">aes</span>(<span class="at">label =</span> ..p.signif..))<span class="sc">+</span></span>
<span id="cb27-92"><a href="#cb27-92" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_test</span>(<span class="dv">20</span>)<span class="sc">+</span></span>
<span id="cb27-93"><a href="#cb27-93" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"mCA/CA (MBHs)"</span>)<span class="sc">+</span></span>
<span id="cb27-94"><a href="#cb27-94" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>,<span class="at">hjust=</span><span class="dv">1</span>,<span class="at">vjust=</span><span class="dv">1</span>))</span>
<span id="cb27-95"><a href="#cb27-95" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">'Plots/mCA_on_MBHs_D3AWT_KO.pdf'</span>,<span class="at">height =</span> <span class="dv">6</span>,<span class="at">width =</span> <span class="dv">3</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></li>
<li><p>Supplementary Figure 2D.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Differential MECP2 binding on MBHS in DNMT3a cKO vs DNMT3a WT MEPC2 CUT&amp;RUN </span></span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Read CNR peaks summit files</span></span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a>MECP2_CNR.noBlacklist.df <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">'results/MECP2/MECP2WT.vs.KO_overlap_MECP2WT.vs.IgG.noBlacklist_filtered.bed'</span>, <span class="at">sep=</span><span class="st">'</span><span class="sc">\t</span><span class="st">'</span>,<span class="at">header =</span> F)</span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(MECP2_CNR.noBlacklist.df)</span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(MECP2_CNR.noBlacklist.df) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">'Chr'</span>,<span class="st">'Start'</span>,<span class="st">'End'</span>,<span class="st">'PeakID'</span>)</span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert to GRanges</span></span>
<span id="cb28-8"><a href="#cb28-8" aria-hidden="true" tabindex="-1"></a>MECP2_CNR.noBlacklist.GR <span class="ot">=</span> MECP2_CNR.noBlacklist.df <span class="sc">%&gt;%</span> <span class="fu">GRanges</span>()</span>
<span id="cb28-9"><a href="#cb28-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-10"><a href="#cb28-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Create SAF format</span></span>
<span id="cb28-11"><a href="#cb28-11" aria-hidden="true" tabindex="-1"></a>MECP2_CNR.noBlacklist.df.saf <span class="ot">=</span> MECP2_CNR.noBlacklist.df <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">c</span>(<span class="dv">4</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>))</span>
<span id="cb28-12"><a href="#cb28-12" aria-hidden="true" tabindex="-1"></a>MECP2_CNR.noBlacklist.df.saf<span class="sc">$</span>Strand <span class="ot">=</span> <span class="st">'*'</span></span>
<span id="cb28-13"><a href="#cb28-13" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(MECP2_CNR.noBlacklist.df.saf) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"GeneID"</span>,<span class="st">"Chr"</span>,<span class="st">"Start"</span>,<span class="st">"End"</span>,<span class="st">"Strand"</span>)</span>
<span id="cb28-14"><a href="#cb28-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-15"><a href="#cb28-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-16"><a href="#cb28-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Read CUT&amp;RUN samples bam files</span></span>
<span id="cb28-17"><a href="#cb28-17" aria-hidden="true" tabindex="-1"></a>bams <span class="ot">=</span> <span class="fu">dir</span>(<span class="st">"../CUT_and_RUN/results/bam/"</span>, </span>
<span id="cb28-18"><a href="#cb28-18" aria-hidden="true" tabindex="-1"></a>            <span class="at">full.names =</span> <span class="cn">TRUE</span>, <span class="at">pattern =</span> <span class="st">"*[0-9].final.bam$"</span>)</span>
<span id="cb28-19"><a href="#cb28-19" aria-hidden="true" tabindex="-1"></a>bams <span class="ot">=</span>  bams[<span class="sc">!</span>(<span class="fu">grepl</span>(<span class="st">'IgG'</span>,bams))]</span>
<span id="cb28-20"><a href="#cb28-20" aria-hidden="true" tabindex="-1"></a>MECP2.bams <span class="ot">=</span> bams[<span class="fu">grepl</span>(<span class="st">'_Mecp2'</span>,bams)]</span>
<span id="cb28-21"><a href="#cb28-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-22"><a href="#cb28-22" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract raw reads mapped to MECP2 peaks</span></span>
<span id="cb28-23"><a href="#cb28-23" aria-hidden="true" tabindex="-1"></a>MECP2.fcResults <span class="ot">=</span> <span class="fu">featureCounts</span>(MECP2.bams, </span>
<span id="cb28-24"><a href="#cb28-24" aria-hidden="true" tabindex="-1"></a>                           <span class="at">annot.ext =</span> MECP2_CNR.noBlacklist.df.saf,</span>
<span id="cb28-25"><a href="#cb28-25" aria-hidden="true" tabindex="-1"></a>                           <span class="at">isPairedEnd =</span> <span class="cn">TRUE</span>,</span>
<span id="cb28-26"><a href="#cb28-26" aria-hidden="true" tabindex="-1"></a>                           <span class="at">countMultiMappingReads =</span> <span class="cn">FALSE</span>,</span>
<span id="cb28-27"><a href="#cb28-27" aria-hidden="true" tabindex="-1"></a>                           <span class="at">maxFragLength =</span> <span class="dv">100</span>,<span class="at">nthreads =</span> <span class="dv">15</span>)</span>
<span id="cb28-28"><a href="#cb28-28" aria-hidden="true" tabindex="-1"></a>myCounts <span class="ot">=</span> MECP2.fcResults<span class="sc">$</span>counts <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() </span>
<span id="cb28-29"><a href="#cb28-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-30"><a href="#cb28-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-31"><a href="#cb28-31" aria-hidden="true" tabindex="-1"></a><span class="co"># prepeare Metadata of KO and WT samples</span></span>
<span id="cb28-32"><a href="#cb28-32" aria-hidden="true" tabindex="-1"></a>metaData <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">Group=</span><span class="fu">colnames</span>(myCounts), <span class="at">row.names =</span> <span class="fu">colnames</span>(myCounts))</span>
<span id="cb28-33"><a href="#cb28-33" aria-hidden="true" tabindex="-1"></a>metaData<span class="sc">$</span>Group <span class="ot">=</span> <span class="fu">c</span>(<span class="st">'MECP2_KO'</span>,<span class="st">'WT'</span>,<span class="st">'DNMT3A_KO'</span>,<span class="st">'WT'</span>,<span class="st">'MECP2_KO'</span>,<span class="st">'WT'</span>,<span class="st">'DNMT3A_KO'</span>,<span class="st">'WT'</span>)</span>
<span id="cb28-34"><a href="#cb28-34" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(metaData) <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">'_CNR.*'</span>,<span class="st">''</span>,<span class="fu">rownames</span>(metaData))</span>
<span id="cb28-35"><a href="#cb28-35" aria-hidden="true" tabindex="-1"></a>metaData<span class="sc">$</span>Sample <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">'_Mecp2'</span>,<span class="st">''</span>,<span class="fu">rownames</span>(metaData))</span>
<span id="cb28-36"><a href="#cb28-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-37"><a href="#cb28-37" aria-hidden="true" tabindex="-1"></a><span class="co"># change column names of count data based on metadta rownames</span></span>
<span id="cb28-38"><a href="#cb28-38" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(myCounts) <span class="ot">=</span> <span class="fu">rownames</span>(metaData)</span>
<span id="cb28-39"><a href="#cb28-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-40"><a href="#cb28-40" aria-hidden="true" tabindex="-1"></a><span class="co"># Run DESEq2</span></span>
<span id="cb28-41"><a href="#cb28-41" aria-hidden="true" tabindex="-1"></a>MECP2_dds <span class="ot">=</span> <span class="fu">DESeqDataSetFromMatrix</span>(myCounts[,<span class="fu">c</span>(<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">7</span>,<span class="dv">8</span>)], </span>
<span id="cb28-42"><a href="#cb28-42" aria-hidden="true" tabindex="-1"></a>                                    metaData[<span class="fu">c</span>(<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">7</span>,<span class="dv">8</span>),], <span class="sc">~</span>Group)</span>
<span id="cb28-43"><a href="#cb28-43" aria-hidden="true" tabindex="-1"></a>MECP2_dds <span class="ot">=</span> <span class="fu">DESeq</span>(MECP2_dds)</span>
<span id="cb28-44"><a href="#cb28-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-45"><a href="#cb28-45" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract variance stabilized transformed values</span></span>
<span id="cb28-46"><a href="#cb28-46" aria-hidden="true" tabindex="-1"></a>MECP2_vst <span class="ot">=</span> <span class="fu">vst</span>(MECP2_dds)</span>
<span id="cb28-47"><a href="#cb28-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-48"><a href="#cb28-48" aria-hidden="true" tabindex="-1"></a><span class="co"># PCA plot</span></span>
<span id="cb28-49"><a href="#cb28-49" aria-hidden="true" tabindex="-1"></a>MECP2.PCA <span class="ot">=</span> <span class="fu">plotPCA</span>(MECP2_vst,<span class="at">intgroup=</span><span class="fu">c</span>(<span class="st">'Group'</span>,<span class="st">'Sample'</span>),<span class="at">returnData=</span><span class="cn">TRUE</span>)</span>
<span id="cb28-50"><a href="#cb28-50" aria-hidden="true" tabindex="-1"></a>percentVar <span class="ot">=</span> <span class="fu">round</span>(<span class="dv">100</span> <span class="sc">*</span> <span class="fu">attr</span>(MECP2.PCA, <span class="st">"percentVar"</span>))</span>
<span id="cb28-51"><a href="#cb28-51" aria-hidden="true" tabindex="-1"></a>MECP2.PCA<span class="sc">$</span>Sample[<span class="dv">4</span>] <span class="ot">=</span> <span class="st">'Adult_CTX_DNMT3AWT'</span></span>
<span id="cb28-52"><a href="#cb28-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-53"><a href="#cb28-53" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(MECP2.PCA, <span class="fu">aes</span>(PC1, PC2, <span class="at">color=</span>Group,<span class="at">shape=</span>Sample)) <span class="sc">+</span> </span>
<span id="cb28-54"><a href="#cb28-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">6</span>,<span class="at">stroke =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb28-55"><a href="#cb28-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_shape_manual</span>(<span class="at">values=</span><span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>)<span class="sc">+</span></span>
<span id="cb28-56"><a href="#cb28-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="fu">paste0</span>(<span class="st">"PC1: "</span>,percentVar[<span class="dv">1</span>],<span class="st">"% variance"</span>)) <span class="sc">+</span></span>
<span id="cb28-57"><a href="#cb28-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="fu">paste0</span>(<span class="st">"PC2: "</span>,percentVar[<span class="dv">2</span>],<span class="st">"% variance"</span>)) <span class="sc">+</span></span>
<span id="cb28-58"><a href="#cb28-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_test</span>(<span class="dv">20</span>)</span>
<span id="cb28-59"><a href="#cb28-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-60"><a href="#cb28-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-61"><a href="#cb28-61" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract differential MECP2 peaks</span></span>
<span id="cb28-62"><a href="#cb28-62" aria-hidden="true" tabindex="-1"></a>DNMT3AKOvsWT <span class="ot">=</span> <span class="fu">results</span>(MECP2_dds,<span class="at">contrast=</span><span class="fu">c</span>(<span class="st">"Group"</span>,<span class="st">"DNMT3A_KO"</span>,<span class="st">"WT"</span>))</span>
<span id="cb28-63"><a href="#cb28-63" aria-hidden="true" tabindex="-1"></a>DNMT3AKOvsWT.diff.peaks <span class="ot">=</span> DNMT3AKOvsWT <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>()</span>
<span id="cb28-64"><a href="#cb28-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-65"><a href="#cb28-65" aria-hidden="true" tabindex="-1"></a>DNMT3AKOvsWT.diff.peaks  <span class="ot">=</span> DNMT3AKOvsWT.diff.peaks <span class="sc">%&gt;%</span> </span>
<span id="cb28-66"><a href="#cb28-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">DEG =</span> <span class="fu">case_when</span>(log2FoldChange <span class="sc">&gt;</span> <span class="dv">0</span> <span class="sc">&amp;</span> pvalue <span class="sc">&lt;</span> <span class="fl">0.1</span> <span class="sc">~</span> <span class="st">'Up'</span>,</span>
<span id="cb28-67"><a href="#cb28-67" aria-hidden="true" tabindex="-1"></a>                         log2FoldChange <span class="sc">&lt;</span> <span class="dv">0</span> <span class="sc">&amp;</span> pvalue <span class="sc">&lt;</span> <span class="fl">0.1</span>  <span class="sc">~</span> <span class="st">'Down'</span>,</span>
<span id="cb28-68"><a href="#cb28-68" aria-hidden="true" tabindex="-1"></a>                         pvalue <span class="sc">&gt;=</span><span class="fl">0.1</span> <span class="sc">|</span> <span class="fu">is.na</span>(pvalue)  <span class="sc">~</span><span class="st">'No Change'</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb28-69"><a href="#cb28-69" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">DEG =</span> <span class="fu">factor</span>(DEG,<span class="at">levels=</span><span class="fu">c</span>(<span class="st">'Up'</span>,<span class="st">'No Change'</span>,<span class="st">'Down'</span>)))</span>
<span id="cb28-70"><a href="#cb28-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-71"><a href="#cb28-71" aria-hidden="true" tabindex="-1"></a><span class="co"># plot MA plot</span></span>
<span id="cb28-72"><a href="#cb28-72" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(DNMT3AKOvsWT.diff.peaks)<span class="sc">+</span></span>
<span id="cb28-73"><a href="#cb28-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">y=</span>log2FoldChange,<span class="at">x=</span><span class="fu">log</span>(baseMean),<span class="at">color=</span>DEG),<span class="at">size=</span><span class="dv">0</span>)<span class="sc">+</span></span>
<span id="cb28-74"><a href="#cb28-74" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="st">"Differential MECP2 peaks</span><span class="sc">\n</span><span class="st"> (DNMT3A KO vs WT)"</span>,</span>
<span id="cb28-75"><a href="#cb28-75" aria-hidden="true" tabindex="-1"></a>                     <span class="at">values=</span><span class="fu">c</span>(<span class="st">'blue'</span>,<span class="st">'grey'</span>,<span class="st">'red'</span>))<span class="sc">+</span></span>
<span id="cb28-76"><a href="#cb28-76" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_test</span>(<span class="dv">15</span>)<span class="sc">+</span></span>
<span id="cb28-77"><a href="#cb28-77" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">colour =</span> <span class="fu">guide_legend</span>(<span class="at">override.aes =</span> <span class="fu">list</span>(<span class="at">size=</span><span class="dv">2</span>)))<span class="sc">+</span></span>
<span id="cb28-78"><a href="#cb28-78" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>,<span class="at">linetype=</span><span class="st">'dashed'</span>)<span class="sc">+</span></span>
<span id="cb28-79"><a href="#cb28-79" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylim</span>(<span class="sc">-</span><span class="dv">6</span>,<span class="dv">2</span>)<span class="sc">+</span></span>
<span id="cb28-80"><a href="#cb28-80" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="fl">5.7</span>, <span class="at">y =</span> <span class="sc">-</span><span class="dv">6</span>, </span>
<span id="cb28-81"><a href="#cb28-81" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> DNMT3AKOvsWT.diff.peaks <span class="sc">%&gt;%</span> </span>
<span id="cb28-82"><a href="#cb28-82" aria-hidden="true" tabindex="-1"></a>             dplyr<span class="sc">::</span><span class="fu">filter</span>(DEG<span class="sc">==</span><span class="st">'Down'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb28-83"><a href="#cb28-83" aria-hidden="true" tabindex="-1"></a>             <span class="fu">dim</span>() <span class="sc">%&gt;%</span> .[<span class="dv">1</span>], </span>
<span id="cb28-84"><a href="#cb28-84" aria-hidden="true" tabindex="-1"></a>           <span class="at">color=</span><span class="st">"red"</span>,<span class="at">size=</span><span class="dv">4</span>)<span class="sc">+</span></span>
<span id="cb28-85"><a href="#cb28-85" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="fl">5.7</span>, <span class="at">y =</span> <span class="dv">2</span>,  </span>
<span id="cb28-86"><a href="#cb28-86" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> DNMT3AKOvsWT.diff.peaks <span class="sc">%&gt;%</span> </span>
<span id="cb28-87"><a href="#cb28-87" aria-hidden="true" tabindex="-1"></a>             dplyr<span class="sc">::</span><span class="fu">filter</span>(DEG<span class="sc">==</span><span class="st">'Up'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb28-88"><a href="#cb28-88" aria-hidden="true" tabindex="-1"></a>             <span class="fu">dim</span>() <span class="sc">%&gt;%</span> .[<span class="dv">1</span>], </span>
<span id="cb28-89"><a href="#cb28-89" aria-hidden="true" tabindex="-1"></a>           <span class="at">color=</span><span class="st">"blue"</span>,<span class="at">size=</span><span class="dv">4</span>)</span>
<span id="cb28-90"><a href="#cb28-90" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">filename =</span> <span class="st">'Plots/MECP2_in_DNMT3AKO_vs_WT_volcanoPlot.pdf'</span>,<span class="at">height =</span> <span class="dv">4</span>, <span class="at">width =</span> <span class="dv">7</span>,<span class="at">units =</span> <span class="st">'in'</span>)</span>
<span id="cb28-91"><a href="#cb28-91" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-92"><a href="#cb28-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-93"><a href="#cb28-93" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-94"><a href="#cb28-94" aria-hidden="true" tabindex="-1"></a>DNMT3AKOvsWT.diff.up.peaks <span class="ot">=</span> DNMT3AKOvsWT.diff.peaks <span class="sc">%&gt;%</span> <span class="fu">filter</span>(DEG <span class="sc">==</span> <span class="st">'Up'</span>)</span>
<span id="cb28-95"><a href="#cb28-95" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-96"><a href="#cb28-96" aria-hidden="true" tabindex="-1"></a>DNMT3AKOvsWT.diff.down.peaks <span class="ot">=</span> DNMT3AKOvsWT.diff.peaks <span class="sc">%&gt;%</span> <span class="fu">filter</span>(DEG <span class="sc">==</span> <span class="st">'Down'</span>)</span>
<span id="cb28-97"><a href="#cb28-97" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-98"><a href="#cb28-98" aria-hidden="true" tabindex="-1"></a>MECP2_CNR.noBlacklist.df.DNMT3AKOvsWT.unchanged.df <span class="ot">=</span> MECP2_CNR.noBlacklist.df <span class="sc">%&gt;%</span> </span>
<span id="cb28-99"><a href="#cb28-99" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span>(PeakID <span class="sc">%in%</span> <span class="fu">rownames</span>(DNMT3AKOvsWT.diff.down.peaks ))) <span class="sc">%&gt;%</span> </span>
<span id="cb28-100"><a href="#cb28-100" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span>(PeakID <span class="sc">%in%</span> <span class="fu">rownames</span>(DNMT3AKOvsWT.diff.up.peaks )))</span>
<span id="cb28-101"><a href="#cb28-101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-102"><a href="#cb28-102" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-103"><a href="#cb28-103" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-104"><a href="#cb28-104" aria-hidden="true" tabindex="-1"></a>MECP2_CNR.noBlacklist.df.DNMT3AKOvsWT.up.df <span class="ot">=</span> MECP2_CNR.noBlacklist.df <span class="sc">%&gt;%</span> </span>
<span id="cb28-105"><a href="#cb28-105" aria-hidden="true" tabindex="-1"></a>                                                    dplyr<span class="sc">::</span><span class="fu">filter</span>(PeakID <span class="sc">%in%</span> </span>
<span id="cb28-106"><a href="#cb28-106" aria-hidden="true" tabindex="-1"></a>                                                                  <span class="fu">rownames</span>(DNMT3AKOvsWT.diff.up.peaks)) </span>
<span id="cb28-107"><a href="#cb28-107" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(MECP2_CNR.noBlacklist.df.DNMT3AKOvsWT.up.df, </span>
<span id="cb28-108"><a href="#cb28-108" aria-hidden="true" tabindex="-1"></a>            <span class="at">file =</span> <span class="st">"results/MECP2/MECP2WT.vs.KO_overlap_MECP2WT.vs.IgG.noBlacklist_filtered_DNMT3AKO_vs_WT.up.bed"</span>,</span>
<span id="cb28-109"><a href="#cb28-109" aria-hidden="true" tabindex="-1"></a>            <span class="at">sep=</span><span class="st">'</span><span class="sc">\t</span><span class="st">'</span>,<span class="at">quote =</span> <span class="cn">FALSE</span>,<span class="at">row.names =</span> F,<span class="at">col.names =</span> F)</span>
<span id="cb28-110"><a href="#cb28-110" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-111"><a href="#cb28-111" aria-hidden="true" tabindex="-1"></a>MECP2_CNR.noBlacklist.df.DNMT3AKOvsWT.down.df <span class="ot">=</span> MECP2_CNR.noBlacklist.df <span class="sc">%&gt;%</span> </span>
<span id="cb28-112"><a href="#cb28-112" aria-hidden="true" tabindex="-1"></a>                                                    dplyr<span class="sc">::</span><span class="fu">filter</span>(PeakID <span class="sc">%in%</span> </span>
<span id="cb28-113"><a href="#cb28-113" aria-hidden="true" tabindex="-1"></a>                                                                  <span class="fu">rownames</span>(DNMT3AKOvsWT.diff.down.peaks)) </span>
<span id="cb28-114"><a href="#cb28-114" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(MECP2_CNR.noBlacklist.df.DNMT3AKOvsWT.down.df, </span>
<span id="cb28-115"><a href="#cb28-115" aria-hidden="true" tabindex="-1"></a>            <span class="at">file =</span> <span class="st">"results/MECP2/MECP2WT.vs.KO_overlap_MECP2WT.vs.IgG.noBlacklist_filtered_DNMT3AKO_vs_WT.down.bed"</span>,</span>
<span id="cb28-116"><a href="#cb28-116" aria-hidden="true" tabindex="-1"></a>            <span class="at">sep=</span><span class="st">'</span><span class="sc">\t</span><span class="st">'</span>,<span class="at">quote =</span> <span class="cn">FALSE</span>,<span class="at">row.names =</span> F,<span class="at">col.names =</span> F)</span>
<span id="cb28-117"><a href="#cb28-117" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-118"><a href="#cb28-118" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-119"><a href="#cb28-119" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(MECP2_CNR.noBlacklist.df.DNMT3AKOvsWT.unchanged.df, </span>
<span id="cb28-120"><a href="#cb28-120" aria-hidden="true" tabindex="-1"></a>            <span class="at">file =</span> <span class="st">"results/MECP2/MECP2WT.vs.KO_overlap_MECP2WT.vs.IgG.noBlacklist_filtered_DNMT3AKO_vs_WT.unchanged.bed"</span>,</span>
<span id="cb28-121"><a href="#cb28-121" aria-hidden="true" tabindex="-1"></a>            <span class="at">sep=</span><span class="st">'</span><span class="sc">\t</span><span class="st">'</span>,<span class="at">quote =</span> <span class="cn">FALSE</span>,<span class="at">row.names =</span> F,<span class="at">col.names =</span> F)</span>
<span id="cb28-122"><a href="#cb28-122" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-123"><a href="#cb28-123" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-124"><a href="#cb28-124" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(MECP2_CNR.noBlacklist.df,</span>
<span id="cb28-125"><a href="#cb28-125" aria-hidden="true" tabindex="-1"></a>     MECP2_CNR.noBlacklist.GR,</span>
<span id="cb28-126"><a href="#cb28-126" aria-hidden="true" tabindex="-1"></a>     MECP2_CNR.noBlacklist.df.saf,</span>
<span id="cb28-127"><a href="#cb28-127" aria-hidden="true" tabindex="-1"></a>     MECP2_dds,</span>
<span id="cb28-128"><a href="#cb28-128" aria-hidden="true" tabindex="-1"></a>     MECP2_vst,</span>
<span id="cb28-129"><a href="#cb28-129" aria-hidden="true" tabindex="-1"></a>     WTvsMECP2KO,</span>
<span id="cb28-130"><a href="#cb28-130" aria-hidden="true" tabindex="-1"></a>     DNMT3AKOvsWT,</span>
<span id="cb28-131"><a href="#cb28-131" aria-hidden="true" tabindex="-1"></a>     fcResults,<span class="at">file =</span> <span class="st">"results/RData/MECP2_D3A_KOvsWT.RData"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></li>
<li><p>Supplementary Figure 2I.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Read MECP2 peaks overlapping with gain and loss in DNMT3A KO final peaks</span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>MEP2_peakList <span class="ot">=</span> <span class="fu">list.files</span>(<span class="st">"results/MECP2/"</span>,<span class="at">pattern =</span> <span class="st">"MECP2_u.*.noBlacklist.bed"</span>,<span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>MEP2_peakListNames <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">'.noBlacklist.bed'</span>,<span class="st">''</span>,<span class="fu">basename</span>(MEP2_peakList))</span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a>MEP2_peakList <span class="ot">=</span> <span class="fu">lapply</span>(MEP2_peakList, <span class="cf">function</span>(x) {<span class="fu">read.csv</span>(x,<span class="at">sep =</span> <span class="st">'</span><span class="sc">\t</span><span class="st">'</span>,<span class="at">header =</span> F)})</span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a>newColnames <span class="ot">=</span> <span class="fu">c</span>(<span class="st">'Chr'</span>,<span class="st">'Start'</span>,<span class="st">'End'</span>)</span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a>MEP2_peakList <span class="ot">=</span> <span class="fu">lapply</span>(MEP2_peakList, setNames, newColnames)</span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(MEP2_peakList) <span class="ot">=</span> MEP2_peakListNames</span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a>MEP2_peakList.GR <span class="ot">=</span> <span class="fu">lapply</span>(MEP2_peakList, <span class="cf">function</span>(x) {<span class="fu">GRanges</span>(x)})</span>
<span id="cb29-9"><a href="#cb29-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-10"><a href="#cb29-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-11"><a href="#cb29-11" aria-hidden="true" tabindex="-1"></a><span class="co"># mCA over Diff MECP2 peaks (DNMT3A KO vs WT)</span></span>
<span id="cb29-12"><a href="#cb29-12" aria-hidden="true" tabindex="-1"></a>MECP2_mCA_mCG_GC <span class="ot">=</span> <span class="fu">list</span>()</span>
<span id="cb29-13"><a href="#cb29-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-14"><a href="#cb29-14" aria-hidden="true" tabindex="-1"></a>MEP2_peakList.GR_CTX_CA_meth <span class="ot">=</span> <span class="fu">lapply</span>(MEP2_peakList.GR,<span class="cf">function</span>(x) <span class="fu">regionCounts</span>(CTX_CA,x))</span>
<span id="cb29-15"><a href="#cb29-15" aria-hidden="true" tabindex="-1"></a>MEP2_peakList.GR_CTX_CA_meth.df <span class="ot">=</span> <span class="fu">lapply</span>(MEP2_peakList.GR_CTX_CA_meth,getData)[<span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>)]</span>
<span id="cb29-16"><a href="#cb29-16" aria-hidden="true" tabindex="-1"></a>MEP2_peakList.GR_CTX_CA_meth.df <span class="ot">=</span> <span class="fu">lapply</span>(MEP2_peakList.GR_CTX_CA_meth.df, </span>
<span id="cb29-17"><a href="#cb29-17" aria-hidden="true" tabindex="-1"></a>                                         <span class="cf">function</span>(x) { x <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">meth=</span>numCs<span class="sc">/</span>coverage)})</span>
<span id="cb29-18"><a href="#cb29-18" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(MEP2_peakList.GR_CTX_CA_meth.df) <span class="ot">=</span> MEP2_peakListNames</span>
<span id="cb29-19"><a href="#cb29-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-20"><a href="#cb29-20" aria-hidden="true" tabindex="-1"></a>MECP2_mCA_mCG_GC[[<span class="dv">1</span>]] <span class="ot">=</span> <span class="fu">do.call</span>(rbind, MEP2_peakList.GR_CTX_CA_meth.df) <span class="sc">%&gt;%</span> </span>
<span id="cb29-21"><a href="#cb29-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">'Group'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb29-22"><a href="#cb29-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Group=</span><span class="fu">gsub</span>(<span class="st">'</span><span class="sc">\\</span><span class="st">.[0-9]*'</span>,<span class="st">''</span>,Group)) <span class="sc">%&gt;%</span> </span>
<span id="cb29-23"><a href="#cb29-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">MECP2 =</span> <span class="fu">factor</span>(<span class="fu">case_when</span>(</span>
<span id="cb29-24"><a href="#cb29-24" aria-hidden="true" tabindex="-1"></a>    Group <span class="sc">==</span> <span class="st">'MECP2_unchaged_DNMT3AKO_vs_WT'</span> <span class="sc">~</span> <span class="st">'No change'</span>,</span>
<span id="cb29-25"><a href="#cb29-25" aria-hidden="true" tabindex="-1"></a>    Group <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">'MECP2_up_DNMT3AKO_vs_WT'</span>,<span class="st">'MECP2_up_unique_DNMT3AKO_vs_WT'</span>)<span class="sc">~</span> <span class="st">'gain'</span>,</span>
<span id="cb29-26"><a href="#cb29-26" aria-hidden="true" tabindex="-1"></a>    Group <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">'MECP2_up_DNMT3AWT_vs_KO'</span>,<span class="st">'MECP2_up_unique_DNMT3AWT_vs_KO'</span>)<span class="sc">~</span><span class="st">'loss'</span>),</span>
<span id="cb29-27"><a href="#cb29-27" aria-hidden="true" tabindex="-1"></a>                        <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">'gain'</span>,<span class="st">'No change'</span>,<span class="st">'loss'</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb29-28"><a href="#cb29-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>MECP2,<span class="at">y=</span>meth))<span class="sc">+</span><span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb29-29"><a href="#cb29-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>(<span class="at">comparisons =</span> <span class="fu">list</span>(<span class="fu">c</span>(<span class="st">'No change'</span>,<span class="st">'gain'</span>),</span>
<span id="cb29-30"><a href="#cb29-30" aria-hidden="true" tabindex="-1"></a>                                        <span class="fu">c</span>(<span class="st">'gain'</span>,<span class="st">'loss'</span>),</span>
<span id="cb29-31"><a href="#cb29-31" aria-hidden="true" tabindex="-1"></a>                                        <span class="fu">c</span>(<span class="st">'No change'</span>,<span class="st">'loss'</span>))) <span class="sc">+</span></span>
<span id="cb29-32"><a href="#cb29-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">'mCA'</span>)<span class="sc">+</span></span>
<span id="cb29-33"><a href="#cb29-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">'mCA'</span>)<span class="sc">+</span></span>
<span id="cb29-34"><a href="#cb29-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">'MBHs (DNMT3A KO vs WT)'</span>)<span class="sc">+</span></span>
<span id="cb29-35"><a href="#cb29-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_test</span>(<span class="dv">20</span>)<span class="sc">+</span></span>
<span id="cb29-36"><a href="#cb29-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>,<span class="at">vjust =</span> <span class="dv">1</span>,<span class="at">hjust =</span> <span class="dv">1</span>))</span>
<span id="cb29-37"><a href="#cb29-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-38"><a href="#cb29-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-39"><a href="#cb29-39" aria-hidden="true" tabindex="-1"></a><span class="co"># mCG over Diff MECP2 peaks (DNMT3A KO vs WT)</span></span>
<span id="cb29-40"><a href="#cb29-40" aria-hidden="true" tabindex="-1"></a>MEP2_peakList.GR_CTX_CG_meth <span class="ot">=</span> <span class="fu">lapply</span>(MEP2_peakList.GR,<span class="cf">function</span>(x) <span class="fu">regionCounts</span>(CTX_CG,x))</span>
<span id="cb29-41"><a href="#cb29-41" aria-hidden="true" tabindex="-1"></a>MEP2_peakList.GR_CTX_CG_meth.df <span class="ot">=</span> <span class="fu">lapply</span>(MEP2_peakList.GR_CTX_CG_meth,getData)[<span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>)]</span>
<span id="cb29-42"><a href="#cb29-42" aria-hidden="true" tabindex="-1"></a>MEP2_peakList.GR_CTX_CG_meth.df <span class="ot">=</span> <span class="fu">lapply</span>(MEP2_peakList.GR_CTX_CG_meth.df, </span>
<span id="cb29-43"><a href="#cb29-43" aria-hidden="true" tabindex="-1"></a>                                         <span class="cf">function</span>(x) { x <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">meth=</span>numCs<span class="sc">/</span>coverage)})</span>
<span id="cb29-44"><a href="#cb29-44" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(MEP2_peakList.GR_CTX_CG_meth.df) <span class="ot">=</span> MEP2_peakListNames</span>
<span id="cb29-45"><a href="#cb29-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-46"><a href="#cb29-46" aria-hidden="true" tabindex="-1"></a>MECP2_mCA_mCG_GC[[<span class="dv">2</span>]] <span class="ot">=</span> <span class="fu">do.call</span>(rbind, MEP2_peakList.GR_CTX_CG_meth.df) <span class="sc">%&gt;%</span> </span>
<span id="cb29-47"><a href="#cb29-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">'Group'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb29-48"><a href="#cb29-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Group=</span><span class="fu">gsub</span>(<span class="st">'</span><span class="sc">\\</span><span class="st">.[0-9]*'</span>,<span class="st">''</span>,Group)) <span class="sc">%&gt;%</span> </span>
<span id="cb29-49"><a href="#cb29-49" aria-hidden="true" tabindex="-1"></a>  <span class="co">#mutate(MECP2 = ifelse(Group %in% names(MEP2_peakList.GR_CTX_CA_meth.df)[c(1,2,4)],'high','low')) %&gt;%</span></span>
<span id="cb29-50"><a href="#cb29-50" aria-hidden="true" tabindex="-1"></a>  <span class="co">#ggplot(aes(x=MECP2,y=meth))+geom_boxplot() +</span></span>
<span id="cb29-51"><a href="#cb29-51" aria-hidden="true" tabindex="-1"></a>  <span class="co">#stat_compare_means(comparisons = list(c('high','low')))+</span></span>
<span id="cb29-52"><a href="#cb29-52" aria-hidden="true" tabindex="-1"></a>  <span class="co">#scale_x_discrete(name = '',labels=c("Increased \nMECP2 binding\n(DNMT3A KO)","Decreased \nMECP2 binding\n(DNMT3A KO)"))+</span></span>
<span id="cb29-53"><a href="#cb29-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">MECP2 =</span> <span class="fu">factor</span>(<span class="fu">case_when</span>(Group <span class="sc">==</span> <span class="st">'MECP2_unchaged_DNMT3AKO_vs_WT'</span> <span class="sc">~</span> <span class="st">'No change'</span>,</span>
<span id="cb29-54"><a href="#cb29-54" aria-hidden="true" tabindex="-1"></a>      Group <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">'MECP2_up_DNMT3AKO_vs_WT'</span>,<span class="st">'MECP2_up_unique_DNMT3AKO_vs_WT'</span>)<span class="sc">~</span> <span class="st">'gain'</span>,</span>
<span id="cb29-55"><a href="#cb29-55" aria-hidden="true" tabindex="-1"></a>      Group <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">'MECP2_up_DNMT3AWT_vs_KO'</span>,<span class="st">'MECP2_up_unique_DNMT3AWT_vs_KO'</span>)<span class="sc">~</span><span class="st">'loss'</span>),</span>
<span id="cb29-56"><a href="#cb29-56" aria-hidden="true" tabindex="-1"></a>                        <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">'gain'</span>,<span class="st">'No change'</span>,<span class="st">'loss'</span>))) <span class="sc">%&gt;%</span> <span class="co">#head()</span></span>
<span id="cb29-57"><a href="#cb29-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>MECP2,<span class="at">y=</span>meth))<span class="sc">+</span><span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb29-58"><a href="#cb29-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>(<span class="at">comparisons =</span> <span class="fu">list</span>(<span class="fu">c</span>(<span class="st">'No change'</span>,<span class="st">'gain'</span>),</span>
<span id="cb29-59"><a href="#cb29-59" aria-hidden="true" tabindex="-1"></a>                                        <span class="fu">c</span>(<span class="st">'gain'</span>,<span class="st">'loss'</span>),</span>
<span id="cb29-60"><a href="#cb29-60" aria-hidden="true" tabindex="-1"></a>                                        <span class="fu">c</span>(<span class="st">'No change'</span>,<span class="st">'loss'</span>))) <span class="sc">+</span></span>
<span id="cb29-61"><a href="#cb29-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-62"><a href="#cb29-62" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">'mCG'</span>)<span class="sc">+</span></span>
<span id="cb29-63"><a href="#cb29-63" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">'mCG'</span>)<span class="sc">+</span></span>
<span id="cb29-64"><a href="#cb29-64" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">'MBHs (DNMT3A KO vs WT)'</span>)<span class="sc">+</span></span>
<span id="cb29-65"><a href="#cb29-65" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_test</span>(<span class="dv">20</span>)<span class="sc">+</span></span>
<span id="cb29-66"><a href="#cb29-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>,<span class="at">vjust =</span> <span class="dv">1</span>,<span class="at">hjust =</span> <span class="dv">1</span>))</span>
<span id="cb29-67"><a href="#cb29-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-68"><a href="#cb29-68" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">"Plots/mCA_CG_GC_on_MECP2_peaks.pdf"</span>,<span class="at">height =</span> <span class="dv">10</span>,<span class="at">width =</span> <span class="dv">12</span>)</span>
<span id="cb29-69"><a href="#cb29-69" aria-hidden="true" tabindex="-1"></a><span class="fu">do.call</span>(<span class="st">"grid.arrange"</span>,<span class="fu">c</span>(MECP2_mCA_mCG_GC,<span class="at">ncol=</span><span class="dv">2</span>))</span>
<span id="cb29-70"><a href="#cb29-70" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></li>
<li><p>Supplementary Figure 2I.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a>D3A_BSseq_methylRaw_repMerged.list <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">"results/RData/GSE103214_BSseq_Cortex_10wk_repMerged_mm10.RDS"</span>)</span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>MECP2_mCG_D3A <span class="ot">=</span> <span class="fu">list</span>()</span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a>MECP2_mCG_D3A <span class="ot">=</span> <span class="fu">lapply</span>(D3A_BSseq_methylRaw_repMerged.list,<span class="cf">function</span>(x) <span class="fu">regionCounts</span>(x,MECP2_peaks.GR))</span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a>MECP2_mCG_D3A.df <span class="ot">=</span> <span class="fu">lapply</span>(MECP2_mCG_D3A,getData)</span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a>MECP2_mCG_D3A.df <span class="ot">=</span> <span class="fu">lapply</span>(MECP2_mCG_D3A.df, <span class="cf">function</span>(x) { </span>
<span id="cb30-7"><a href="#cb30-7" aria-hidden="true" tabindex="-1"></a>  x <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">meth=</span>numCs<span class="sc">/</span>coverage,</span>
<span id="cb30-8"><a href="#cb30-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">ID =</span> <span class="fu">paste0</span>(chr,<span class="st">'_'</span>,start,<span class="st">'_'</span>,end))})</span>
<span id="cb30-9"><a href="#cb30-9" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="fu">seq_along</span>(MECP2_mCG_D3A.df)){</span>
<span id="cb30-10"><a href="#cb30-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(MECP2_mCG_D3A.df[[i]])[<span class="dv">8</span>] <span class="ot">=</span> <span class="fu">names</span>(MECP2_mCG_D3A.df)[i]</span>
<span id="cb30-11"><a href="#cb30-11" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb30-12"><a href="#cb30-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-13"><a href="#cb30-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-14"><a href="#cb30-14" aria-hidden="true" tabindex="-1"></a><span class="fu">merge</span>(MECP2_mCG_D3A.df[[<span class="dv">2</span>]],MECP2_mCG_D3A.df[[<span class="dv">4</span>]],<span class="at">by=</span><span class="st">'ID'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb30-15"><a href="#cb30-15" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">9</span>,<span class="dv">17</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb30-16"><a href="#cb30-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(GSM2757476_Dnmt3aWT_Cortex_BS_10wk_CG <span class="sc">&gt;</span><span class="fl">0.3</span> <span class="sc">&amp;</span> GSM2757478_Dnmt3acKO_Cortex_BS_10wk_CG <span class="sc">&lt;</span><span class="fl">0.1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb30-17"><a href="#cb30-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">merge</span>(.,MECP2_mCG_D3A.df[[<span class="dv">1</span>]],<span class="at">by=</span><span class="st">'ID'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb30-18"><a href="#cb30-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">merge</span>(.,MECP2_mCG_D3A.df[[<span class="dv">3</span>]],<span class="at">by=</span><span class="st">'ID'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb30-19"><a href="#cb30-19" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">1</span>,<span class="dv">5</span>,<span class="dv">6</span>,<span class="dv">14</span>,<span class="dv">22</span>) <span class="sc">%&gt;%</span>  <span class="fu">melt</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb30-20"><a href="#cb30-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">methylation =</span> <span class="fu">case_when</span>(<span class="fu">grepl</span>(<span class="st">'CG'</span>,variable)<span class="sc">~</span><span class="st">'mCG/CG'</span>,</span>
<span id="cb30-21"><a href="#cb30-21" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">grepl</span>(<span class="st">'CA'</span>,variable)<span class="sc">~</span><span class="st">'mCA/CA'</span>),</span>
<span id="cb30-22"><a href="#cb30-22" aria-hidden="true" tabindex="-1"></a>         <span class="at">genotype =</span> <span class="fu">case_when</span>(<span class="fu">grepl</span>(<span class="st">'Dnmt3aWT'</span>,variable)<span class="sc">~</span><span class="st">'DNMT3A_WT'</span>,</span>
<span id="cb30-23"><a href="#cb30-23" aria-hidden="true" tabindex="-1"></a>                              <span class="fu">grepl</span>(<span class="st">'Dnmt3acKO'</span>,variable)<span class="sc">~</span><span class="st">'DNMT3A_cKO'</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb30-24"><a href="#cb30-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">genotype =</span> <span class="fu">factor</span>(genotype, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">'DNMT3A_WT'</span>,<span class="st">'DNMT3A_cKO'</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb30-25"><a href="#cb30-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>genotype,<span class="at">y=</span>value))<span class="sc">+</span></span>
<span id="cb30-26"><a href="#cb30-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>()<span class="sc">+</span></span>
<span id="cb30-27"><a href="#cb30-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>(<span class="at">comparisons =</span> <span class="fu">list</span>(<span class="fu">c</span>(<span class="st">'DNMT3A_WT'</span>,<span class="st">'DNMT3A_cKO'</span>)),<span class="fu">aes</span>(<span class="at">label =</span> ..p.signif..))<span class="sc">+</span></span>
<span id="cb30-28"><a href="#cb30-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>methylation,<span class="at">scales =</span> <span class="st">'free_y'</span>)<span class="sc">+</span></span>
<span id="cb30-29"><a href="#cb30-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_test</span>(<span class="dv">20</span>)<span class="sc">+</span></span>
<span id="cb30-30"><a href="#cb30-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>,<span class="at">hjust=</span><span class="dv">1</span>,<span class="at">vjust=</span><span class="dv">1</span>))</span>
<span id="cb30-31"><a href="#cb30-31" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">'Plots/MECP2_peaks_with_diff_mCG_D3AKO_vs_WT.pdf'</span>,<span class="at">height =</span> <span class="dv">6</span>,<span class="at">width =</span> <span class="dv">6</span>)</span>
<span id="cb30-32"><a href="#cb30-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-33"><a href="#cb30-33" aria-hidden="true" tabindex="-1"></a><span class="fu">merge</span>(MECP2_mCG_D3A.df[[<span class="dv">2</span>]],MECP2_mCG_D3A.df[[<span class="dv">4</span>]],<span class="at">by=</span><span class="st">'ID'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb30-34"><a href="#cb30-34" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">9</span>,<span class="dv">17</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb30-35"><a href="#cb30-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(GSM2757476_Dnmt3aWT_Cortex_BS_10wk_CG <span class="sc">&gt;</span><span class="fl">0.3</span> <span class="sc">&amp;</span> GSM2757478_Dnmt3acKO_Cortex_BS_10wk_CG <span class="sc">&lt;</span><span class="fl">0.1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb30-36"><a href="#cb30-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write.table</span>(.[,<span class="fu">c</span>(<span class="dv">2</span><span class="sc">:</span><span class="dv">4</span>)],<span class="at">file =</span> <span class="st">'results/MECP2/MECP2_peaks_with_diff_mCG_D3AKO_vs_WT.bed'</span>,</span>
<span id="cb30-37"><a href="#cb30-37" aria-hidden="true" tabindex="-1"></a>              <span class="at">quote =</span> <span class="cn">FALSE</span>,<span class="at">sep =</span> <span class="st">'</span><span class="sc">\t</span><span class="st">'</span>,<span class="at">row.names =</span> <span class="cn">FALSE</span>,<span class="at">col.names =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></li>
</ol>
</section>
<section id="figure-3-mecp2-modulates-enhancer-activity-at-mbh-sites." class="level3">
<h3 class="anchored" data-anchor-id="figure-3-mecp2-modulates-enhancer-activity-at-mbh-sites."><strong>Figure 3: MECP2 modulates enhancer activity at MBH sites.</strong></h3>
<ol type="1">
<li><p>Supplementary Figure 3A.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="co"># read 8wk cortex encode data from UCSC</span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>encode_cortex <span class="ot">=</span> <span class="fu">list.files</span>(<span class="st">"data/ENCODE"</span>,<span class="at">pattern =</span> <span class="st">"*.broadPeak.bed.mm10.bed"</span>,<span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(encode_cortex) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"H3K27ac"</span>,<span class="st">"H3K4me1"</span>,<span class="st">"H3K4me3"</span>,<span class="st">"CTCF"</span>,<span class="st">"POLR2A"</span>)</span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a><span class="co"># read 8wk cortex encode data from Stroud et al</span></span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a>stroud_cortex <span class="ot">=</span> <span class="fu">list.files</span>(<span class="st">"data/Stroud_2017_Cell_data/"</span>,</span>
<span id="cb31-7"><a href="#cb31-7" aria-hidden="true" tabindex="-1"></a>                           <span class="at">pattern =</span> <span class="st">"*.mm10noChrM.bed$"</span>,<span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb31-8"><a href="#cb31-8" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(stroud_cortex) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"H3K27me3"</span>,<span class="st">"H3K9me3"</span>)</span>
<span id="cb31-9"><a href="#cb31-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-10"><a href="#cb31-10" aria-hidden="true" tabindex="-1"></a>cortex <span class="ot">=</span> <span class="fu">c</span>(encode_cortex,stroud_cortex)</span>
<span id="cb31-11"><a href="#cb31-11" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(cortex) <span class="ot">=</span> <span class="fu">paste0</span>(<span class="fu">names</span>(cortex), <span class="st">"_Adult.cortex"</span>)</span>
<span id="cb31-12"><a href="#cb31-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-13"><a href="#cb31-13" aria-hidden="true" tabindex="-1"></a>encode_forebrain <span class="ot">=</span> <span class="fu">list.files</span>(<span class="st">"data/ENCODE"</span>,<span class="at">pattern =</span> <span class="st">"*_brain.*.bed|*forebrain.*.bed"</span>,</span>
<span id="cb31-14"><a href="#cb31-14" aria-hidden="true" tabindex="-1"></a>                              <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb31-15"><a href="#cb31-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-16"><a href="#cb31-16" aria-hidden="true" tabindex="-1"></a>encode_histone_Metadta <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">"data/ENCODE/Histone_brain_mm10_ENCODE_experiment_report.tsv"</span>,</span>
<span id="cb31-17"><a href="#cb31-17" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>,<span class="at">skip =</span> <span class="dv">1</span>) </span>
<span id="cb31-18"><a href="#cb31-18" aria-hidden="true" tabindex="-1"></a>encode_TF_Metadta <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">"data/ENCODE/TFs_experiment_report_2024_4_10_17h_45m.tsv"</span>,</span>
<span id="cb31-19"><a href="#cb31-19" aria-hidden="true" tabindex="-1"></a>                             <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>,<span class="at">skip =</span> <span class="dv">1</span>) </span>
<span id="cb31-20"><a href="#cb31-20" aria-hidden="true" tabindex="-1"></a>encode_metadata <span class="ot">=</span> <span class="fu">rbind</span>(encode_TF_Metadta,encode_histone_Metadta)</span>
<span id="cb31-21"><a href="#cb31-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-22"><a href="#cb31-22" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(encode_forebrain)){</span>
<span id="cb31-23"><a href="#cb31-23" aria-hidden="true" tabindex="-1"></a>  id <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">'.*_ChIP-seq_.*brain_|.bed'</span>,<span class="st">''</span>,<span class="fu">basename</span>(encode_forebrain[i]))</span>
<span id="cb31-24"><a href="#cb31-24" aria-hidden="true" tabindex="-1"></a>  sampleName <span class="ot">=</span> encode_metadata <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">grepl</span>(id,Files)) <span class="sc">%&gt;%</span> </span>
<span id="cb31-25"><a href="#cb31-25" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="st">"Target.of.assay"</span>,<span class="st">"Life.stage"</span>,<span class="st">"Biosample.age"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb31-26"><a href="#cb31-26" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">group =</span> <span class="fu">paste0</span>(Target.of.assay,<span class="st">'_'</span>,Life.stage,<span class="st">'('</span>,Biosample.age,<span class="st">')'</span>)) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(group)</span>
<span id="cb31-27"><a href="#cb31-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(sampleName)</span>
<span id="cb31-28"><a href="#cb31-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(encode_forebrain)[i] <span class="ot">=</span> sampleName</span>
<span id="cb31-29"><a href="#cb31-29" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb31-30"><a href="#cb31-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-31"><a href="#cb31-31" aria-hidden="true" tabindex="-1"></a><span class="co"># read MECP2 peaks</span></span>
<span id="cb31-32"><a href="#cb31-32" aria-hidden="true" tabindex="-1"></a>mecp2  <span class="ot">=</span> <span class="st">"results/MECP2/MECP2WT.vs.KO_overlap_MECP2WT.vs.IgG.noBlacklist_filtered_chr1to19XY.bed"</span></span>
<span id="cb31-33"><a href="#cb31-33" aria-hidden="true" tabindex="-1"></a>MECP2_peaks.GR <span class="ot">=</span> <span class="fu">readPeakFile</span>(mecp2)</span>
<span id="cb31-34"><a href="#cb31-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-35"><a href="#cb31-35" aria-hidden="true" tabindex="-1"></a><span class="co"># my target </span></span>
<span id="cb31-36"><a href="#cb31-36" aria-hidden="true" tabindex="-1"></a>target <span class="ot">=</span> <span class="fu">c</span>(cortex,encode_forebrain)</span>
<span id="cb31-37"><a href="#cb31-37" aria-hidden="true" tabindex="-1"></a>GR <span class="ot">=</span> <span class="fu">GRangesList</span>(<span class="fu">lapply</span>(target, readPeakFile))</span>
<span id="cb31-38"><a href="#cb31-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-39"><a href="#cb31-39" aria-hidden="true" tabindex="-1"></a><span class="co"># chromosome length</span></span>
<span id="cb31-40"><a href="#cb31-40" aria-hidden="true" tabindex="-1"></a>chrLen <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">"data/mm10/mm10.chrom.sizes"</span>,<span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>,<span class="at">header =</span> F) <span class="sc">%&gt;%</span> </span>
<span id="cb31-41"><a href="#cb31-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(V1 <span class="sc">%in%</span> <span class="fu">c</span>(<span class="fu">paste0</span>(<span class="st">'chr'</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">19</span>),<span class="st">"chrX"</span>,<span class="st">"chrY"</span>))</span>
<span id="cb31-42"><a href="#cb31-42" aria-hidden="true" tabindex="-1"></a>chrLen.vec <span class="ot">=</span> chrLen<span class="sc">$</span>V2</span>
<span id="cb31-43"><a href="#cb31-43" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(chrLen.vec) <span class="ot">=</span> chrLen<span class="sc">$</span>V1</span>
<span id="cb31-44"><a href="#cb31-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-45"><a href="#cb31-45" aria-hidden="true" tabindex="-1"></a>GR.mecp2_overlap <span class="ot">=</span> <span class="fu">list</span>()</span>
<span id="cb31-46"><a href="#cb31-46" aria-hidden="true" tabindex="-1"></a>GR.mecp2_overlap.test <span class="ot">=</span> <span class="fu">list</span>()</span>
<span id="cb31-47"><a href="#cb31-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-48"><a href="#cb31-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-49"><a href="#cb31-49" aria-hidden="true" tabindex="-1"></a>pn.area <span class="ot">=</span> <span class="dv">100</span></span>
<span id="cb31-50"><a href="#cb31-50" aria-hidden="true" tabindex="-1"></a>pn.dist <span class="ot">=</span> <span class="dv">100</span></span>
<span id="cb31-51"><a href="#cb31-51" aria-hidden="true" tabindex="-1"></a>pn.jacc <span class="ot">=</span> <span class="dv">100</span></span>
<span id="cb31-52"><a href="#cb31-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-53"><a href="#cb31-53" aria-hidden="true" tabindex="-1"></a>GR.mecp2_overlap.test <span class="ot">=</span> <span class="fu">mclapply</span>(GR,<span class="at">mc.cores =</span> <span class="dv">45</span>,<span class="cf">function</span>(x){</span>
<span id="cb31-54"><a href="#cb31-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">GenometriCorrelation</span>(MECP2_peaks.GR, x, <span class="at">chromosomes.length =</span> chrLen.vec,</span>
<span id="cb31-55"><a href="#cb31-55" aria-hidden="true" tabindex="-1"></a>                       <span class="at">permut.number =</span> <span class="dv">0</span>, <span class="at">keep.distributions =</span> <span class="cn">TRUE</span>, </span>
<span id="cb31-56"><a href="#cb31-56" aria-hidden="true" tabindex="-1"></a>                       <span class="at">showProgressBar =</span> <span class="cn">TRUE</span>,<span class="at">awhole.only=</span><span class="cn">TRUE</span>,</span>
<span id="cb31-57"><a href="#cb31-57" aria-hidden="true" tabindex="-1"></a>                       <span class="at">ecdf.area.permut.number =</span> pn.area,</span>
<span id="cb31-58"><a href="#cb31-58" aria-hidden="true" tabindex="-1"></a>                       <span class="at">mean.distance.permut.number =</span> pn.dist,</span>
<span id="cb31-59"><a href="#cb31-59" aria-hidden="true" tabindex="-1"></a>                       <span class="at">jaccard.measure.permut.number =</span> pn.jacc)</span>
<span id="cb31-60"><a href="#cb31-60" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb31-61"><a href="#cb31-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-62"><a href="#cb31-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-63"><a href="#cb31-63" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(GR.mecp2_overlap.test)){</span>
<span id="cb31-64"><a href="#cb31-64" aria-hidden="true" tabindex="-1"></a>  test1 <span class="ot">=</span> <span class="fu">unlist</span>(GR.mecp2_overlap.test[[i]]<span class="sc">$</span>awhole)</span>
<span id="cb31-65"><a href="#cb31-65" aria-hidden="true" tabindex="-1"></a>  GR.mecp2_overlap[[i]] <span class="ot">=</span> test1 <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb31-66"><a href="#cb31-66" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rownames_to_column</span>(<span class="st">'param'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb31-67"><a href="#cb31-67" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span>(<span class="fu">grepl</span>(<span class="st">".data|null"</span>,param))) </span>
<span id="cb31-68"><a href="#cb31-68" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(GR.mecp2_overlap[[i]])[<span class="dv">2</span>] <span class="ot">=</span> <span class="fu">names</span>(target)[i]</span>
<span id="cb31-69"><a href="#cb31-69" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb31-70"><a href="#cb31-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-71"><a href="#cb31-71" aria-hidden="true" tabindex="-1"></a><span class="co"># generate graph</span></span>
<span id="cb31-72"><a href="#cb31-72" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(GR.mecp2_overlap.test)){</span>
<span id="cb31-73"><a href="#cb31-73" aria-hidden="true" tabindex="-1"></a><span class="fu">graphical.report</span>(GR.mecp2_overlap.test[[i]],</span>
<span id="cb31-74"><a href="#cb31-74" aria-hidden="true" tabindex="-1"></a>                 <span class="at">pdffile =</span> <span class="fu">paste0</span>(<span class="st">"results/ENCODE/ENCODE_MECP2_overlap_GenometricCorr_"</span>,</span>
<span id="cb31-75"><a href="#cb31-75" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">names</span>(GR.mecp2_overlap.test)[i],<span class="st">".pdf"</span>), </span>
<span id="cb31-76"><a href="#cb31-76" aria-hidden="true" tabindex="-1"></a>                 <span class="at">show.all =</span> <span class="cn">TRUE</span>)</span>
<span id="cb31-77"><a href="#cb31-77" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb31-78"><a href="#cb31-78" aria-hidden="true" tabindex="-1"></a>GR.mecp2_overlap</span>
<span id="cb31-79"><a href="#cb31-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-80"><a href="#cb31-80" aria-hidden="true" tabindex="-1"></a><span class="co"># change 2nd column name</span></span>
<span id="cb31-81"><a href="#cb31-81" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(GR.mecp2_overlap)){</span>
<span id="cb31-82"><a href="#cb31-82" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(GR.mecp2_overlap[[<span class="dv">1</span>]])[<span class="dv">2</span>] <span class="ot">=</span> <span class="st">"value"</span></span>
<span id="cb31-83"><a href="#cb31-83" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb31-84"><a href="#cb31-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-85"><a href="#cb31-85" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(GR.mecp2_overlap) <span class="ot">=</span> <span class="fu">names</span>(GR)</span>
<span id="cb31-86"><a href="#cb31-86" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-87"><a href="#cb31-87" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-88"><a href="#cb31-88" aria-hidden="true" tabindex="-1"></a><span class="fu">rbindlist</span>(GR.mecp2_overlap,<span class="at">idcol =</span> <span class="st">"Group"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb31-89"><a href="#cb31-89" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dcast</span>(Group<span class="sc">~</span>param,<span class="at">value.var =</span> <span class="st">"value"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb31-90"><a href="#cb31-90" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write.table</span>(<span class="st">"results/ENCODE/ENCODE_MECP2_overlap_GenometricCorr_stats.tsv"</span>,</span>
<span id="cb31-91"><a href="#cb31-91" aria-hidden="true" tabindex="-1"></a>              <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>,<span class="at">row.names =</span> <span class="cn">FALSE</span>,<span class="at">quote =</span> <span class="cn">FALSE</span>)</span>
<span id="cb31-92"><a href="#cb31-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-93"><a href="#cb31-93" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-94"><a href="#cb31-94" aria-hidden="true" tabindex="-1"></a>jaccard <span class="ot">=</span> <span class="fu">rbindlist</span>(GR.mecp2_overlap,<span class="at">idcol =</span> <span class="st">"Group"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb31-95"><a href="#cb31-95" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dcast</span>(Group<span class="sc">~</span>param,<span class="at">value.var =</span> <span class="st">"value"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb31-96"><a href="#cb31-96" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">1</span>,<span class="dv">3</span>,<span class="dv">5</span>)</span>
<span id="cb31-97"><a href="#cb31-97" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-98"><a href="#cb31-98" aria-hidden="true" tabindex="-1"></a>jaccard <span class="ot">=</span> jaccard <span class="sc">%&gt;%</span> <span class="fu">separate</span>(Group,<span class="fu">c</span>(<span class="st">'Factor'</span>,<span class="st">'tissue'</span>),<span class="at">sep =</span> <span class="st">"_"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb31-99"><a href="#cb31-99" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">jaccard.measure.test.direction =</span> <span class="fu">ifelse</span>(jaccard.measure.test.direction <span class="sc">==</span> <span class="st">'attraction'</span>, </span>
<span id="cb31-100"><a href="#cb31-100" aria-hidden="true" tabindex="-1"></a>                                                 <span class="st">'*'</span>,<span class="st">'*'</span>)) </span>
<span id="cb31-101"><a href="#cb31-101" aria-hidden="true" tabindex="-1"></a>jaccard<span class="sc">$</span>jaccard.measure <span class="ot">=</span> <span class="fu">as.numeric</span>(jaccard<span class="sc">$</span>jaccard.measure)</span>
<span id="cb31-102"><a href="#cb31-102" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-103"><a href="#cb31-103" aria-hidden="true" tabindex="-1"></a>jaccard_index <span class="ot">=</span> <span class="fu">dcast</span>(jaccard,</span>
<span id="cb31-104"><a href="#cb31-104" aria-hidden="true" tabindex="-1"></a>                      <span class="at">formula =</span> Factor<span class="sc">~</span>tissue,<span class="at">value.var =</span> <span class="st">"jaccard.measure"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb31-105"><a href="#cb31-105" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="st">"Factor"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb31-106"><a href="#cb31-106" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">2</span><span class="sc">:</span><span class="dv">8</span>,<span class="dv">9</span>,<span class="dv">1</span>)</span>
<span id="cb31-107"><a href="#cb31-107" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-108"><a href="#cb31-108" aria-hidden="true" tabindex="-1"></a>jaccard_direction<span class="ot">=</span> <span class="fu">dcast</span>(jaccard,</span>
<span id="cb31-109"><a href="#cb31-109" aria-hidden="true" tabindex="-1"></a>                         <span class="at">formula =</span> Factor<span class="sc">~</span>tissue,</span>
<span id="cb31-110"><a href="#cb31-110" aria-hidden="true" tabindex="-1"></a>                         <span class="at">value.var =</span> <span class="st">"jaccard.measure.test.direction"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb31-111"><a href="#cb31-111" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="st">"Factor"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb31-112"><a href="#cb31-112" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">2</span><span class="sc">:</span><span class="dv">8</span>,<span class="dv">9</span>,<span class="dv">1</span>)</span>
<span id="cb31-113"><a href="#cb31-113" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-114"><a href="#cb31-114" aria-hidden="true" tabindex="-1"></a>jaccard_direction[<span class="fu">is.na</span>(jaccard_direction)] <span class="ot">=</span> <span class="st">" "</span></span>
<span id="cb31-115"><a href="#cb31-115" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-116"><a href="#cb31-116" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">"Plots/ENCODE_dataOverlap_with_MECP2.pdf"</span>,<span class="at">height =</span> <span class="dv">4</span>,<span class="at">width =</span> <span class="dv">6</span>)</span>
<span id="cb31-117"><a href="#cb31-117" aria-hidden="true" tabindex="-1"></a><span class="fu">draw</span>(<span class="fu">Heatmap</span>(jaccard_index,<span class="at">name =</span> <span class="st">"Jaccard measure"</span>,</span>
<span id="cb31-118"><a href="#cb31-118" aria-hidden="true" tabindex="-1"></a>             <span class="at">cluster_columns =</span> <span class="cn">FALSE</span>,</span>
<span id="cb31-119"><a href="#cb31-119" aria-hidden="true" tabindex="-1"></a>             <span class="at">cluster_rows =</span> <span class="cn">TRUE</span>,</span>
<span id="cb31-120"><a href="#cb31-120" aria-hidden="true" tabindex="-1"></a>             <span class="at">rect_gp =</span> <span class="fu">gpar</span>(<span class="at">col =</span> <span class="st">"black"</span>),</span>
<span id="cb31-121"><a href="#cb31-121" aria-hidden="true" tabindex="-1"></a>             <span class="at">column_names_side =</span> <span class="st">"top"</span>,</span>
<span id="cb31-122"><a href="#cb31-122" aria-hidden="true" tabindex="-1"></a>             <span class="at">column_names_rot =</span> <span class="dv">30</span>,</span>
<span id="cb31-123"><a href="#cb31-123" aria-hidden="true" tabindex="-1"></a>             <span class="at">column_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">8</span>),</span>
<span id="cb31-124"><a href="#cb31-124" aria-hidden="true" tabindex="-1"></a>             <span class="at">row_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">8</span>)),</span>
<span id="cb31-125"><a href="#cb31-125" aria-hidden="true" tabindex="-1"></a>             <span class="co">#cell_fun = function(j, i, x, y, width, height, fill) {</span></span>
<span id="cb31-126"><a href="#cb31-126" aria-hidden="true" tabindex="-1"></a>             <span class="co">#  grid.text(jaccard_direction[i, j], x, y, gp = gpar(fontsize = 12))},</span></span>
<span id="cb31-127"><a href="#cb31-127" aria-hidden="true" tabindex="-1"></a>     <span class="at">heatmap_legend_side=</span><span class="st">"right"</span>)</span>
<span id="cb31-128"><a href="#cb31-128" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb31-129"><a href="#cb31-129" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-130"><a href="#cb31-130" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(GR.mecp2_overlap,GR.mecp2_overlap.test,</span>
<span id="cb31-131"><a href="#cb31-131" aria-hidden="true" tabindex="-1"></a>     <span class="at">file =</span> <span class="st">"results/RData/MECP2_ENCODE_geometricCorr.RData"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></li>
<li><p>Supplementary Figure 3B ( All H3K27ac peaks).</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Read H3K27ac CNR peaks summit files</span></span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>CNR_peaks_file <span class="ot">=</span> <span class="fu">list.dirs</span>(<span class="st">"../CUT_and_RUN/results/macs2_Callpeak/"</span>)[<span class="sc">!</span>(<span class="fu">grepl</span>(<span class="st">'tag|_macs2|_Macs2'</span>,<span class="fu">list.dirs</span>(<span class="st">"../CUT_and_RUN/results/macs2_Callpeak/"</span>)))][<span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a>CNR_peak_summit_files <span class="ot">=</span> <span class="fu">lapply</span>(CNR_peaks_file, </span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a>                               <span class="cf">function</span>(x) <span class="fu">paste0</span>(x,<span class="st">"/"</span>,<span class="fu">basename</span>(x), <span class="st">"_summits.bed"</span>)) <span class="sc">%&gt;%</span> <span class="fu">unlist</span>()</span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a>CNR_peak_summit.bed <span class="ot">=</span> <span class="fu">lapply</span>(CNR_peak_summit_files, import.bed)</span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-7"><a href="#cb32-7" aria-hidden="true" tabindex="-1"></a>CNR_filename <span class="ot">=</span> <span class="fu">basename</span>(CNR_peaks_file)</span>
<span id="cb32-8"><a href="#cb32-8" aria-hidden="true" tabindex="-1"></a>CNR_filename <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">'_CNR_.*'</span>,<span class="st">''</span>,CNR_filename)</span>
<span id="cb32-9"><a href="#cb32-9" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(CNR_peak_summit.bed) <span class="ot">=</span> CNR_filename</span>
<span id="cb32-10"><a href="#cb32-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-11"><a href="#cb32-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Extend 500bp up and downstream to peak Summit</span></span>
<span id="cb32-12"><a href="#cb32-12" aria-hidden="true" tabindex="-1"></a>CNR_peak_summit.pm500.bed <span class="ot">=</span> <span class="fu">lapply</span>(CNR_peak_summit.bed, resize ,<span class="at">width =</span> <span class="dv">500</span>,<span class="at">fix =</span> <span class="st">'center'</span>)</span>
<span id="cb32-13"><a href="#cb32-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-14"><a href="#cb32-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-15"><a href="#cb32-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter H3K27ac CNR data</span></span>
<span id="cb32-16"><a href="#cb32-16" aria-hidden="true" tabindex="-1"></a>H3K27ac_CNR.GR <span class="ot">=</span> CNR_peak_summit.pm500.bed[<span class="fu">grepl</span>(<span class="st">'_H3K27ac'</span>,<span class="fu">names</span>(CNR_peak_summit.pm500.bed))]</span>
<span id="cb32-17"><a href="#cb32-17" aria-hidden="true" tabindex="-1"></a>H3K27ac_CNR.GR <span class="ot">=</span> H3K27ac_CNR.GR[<span class="fu">grepl</span>(<span class="st">'MECP2WT'</span>,<span class="fu">names</span>(H3K27ac_CNR.GR))]</span>
<span id="cb32-18"><a href="#cb32-18" aria-hidden="true" tabindex="-1"></a>H3K27ac_CNR.df <span class="ot">=</span> <span class="fu">lapply</span>(H3K27ac_CNR.GR, <span class="cf">function</span>(x) x <span class="sc">|&gt;</span> <span class="fu">as.data.frame</span>()) </span>
<span id="cb32-19"><a href="#cb32-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-20"><a href="#cb32-20" aria-hidden="true" tabindex="-1"></a>H3K27ac_CNR.df <span class="ot">=</span> <span class="fu">do.call</span>(rbind,H3K27ac_CNR.df) <span class="sc">%&gt;%</span> </span>
<span id="cb32-21"><a href="#cb32-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(score <span class="sc">&gt;</span><span class="dv">4</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb32-22"><a href="#cb32-22" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>,<span class="dv">5</span>)</span>
<span id="cb32-23"><a href="#cb32-23" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(H3K27ac_CNR.df) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">'chr'</span>,<span class="st">'start'</span>,<span class="st">'end'</span>,<span class="st">'strand'</span>)</span>
<span id="cb32-24"><a href="#cb32-24" aria-hidden="true" tabindex="-1"></a>H3K27ac_CNR_filtered.GR <span class="ot">=</span> H3K27ac_CNR.df <span class="sc">%&gt;%</span> <span class="fu">GRanges</span>() <span class="sc">%&gt;%</span> GenomicRanges<span class="sc">::</span><span class="fu">reduce</span>()</span>
<span id="cb32-25"><a href="#cb32-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-26"><a href="#cb32-26" aria-hidden="true" tabindex="-1"></a>GR.colnames <span class="ot">=</span> <span class="fu">c</span>(<span class="st">'chr'</span>,<span class="st">'start'</span>,<span class="st">'end'</span>)</span>
<span id="cb32-27"><a href="#cb32-27" aria-hidden="true" tabindex="-1"></a>mm10.blacklisted.Regions <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">"../CUT_and_RUN/data/mm10/mm10.blacklist.bed"</span>,</span>
<span id="cb32-28"><a href="#cb32-28" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">sep =</span> <span class="st">'</span><span class="sc">\t</span><span class="st">'</span>,<span class="at">header =</span> F) <span class="sc">%&gt;%</span> </span>
<span id="cb32-29"><a href="#cb32-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename_with</span>(<span class="sc">~</span>GR.colnames) <span class="sc">%&gt;%</span> </span>
<span id="cb32-30"><a href="#cb32-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">GRanges</span>()</span>
<span id="cb32-31"><a href="#cb32-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-32"><a href="#cb32-32" aria-hidden="true" tabindex="-1"></a>H3K27ac_CNR_filtered.GR.noBlacklist <span class="ot">=</span> <span class="fu">subsetByOverlaps</span>(H3K27ac_CNR_filtered.GR, </span>
<span id="cb32-33"><a href="#cb32-33" aria-hidden="true" tabindex="-1"></a>                                                       mm10.blacklisted.Regions, <span class="at">invert =</span> <span class="cn">TRUE</span>)</span>
<span id="cb32-34"><a href="#cb32-34" aria-hidden="true" tabindex="-1"></a>H3K27ac_CNR_filtered.GR.noBlacklist.df <span class="ot">=</span> <span class="fu">as.data.frame</span>(H3K27ac_CNR_filtered.GR.noBlacklist) <span class="sc">%&gt;%</span> </span>
<span id="cb32-35"><a href="#cb32-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">PeakID=</span><span class="fu">paste0</span>(<span class="st">'Peak'</span>,<span class="fu">row_number</span>())) </span>
<span id="cb32-36"><a href="#cb32-36" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(<span class="at">file =</span> <span class="st">"results/H3K27ac/H3K27ac_MECP2WT_filtered_peaks.bed"</span>,</span>
<span id="cb32-37"><a href="#cb32-37" aria-hidden="true" tabindex="-1"></a>            H3K27ac_CNR_filtered.GR.noBlacklist.df[,<span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>,<span class="dv">6</span>)],<span class="at">sep=</span><span class="st">'</span><span class="sc">\t</span><span class="st">'</span>,</span>
<span id="cb32-38"><a href="#cb32-38" aria-hidden="true" tabindex="-1"></a>            <span class="at">quote =</span> <span class="cn">FALSE</span>,<span class="at">row.names =</span> F,<span class="at">col.names =</span> F)</span>
<span id="cb32-39"><a href="#cb32-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-40"><a href="#cb32-40" aria-hidden="true" tabindex="-1"></a><span class="co"># create SAF format</span></span>
<span id="cb32-41"><a href="#cb32-41" aria-hidden="true" tabindex="-1"></a>H3K27ac_CNR.noBlacklist.df.saf <span class="ot">=</span> H3K27ac_CNR_filtered.GR.noBlacklist.df <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">c</span>(<span class="dv">6</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>))</span>
<span id="cb32-42"><a href="#cb32-42" aria-hidden="true" tabindex="-1"></a>H3K27ac_CNR.noBlacklist.df.saf<span class="sc">$</span>Strand <span class="ot">=</span> <span class="st">'*'</span></span>
<span id="cb32-43"><a href="#cb32-43" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(H3K27ac_CNR.noBlacklist.df.saf) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"GeneID"</span>,<span class="st">"Chr"</span>,<span class="st">"Start"</span>,<span class="st">"End"</span>,<span class="st">"Strand"</span>)</span>
<span id="cb32-44"><a href="#cb32-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-45"><a href="#cb32-45" aria-hidden="true" tabindex="-1"></a><span class="co"># List alignment files</span></span>
<span id="cb32-46"><a href="#cb32-46" aria-hidden="true" tabindex="-1"></a>bams <span class="ot">=</span> <span class="fu">dir</span>(<span class="st">"../CUT_and_RUN/results/"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>, <span class="at">pattern =</span> <span class="st">"*.final.bam"</span>)</span>
<span id="cb32-47"><a href="#cb32-47" aria-hidden="true" tabindex="-1"></a>bams <span class="ot">=</span>  bams[<span class="sc">!</span>(<span class="fu">grepl</span>(<span class="st">'IgG'</span>,bams))]</span>
<span id="cb32-48"><a href="#cb32-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-49"><a href="#cb32-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-50"><a href="#cb32-50" aria-hidden="true" tabindex="-1"></a>bams <span class="ot">=</span> <span class="fu">dir</span>(<span class="st">"../CUT_and_RUN/results/bam/"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>, <span class="at">pattern =</span> <span class="st">"*.final.bam"</span>)</span>
<span id="cb32-51"><a href="#cb32-51" aria-hidden="true" tabindex="-1"></a>bams <span class="ot">=</span>  bams[<span class="sc">!</span>(<span class="fu">grepl</span>(<span class="st">'IgG|.bai|.sort'</span>,bams))]</span>
<span id="cb32-52"><a href="#cb32-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-53"><a href="#cb32-53" aria-hidden="true" tabindex="-1"></a>H3K27ac.bams <span class="ot">=</span> bams[<span class="fu">grepl</span>(<span class="st">'_H3K27ac'</span>,bams)][<span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb32-54"><a href="#cb32-54" aria-hidden="true" tabindex="-1"></a><span class="co">#bamsToCount = subset(bamsToCount,!grepl("*_2h*", bamsToCount))</span></span>
<span id="cb32-55"><a href="#cb32-55" aria-hidden="true" tabindex="-1"></a>H3K27ac.fcResults <span class="ot">=</span> <span class="fu">featureCounts</span>(H3K27ac.bams, </span>
<span id="cb32-56"><a href="#cb32-56" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">annot.ext =</span> H3K27ac_CNR.noBlacklist.df.saf,</span>
<span id="cb32-57"><a href="#cb32-57" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">isPairedEnd =</span> <span class="cn">TRUE</span>,</span>
<span id="cb32-58"><a href="#cb32-58" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">countMultiMappingReads =</span> <span class="cn">FALSE</span>,</span>
<span id="cb32-59"><a href="#cb32-59" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">maxFragLength =</span> <span class="dv">100</span>,<span class="at">nthreads =</span> <span class="dv">15</span>)</span>
<span id="cb32-60"><a href="#cb32-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-61"><a href="#cb32-61" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(H3K27ac.fcResults,<span class="at">file =</span> <span class="st">"results/H3K27ac/H3K27ac.RData"</span>)</span>
<span id="cb32-62"><a href="#cb32-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-63"><a href="#cb32-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-64"><a href="#cb32-64" aria-hidden="true" tabindex="-1"></a>H3K27ac.myCounts <span class="ot">=</span> H3K27ac.fcResults<span class="sc">$</span>counts <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() </span>
<span id="cb32-65"><a href="#cb32-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-66"><a href="#cb32-66" aria-hidden="true" tabindex="-1"></a><span class="co"># Metadata</span></span>
<span id="cb32-67"><a href="#cb32-67" aria-hidden="true" tabindex="-1"></a>H3K27ac.metaData <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">Group=</span><span class="fu">colnames</span>(H3K27ac.myCounts), </span>
<span id="cb32-68"><a href="#cb32-68" aria-hidden="true" tabindex="-1"></a>                              <span class="at">row.names =</span> <span class="fu">colnames</span>(H3K27ac.myCounts))</span>
<span id="cb32-69"><a href="#cb32-69" aria-hidden="true" tabindex="-1"></a>H3K27ac.metaData<span class="sc">$</span>Group <span class="ot">=</span> <span class="fu">c</span>(<span class="st">'MECP2_KO'</span>,<span class="st">'WT'</span>,<span class="st">'DNMT3A_KO'</span>,<span class="st">'WT'</span>,</span>
<span id="cb32-70"><a href="#cb32-70" aria-hidden="true" tabindex="-1"></a>                           <span class="st">'MECP2_KO'</span>,<span class="st">'WT'</span>,<span class="st">'DNMT3A_KO'</span>,<span class="st">'WT'</span>)</span>
<span id="cb32-71"><a href="#cb32-71" aria-hidden="true" tabindex="-1"></a>H3K27ac.metaData<span class="sc">$</span>Batch <span class="ot">=</span> <span class="fu">rownames</span>(H3K27ac.metaData)</span>
<span id="cb32-72"><a href="#cb32-72" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(H3K27ac.metaData) <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">'_CNR_.*'</span>,<span class="st">''</span>,<span class="fu">rownames</span>(H3K27ac.metaData))</span>
<span id="cb32-73"><a href="#cb32-73" aria-hidden="true" tabindex="-1"></a>H3K27ac.metaData<span class="sc">$</span>Sample <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">'_H3K27ac'</span>,<span class="st">''</span>,<span class="fu">rownames</span>(H3K27ac.metaData))</span>
<span id="cb32-74"><a href="#cb32-74" aria-hidden="true" tabindex="-1"></a>H3K27ac.metaData<span class="sc">$</span>Batch <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">'.*_CTX_.*_CNR_|.final.bam'</span>,<span class="st">''</span>,H3K27ac.metaData<span class="sc">$</span>Batch)</span>
<span id="cb32-75"><a href="#cb32-75" aria-hidden="true" tabindex="-1"></a>H3K27ac.metaData<span class="sc">$</span>Sample <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">'_Mecp2'</span>,<span class="st">''</span>,<span class="fu">rownames</span>(H3K27ac.metaData))</span>
<span id="cb32-76"><a href="#cb32-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-77"><a href="#cb32-77" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(H3K27ac.myCounts) <span class="ot">=</span> <span class="fu">rownames</span>(H3K27ac.metaData)</span>
<span id="cb32-78"><a href="#cb32-78" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-79"><a href="#cb32-79" aria-hidden="true" tabindex="-1"></a><span class="co"># # Select only MECP2 WT and KO samples</span></span>
<span id="cb32-80"><a href="#cb32-80" aria-hidden="true" tabindex="-1"></a>y <span class="ot">=</span> <span class="fu">DGEList</span>(<span class="at">counts =</span> H3K27ac.myCounts[,<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">5</span>,<span class="dv">6</span>)])</span>
<span id="cb32-81"><a href="#cb32-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-82"><a href="#cb32-82" aria-hidden="true" tabindex="-1"></a>y<span class="sc">$</span>samples<span class="sc">$</span>Group <span class="ot">=</span> H3K27ac.metaData[<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">5</span>,<span class="dv">6</span>),]<span class="sc">$</span>Group</span>
<span id="cb32-83"><a href="#cb32-83" aria-hidden="true" tabindex="-1"></a>y<span class="sc">$</span>samples<span class="sc">$</span>Batch <span class="ot">=</span> H3K27ac.metaData[<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">5</span>,<span class="dv">6</span>),]<span class="sc">$</span>Batch</span>
<span id="cb32-84"><a href="#cb32-84" aria-hidden="true" tabindex="-1"></a>y<span class="sc">$</span>samples<span class="sc">$</span>Sample <span class="ot">=</span> H3K27ac.metaData[<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">5</span>,<span class="dv">6</span>),]<span class="sc">$</span>Sample</span>
<span id="cb32-85"><a href="#cb32-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-86"><a href="#cb32-86" aria-hidden="true" tabindex="-1"></a><span class="co"># Normalize and obtain logcounts for QC</span></span>
<span id="cb32-87"><a href="#cb32-87" aria-hidden="true" tabindex="-1"></a>y <span class="ot">=</span> <span class="fu">calcNormFactors</span>(y)</span>
<span id="cb32-88"><a href="#cb32-88" aria-hidden="true" tabindex="-1"></a>logCPMs <span class="ot">=</span> <span class="fu">cpm</span>(y, <span class="at">log =</span> <span class="cn">TRUE</span>)</span>
<span id="cb32-89"><a href="#cb32-89" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-90"><a href="#cb32-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-91"><a href="#cb32-91" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate rowwise variance</span></span>
<span id="cb32-92"><a href="#cb32-92" aria-hidden="true" tabindex="-1"></a>rv <span class="ot">=</span> <span class="fu">apply</span>(logCPMs, <span class="dv">1</span>, var)</span>
<span id="cb32-93"><a href="#cb32-93" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-94"><a href="#cb32-94" aria-hidden="true" tabindex="-1"></a><span class="co"># Sort decreasingly and take top 1000</span></span>
<span id="cb32-95"><a href="#cb32-95" aria-hidden="true" tabindex="-1"></a>o <span class="ot">=</span> <span class="fu">order</span>(rv, <span class="at">decreasing=</span><span class="cn">TRUE</span>)</span>
<span id="cb32-96"><a href="#cb32-96" aria-hidden="true" tabindex="-1"></a>top1000 <span class="ot">=</span> <span class="fu">head</span>(o, <span class="dv">1000</span>)</span>
<span id="cb32-97"><a href="#cb32-97" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-98"><a href="#cb32-98" aria-hidden="true" tabindex="-1"></a><span class="co"># From the logCPMs subset for the top-1000</span></span>
<span id="cb32-99"><a href="#cb32-99" aria-hidden="true" tabindex="-1"></a>logCPM_top1000 <span class="ot">=</span> logCPMs[top1000,]</span>
<span id="cb32-100"><a href="#cb32-100" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-101"><a href="#cb32-101" aria-hidden="true" tabindex="-1"></a><span class="co"># Run PCA</span></span>
<span id="cb32-102"><a href="#cb32-102" aria-hidden="true" tabindex="-1"></a>pca <span class="ot">=</span> <span class="fu">prcomp</span>(<span class="fu">t</span>(logCPM_top1000))</span>
<span id="cb32-103"><a href="#cb32-103" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-104"><a href="#cb32-104" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine PCA coordinates with the metadata from the DGEList</span></span>
<span id="cb32-105"><a href="#cb32-105" aria-hidden="true" tabindex="-1"></a>to_plot <span class="ot">=</span> <span class="fu">data.frame</span>(pca<span class="sc">$</span>x, y<span class="sc">$</span>samples)</span>
<span id="cb32-106"><a href="#cb32-106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-107"><a href="#cb32-107" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate how many % of total variance is explained by each principal component</span></span>
<span id="cb32-108"><a href="#cb32-108" aria-hidden="true" tabindex="-1"></a>percentVar <span class="ot">=</span> pca<span class="sc">$</span>sdev<span class="sc">^</span><span class="dv">2</span> <span class="sc">/</span> <span class="fu">sum</span>( pca<span class="sc">$</span>sdev<span class="sc">^</span><span class="dv">2</span> )<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb32-109"><a href="#cb32-109" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-110"><a href="#cb32-110" aria-hidden="true" tabindex="-1"></a><span class="co"># We focus here on PC1 and PC2</span></span>
<span id="cb32-111"><a href="#cb32-111" aria-hidden="true" tabindex="-1"></a>use.pcs <span class="ot">=</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>)</span>
<span id="cb32-112"><a href="#cb32-112" aria-hidden="true" tabindex="-1"></a>labs <span class="ot">=</span> <span class="fu">paste0</span>(<span class="fu">paste0</span>(<span class="st">"PC"</span>, use.pcs, <span class="st">" - "</span>), </span>
<span id="cb32-113"><a href="#cb32-113" aria-hidden="true" tabindex="-1"></a>               <span class="fu">paste0</span>(<span class="st">"Var.expl = "</span>, <span class="fu">round</span>(percentVar[use.pcs], <span class="dv">2</span>), <span class="st">"%"</span>))</span>
<span id="cb32-114"><a href="#cb32-114" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-115"><a href="#cb32-115" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(to_plot, <span class="fu">aes</span>(<span class="at">x=</span>PC1, <span class="at">y=</span>PC2, <span class="at">color=</span>Group, <span class="at">shape=</span>Sample)) <span class="sc">+</span> </span>
<span id="cb32-116"><a href="#cb32-116" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb32-117"><a href="#cb32-117" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_shape_manual</span>(<span class="at">values=</span><span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>)<span class="sc">+</span></span>
<span id="cb32-118"><a href="#cb32-118" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(labs[<span class="dv">1</span>]) <span class="sc">+</span> <span class="fu">ylab</span>(labs[<span class="dv">2</span>])</span>
<span id="cb32-119"><a href="#cb32-119" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-120"><a href="#cb32-120" aria-hidden="true" tabindex="-1"></a><span class="co"># correct for the Batch </span></span>
<span id="cb32-121"><a href="#cb32-121" aria-hidden="true" tabindex="-1"></a>batch <span class="ot">=</span> <span class="fu">factor</span>(y<span class="sc">$</span>samples<span class="sc">$</span>Batch)</span>
<span id="cb32-122"><a href="#cb32-122" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-123"><a href="#cb32-123" aria-hidden="true" tabindex="-1"></a>logCPMs_corrected <span class="ot">=</span> limma<span class="sc">::</span><span class="fu">removeBatchEffect</span>(logCPMs, <span class="at">batch =</span> batch)</span>
<span id="cb32-124"><a href="#cb32-124" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-125"><a href="#cb32-125" aria-hidden="true" tabindex="-1"></a><span class="co"># repeat PCA as before, using the same genes</span></span>
<span id="cb32-126"><a href="#cb32-126" aria-hidden="true" tabindex="-1"></a>logCPM_corrected_top1000 <span class="ot">=</span> logCPMs_corrected[top1000,]</span>
<span id="cb32-127"><a href="#cb32-127" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-128"><a href="#cb32-128" aria-hidden="true" tabindex="-1"></a><span class="co"># Run PCA</span></span>
<span id="cb32-129"><a href="#cb32-129" aria-hidden="true" tabindex="-1"></a>pca <span class="ot">=</span> <span class="fu">prcomp</span>(<span class="fu">t</span>(logCPM_corrected_top1000))</span>
<span id="cb32-130"><a href="#cb32-130" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-131"><a href="#cb32-131" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine PCA coordinates with the metadata from the DGEList</span></span>
<span id="cb32-132"><a href="#cb32-132" aria-hidden="true" tabindex="-1"></a>to_plot <span class="ot">=</span> <span class="fu">data.frame</span>(pca<span class="sc">$</span>x, y<span class="sc">$</span>samples)</span>
<span id="cb32-133"><a href="#cb32-133" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-134"><a href="#cb32-134" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate how many % of total variance is explained by each principal component</span></span>
<span id="cb32-135"><a href="#cb32-135" aria-hidden="true" tabindex="-1"></a>percentVar <span class="ot">=</span> pca<span class="sc">$</span>sdev<span class="sc">^</span><span class="dv">2</span> <span class="sc">/</span> <span class="fu">sum</span>( pca<span class="sc">$</span>sdev<span class="sc">^</span><span class="dv">2</span> )<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb32-136"><a href="#cb32-136" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-137"><a href="#cb32-137" aria-hidden="true" tabindex="-1"></a><span class="co"># We focus here on PC1 and PC2</span></span>
<span id="cb32-138"><a href="#cb32-138" aria-hidden="true" tabindex="-1"></a>use.pcs <span class="ot">=</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>)</span>
<span id="cb32-139"><a href="#cb32-139" aria-hidden="true" tabindex="-1"></a>labs <span class="ot">=</span> <span class="fu">paste0</span>(<span class="st">"PC"</span>, use.pcs, <span class="st">" - "</span>,<span class="fu">round</span>(percentVar[use.pcs], <span class="dv">2</span>), <span class="st">"%"</span>)</span>
<span id="cb32-140"><a href="#cb32-140" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-141"><a href="#cb32-141" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(to_plot, <span class="fu">aes</span>(<span class="at">x=</span>PC1, <span class="at">y=</span>PC2, <span class="at">color=</span>Group, <span class="at">shape=</span>Sample)) <span class="sc">+</span> </span>
<span id="cb32-142"><a href="#cb32-142" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">5</span>,<span class="at">stroke =</span> <span class="fl">1.8</span>) <span class="sc">+</span></span>
<span id="cb32-143"><a href="#cb32-143" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_shape_manual</span>(<span class="at">values=</span><span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>)<span class="sc">+</span></span>
<span id="cb32-144"><a href="#cb32-144" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(labs[<span class="dv">1</span>]) <span class="sc">+</span> <span class="fu">ylab</span>(labs[<span class="dv">2</span>]) <span class="sc">+</span></span>
<span id="cb32-145"><a href="#cb32-145" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_test</span>(<span class="dv">10</span>) <span class="sc">+</span></span>
<span id="cb32-146"><a href="#cb32-146" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">colour =</span> <span class="fu">guide_legend</span>(<span class="at">override.aes =</span> <span class="fu">list</span>(<span class="at">size=</span><span class="dv">2</span>)))<span class="sc">+</span></span>
<span id="cb32-147"><a href="#cb32-147" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"PCA"</span>) </span>
<span id="cb32-148"><a href="#cb32-148" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"Plots/H3K27ac_MECP2KO_vs_WT_PCA.pdf"</span>,<span class="at">units =</span> <span class="st">'cm'</span>,<span class="at">height =</span> <span class="dv">8</span>,<span class="at">width =</span> <span class="dv">14</span>)</span>
<span id="cb32-149"><a href="#cb32-149" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-150"><a href="#cb32-150" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-151"><a href="#cb32-151" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-152"><a href="#cb32-152" aria-hidden="true" tabindex="-1"></a><span class="co"># Include Batch in the design matrix</span></span>
<span id="cb32-153"><a href="#cb32-153" aria-hidden="true" tabindex="-1"></a>design <span class="ot">=</span> <span class="fu">model.matrix</span>(<span class="sc">~</span>Batch<span class="sc">+</span>Group, y<span class="sc">$</span>samples)</span>
<span id="cb32-154"><a href="#cb32-154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-155"><a href="#cb32-155" aria-hidden="true" tabindex="-1"></a><span class="co"># QLF workflow from edgeR</span></span>
<span id="cb32-156"><a href="#cb32-156" aria-hidden="true" tabindex="-1"></a>y <span class="ot">=</span> <span class="fu">estimateDisp</span>(y, design)</span>
<span id="cb32-157"><a href="#cb32-157" aria-hidden="true" tabindex="-1"></a>fit <span class="ot">=</span> <span class="fu">glmQLFit</span>(y, design)</span>
<span id="cb32-158"><a href="#cb32-158" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-159"><a href="#cb32-159" aria-hidden="true" tabindex="-1"></a><span class="co"># include column (factor) for which differential enhancer need to tested</span></span>
<span id="cb32-160"><a href="#cb32-160" aria-hidden="true" tabindex="-1"></a>fit  <span class="ot">=</span> <span class="fu">glmQLFTest</span>(fit, <span class="at">coef =</span> <span class="dv">3</span>)</span>
<span id="cb32-161"><a href="#cb32-161" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-162"><a href="#cb32-162" aria-hidden="true" tabindex="-1"></a>tt <span class="ot">=</span> <span class="fu">topTable</span>(fit, <span class="at">coef=</span><span class="dv">3</span>)  <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">'Gene'</span>)</span>
<span id="cb32-163"><a href="#cb32-163" aria-hidden="true" tabindex="-1"></a>cp <span class="ot">=</span> <span class="fu">cpm</span>(y) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">'Gene'</span>)</span>
<span id="cb32-164"><a href="#cb32-164" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-165"><a href="#cb32-165" aria-hidden="true" tabindex="-1"></a><span class="co">#/ 1/2 is "A" (=the reference) and 3/4 is "B", positive logFC mean higher in B,</span></span>
<span id="cb32-166"><a href="#cb32-166" aria-hidden="true" tabindex="-1"></a><span class="fu">merge</span>(<span class="at">x=</span>tt, <span class="at">y=</span>cp, <span class="at">by=</span><span class="st">"Gene"</span>)[<span class="dv">2</span>,<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">8</span>,<span class="dv">9</span>,<span class="dv">10</span>,<span class="dv">11</span>)]</span>
<span id="cb32-167"><a href="#cb32-167" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-168"><a href="#cb32-168" aria-hidden="true" tabindex="-1"></a><span class="co"># get stats as a data.frame</span></span>
<span id="cb32-169"><a href="#cb32-169" aria-hidden="true" tabindex="-1"></a>DiffEnhacer <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="fu">topTags</span>(fit, <span class="at">n=</span><span class="cn">Inf</span>))</span>
<span id="cb32-170"><a href="#cb32-170" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-171"><a href="#cb32-171" aria-hidden="true" tabindex="-1"></a><span class="co"># Classify enhancer into significantly up and down</span></span>
<span id="cb32-172"><a href="#cb32-172" aria-hidden="true" tabindex="-1"></a>DiffEnhancer.mod <span class="ot">=</span> DiffEnhacer <span class="sc">%&gt;%</span> </span>
<span id="cb32-173"><a href="#cb32-173" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Diff_H3K27ac=</span><span class="fu">factor</span>(<span class="fu">case_when</span>(logFC<span class="sc">&gt;</span><span class="dv">0</span> <span class="sc">&amp;</span> FDR<span class="sc">&lt;</span><span class="fl">0.1</span> <span class="sc">&amp;</span> logCPM <span class="sc">&gt;</span> <span class="dv">2</span> <span class="sc">~</span> <span class="st">"Up"</span>,</span>
<span id="cb32-174"><a href="#cb32-174" aria-hidden="true" tabindex="-1"></a>                                       logFC<span class="sc">&lt;</span><span class="dv">0</span> <span class="sc">&amp;</span> FDR<span class="sc">&lt;</span><span class="fl">0.1</span> <span class="sc">&amp;</span> logCPM <span class="sc">&gt;</span> <span class="dv">2</span> <span class="sc">~</span> <span class="st">"Down"</span>,</span>
<span id="cb32-175"><a href="#cb32-175" aria-hidden="true" tabindex="-1"></a>                                       <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"not significant"</span>),</span>
<span id="cb32-176"><a href="#cb32-176" aria-hidden="true" tabindex="-1"></a>                             <span class="at">levels=</span><span class="fu">c</span>(<span class="st">"Up"</span>, <span class="st">"not significant"</span>, <span class="st">"Down"</span>)))</span>
<span id="cb32-177"><a href="#cb32-177" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-178"><a href="#cb32-178" aria-hidden="true" tabindex="-1"></a><span class="co"># MA-plot</span></span>
<span id="cb32-179"><a href="#cb32-179" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(DiffEnhancer.mod <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Diff_H3K27ac <span class="sc">!=</span> <span class="st">'not significant'</span>), </span>
<span id="cb32-180"><a href="#cb32-180" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x=</span>logCPM, <span class="at">y=</span>logFC, <span class="at">color=</span>Diff_H3K27ac)) <span class="sc">+</span></span>
<span id="cb32-181"><a href="#cb32-181" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">0</span>) <span class="sc">+</span></span>
<span id="cb32-182"><a href="#cb32-182" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values=</span><span class="fu">c</span>( <span class="st">"blue"</span>,<span class="st">"red"</span>)) <span class="sc">+</span></span>
<span id="cb32-183"><a href="#cb32-183" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"H3K27ac (MECP2 WT vs KO)"</span>) <span class="sc">+</span></span>
<span id="cb32-184"><a href="#cb32-184" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_test</span>(<span class="dv">15</span>) <span class="sc">+</span></span>
<span id="cb32-185"><a href="#cb32-185" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">colour =</span> <span class="fu">guide_legend</span>(<span class="at">override.aes =</span> <span class="fu">list</span>(<span class="at">size=</span><span class="dv">2</span>)))<span class="sc">+</span></span>
<span id="cb32-186"><a href="#cb32-186" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">2</span>,<span class="at">linetype =</span> <span class="st">'dashed'</span>)<span class="sc">+</span></span>
<span id="cb32-187"><a href="#cb32-187" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="fl">6.5</span>, <span class="at">y =</span> <span class="sc">-</span><span class="dv">1</span>, </span>
<span id="cb32-188"><a href="#cb32-188" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> DiffEnhancer.mod <span class="sc">%&gt;%</span> </span>
<span id="cb32-189"><a href="#cb32-189" aria-hidden="true" tabindex="-1"></a>             dplyr<span class="sc">::</span><span class="fu">filter</span>(Diff_H3K27ac<span class="sc">==</span><span class="st">'Down'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb32-190"><a href="#cb32-190" aria-hidden="true" tabindex="-1"></a>             <span class="fu">dim</span>() <span class="sc">%&gt;%</span> .[<span class="dv">1</span>], <span class="at">color=</span><span class="st">"red"</span>,<span class="at">size=</span><span class="dv">4</span>)<span class="sc">+</span></span>
<span id="cb32-191"><a href="#cb32-191" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="fl">6.5</span>, <span class="at">y =</span> <span class="dv">1</span>,  </span>
<span id="cb32-192"><a href="#cb32-192" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> DiffEnhancer.mod <span class="sc">%&gt;%</span> </span>
<span id="cb32-193"><a href="#cb32-193" aria-hidden="true" tabindex="-1"></a>             dplyr<span class="sc">::</span><span class="fu">filter</span>(Diff_H3K27ac<span class="sc">==</span><span class="st">'Up'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb32-194"><a href="#cb32-194" aria-hidden="true" tabindex="-1"></a>             <span class="fu">dim</span>() <span class="sc">%&gt;%</span> .[<span class="dv">1</span>], <span class="at">color=</span><span class="st">"blue"</span>,<span class="at">size=</span><span class="dv">4</span>)<span class="sc">+</span></span>
<span id="cb32-195"><a href="#cb32-195" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">'logFC (MECP2 WT vs KO)'</span>)</span>
<span id="cb32-196"><a href="#cb32-196" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"Plots/H3K27ac_MECP2WT_vs_KO.pdf"</span>,<span class="at">units =</span> <span class="st">'cm'</span>,<span class="at">height =</span> <span class="dv">10</span>,<span class="at">width =</span> <span class="dv">14</span>)</span>
<span id="cb32-197"><a href="#cb32-197" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-198"><a href="#cb32-198" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-199"><a href="#cb32-199" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter Enahncer based on logCPM value</span></span>
<span id="cb32-200"><a href="#cb32-200" aria-hidden="true" tabindex="-1"></a>Enhancer.Up_in_MECP2.KO  <span class="ot">=</span> DiffEnhancer.mod <span class="sc">%&gt;%</span> <span class="fu">filter</span>(logCPM <span class="sc">&gt;</span><span class="dv">2</span> <span class="sc">&amp;</span> Diff_H3K27ac <span class="sc">==</span> <span class="st">'Up'</span>)</span>
<span id="cb32-201"><a href="#cb32-201" aria-hidden="true" tabindex="-1"></a>Enhancer.Down_in_MECP2.KO  <span class="ot">=</span> DiffEnhancer.mod <span class="sc">%&gt;%</span> <span class="fu">filter</span>(logCPM <span class="sc">&gt;</span><span class="dv">2</span> <span class="sc">&amp;</span> Diff_H3K27ac <span class="sc">==</span> <span class="st">'Down'</span>)</span>
<span id="cb32-202"><a href="#cb32-202" aria-hidden="true" tabindex="-1"></a>Enhancer.Unchanged_in_MECP2.KO  <span class="ot">=</span> DiffEnhancer.mod <span class="sc">%&gt;%</span> <span class="fu">filter</span>(logCPM <span class="sc">&gt;</span><span class="dv">2</span> <span class="sc">&amp;</span> Diff_H3K27ac <span class="sc">==</span> <span class="st">'not significant'</span>)</span>
<span id="cb32-203"><a href="#cb32-203" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-204"><a href="#cb32-204" aria-hidden="true" tabindex="-1"></a>H3K27ac_CNR_filtered.GR.noBlacklist.df <span class="sc">%&gt;%</span> </span>
<span id="cb32-205"><a href="#cb32-205" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(PeakID <span class="sc">%in%</span> <span class="fu">rownames</span>(Enhancer.Up_in_MECP2.KO)) <span class="sc">%&gt;%</span> </span>
<span id="cb32-206"><a href="#cb32-206" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>,<span class="dv">6</span>) <span class="sc">%&gt;%</span></span>
<span id="cb32-207"><a href="#cb32-207" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write.table</span>(<span class="at">file =</span> <span class="st">"results/H3K27ac/H3K27ac_up_MECP2WT.vs.KO.bed"</span>,<span class="at">sep=</span><span class="st">'</span><span class="sc">\t</span><span class="st">'</span>,</span>
<span id="cb32-208"><a href="#cb32-208" aria-hidden="true" tabindex="-1"></a>              <span class="at">quote =</span> <span class="cn">FALSE</span>,<span class="at">row.names =</span> F,<span class="at">col.names =</span> F)</span>
<span id="cb32-209"><a href="#cb32-209" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-210"><a href="#cb32-210" aria-hidden="true" tabindex="-1"></a>H3K27ac_CNR_filtered.GR.noBlacklist.df <span class="sc">%&gt;%</span> </span>
<span id="cb32-211"><a href="#cb32-211" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(PeakID <span class="sc">%in%</span> <span class="fu">rownames</span>(Enhancer.Down_in_MECP2.KO)) <span class="sc">%&gt;%</span> </span>
<span id="cb32-212"><a href="#cb32-212" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>,<span class="dv">6</span>) <span class="sc">%&gt;%</span></span>
<span id="cb32-213"><a href="#cb32-213" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write.table</span>(<span class="at">file =</span> <span class="st">"results/H3K27ac/H3K27ac_down_MECP2WT.vs.KO.bed"</span>,<span class="at">sep=</span><span class="st">'</span><span class="sc">\t</span><span class="st">'</span>,</span>
<span id="cb32-214"><a href="#cb32-214" aria-hidden="true" tabindex="-1"></a>              <span class="at">quote =</span> <span class="cn">FALSE</span>,<span class="at">row.names =</span> F,<span class="at">col.names =</span> F)</span>
<span id="cb32-215"><a href="#cb32-215" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-216"><a href="#cb32-216" aria-hidden="true" tabindex="-1"></a>H3K27ac_CNR_filtered.GR.noBlacklist.df <span class="sc">%&gt;%</span> </span>
<span id="cb32-217"><a href="#cb32-217" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(PeakID <span class="sc">%in%</span> <span class="fu">rownames</span>(Enhancer.Unchanged_in_MECP2.KO)) <span class="sc">%&gt;%</span> </span>
<span id="cb32-218"><a href="#cb32-218" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>,<span class="dv">6</span>) <span class="sc">%&gt;%</span></span>
<span id="cb32-219"><a href="#cb32-219" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write.table</span>(<span class="at">file =</span> <span class="st">"results/H3K27ac/H3K27ac_unchanged_MECP2WT.vs.KO.bed"</span>,<span class="at">sep=</span><span class="st">'</span><span class="sc">\t</span><span class="st">'</span>,</span>
<span id="cb32-220"><a href="#cb32-220" aria-hidden="true" tabindex="-1"></a>              <span class="at">quote =</span> <span class="cn">FALSE</span>,<span class="at">row.names =</span> F,<span class="at">col.names =</span> F)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></li>
<li><p>Supplementary Figure 3B ( MECP2-H3K27ac overlapping peaks).</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Read CNR peaks summit files</span></span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>CNR_peaks_file <span class="ot">=</span> <span class="fu">list.dirs</span>(<span class="st">"../CUT_and_RUN/results/macs2_Callpeak/"</span>)[<span class="sc">!</span>(<span class="fu">grepl</span>(<span class="st">'tag|_macs2|_Macs2'</span>,<span class="fu">list.dirs</span>(<span class="st">"../CUT_and_RUN/results/macs2_Callpeak/"</span>)))][<span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a>CNR_peak_summit_files <span class="ot">=</span> <span class="fu">lapply</span>(CNR_peaks_file, <span class="cf">function</span>(x) <span class="fu">paste0</span>(x,<span class="st">"/"</span>,<span class="fu">basename</span>(x), <span class="st">"_summits.bed"</span>)) <span class="sc">%&gt;%</span> <span class="fu">unlist</span>()</span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a>CNR_peak_summit.bed <span class="ot">=</span> <span class="fu">lapply</span>(CNR_peak_summit_files, import.bed)</span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-7"><a href="#cb33-7" aria-hidden="true" tabindex="-1"></a>CNR_filename <span class="ot">=</span> <span class="fu">basename</span>(CNR_peaks_file)</span>
<span id="cb33-8"><a href="#cb33-8" aria-hidden="true" tabindex="-1"></a>CNR_filename <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">'_CNR_.*'</span>,<span class="st">''</span>,CNR_filename)</span>
<span id="cb33-9"><a href="#cb33-9" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(CNR_peak_summit.bed) <span class="ot">=</span> CNR_filename</span>
<span id="cb33-10"><a href="#cb33-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-11"><a href="#cb33-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Extend 500bp up and downstream to peak Summit</span></span>
<span id="cb33-12"><a href="#cb33-12" aria-hidden="true" tabindex="-1"></a>CNR_peak_summit.pm500.bed <span class="ot">&lt;-</span> <span class="fu">lapply</span>(CNR_peak_summit.bed, resize ,<span class="at">width =</span> <span class="dv">500</span>,<span class="at">fix =</span> <span class="st">'center'</span>)</span>
<span id="cb33-13"><a href="#cb33-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-14"><a href="#cb33-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-15"><a href="#cb33-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter H3K27ac CNR data</span></span>
<span id="cb33-16"><a href="#cb33-16" aria-hidden="true" tabindex="-1"></a>H3K27ac_CNR.GR <span class="ot">=</span> CNR_peak_summit.pm500.bed[<span class="fu">grepl</span>(<span class="st">'_H3K27ac'</span>,<span class="fu">names</span>(CNR_peak_summit.pm500.bed))]</span>
<span id="cb33-17"><a href="#cb33-17" aria-hidden="true" tabindex="-1"></a>H3K27ac_CNR.GR <span class="ot">=</span> H3K27ac_CNR.GR[<span class="fu">grepl</span>(<span class="st">'MECP2WT'</span>,<span class="fu">names</span>(H3K27ac_CNR.GR))]</span>
<span id="cb33-18"><a href="#cb33-18" aria-hidden="true" tabindex="-1"></a>H3K27ac_CNR.df <span class="ot">=</span> <span class="fu">lapply</span>(H3K27ac_CNR.GR, <span class="cf">function</span>(x) x <span class="sc">|&gt;</span> <span class="fu">as.data.frame</span>()) </span>
<span id="cb33-19"><a href="#cb33-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-20"><a href="#cb33-20" aria-hidden="true" tabindex="-1"></a>H3K27ac_CNR.df <span class="ot">=</span> <span class="fu">do.call</span>(rbind,H3K27ac_CNR.df) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(score <span class="sc">&gt;</span><span class="dv">4</span>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>,<span class="dv">5</span>)</span>
<span id="cb33-21"><a href="#cb33-21" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(H3K27ac_CNR.df) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">'chr'</span>,<span class="st">'start'</span>,<span class="st">'end'</span>,<span class="st">'strand'</span>)</span>
<span id="cb33-22"><a href="#cb33-22" aria-hidden="true" tabindex="-1"></a>H3K27ac_CNR_filtered.GR <span class="ot">=</span> H3K27ac_CNR.df <span class="sc">%&gt;%</span> <span class="fu">GRanges</span>() <span class="sc">%&gt;%</span> GenomicRanges<span class="sc">::</span><span class="fu">reduce</span>()</span>
<span id="cb33-23"><a href="#cb33-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-24"><a href="#cb33-24" aria-hidden="true" tabindex="-1"></a>GR.colnames <span class="ot">=</span> <span class="fu">c</span>(<span class="st">'chr'</span>,<span class="st">'start'</span>,<span class="st">'end'</span>)</span>
<span id="cb33-25"><a href="#cb33-25" aria-hidden="true" tabindex="-1"></a>mm10.blacklisted.Regions <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">"../CUT_and_RUN/data/mm10/mm10.blacklist.bed"</span>,<span class="at">sep =</span> <span class="st">'</span><span class="sc">\t</span><span class="st">'</span>,<span class="at">header =</span> F) <span class="sc">%&gt;%</span> </span>
<span id="cb33-26"><a href="#cb33-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename_with</span>(<span class="sc">~</span>GR.colnames) <span class="sc">%&gt;%</span> </span>
<span id="cb33-27"><a href="#cb33-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">GRanges</span>()</span>
<span id="cb33-28"><a href="#cb33-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-29"><a href="#cb33-29" aria-hidden="true" tabindex="-1"></a>H3K27ac_CNR_filtered.GR.noBlacklist <span class="ot">=</span> <span class="fu">subsetByOverlaps</span>(H3K27ac_CNR_filtered.GR, mm10.blacklisted.Regions, <span class="at">invert =</span> <span class="cn">TRUE</span>)</span>
<span id="cb33-30"><a href="#cb33-30" aria-hidden="true" tabindex="-1"></a>H3K27ac_CNR_filtered.GR.noBlacklist.df <span class="ot">=</span> <span class="fu">as.data.frame</span>(H3K27ac_CNR_filtered.GR.noBlacklist) <span class="sc">%&gt;%</span> </span>
<span id="cb33-31"><a href="#cb33-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">PeakID=</span><span class="fu">paste0</span>(<span class="st">'Peak'</span>,<span class="fu">row_number</span>())) </span>
<span id="cb33-32"><a href="#cb33-32" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(<span class="at">file =</span> <span class="st">"results/H3K27ac/H3K27ac_MECP2WT_filtered_peaks.bed"</span>,</span>
<span id="cb33-33"><a href="#cb33-33" aria-hidden="true" tabindex="-1"></a>            H3K27ac_CNR_filtered.GR.noBlacklist.df[,<span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>,<span class="dv">6</span>)],<span class="at">sep=</span><span class="st">'</span><span class="sc">\t</span><span class="st">'</span>,</span>
<span id="cb33-34"><a href="#cb33-34" aria-hidden="true" tabindex="-1"></a>            <span class="at">quote =</span> <span class="cn">FALSE</span>,<span class="at">row.names =</span> F,<span class="at">col.names =</span> F)</span>
<span id="cb33-35"><a href="#cb33-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-36"><a href="#cb33-36" aria-hidden="true" tabindex="-1"></a><span class="co"># Overlap "results/H3K27ac/H3K27ac_MECP2WT_filtered_peaks.bed" with MECP2 peaks using bedtools</span></span>
<span id="cb33-37"><a href="#cb33-37" aria-hidden="true" tabindex="-1"></a><span class="co"># to extract overlapping peaks</span></span>
<span id="cb33-38"><a href="#cb33-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-39"><a href="#cb33-39" aria-hidden="true" tabindex="-1"></a>H3K27ac_CNR_filtered.GR.noBlacklist.df <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">'results/MECP2_H3K27ac_overlap/MECP2WT_H3K27ac_peakOverlap/MECP2_H3K27ac_overlap.bed'</span>,</span>
<span id="cb33-40"><a href="#cb33-40" aria-hidden="true" tabindex="-1"></a>                       <span class="at">sep=</span><span class="st">'</span><span class="sc">\t</span><span class="st">'</span>,<span class="at">header =</span> F)</span>
<span id="cb33-41"><a href="#cb33-41" aria-hidden="true" tabindex="-1"></a>H3K27ac_CNR_filtered.GR.noBlacklist.df<span class="sc">$</span>GeneID<span class="ot">=</span> <span class="fu">paste0</span>(H3K27ac_CNR_filtered.GR.noBlacklist.df<span class="sc">$</span>V1,<span class="st">'_'</span>,</span>
<span id="cb33-42"><a href="#cb33-42" aria-hidden="true" tabindex="-1"></a>                                                     H3K27ac_CNR_filtered.GR.noBlacklist.df<span class="sc">$</span>V2,<span class="st">'_'</span>,</span>
<span id="cb33-43"><a href="#cb33-43" aria-hidden="true" tabindex="-1"></a>                                                     H3K27ac_CNR_filtered.GR.noBlacklist.df<span class="sc">$</span>V3)</span>
<span id="cb33-44"><a href="#cb33-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-45"><a href="#cb33-45" aria-hidden="true" tabindex="-1"></a>H3K27ac_CNR.noBlacklist.df.saf <span class="ot">=</span> H3K27ac_CNR_filtered.GR.noBlacklist.df <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">c</span>(<span class="dv">4</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>))</span>
<span id="cb33-46"><a href="#cb33-46" aria-hidden="true" tabindex="-1"></a>H3K27ac_CNR.noBlacklist.df.saf<span class="sc">$</span>Strand <span class="ot">=</span> <span class="st">'*'</span></span>
<span id="cb33-47"><a href="#cb33-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-48"><a href="#cb33-48" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(H3K27ac_CNR.noBlacklist.df.saf) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"GeneID"</span>,<span class="st">"Chr"</span>,<span class="st">"Start"</span>,<span class="st">"End"</span>,<span class="st">"Strand"</span>)</span>
<span id="cb33-49"><a href="#cb33-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-50"><a href="#cb33-50" aria-hidden="true" tabindex="-1"></a>bams <span class="ot">=</span> <span class="fu">dir</span>(<span class="st">"../CUT_and_RUN/results/"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>, <span class="at">pattern =</span> <span class="st">"*.final.bam"</span>)</span>
<span id="cb33-51"><a href="#cb33-51" aria-hidden="true" tabindex="-1"></a>bams <span class="ot">=</span>  bams[<span class="sc">!</span>(<span class="fu">grepl</span>(<span class="st">'IgG'</span>,bams))]</span>
<span id="cb33-52"><a href="#cb33-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-53"><a href="#cb33-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-54"><a href="#cb33-54" aria-hidden="true" tabindex="-1"></a>bams <span class="ot">=</span> <span class="fu">dir</span>(<span class="st">"../CUT_and_RUN/results/bam/"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>, <span class="at">pattern =</span> <span class="st">"*.final.bam"</span>)</span>
<span id="cb33-55"><a href="#cb33-55" aria-hidden="true" tabindex="-1"></a>bams <span class="ot">=</span>  bams[<span class="sc">!</span>(<span class="fu">grepl</span>(<span class="st">'IgG|.bai|.sort'</span>,bams))]</span>
<span id="cb33-56"><a href="#cb33-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-57"><a href="#cb33-57" aria-hidden="true" tabindex="-1"></a>H3K27ac.bams <span class="ot">=</span> bams[<span class="fu">grepl</span>(<span class="st">'_H3K27ac'</span>,bams)][<span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb33-58"><a href="#cb33-58" aria-hidden="true" tabindex="-1"></a><span class="co">#bamsToCount = subset(bamsToCount,!grepl("*_2h*", bamsToCount))</span></span>
<span id="cb33-59"><a href="#cb33-59" aria-hidden="true" tabindex="-1"></a>H3K27ac.fcResults <span class="ot">=</span> <span class="fu">featureCounts</span>(H3K27ac.bams, </span>
<span id="cb33-60"><a href="#cb33-60" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">annot.ext =</span> H3K27ac_CNR.noBlacklist.df.saf,</span>
<span id="cb33-61"><a href="#cb33-61" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">isPairedEnd =</span> <span class="cn">TRUE</span>,</span>
<span id="cb33-62"><a href="#cb33-62" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">countMultiMappingReads =</span> <span class="cn">FALSE</span>,</span>
<span id="cb33-63"><a href="#cb33-63" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">maxFragLength =</span> <span class="dv">100</span>,<span class="at">nthreads =</span> <span class="dv">15</span>)</span>
<span id="cb33-64"><a href="#cb33-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-65"><a href="#cb33-65" aria-hidden="true" tabindex="-1"></a><span class="co">#save(H3K27ac.fcResults,file = "results/H3K27ac/H3K27ac.RData")</span></span>
<span id="cb33-66"><a href="#cb33-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-67"><a href="#cb33-67" aria-hidden="true" tabindex="-1"></a><span class="co">#load("results/H3K27ac/H3K27ac.RData",verbose = TRUE)</span></span>
<span id="cb33-68"><a href="#cb33-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-69"><a href="#cb33-69" aria-hidden="true" tabindex="-1"></a>H3K27ac.myCounts <span class="ot">=</span> H3K27ac.fcResults<span class="sc">$</span>counts <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() </span>
<span id="cb33-70"><a href="#cb33-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-71"><a href="#cb33-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-72"><a href="#cb33-72" aria-hidden="true" tabindex="-1"></a><span class="co"># Metadata</span></span>
<span id="cb33-73"><a href="#cb33-73" aria-hidden="true" tabindex="-1"></a>H3K27ac.metaData <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">Group=</span><span class="fu">colnames</span>(H3K27ac.myCounts), <span class="at">row.names =</span> <span class="fu">colnames</span>(H3K27ac.myCounts))</span>
<span id="cb33-74"><a href="#cb33-74" aria-hidden="true" tabindex="-1"></a>H3K27ac.metaData<span class="sc">$</span>Group <span class="ot">=</span> <span class="fu">c</span>(<span class="st">'MECP2_KO'</span>,<span class="st">'WT'</span>,<span class="st">'DNMT3A_KO'</span>,<span class="st">'WT'</span>,<span class="st">'MECP2_KO'</span>,<span class="st">'WT'</span>,<span class="st">'DNMT3A_KO'</span>,<span class="st">'WT'</span>)</span>
<span id="cb33-75"><a href="#cb33-75" aria-hidden="true" tabindex="-1"></a>H3K27ac.metaData<span class="sc">$</span>Batch <span class="ot">=</span> <span class="fu">rownames</span>(H3K27ac.metaData)</span>
<span id="cb33-76"><a href="#cb33-76" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(H3K27ac.metaData) <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">'_CNR_.*'</span>,<span class="st">''</span>,<span class="fu">rownames</span>(H3K27ac.metaData))</span>
<span id="cb33-77"><a href="#cb33-77" aria-hidden="true" tabindex="-1"></a>H3K27ac.metaData<span class="sc">$</span>Sample <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">'_H3K27ac'</span>,<span class="st">''</span>,<span class="fu">rownames</span>(H3K27ac.metaData))</span>
<span id="cb33-78"><a href="#cb33-78" aria-hidden="true" tabindex="-1"></a>H3K27ac.metaData<span class="sc">$</span>Batch <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">'.*_CTX_.*_CNR_|.final.bam'</span>,<span class="st">''</span>,H3K27ac.metaData<span class="sc">$</span>Batch)</span>
<span id="cb33-79"><a href="#cb33-79" aria-hidden="true" tabindex="-1"></a>H3K27ac.metaData<span class="sc">$</span>Sample <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">'_Mecp2'</span>,<span class="st">''</span>,<span class="fu">rownames</span>(H3K27ac.metaData))</span>
<span id="cb33-80"><a href="#cb33-80" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-81"><a href="#cb33-81" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(H3K27ac.myCounts) <span class="ot">=</span> <span class="fu">rownames</span>(H3K27ac.metaData)</span>
<span id="cb33-82"><a href="#cb33-82" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-83"><a href="#cb33-83" aria-hidden="true" tabindex="-1"></a><span class="co"># # Select only MECP2 WT and KO samples</span></span>
<span id="cb33-84"><a href="#cb33-84" aria-hidden="true" tabindex="-1"></a>y <span class="ot">=</span> <span class="fu">DGEList</span>(<span class="at">counts =</span> H3K27ac.myCounts[,<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">5</span>,<span class="dv">6</span>)])</span>
<span id="cb33-85"><a href="#cb33-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-86"><a href="#cb33-86" aria-hidden="true" tabindex="-1"></a>y<span class="sc">$</span>samples<span class="sc">$</span>Group <span class="ot">=</span> H3K27ac.metaData[<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">5</span>,<span class="dv">6</span>),]<span class="sc">$</span>Group</span>
<span id="cb33-87"><a href="#cb33-87" aria-hidden="true" tabindex="-1"></a>y<span class="sc">$</span>samples<span class="sc">$</span>Batch <span class="ot">=</span> H3K27ac.metaData[<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">5</span>,<span class="dv">6</span>),]<span class="sc">$</span>Batch</span>
<span id="cb33-88"><a href="#cb33-88" aria-hidden="true" tabindex="-1"></a>y<span class="sc">$</span>samples<span class="sc">$</span>Sample <span class="ot">=</span> H3K27ac.metaData[<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">5</span>,<span class="dv">6</span>),]<span class="sc">$</span>Sample</span>
<span id="cb33-89"><a href="#cb33-89" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-90"><a href="#cb33-90" aria-hidden="true" tabindex="-1"></a><span class="co"># Normalize and obtain logcounts for QC</span></span>
<span id="cb33-91"><a href="#cb33-91" aria-hidden="true" tabindex="-1"></a>y <span class="ot">=</span> <span class="fu">calcNormFactors</span>(y)</span>
<span id="cb33-92"><a href="#cb33-92" aria-hidden="true" tabindex="-1"></a>logCPMs <span class="ot">=</span> <span class="fu">cpm</span>(y, <span class="at">log =</span> <span class="cn">TRUE</span>)</span>
<span id="cb33-93"><a href="#cb33-93" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-94"><a href="#cb33-94" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-95"><a href="#cb33-95" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate rowwise variance</span></span>
<span id="cb33-96"><a href="#cb33-96" aria-hidden="true" tabindex="-1"></a>rv <span class="ot">=</span> <span class="fu">apply</span>(logCPMs, <span class="dv">1</span>, var)</span>
<span id="cb33-97"><a href="#cb33-97" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-98"><a href="#cb33-98" aria-hidden="true" tabindex="-1"></a><span class="co"># Sort decreasingly and take top 1000</span></span>
<span id="cb33-99"><a href="#cb33-99" aria-hidden="true" tabindex="-1"></a>o <span class="ot">=</span> <span class="fu">order</span>(rv, <span class="at">decreasing=</span><span class="cn">TRUE</span>)</span>
<span id="cb33-100"><a href="#cb33-100" aria-hidden="true" tabindex="-1"></a>top1000 <span class="ot">=</span> <span class="fu">head</span>(o, <span class="dv">1000</span>)</span>
<span id="cb33-101"><a href="#cb33-101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-102"><a href="#cb33-102" aria-hidden="true" tabindex="-1"></a><span class="co"># From the logCPMs subset for the top-1000</span></span>
<span id="cb33-103"><a href="#cb33-103" aria-hidden="true" tabindex="-1"></a>logCPM_top1000 <span class="ot">=</span> logCPMs[top1000,]</span>
<span id="cb33-104"><a href="#cb33-104" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-105"><a href="#cb33-105" aria-hidden="true" tabindex="-1"></a><span class="co"># Run PCA</span></span>
<span id="cb33-106"><a href="#cb33-106" aria-hidden="true" tabindex="-1"></a>pca <span class="ot">=</span> <span class="fu">prcomp</span>(<span class="fu">t</span>(logCPM_top1000))</span>
<span id="cb33-107"><a href="#cb33-107" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-108"><a href="#cb33-108" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine PCA coordinates with the metadata from the DGEList</span></span>
<span id="cb33-109"><a href="#cb33-109" aria-hidden="true" tabindex="-1"></a>to_plot <span class="ot">=</span> <span class="fu">data.frame</span>(pca<span class="sc">$</span>x, y<span class="sc">$</span>samples)</span>
<span id="cb33-110"><a href="#cb33-110" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-111"><a href="#cb33-111" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate how many % of total variance is explained by each principal component</span></span>
<span id="cb33-112"><a href="#cb33-112" aria-hidden="true" tabindex="-1"></a>percentVar <span class="ot">=</span> pca<span class="sc">$</span>sdev<span class="sc">^</span><span class="dv">2</span> <span class="sc">/</span> <span class="fu">sum</span>( pca<span class="sc">$</span>sdev<span class="sc">^</span><span class="dv">2</span> )<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb33-113"><a href="#cb33-113" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-114"><a href="#cb33-114" aria-hidden="true" tabindex="-1"></a><span class="co"># We focus here on PC1 and PC2</span></span>
<span id="cb33-115"><a href="#cb33-115" aria-hidden="true" tabindex="-1"></a>use.pcs <span class="ot">=</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>)</span>
<span id="cb33-116"><a href="#cb33-116" aria-hidden="true" tabindex="-1"></a>labs <span class="ot">=</span> <span class="fu">paste0</span>(<span class="fu">paste0</span>(<span class="st">"PC"</span>, use.pcs, <span class="st">" - "</span>), <span class="fu">paste0</span>(<span class="st">"Var.expl = "</span>, <span class="fu">round</span>(percentVar[use.pcs], <span class="dv">2</span>), <span class="st">"%"</span>))</span>
<span id="cb33-117"><a href="#cb33-117" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-118"><a href="#cb33-118" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(to_plot, <span class="fu">aes</span>(<span class="at">x=</span>PC1, <span class="at">y=</span>PC2, <span class="at">color=</span>Group, <span class="at">shape=</span>Sample)) <span class="sc">+</span> </span>
<span id="cb33-119"><a href="#cb33-119" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb33-120"><a href="#cb33-120" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_shape_manual</span>(<span class="at">values=</span><span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>)<span class="sc">+</span></span>
<span id="cb33-121"><a href="#cb33-121" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(labs[<span class="dv">1</span>]) <span class="sc">+</span> <span class="fu">ylab</span>(labs[<span class="dv">2</span>])</span>
<span id="cb33-122"><a href="#cb33-122" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-123"><a href="#cb33-123" aria-hidden="true" tabindex="-1"></a><span class="co"># correct for the Batch </span></span>
<span id="cb33-124"><a href="#cb33-124" aria-hidden="true" tabindex="-1"></a>batch <span class="ot">=</span> <span class="fu">factor</span>(y<span class="sc">$</span>samples<span class="sc">$</span>Batch)</span>
<span id="cb33-125"><a href="#cb33-125" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-126"><a href="#cb33-126" aria-hidden="true" tabindex="-1"></a>logCPMs_corrected <span class="ot">=</span> limma<span class="sc">::</span><span class="fu">removeBatchEffect</span>(logCPMs, <span class="at">batch =</span> batch)</span>
<span id="cb33-127"><a href="#cb33-127" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-128"><a href="#cb33-128" aria-hidden="true" tabindex="-1"></a><span class="co"># repeat PCA as before, using the same genes</span></span>
<span id="cb33-129"><a href="#cb33-129" aria-hidden="true" tabindex="-1"></a>logCPM_corrected_top1000 <span class="ot">=</span> logCPMs_corrected[top1000,]</span>
<span id="cb33-130"><a href="#cb33-130" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-131"><a href="#cb33-131" aria-hidden="true" tabindex="-1"></a><span class="co"># Run PCA</span></span>
<span id="cb33-132"><a href="#cb33-132" aria-hidden="true" tabindex="-1"></a>pca <span class="ot">=</span> <span class="fu">prcomp</span>(<span class="fu">t</span>(logCPM_corrected_top1000))</span>
<span id="cb33-133"><a href="#cb33-133" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-134"><a href="#cb33-134" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine PCA coordinates with the metadata from the DGEList</span></span>
<span id="cb33-135"><a href="#cb33-135" aria-hidden="true" tabindex="-1"></a>to_plot <span class="ot">=</span> <span class="fu">data.frame</span>(pca<span class="sc">$</span>x, y<span class="sc">$</span>samples)</span>
<span id="cb33-136"><a href="#cb33-136" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-137"><a href="#cb33-137" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate how many % of total variance is explained by each principal component</span></span>
<span id="cb33-138"><a href="#cb33-138" aria-hidden="true" tabindex="-1"></a>percentVar <span class="ot">=</span> pca<span class="sc">$</span>sdev<span class="sc">^</span><span class="dv">2</span> <span class="sc">/</span> <span class="fu">sum</span>( pca<span class="sc">$</span>sdev<span class="sc">^</span><span class="dv">2</span> )<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb33-139"><a href="#cb33-139" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-140"><a href="#cb33-140" aria-hidden="true" tabindex="-1"></a><span class="co"># We focus here on PC1 and PC2</span></span>
<span id="cb33-141"><a href="#cb33-141" aria-hidden="true" tabindex="-1"></a>use.pcs <span class="ot">=</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>)</span>
<span id="cb33-142"><a href="#cb33-142" aria-hidden="true" tabindex="-1"></a>labs <span class="ot">=</span> <span class="fu">paste0</span>(<span class="st">"PC"</span>, use.pcs, <span class="st">" - "</span>,<span class="fu">round</span>(percentVar[use.pcs], <span class="dv">2</span>), <span class="st">"%"</span>)</span>
<span id="cb33-143"><a href="#cb33-143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-144"><a href="#cb33-144" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(to_plot, <span class="fu">aes</span>(<span class="at">x=</span>PC1, <span class="at">y=</span>PC2, <span class="at">color=</span>Group, <span class="at">shape=</span>Sample)) <span class="sc">+</span> </span>
<span id="cb33-145"><a href="#cb33-145" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">5</span>,<span class="at">stroke =</span> <span class="fl">1.8</span>) <span class="sc">+</span></span>
<span id="cb33-146"><a href="#cb33-146" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_shape_manual</span>(<span class="at">values=</span><span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>)<span class="sc">+</span></span>
<span id="cb33-147"><a href="#cb33-147" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(labs[<span class="dv">1</span>]) <span class="sc">+</span> <span class="fu">ylab</span>(labs[<span class="dv">2</span>]) <span class="sc">+</span></span>
<span id="cb33-148"><a href="#cb33-148" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_test</span>(<span class="dv">10</span>) <span class="sc">+</span></span>
<span id="cb33-149"><a href="#cb33-149" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">colour =</span> <span class="fu">guide_legend</span>(<span class="at">override.aes =</span> <span class="fu">list</span>(<span class="at">size=</span><span class="dv">2</span>)))<span class="sc">+</span></span>
<span id="cb33-150"><a href="#cb33-150" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"PCA"</span>) </span>
<span id="cb33-151"><a href="#cb33-151" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"Plots/H3K27ac_MECP2KO_vs_WT_PCA.pdf"</span>,<span class="at">units =</span> <span class="st">'cm'</span>,<span class="at">height =</span> <span class="dv">8</span>,<span class="at">width =</span> <span class="dv">14</span>)</span>
<span id="cb33-152"><a href="#cb33-152" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-153"><a href="#cb33-153" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-154"><a href="#cb33-154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-155"><a href="#cb33-155" aria-hidden="true" tabindex="-1"></a><span class="co"># Include Batch in the design matrix</span></span>
<span id="cb33-156"><a href="#cb33-156" aria-hidden="true" tabindex="-1"></a>design <span class="ot">=</span> <span class="fu">model.matrix</span>(<span class="sc">~</span>Batch<span class="sc">+</span>Group, y<span class="sc">$</span>samples)</span>
<span id="cb33-157"><a href="#cb33-157" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-158"><a href="#cb33-158" aria-hidden="true" tabindex="-1"></a><span class="co"># QLF workflow from edgeR</span></span>
<span id="cb33-159"><a href="#cb33-159" aria-hidden="true" tabindex="-1"></a>y <span class="ot">=</span> <span class="fu">estimateDisp</span>(y, design)</span>
<span id="cb33-160"><a href="#cb33-160" aria-hidden="true" tabindex="-1"></a>fit <span class="ot">=</span> <span class="fu">glmQLFit</span>(y, design)</span>
<span id="cb33-161"><a href="#cb33-161" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-162"><a href="#cb33-162" aria-hidden="true" tabindex="-1"></a><span class="co"># include column (factor) for which differential enhancer need to tested</span></span>
<span id="cb33-163"><a href="#cb33-163" aria-hidden="true" tabindex="-1"></a>fit  <span class="ot">=</span> <span class="fu">glmQLFTest</span>(fit, <span class="at">coef =</span> <span class="dv">3</span>)</span>
<span id="cb33-164"><a href="#cb33-164" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-165"><a href="#cb33-165" aria-hidden="true" tabindex="-1"></a>tt <span class="ot">=</span> <span class="fu">topTable</span>(fit, <span class="at">coef=</span><span class="dv">3</span>)  <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">'Gene'</span>)</span>
<span id="cb33-166"><a href="#cb33-166" aria-hidden="true" tabindex="-1"></a>cp <span class="ot">=</span> <span class="fu">cpm</span>(y) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">'Gene'</span>)</span>
<span id="cb33-167"><a href="#cb33-167" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-168"><a href="#cb33-168" aria-hidden="true" tabindex="-1"></a><span class="co">#/ 1/2 is "A" (=the reference) and 3/4 is "B", positive logFC mean higher in B,</span></span>
<span id="cb33-169"><a href="#cb33-169" aria-hidden="true" tabindex="-1"></a><span class="fu">merge</span>(<span class="at">x=</span>tt, <span class="at">y=</span>cp, <span class="at">by=</span><span class="st">"Gene"</span>)[<span class="dv">2</span>,<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">8</span>,<span class="dv">9</span>,<span class="dv">10</span>,<span class="dv">11</span>)]</span>
<span id="cb33-170"><a href="#cb33-170" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-171"><a href="#cb33-171" aria-hidden="true" tabindex="-1"></a><span class="co"># get stats as a data.frame</span></span>
<span id="cb33-172"><a href="#cb33-172" aria-hidden="true" tabindex="-1"></a>DiffEnhacer <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="fu">topTags</span>(fit, <span class="at">n=</span><span class="cn">Inf</span>))</span>
<span id="cb33-173"><a href="#cb33-173" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-174"><a href="#cb33-174" aria-hidden="true" tabindex="-1"></a><span class="co"># Classify enhancer into significantly up and down</span></span>
<span id="cb33-175"><a href="#cb33-175" aria-hidden="true" tabindex="-1"></a>DiffEnhancer.mod <span class="ot">=</span> DiffEnhacer <span class="sc">%&gt;%</span> </span>
<span id="cb33-176"><a href="#cb33-176" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Diff_H3K27ac=</span><span class="fu">factor</span>(<span class="fu">case_when</span>(logFC<span class="sc">&gt;</span><span class="dv">0</span> <span class="sc">&amp;</span> FDR<span class="sc">&lt;</span><span class="fl">0.1</span> <span class="sc">&amp;</span> logCPM <span class="sc">&gt;</span> <span class="dv">2</span> <span class="sc">~</span> <span class="st">"Up"</span>,</span>
<span id="cb33-177"><a href="#cb33-177" aria-hidden="true" tabindex="-1"></a>                                       logFC<span class="sc">&lt;</span><span class="dv">0</span> <span class="sc">&amp;</span> FDR<span class="sc">&lt;</span><span class="fl">0.1</span> <span class="sc">&amp;</span> logCPM <span class="sc">&gt;</span> <span class="dv">2</span> <span class="sc">~</span> <span class="st">"Down"</span>,</span>
<span id="cb33-178"><a href="#cb33-178" aria-hidden="true" tabindex="-1"></a>                                       <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"not significant"</span>),</span>
<span id="cb33-179"><a href="#cb33-179" aria-hidden="true" tabindex="-1"></a>                             <span class="at">levels=</span><span class="fu">c</span>(<span class="st">"Up"</span>, <span class="st">"not significant"</span>, <span class="st">"Down"</span>)))</span>
<span id="cb33-180"><a href="#cb33-180" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-181"><a href="#cb33-181" aria-hidden="true" tabindex="-1"></a><span class="co"># MA-plot</span></span>
<span id="cb33-182"><a href="#cb33-182" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(DiffEnhancer.mod,<span class="co"># %&gt;% filter(Diff_H3K27ac != 'not significant'), </span></span>
<span id="cb33-183"><a href="#cb33-183" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x=</span>logCPM, <span class="at">y=</span>logFC, <span class="at">color=</span>Diff_H3K27ac)) <span class="sc">+</span></span>
<span id="cb33-184"><a href="#cb33-184" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">0</span>) <span class="sc">+</span></span>
<span id="cb33-185"><a href="#cb33-185" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values=</span><span class="fu">c</span>( <span class="st">"blue"</span>,<span class="st">'grey'</span>,<span class="st">"red"</span>)) <span class="sc">+</span></span>
<span id="cb33-186"><a href="#cb33-186" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"H3K27ac (MECP2 WT vs KO)"</span>) <span class="sc">+</span></span>
<span id="cb33-187"><a href="#cb33-187" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_test</span>(<span class="dv">15</span>) <span class="sc">+</span></span>
<span id="cb33-188"><a href="#cb33-188" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">colour =</span> <span class="fu">guide_legend</span>(<span class="at">override.aes =</span> <span class="fu">list</span>(<span class="at">size=</span><span class="dv">2</span>)))<span class="sc">+</span></span>
<span id="cb33-189"><a href="#cb33-189" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_vline(xintercept = 2,linetype = 'dashed')+</span></span>
<span id="cb33-190"><a href="#cb33-190" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="fl">6.5</span>, <span class="at">y =</span> <span class="sc">-</span><span class="dv">1</span>, </span>
<span id="cb33-191"><a href="#cb33-191" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> DiffEnhancer.mod <span class="sc">%&gt;%</span> </span>
<span id="cb33-192"><a href="#cb33-192" aria-hidden="true" tabindex="-1"></a>             dplyr<span class="sc">::</span><span class="fu">filter</span>(Diff_H3K27ac<span class="sc">==</span><span class="st">'Down'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb33-193"><a href="#cb33-193" aria-hidden="true" tabindex="-1"></a>             <span class="fu">dim</span>() <span class="sc">%&gt;%</span> .[<span class="dv">1</span>], <span class="at">color=</span><span class="st">"red"</span>,<span class="at">size=</span><span class="dv">4</span>)<span class="sc">+</span></span>
<span id="cb33-194"><a href="#cb33-194" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="fl">6.5</span>, <span class="at">y =</span> <span class="dv">1</span>,  </span>
<span id="cb33-195"><a href="#cb33-195" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> DiffEnhancer.mod <span class="sc">%&gt;%</span> </span>
<span id="cb33-196"><a href="#cb33-196" aria-hidden="true" tabindex="-1"></a>             dplyr<span class="sc">::</span><span class="fu">filter</span>(Diff_H3K27ac<span class="sc">==</span><span class="st">'Up'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb33-197"><a href="#cb33-197" aria-hidden="true" tabindex="-1"></a>             <span class="fu">dim</span>() <span class="sc">%&gt;%</span> .[<span class="dv">1</span>], <span class="at">color=</span><span class="st">"blue"</span>,<span class="at">size=</span><span class="dv">4</span>)<span class="sc">+</span></span>
<span id="cb33-198"><a href="#cb33-198" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">'logFC (MECP2 WT vs KO)'</span>)</span>
<span id="cb33-199"><a href="#cb33-199" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"Plots/H3K27ac_MECP2_MECP2WT_vs_KO.pdf"</span>,<span class="at">units =</span> <span class="st">'cm'</span>,<span class="at">height =</span> <span class="dv">10</span>,<span class="at">width =</span> <span class="dv">14</span>)</span>
<span id="cb33-200"><a href="#cb33-200" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-201"><a href="#cb33-201" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-202"><a href="#cb33-202" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter Enahncer based on logCPM value</span></span>
<span id="cb33-203"><a href="#cb33-203" aria-hidden="true" tabindex="-1"></a>Enhacer.Up_in_MECP2.KO  <span class="ot">=</span> DiffEnhancer.mod <span class="sc">%&gt;%</span> <span class="fu">filter</span>(logCPM <span class="sc">&gt;</span><span class="dv">2</span> <span class="sc">&amp;</span> Diff_H3K27ac <span class="sc">==</span> <span class="st">'Up'</span>)</span>
<span id="cb33-204"><a href="#cb33-204" aria-hidden="true" tabindex="-1"></a>Enhacer.Down_in_MECP2.KO  <span class="ot">=</span> DiffEnhancer.mod <span class="sc">%&gt;%</span> <span class="fu">filter</span>(logCPM <span class="sc">&gt;</span><span class="dv">2</span> <span class="sc">&amp;</span> Diff_H3K27ac <span class="sc">==</span> <span class="st">'Down'</span>)</span>
<span id="cb33-205"><a href="#cb33-205" aria-hidden="true" tabindex="-1"></a>Enhacer.Unchanged_in_MECP2.KO  <span class="ot">=</span> DiffEnhancer.mod <span class="sc">%&gt;%</span> <span class="fu">filter</span>(logCPM <span class="sc">&gt;</span><span class="dv">2</span> <span class="sc">&amp;</span> Diff_H3K27ac <span class="sc">==</span> <span class="st">'not significant'</span>)</span>
<span id="cb33-206"><a href="#cb33-206" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-207"><a href="#cb33-207" aria-hidden="true" tabindex="-1"></a>H3K27ac_CNR.noBlacklist.df.saf <span class="sc">%&gt;%</span> </span>
<span id="cb33-208"><a href="#cb33-208" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(GeneID <span class="sc">%in%</span> <span class="fu">rownames</span>(Enhacer.Up_in_MECP2.KO)) <span class="sc">%&gt;%</span> </span>
<span id="cb33-209"><a href="#cb33-209" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">1</span>) <span class="sc">%&gt;%</span> <span class="fu">dim</span>()</span>
<span id="cb33-210"><a href="#cb33-210" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write.table</span>(<span class="at">file =</span> <span class="st">"results/H3K27ac/H3K27ac_MECP2_up_MECP2WT.vs.KO.bed"</span>,<span class="at">sep=</span><span class="st">'</span><span class="sc">\t</span><span class="st">'</span>,</span>
<span id="cb33-211"><a href="#cb33-211" aria-hidden="true" tabindex="-1"></a>              <span class="at">quote =</span> <span class="cn">FALSE</span>,<span class="at">row.names =</span> F,<span class="at">col.names =</span> F)</span>
<span id="cb33-212"><a href="#cb33-212" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-213"><a href="#cb33-213" aria-hidden="true" tabindex="-1"></a>H3K27ac_CNR.noBlacklist.df.saf <span class="sc">%&gt;%</span> </span>
<span id="cb33-214"><a href="#cb33-214" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(GeneID <span class="sc">%in%</span> <span class="fu">rownames</span>(Enhacer.Down_in_MECP2.KO)) <span class="sc">%&gt;%</span> </span>
<span id="cb33-215"><a href="#cb33-215" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb33-216"><a href="#cb33-216" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write.table</span>(<span class="at">file =</span> <span class="st">"results/H3K27ac/H3K27ac_MECP2_down_MECP2WT.vs.KO.bed"</span>,<span class="at">sep=</span><span class="st">'</span><span class="sc">\t</span><span class="st">'</span>,</span>
<span id="cb33-217"><a href="#cb33-217" aria-hidden="true" tabindex="-1"></a>              <span class="at">quote =</span> <span class="cn">FALSE</span>,<span class="at">row.names =</span> F,<span class="at">col.names =</span> F)</span>
<span id="cb33-218"><a href="#cb33-218" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-219"><a href="#cb33-219" aria-hidden="true" tabindex="-1"></a>H3K27ac_CNR.noBlacklist.df.saf <span class="sc">%&gt;%</span> </span>
<span id="cb33-220"><a href="#cb33-220" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(GeneID <span class="sc">%in%</span> <span class="fu">rownames</span>(Enhacer.Unchanged_in_MECP2.KO)) <span class="sc">%&gt;%</span> </span>
<span id="cb33-221"><a href="#cb33-221" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb33-222"><a href="#cb33-222" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write.table</span>(<span class="at">file =</span> <span class="st">"results/H3K27ac/H3K27ac_MECP2_unchanged_MECP2WT.vs.KO.bed"</span>,<span class="at">sep=</span><span class="st">'</span><span class="sc">\t</span><span class="st">'</span>,</span>
<span id="cb33-223"><a href="#cb33-223" aria-hidden="true" tabindex="-1"></a>              <span class="at">quote =</span> <span class="cn">FALSE</span>,<span class="at">row.names =</span> F,<span class="at">col.names =</span> F)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></li>
<li><p>Figure 3D.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"results/RData/Boxer_RNAseq_data.RData"</span>)</span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a><span class="co"># read MECP2 peaks with Differential enhancer </span></span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a>MECP2_h3k27ac <span class="ot">=</span> <span class="fu">list.files</span>(<span class="st">'results/MECP2_H3K27ac_overlap/'</span>,<span class="at">pattern =</span> <span class="st">"*.bed"</span>,</span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a>                           <span class="at">full.names =</span> <span class="cn">TRUE</span>)[<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">4</span>,<span class="dv">5</span>)]</span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a>MECP2_h3k27ac.df <span class="ot">=</span> <span class="fu">lapply</span>(MECP2_h3k27ac, readPeakFile)</span>
<span id="cb34-7"><a href="#cb34-7" aria-hidden="true" tabindex="-1"></a>MECP2_h3k27ac.df.anno <span class="ot">=</span> <span class="fu">lapply</span>(MECP2_h3k27ac.df, annotatePeak, <span class="at">tssRegion=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">1000</span>, <span class="dv">1000</span>),</span>
<span id="cb34-8"><a href="#cb34-8" aria-hidden="true" tabindex="-1"></a>                               <span class="at">TxDb=</span>txdb, <span class="at">annoDb=</span><span class="st">"org.Mm.eg.db"</span>)</span>
<span id="cb34-9"><a href="#cb34-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-10"><a href="#cb34-10" aria-hidden="true" tabindex="-1"></a>MECP2_h3k27ac.df.anno <span class="ot">=</span> <span class="fu">lapply</span>(MECP2_h3k27ac.df.anno, as.data.frame)</span>
<span id="cb34-11"><a href="#cb34-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-12"><a href="#cb34-12" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(MECP2_h3k27ac.df.anno) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">'Gain'</span>,<span class="st">'No Change'</span>,<span class="st">'Loss'</span>)</span>
<span id="cb34-13"><a href="#cb34-13" aria-hidden="true" tabindex="-1"></a>MECP2_h3k27ac.df <span class="ot">=</span> <span class="fu">rbindlist</span>(MECP2_h3k27ac.df.anno,<span class="at">idcol =</span> <span class="st">'Group'</span>)</span>
<span id="cb34-14"><a href="#cb34-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-15"><a href="#cb34-15" aria-hidden="true" tabindex="-1"></a><span class="co">#MECP2_h3k27ac.df = merge(MECP2_h3k27ac.df %&gt;% select(1,7,17), DiffEnhancer.mod %&gt;% rownames_to_column(), by.x ='V4',by.y='rowname')</span></span>
<span id="cb34-16"><a href="#cb34-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-17"><a href="#cb34-17" aria-hidden="true" tabindex="-1"></a>diff_h3k27c_mecp2_genes <span class="ot">=</span> <span class="fu">merge</span>(MECP2_h3k27ac.df[,<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">16</span>)] <span class="sc">%&gt;%</span> <span class="fu">distinct</span>() <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(ENSEMBL)) ,</span>
<span id="cb34-18"><a href="#cb34-18" aria-hidden="true" tabindex="-1"></a>                                RNAseq.list[[<span class="dv">1</span>]],<span class="at">by=</span><span class="st">'ENSEMBL'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb34-19"><a href="#cb34-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-20"><a href="#cb34-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(FDR <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb34-21"><a href="#cb34-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Group =</span> <span class="fu">factor</span>(Group,<span class="at">levels=</span><span class="fu">c</span>(<span class="st">'Loss'</span>,<span class="st">'No Change'</span>,<span class="st">'Gain'</span>))) <span class="co">#%&gt;%</span></span>
<span id="cb34-22"><a href="#cb34-22" aria-hidden="true" tabindex="-1"></a>  <span class="co">#filter(Group != 'No Change')</span></span>
<span id="cb34-23"><a href="#cb34-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-24"><a href="#cb34-24" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(diff_h3k27c_mecp2_genes,(<span class="fu">aes</span>(<span class="at">x=</span>Group,<span class="at">y=</span>logFC))) <span class="sc">+</span></span>
<span id="cb34-25"><a href="#cb34-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>()<span class="sc">+</span></span>
<span id="cb34-26"><a href="#cb34-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_test</span>(<span class="dv">15</span>)<span class="sc">+</span></span>
<span id="cb34-27"><a href="#cb34-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>(<span class="at">comparisons =</span> <span class="fu">list</span>(<span class="fu">c</span>(<span class="st">'Gain'</span>,<span class="st">'No Change'</span>),</span>
<span id="cb34-28"><a href="#cb34-28" aria-hidden="true" tabindex="-1"></a>                                          <span class="fu">c</span>(<span class="st">'Gain'</span>,<span class="st">'Loss'</span>),</span>
<span id="cb34-29"><a href="#cb34-29" aria-hidden="true" tabindex="-1"></a>                                          <span class="fu">c</span>(<span class="st">'Loss'</span>,<span class="st">'No Change'</span>)))<span class="sc">+</span></span>
<span id="cb34-30"><a href="#cb34-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>,<span class="at">vjust=</span><span class="dv">1</span>,<span class="at">hjust=</span><span class="dv">1</span>))<span class="sc">+</span></span>
<span id="cb34-31"><a href="#cb34-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">'H3K27ac (MECP2 KO vs WT)'</span>)<span class="sc">+</span></span>
<span id="cb34-32"><a href="#cb34-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">'logFC (MECP2 KO vs WT WholeCell RNA-seq)'</span>)</span>
<span id="cb34-33"><a href="#cb34-33" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">'Plots/MBHs_enhacer_regulated_genes.pdf'</span>,<span class="at">height =</span> <span class="dv">6</span>,<span class="at">width =</span> <span class="dv">4</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></li>
</ol>
</section>
<section id="figure-4-mecp2-preferentially-represses-genes-harboring-multiple-mbh-sites." class="level3">
<h3 class="anchored" data-anchor-id="figure-4-mecp2-preferentially-represses-genes-harboring-multiple-mbh-sites."><strong>Figure 4: MECP2 preferentially represses genes harboring multiple MBH sites.</strong></h3>
<ol type="1">
<li><p>Figure 4A.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">'results/MECP2/MECP2_regulatedGenes.RData'</span>)</span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a>MECP2_regulatedGenes.statistics <span class="ot">=</span> <span class="fu">data.frame</span>(</span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a>  <span class="st">"RNA-seq"</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">'wholeCell'</span>,<span class="dv">6</span>),<span class="fu">rep</span>(<span class="st">'Nuclear'</span>,<span class="dv">6</span>),<span class="fu">rep</span>(<span class="st">'Chromatin'</span>,<span class="dv">6</span>)),</span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a>  <span class="st">"MECP2"</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">'Bound'</span>,<span class="dv">3</span>),<span class="fu">rep</span>(<span class="st">'Unbound'</span>,<span class="dv">3</span>)),<span class="dv">3</span>)),</span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Status"</span> <span class="ot">=</span> <span class="fu">rep</span>(<span class="fu">c</span>(<span class="st">"Repressed"</span>,<span class="st">"Activated"</span>,<span class="st">"Unchanged"</span>),<span class="dv">6</span>),</span>
<span id="cb35-6"><a href="#cb35-6" aria-hidden="true" tabindex="-1"></a>  <span class="st">"DEGs"</span> <span class="ot">=</span> <span class="fu">c</span>(</span>
<span id="cb35-7"><a href="#cb35-7" aria-hidden="true" tabindex="-1"></a>Mecp2.Bound.meth.expressed.wholeCell   <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(DEG <span class="sc">==</span> <span class="st">'Repressed'</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(ENSEMBL) <span class="sc">%&gt;%</span> <span class="fu">unique</span>() <span class="sc">%&gt;%</span> <span class="fu">length</span>(),</span>
<span id="cb35-8"><a href="#cb35-8" aria-hidden="true" tabindex="-1"></a>Mecp2.Bound.meth.expressed.wholeCell   <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(DEG <span class="sc">==</span> <span class="st">'Activated'</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(ENSEMBL) <span class="sc">%&gt;%</span> <span class="fu">unique</span>() <span class="sc">%&gt;%</span> <span class="fu">length</span>(),</span>
<span id="cb35-9"><a href="#cb35-9" aria-hidden="true" tabindex="-1"></a>Mecp2.Bound.meth.expressed.wholeCell   <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(DEG <span class="sc">==</span> <span class="st">'Unchanged'</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(ENSEMBL) <span class="sc">%&gt;%</span> <span class="fu">unique</span>() <span class="sc">%&gt;%</span> <span class="fu">length</span>(),</span>
<span id="cb35-10"><a href="#cb35-10" aria-hidden="true" tabindex="-1"></a>Mecp2.UnBound.meth.expressed.wholeCell <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(DEG <span class="sc">==</span> <span class="st">'Repressed'</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(ENSEMBL) <span class="sc">%&gt;%</span> <span class="fu">unique</span>() <span class="sc">%&gt;%</span> <span class="fu">length</span>(),</span>
<span id="cb35-11"><a href="#cb35-11" aria-hidden="true" tabindex="-1"></a>Mecp2.UnBound.meth.expressed.wholeCell <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(DEG <span class="sc">==</span> <span class="st">'Activated'</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(ENSEMBL) <span class="sc">%&gt;%</span> <span class="fu">unique</span>() <span class="sc">%&gt;%</span> <span class="fu">length</span>(),</span>
<span id="cb35-12"><a href="#cb35-12" aria-hidden="true" tabindex="-1"></a>Mecp2.UnBound.meth.expressed.wholeCell <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(DEG <span class="sc">==</span> <span class="st">'Unchanged'</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(ENSEMBL) <span class="sc">%&gt;%</span> <span class="fu">unique</span>() <span class="sc">%&gt;%</span> <span class="fu">length</span>(),</span>
<span id="cb35-13"><a href="#cb35-13" aria-hidden="true" tabindex="-1"></a>Mecp2.Bound.meth.expressed.nuclear     <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(DEG <span class="sc">==</span> <span class="st">'Repressed'</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(ENSEMBL) <span class="sc">%&gt;%</span> <span class="fu">unique</span>() <span class="sc">%&gt;%</span> <span class="fu">length</span>(),</span>
<span id="cb35-14"><a href="#cb35-14" aria-hidden="true" tabindex="-1"></a>Mecp2.Bound.meth.expressed.nuclear     <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(DEG <span class="sc">==</span> <span class="st">'Activated'</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(ENSEMBL) <span class="sc">%&gt;%</span> <span class="fu">unique</span>() <span class="sc">%&gt;%</span> <span class="fu">length</span>(),</span>
<span id="cb35-15"><a href="#cb35-15" aria-hidden="true" tabindex="-1"></a>Mecp2.Bound.meth.expressed.nuclear     <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(DEG <span class="sc">==</span> <span class="st">'Unchanged'</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(ENSEMBL) <span class="sc">%&gt;%</span> <span class="fu">unique</span>() <span class="sc">%&gt;%</span> <span class="fu">length</span>(),</span>
<span id="cb35-16"><a href="#cb35-16" aria-hidden="true" tabindex="-1"></a>Mecp2.UnBound.meth.expressed.nuclear   <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(DEG <span class="sc">==</span> <span class="st">'Repressed'</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(ENSEMBL) <span class="sc">%&gt;%</span> <span class="fu">unique</span>() <span class="sc">%&gt;%</span> <span class="fu">length</span>(),</span>
<span id="cb35-17"><a href="#cb35-17" aria-hidden="true" tabindex="-1"></a>Mecp2.UnBound.meth.expressed.nuclear   <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(DEG <span class="sc">==</span> <span class="st">'Activated'</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(ENSEMBL) <span class="sc">%&gt;%</span> <span class="fu">unique</span>() <span class="sc">%&gt;%</span> <span class="fu">length</span>(),</span>
<span id="cb35-18"><a href="#cb35-18" aria-hidden="true" tabindex="-1"></a>Mecp2.UnBound.meth.expressed.nuclear   <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(DEG <span class="sc">==</span> <span class="st">'Unchanged'</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(ENSEMBL) <span class="sc">%&gt;%</span> <span class="fu">unique</span>() <span class="sc">%&gt;%</span> <span class="fu">length</span>(),</span>
<span id="cb35-19"><a href="#cb35-19" aria-hidden="true" tabindex="-1"></a>Mecp2.Bound.meth.expressed.chromatin   <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(DEG <span class="sc">==</span> <span class="st">'Repressed'</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(ENSEMBL) <span class="sc">%&gt;%</span> <span class="fu">unique</span>() <span class="sc">%&gt;%</span> <span class="fu">length</span>(),</span>
<span id="cb35-20"><a href="#cb35-20" aria-hidden="true" tabindex="-1"></a>Mecp2.Bound.meth.expressed.chromatin   <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(DEG <span class="sc">==</span> <span class="st">'Activated'</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(ENSEMBL) <span class="sc">%&gt;%</span> <span class="fu">unique</span>() <span class="sc">%&gt;%</span> <span class="fu">length</span>(),</span>
<span id="cb35-21"><a href="#cb35-21" aria-hidden="true" tabindex="-1"></a>Mecp2.Bound.meth.expressed.chromatin   <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(DEG <span class="sc">==</span> <span class="st">'Unchanged'</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(ENSEMBL) <span class="sc">%&gt;%</span> <span class="fu">unique</span>() <span class="sc">%&gt;%</span> <span class="fu">length</span>(),</span>
<span id="cb35-22"><a href="#cb35-22" aria-hidden="true" tabindex="-1"></a>Mecp2.UnBound.meth.expressed.chromatin <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(DEG <span class="sc">==</span> <span class="st">'Repressed'</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(ENSEMBL) <span class="sc">%&gt;%</span> <span class="fu">unique</span>() <span class="sc">%&gt;%</span> <span class="fu">length</span>(),</span>
<span id="cb35-23"><a href="#cb35-23" aria-hidden="true" tabindex="-1"></a>Mecp2.UnBound.meth.expressed.chromatin <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(DEG <span class="sc">==</span> <span class="st">'Activated'</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(ENSEMBL) <span class="sc">%&gt;%</span> <span class="fu">unique</span>() <span class="sc">%&gt;%</span> <span class="fu">length</span>(),</span>
<span id="cb35-24"><a href="#cb35-24" aria-hidden="true" tabindex="-1"></a>Mecp2.UnBound.meth.expressed.chromatin <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(DEG <span class="sc">==</span> <span class="st">'Unchanged'</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(ENSEMBL) <span class="sc">%&gt;%</span> <span class="fu">unique</span>() <span class="sc">%&gt;%</span> <span class="fu">length</span>()))</span>
<span id="cb35-25"><a href="#cb35-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-26"><a href="#cb35-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-27"><a href="#cb35-27" aria-hidden="true" tabindex="-1"></a>MECP2_regulatedGenes.statistics <span class="sc">%&gt;%</span> </span>
<span id="cb35-28"><a href="#cb35-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Status <span class="sc">!=</span><span class="st">"Unchanged"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb35-29"><a href="#cb35-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">MECP2 =</span> <span class="fu">factor</span>(MECP2, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">'Unbound'</span>,<span class="st">'Bound'</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb35-30"><a href="#cb35-30" aria-hidden="true" tabindex="-1"></a>  <span class="co">#filter(MECP2 != "Unbound") %&gt;%</span></span>
<span id="cb35-31"><a href="#cb35-31" aria-hidden="true" tabindex="-1"></a>  <span class="co">#mutate(DEGs = ifelse(Status == 'Repressed',DEGs,DEGs*-1)) %&gt;% </span></span>
<span id="cb35-32"><a href="#cb35-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>Status,<span class="at">y=</span>DEGs,<span class="at">fill=</span>MECP2))<span class="sc">+</span></span>
<span id="cb35-33"><a href="#cb35-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">'identity'</span>)<span class="sc">+</span></span>
<span id="cb35-34"><a href="#cb35-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>RNA.seq)<span class="sc">+</span></span>
<span id="cb35-35"><a href="#cb35-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb35-36"><a href="#cb35-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylim</span>(<span class="dv">0</span>,<span class="dv">1500</span>)<span class="sc">+</span></span>
<span id="cb35-37"><a href="#cb35-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_test</span>(<span class="dv">20</span>)<span class="sc">+</span></span>
<span id="cb35-38"><a href="#cb35-38" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_hline(yintercept=0)+</span></span>
<span id="cb35-39"><a href="#cb35-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label =</span> DEGs), <span class="at">vjust=</span><span class="fl">0.5</span>,<span class="at">hjust =</span> <span class="fl">0.5</span>,<span class="at">size=</span><span class="dv">5</span>)<span class="sc">+</span></span>
<span id="cb35-40"><a href="#cb35-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">'bulk RNA-seq</span><span class="sc">\n</span><span class="st">(MECP2 KO vs WT)'</span>)<span class="sc">+</span></span>
<span id="cb35-41"><a href="#cb35-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">'Number of DEGs'</span>)</span>
<span id="cb35-42"><a href="#cb35-42" aria-hidden="true" tabindex="-1"></a><span class="co">#ggsave("Plots/MECP2_regulated_DEGs.pdf",height = 2,width = 12)</span></span>
<span id="cb35-43"><a href="#cb35-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-44"><a href="#cb35-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-45"><a href="#cb35-45" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"results/RData/MECP2_peakAnnotation.RData"</span>)</span>
<span id="cb35-46"><a href="#cb35-46" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"results/RData/Boxer_RNAseq_data.RData"</span>)</span>
<span id="cb35-47"><a href="#cb35-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-48"><a href="#cb35-48" aria-hidden="true" tabindex="-1"></a>mecp2.bound.Genes <span class="ot">=</span> mecp2.binding <span class="sc">%&gt;%</span> <span class="fu">pull</span>(SYMBOL) <span class="sc">%&gt;%</span> <span class="fu">unique</span>()</span>
<span id="cb35-49"><a href="#cb35-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-50"><a href="#cb35-50" aria-hidden="true" tabindex="-1"></a>RNAseq_Mecp2KO <span class="ot">=</span> RNAseq.list[<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>)]</span>
<span id="cb35-51"><a href="#cb35-51" aria-hidden="true" tabindex="-1"></a>geneSet1 <span class="ot">=</span> <span class="fu">lapply</span>(RNAseq_Mecp2KO, <span class="cf">function</span>(x) x <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(FDR <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> logFC <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb35-52"><a href="#cb35-52" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">pull</span>(ENSEMBL))</span>
<span id="cb35-53"><a href="#cb35-53" aria-hidden="true" tabindex="-1"></a>geneSet2 <span class="ot">=</span> <span class="fu">lapply</span>(RNAseq_Mecp2KO, <span class="cf">function</span>(x) x <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(FDR <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> logFC <span class="sc">&lt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb35-54"><a href="#cb35-54" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">pull</span>(ENSEMBL))</span>
<span id="cb35-55"><a href="#cb35-55" aria-hidden="true" tabindex="-1"></a>geneSet3 <span class="ot">=</span> <span class="fu">lapply</span>(RNAseq_Mecp2KO, <span class="cf">function</span>(x) x <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(FDR <span class="sc">&gt;</span> <span class="fl">0.05</span> <span class="sc">|</span> <span class="fu">is.na</span>(FDR)) <span class="sc">%&gt;%</span> </span>
<span id="cb35-56"><a href="#cb35-56" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">pull</span>(ENSEMBL))</span>
<span id="cb35-57"><a href="#cb35-57" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(geneSet1) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"MECP2.KO_vs_Control.WholeCell.Repressed"</span>,</span>
<span id="cb35-58"><a href="#cb35-58" aria-hidden="true" tabindex="-1"></a>                    <span class="st">"MECP2.KO_vs_Control.Nuclear.Repressed"</span>,</span>
<span id="cb35-59"><a href="#cb35-59" aria-hidden="true" tabindex="-1"></a>                    <span class="st">"MECP2.KO_vs_Control.Chromatin.Repressed"</span>)</span>
<span id="cb35-60"><a href="#cb35-60" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(geneSet2) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"MECP2.KO_vs_Control.WholeCell.Activated"</span>,</span>
<span id="cb35-61"><a href="#cb35-61" aria-hidden="true" tabindex="-1"></a>                    <span class="st">"MECP2.KO_vs_Control.Nuclear.Activated"</span>,</span>
<span id="cb35-62"><a href="#cb35-62" aria-hidden="true" tabindex="-1"></a>                    <span class="st">"MECP2.KO_vs_Control.Chromatin.Activated"</span>)</span>
<span id="cb35-63"><a href="#cb35-63" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(geneSet3) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"MECP2.KO_vs_Control.WholeCell.Unchanged"</span>,</span>
<span id="cb35-64"><a href="#cb35-64" aria-hidden="true" tabindex="-1"></a>                    <span class="st">"MECP2.KO_vs_Control.Nuclear.Unchanged"</span>,</span>
<span id="cb35-65"><a href="#cb35-65" aria-hidden="true" tabindex="-1"></a>                    <span class="st">"MECP2.KO_vs_Control.Chromatin.Unchanged"</span>)</span>
<span id="cb35-66"><a href="#cb35-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-67"><a href="#cb35-67" aria-hidden="true" tabindex="-1"></a>RNAseq_Mecp2KO_GeneSets <span class="ot">=</span> <span class="fu">c</span>(geneSet1,geneSet2,geneSet3)</span>
<span id="cb35-68"><a href="#cb35-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-69"><a href="#cb35-69" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform Chi-square test for association of bound and unbound genes</span></span>
<span id="cb35-70"><a href="#cb35-70" aria-hidden="true" tabindex="-1"></a>overlap <span class="ot">=</span> <span class="fu">newGOM</span>(<span class="fu">list</span>(<span class="at">MECP2.Target =</span> mecp2.bound.Genes,</span>
<span id="cb35-71"><a href="#cb35-71" aria-hidden="true" tabindex="-1"></a>                      <span class="at">MECP2.Target2 =</span> mecp2.bound.Genes),</span>
<span id="cb35-72"><a href="#cb35-72" aria-hidden="true" tabindex="-1"></a>                      RNAseq_Mecp2KO_GeneSets)</span>
<span id="cb35-73"><a href="#cb35-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-74"><a href="#cb35-74" aria-hidden="true" tabindex="-1"></a><span class="co"># drawHeatmap(overlap)</span></span>
<span id="cb35-75"><a href="#cb35-75" aria-hidden="true" tabindex="-1"></a>MECP2.Bound.DEG <span class="ot">=</span> <span class="fu">getNestedList</span>(overlap,<span class="at">name=</span><span class="st">'intersection'</span>)</span>
<span id="cb35-76"><a href="#cb35-76" aria-hidden="true" tabindex="-1"></a>MECP2.Bound.DEG <span class="ot">=</span> (<span class="fu">lapply</span>(MECP2.Bound.DEG, unlist, <span class="at">use.names=</span><span class="cn">FALSE</span>))</span>
<span id="cb35-77"><a href="#cb35-77" aria-hidden="true" tabindex="-1"></a>MECP2.Bound.DEG <span class="ot">=</span> <span class="fu">lapply</span>(MECP2.Bound.DEG,unique)</span>
<span id="cb35-78"><a href="#cb35-78" aria-hidden="true" tabindex="-1"></a>MECP2.UnBound.DEG <span class="ot">=</span> <span class="fu">lapply</span>(<span class="fu">names</span>(RNAseq_Mecp2KO_GeneSets), <span class="cf">function</span>(x) <span class="fu">setdiff</span>(RNAseq_Mecp2KO_GeneSets[[x]], MECP2.Bound.DEG[[x]]))</span>
<span id="cb35-79"><a href="#cb35-79" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(MECP2.UnBound.DEG) <span class="ot">=</span> <span class="fu">names</span>(RNAseq_Mecp2KO_GeneSets)</span>
<span id="cb35-80"><a href="#cb35-80" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-81"><a href="#cb35-81" aria-hidden="true" tabindex="-1"></a><span class="fu">data.frame</span>(<span class="fu">lengths</span>(MECP2.Bound.DEG),</span>
<span id="cb35-82"><a href="#cb35-82" aria-hidden="true" tabindex="-1"></a><span class="fu">lengths</span>(MECP2.UnBound.DEG)) <span class="sc">%&gt;%</span> <span class="fu">t</span>()</span>
<span id="cb35-83"><a href="#cb35-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-84"><a href="#cb35-84" aria-hidden="true" tabindex="-1"></a><span class="co">#data.frame(lengths(MECP2.Bound.DEG),lengths(MECP2.UnBound.DEG)) |&gt; t() |&gt; </span></span>
<span id="cb35-85"><a href="#cb35-85" aria-hidden="true" tabindex="-1"></a><span class="co">#  write.table(., file = "results/MECP2.KO_vs_Control_Bound_Unbound.genes.tsv",sep="\t",quote = FALSE)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></li>
<li><p>Figure 4B.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Correlate number of MECP2 binding and SE annotated genes with logFC of gene expression</span></span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a>MECP2_SE <span class="ot">=</span> <span class="st">"results/MECP2_SE/MECP2WT_Gateway_SuperEnhancers.bed"</span> </span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>MECP2_SE <span class="ot">=</span> <span class="fu">readPeakFile</span>(MECP2_SE)</span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a>txdb <span class="ot">=</span> TxDb.Mmusculus.UCSC.mm10.knownGene</span>
<span id="cb36-5"><a href="#cb36-5" aria-hidden="true" tabindex="-1"></a>MECP2_SE.peakAnno <span class="ot">=</span> <span class="fu">annotatePeak</span>(MECP2_SE, <span class="at">tssRegion=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">1000</span>, <span class="dv">1000</span>),<span class="at">TxDb=</span>txdb, <span class="at">annoDb=</span><span class="st">"org.Mm.eg.db"</span>)</span>
<span id="cb36-6"><a href="#cb36-6" aria-hidden="true" tabindex="-1"></a>MECP2_SE.peakAnno.df <span class="ot">=</span> MECP2_SE.peakAnno <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>()</span>
<span id="cb36-7"><a href="#cb36-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-8"><a href="#cb36-8" aria-hidden="true" tabindex="-1"></a>y.wholeCell <span class="ot">=</span> <span class="fu">merge</span>(MECP2_SE.peakAnno.df,Mecp2.Bound.meth.expressed.wholeCell,<span class="at">by =</span> <span class="st">'ENSEMBL'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-9"><a href="#cb36-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(FDR <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-10"><a href="#cb36-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(DEG <span class="sc">==</span> <span class="st">'Repressed'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-11"><a href="#cb36-11" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(logFC, Binding) <span class="sc">%&gt;%</span> </span>
<span id="cb36-12"><a href="#cb36-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">MECP2 =</span> <span class="st">'SE'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-13"><a href="#cb36-13" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">1</span>,<span class="dv">3</span>) <span class="sc">%&gt;%</span></span>
<span id="cb36-14"><a href="#cb36-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb36-15"><a href="#cb36-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">RNAseq =</span> <span class="st">'WholeCell'</span>)</span>
<span id="cb36-16"><a href="#cb36-16" aria-hidden="true" tabindex="-1"></a>x.wholeCell <span class="ot">=</span> mecp2.binding.meth.RNAseqKO_wholeCell <span class="sc">%&gt;%</span> <span class="fu">filter</span>(DEG <span class="sc">!=</span> <span class="st">'Unchanged'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-17"><a href="#cb36-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(logFC,Binding) <span class="sc">%&gt;%</span> </span>
<span id="cb36-18"><a href="#cb36-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">MECP2 =</span>dplyr<span class="sc">::</span><span class="fu">n</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb36-19"><a href="#cb36-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">MECP2=</span> <span class="fu">ifelse</span>(MECP2<span class="sc">==</span><span class="dv">1</span> <span class="sc">&amp;</span> Binding <span class="sc">==</span><span class="st">'Unbound'</span>,<span class="dv">0</span>, MECP2)) <span class="sc">%&gt;%</span> </span>
<span id="cb36-20"><a href="#cb36-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb36-21"><a href="#cb36-21" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">1</span>,<span class="dv">3</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-22"><a href="#cb36-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">MECP2 =</span> <span class="fu">ifelse</span>(MECP2 <span class="sc">&gt;=</span><span class="dv">10</span>,<span class="dv">10</span>,MECP2)) <span class="sc">%&gt;%</span> </span>
<span id="cb36-23"><a href="#cb36-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">MECP2 =</span> <span class="fu">factor</span>(MECP2,<span class="at">levels =</span> <span class="fu">as.character</span>(<span class="dv">0</span><span class="sc">:</span><span class="dv">10</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb36-24"><a href="#cb36-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">RNAseq =</span> <span class="st">'WholeCell'</span>)</span>
<span id="cb36-25"><a href="#cb36-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-26"><a href="#cb36-26" aria-hidden="true" tabindex="-1"></a>y.nuclear <span class="ot">=</span> <span class="fu">merge</span>(MECP2_SE.peakAnno.df,Mecp2.Bound.meth.expressed.nuclear,<span class="at">by =</span> <span class="st">'ENSEMBL'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-27"><a href="#cb36-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(FDR <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-28"><a href="#cb36-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(DEG <span class="sc">==</span> <span class="st">'Repressed'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-29"><a href="#cb36-29" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(logFC, Binding) <span class="sc">%&gt;%</span> </span>
<span id="cb36-30"><a href="#cb36-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">MECP2 =</span> <span class="st">'SE'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-31"><a href="#cb36-31" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">1</span>,<span class="dv">3</span>) <span class="sc">%&gt;%</span></span>
<span id="cb36-32"><a href="#cb36-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb36-33"><a href="#cb36-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">RNAseq =</span> <span class="st">'Nuclear'</span>)</span>
<span id="cb36-34"><a href="#cb36-34" aria-hidden="true" tabindex="-1"></a>x.nuclear <span class="ot">=</span> mecp2.binding.meth.RNAseqKO_nuclear <span class="sc">%&gt;%</span> <span class="fu">filter</span>(DEG <span class="sc">!=</span> <span class="st">'Unchanged'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-35"><a href="#cb36-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(logFC,Binding) <span class="sc">%&gt;%</span> </span>
<span id="cb36-36"><a href="#cb36-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">MECP2 =</span>dplyr<span class="sc">::</span><span class="fu">n</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb36-37"><a href="#cb36-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">MECP2=</span> <span class="fu">ifelse</span>(MECP2<span class="sc">==</span><span class="dv">1</span> <span class="sc">&amp;</span> Binding <span class="sc">==</span><span class="st">'Unbound'</span>,<span class="dv">0</span>, MECP2)) <span class="sc">%&gt;%</span> </span>
<span id="cb36-38"><a href="#cb36-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb36-39"><a href="#cb36-39" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">1</span>,<span class="dv">3</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-40"><a href="#cb36-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">MECP2 =</span> <span class="fu">ifelse</span>(MECP2 <span class="sc">&gt;=</span><span class="dv">10</span>,<span class="dv">10</span>,MECP2)) <span class="sc">%&gt;%</span> </span>
<span id="cb36-41"><a href="#cb36-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">MECP2 =</span> <span class="fu">factor</span>(MECP2,<span class="at">levels =</span> <span class="fu">as.character</span>(<span class="dv">0</span><span class="sc">:</span><span class="dv">10</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb36-42"><a href="#cb36-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">RNAseq =</span> <span class="st">'Nuclear'</span>)</span>
<span id="cb36-43"><a href="#cb36-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-44"><a href="#cb36-44" aria-hidden="true" tabindex="-1"></a>y.chromatin <span class="ot">=</span> <span class="fu">merge</span>(MECP2_SE.peakAnno.df,Mecp2.Bound.meth.expressed.chromatin,<span class="at">by =</span> <span class="st">'ENSEMBL'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-45"><a href="#cb36-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(FDR <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-46"><a href="#cb36-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(DEG <span class="sc">==</span> <span class="st">'Repressed'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-47"><a href="#cb36-47" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(logFC, Binding) <span class="sc">%&gt;%</span> </span>
<span id="cb36-48"><a href="#cb36-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">MECP2 =</span> <span class="st">'SE'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-49"><a href="#cb36-49" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">1</span>,<span class="dv">3</span>) <span class="sc">%&gt;%</span></span>
<span id="cb36-50"><a href="#cb36-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb36-51"><a href="#cb36-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">RNAseq =</span> <span class="st">'Chromatin'</span>)</span>
<span id="cb36-52"><a href="#cb36-52" aria-hidden="true" tabindex="-1"></a>x.chromatin <span class="ot">=</span> mecp2.binding.meth.RNAseqKO_chromatin <span class="sc">%&gt;%</span> <span class="fu">filter</span>(DEG <span class="sc">!=</span> <span class="st">'Unchanged'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-53"><a href="#cb36-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(logFC,Binding) <span class="sc">%&gt;%</span> </span>
<span id="cb36-54"><a href="#cb36-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">MECP2 =</span>dplyr<span class="sc">::</span><span class="fu">n</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb36-55"><a href="#cb36-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">MECP2=</span> <span class="fu">ifelse</span>(MECP2<span class="sc">==</span><span class="dv">1</span> <span class="sc">&amp;</span> Binding <span class="sc">==</span><span class="st">'Unbound'</span>,<span class="dv">0</span>, MECP2)) <span class="sc">%&gt;%</span> </span>
<span id="cb36-56"><a href="#cb36-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb36-57"><a href="#cb36-57" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">1</span>,<span class="dv">3</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-58"><a href="#cb36-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">MECP2 =</span> <span class="fu">ifelse</span>(MECP2 <span class="sc">&gt;=</span><span class="dv">10</span>,<span class="dv">10</span>,MECP2)) <span class="sc">%&gt;%</span> </span>
<span id="cb36-59"><a href="#cb36-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">MECP2 =</span> <span class="fu">factor</span>(MECP2,<span class="at">levels =</span> <span class="fu">as.character</span>(<span class="dv">0</span><span class="sc">:</span><span class="dv">10</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb36-60"><a href="#cb36-60" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">RNAseq =</span> <span class="st">'Chromatin'</span>)</span>
<span id="cb36-61"><a href="#cb36-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-62"><a href="#cb36-62" aria-hidden="true" tabindex="-1"></a>df <span class="ot">=</span> <span class="fu">rbind</span>(x.nuclear,y.nuclear,</span>
<span id="cb36-63"><a href="#cb36-63" aria-hidden="true" tabindex="-1"></a>      x.wholeCell,y.wholeCell,</span>
<span id="cb36-64"><a href="#cb36-64" aria-hidden="true" tabindex="-1"></a>      x.chromatin,y.chromatin) <span class="sc">%&gt;%</span> </span>
<span id="cb36-65"><a href="#cb36-65" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(MECP2 <span class="sc">!=</span> <span class="st">'SE'</span>) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>()  <span class="sc">%&gt;%</span> </span>
<span id="cb36-66"><a href="#cb36-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(logFC <span class="sc">&gt;-</span><span class="dv">2</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-67"><a href="#cb36-67" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">MECP2 =</span><span class="fu">as.character</span>(MECP2),</span>
<span id="cb36-68"><a href="#cb36-68" aria-hidden="true" tabindex="-1"></a>         <span class="at">MECP2 =</span> <span class="fu">factor</span>(MECP2,<span class="at">levels=</span><span class="fu">as.character</span>(<span class="fu">c</span>(<span class="dv">0</span><span class="sc">:</span><span class="dv">10</span>))))</span>
<span id="cb36-69"><a href="#cb36-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-70"><a href="#cb36-70" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(df,<span class="fu">aes</span>(<span class="at">x=</span>MECP2 ,<span class="at">y =</span>logFC,<span class="at">color=</span>RNAseq))<span class="sc">+</span><span class="fu">geom_boxplot</span>()<span class="sc">+</span></span>
<span id="cb36-71"><a href="#cb36-71" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_test</span>(<span class="dv">20</span>)</span>
<span id="cb36-72"><a href="#cb36-72" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">'Plots/MECP2_regulated_DEGs_boxplot.pdf'</span>,<span class="at">width =</span> <span class="dv">8</span>,<span class="at">height =</span> <span class="dv">6</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></li>
<li><p>Figure 4D.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>mouse_human_ortholog <span class="ot">=</span> <span class="st">"https://www.informatics.jax.org/downloads/reports/HOM_MouseHumanSequence.rpt"</span></span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>mouse_human_genes <span class="ot">=</span> <span class="fu">read.csv</span>(mouse_human_ortholog,<span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a><span class="co"># separate human and mouse </span></span>
<span id="cb37-5"><a href="#cb37-5" aria-hidden="true" tabindex="-1"></a>mouse <span class="ot">=</span> <span class="fu">split.data.frame</span>(mouse_human_genes,mouse_human_genes<span class="sc">$</span>Common.Organism.Name)[[<span class="dv">2</span>]]</span>
<span id="cb37-6"><a href="#cb37-6" aria-hidden="true" tabindex="-1"></a>human <span class="ot">=</span> <span class="fu">split.data.frame</span>(mouse_human_genes,mouse_human_genes<span class="sc">$</span>Common.Organism.Name)[[<span class="dv">1</span>]]</span>
<span id="cb37-7"><a href="#cb37-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-8"><a href="#cb37-8" aria-hidden="true" tabindex="-1"></a><span class="co"># remove some columns</span></span>
<span id="cb37-9"><a href="#cb37-9" aria-hidden="true" tabindex="-1"></a>mouse <span class="ot">=</span> mouse[,<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">4</span>)]</span>
<span id="cb37-10"><a href="#cb37-10" aria-hidden="true" tabindex="-1"></a>human <span class="ot">=</span> human[,<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">4</span>)]</span>
<span id="cb37-11"><a href="#cb37-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-12"><a href="#cb37-12" aria-hidden="true" tabindex="-1"></a><span class="co"># merge the 2 dataset  (note that the human list is longer than the mouse one)</span></span>
<span id="cb37-13"><a href="#cb37-13" aria-hidden="true" tabindex="-1"></a>mh_data <span class="ot">=</span> <span class="fu">merge.data.frame</span>(mouse,human,<span class="at">by =</span> <span class="st">"DB.Class.Key"</span>,<span class="at">all.y =</span> <span class="cn">TRUE</span>) </span>
<span id="cb37-14"><a href="#cb37-14" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(mh_data,<span class="at">file =</span> <span class="st">'results/RData/Human_mouse_ortholog.RData'</span>)</span>
<span id="cb37-15"><a href="#cb37-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-16"><a href="#cb37-16" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(mh_data) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"DB.Class.Key"</span>,<span class="st">"Mouse"</span>,<span class="st">'Human'</span>)</span>
<span id="cb37-17"><a href="#cb37-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-18"><a href="#cb37-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-19"><a href="#cb37-19" aria-hidden="true" tabindex="-1"></a>mecp2ko_logFC<span class="ot">=</span>  mecp2.binding.meth.RNAseqKO_wholeCell <span class="sc">%&gt;%</span> </span>
<span id="cb37-20"><a href="#cb37-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Meth_CA <span class="sc">==</span><span class="st">'CA'</span> <span class="sc">&amp;</span> Meth_CG <span class="sc">==</span><span class="st">'CG'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb37-21"><a href="#cb37-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Expression<span class="sc">==</span><span class="st">'Expressed'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb37-22"><a href="#cb37-22" aria-hidden="true" tabindex="-1"></a>  <span class="co">#filter(DEG != 'Unchanged') %&gt;% </span></span>
<span id="cb37-23"><a href="#cb37-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(ENSEMBL)) <span class="sc">%&gt;%</span> </span>
<span id="cb37-24"><a href="#cb37-24" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(Gene,logFC,Binding,DEG) <span class="sc">%&gt;%</span> </span>
<span id="cb37-25"><a href="#cb37-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Gene,logFC,Binding) <span class="sc">%&gt;%</span> </span>
<span id="cb37-26"><a href="#cb37-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">Count=</span> dplyr<span class="sc">::</span><span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb37-27"><a href="#cb37-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(logFC)) <span class="sc">%&gt;%</span> </span>
<span id="cb37-28"><a href="#cb37-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Count=</span> <span class="fu">ifelse</span>(Count<span class="sc">==</span><span class="dv">1</span> <span class="sc">&amp;</span> Binding <span class="sc">==</span><span class="st">'Unbound'</span>,<span class="dv">0</span>, Count)) <span class="sc">%&gt;%</span> </span>
<span id="cb37-29"><a href="#cb37-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">merge</span>(.,mh_data,<span class="at">by.x=</span><span class="st">'Gene'</span>,<span class="at">by.y=</span><span class="st">"Mouse"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb37-30"><a href="#cb37-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb37-31"><a href="#cb37-31" aria-hidden="true" tabindex="-1"></a>  <span class="co">#dplyr::select() %&gt;% </span></span>
<span id="cb37-32"><a href="#cb37-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">duplicated</span>(.[,]))</span>
<span id="cb37-33"><a href="#cb37-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-34"><a href="#cb37-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-35"><a href="#cb37-35" aria-hidden="true" tabindex="-1"></a>MECP2_SE <span class="ot">=</span> <span class="st">"results/MECP2_SE/MECP2WT_Gateway_SuperEnhancers.bed"</span> </span>
<span id="cb37-36"><a href="#cb37-36" aria-hidden="true" tabindex="-1"></a>MECP2_SE <span class="ot">=</span> <span class="fu">readPeakFile</span>(MECP2_SE)</span>
<span id="cb37-37"><a href="#cb37-37" aria-hidden="true" tabindex="-1"></a>txdb <span class="ot">=</span> TxDb.Mmusculus.UCSC.mm10.knownGene</span>
<span id="cb37-38"><a href="#cb37-38" aria-hidden="true" tabindex="-1"></a>MECP2_SE.peakAnno <span class="ot">=</span> peakAnno.mecp2 <span class="ot">=</span> <span class="fu">annotatePeak</span>(MECP2_SE, </span>
<span id="cb37-39"><a href="#cb37-39" aria-hidden="true" tabindex="-1"></a>                                                  <span class="at">tssRegion=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">1000</span>, <span class="dv">1000</span>),<span class="at">TxDb=</span>txdb, </span>
<span id="cb37-40"><a href="#cb37-40" aria-hidden="true" tabindex="-1"></a>                                                  <span class="at">annoDb=</span><span class="st">"org.Mm.eg.db"</span>)</span>
<span id="cb37-41"><a href="#cb37-41" aria-hidden="true" tabindex="-1"></a>MECP2_SE.peakAnno.df <span class="ot">=</span> MECP2_SE.peakAnno <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>()</span>
<span id="cb37-42"><a href="#cb37-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-43"><a href="#cb37-43" aria-hidden="true" tabindex="-1"></a><span class="co"># Download Supplementary file 7 (https://doi.org/10.1038/s41593-018-0270-6)</span></span>
<span id="cb37-44"><a href="#cb37-44" aria-hidden="true" tabindex="-1"></a>link <span class="ot">=</span> <span class="st">"https://static-content.springer.com/esm/art%3A10.1038%2Fs41593-018-0270-6/MediaObjects/41593_2018_270_MOESM10_ESM.csv"</span></span>
<span id="cb37-45"><a href="#cb37-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-46"><a href="#cb37-46" aria-hidden="true" tabindex="-1"></a>RTT_cortex <span class="ot">=</span> <span class="fu">read.csv</span>(link,<span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb37-47"><a href="#cb37-47" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(RTT_cortex)</span>
<span id="cb37-48"><a href="#cb37-48" aria-hidden="true" tabindex="-1"></a>mecp2_rperessed <span class="ot">=</span> mecp2ko_logFC <span class="sc">%&gt;%</span></span>
<span id="cb37-49"><a href="#cb37-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">merge</span>(.,RTT_cortex,<span class="at">by.x=</span><span class="st">'Human'</span>,<span class="at">by.y=</span><span class="st">'gene_name'</span>) <span class="co">#%&gt;% </span></span>
<span id="cb37-50"><a href="#cb37-50" aria-hidden="true" tabindex="-1"></a><span class="co">#filter( qval &lt; 0.05) %&gt;% </span></span>
<span id="cb37-51"><a href="#cb37-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-52"><a href="#cb37-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-53"><a href="#cb37-53" aria-hidden="true" tabindex="-1"></a>mecp2_rperessed.all <span class="ot">=</span> mecp2_rperessed <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">Group =</span><span class="st">'All'</span>) </span>
<span id="cb37-54"><a href="#cb37-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-55"><a href="#cb37-55" aria-hidden="true" tabindex="-1"></a>mecp2_rperessed.noMBH <span class="ot">=</span> mecp2_rperessed <span class="sc">%&gt;%</span> <span class="co">#filter(!(Gene %in% mecp2.binding$SYMBOL)) %&gt;% </span></span>
<span id="cb37-56"><a href="#cb37-56" aria-hidden="true" tabindex="-1"></a>  <span class="co">#filter(Count &gt;=7) %&gt;% </span></span>
<span id="cb37-57"><a href="#cb37-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Group =</span> <span class="st">'No MBHs'</span>) <span class="sc">%&gt;%</span> <span class="fu">filter</span>( qval <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="co">#logFC &gt;0)</span></span>
<span id="cb37-58"><a href="#cb37-58" aria-hidden="true" tabindex="-1"></a>mecp2_rperessed<span class="fl">.7</span>MBH <span class="ot">=</span> mecp2_rperessed <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Gene <span class="sc">%in%</span> MECP2_SE.peakAnno.df<span class="sc">$</span>SYMBOL) <span class="sc">%&gt;%</span> </span>
<span id="cb37-59"><a href="#cb37-59" aria-hidden="true" tabindex="-1"></a>  <span class="co">#filter(Count &gt;=7) %&gt;% </span></span>
<span id="cb37-60"><a href="#cb37-60" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Group =</span> <span class="st">'MBH Cluster'</span>) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(logFC <span class="sc">&gt;</span> <span class="dv">0</span> <span class="sc">&amp;</span> qval <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="co">#logFC &gt;0)# &amp;</span></span>
<span id="cb37-61"><a href="#cb37-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-62"><a href="#cb37-62" aria-hidden="true" tabindex="-1"></a>Rtt.plt <span class="ot">=</span> <span class="fu">rbind</span>(mecp2_rperessed.all,mecp2_rperessed<span class="fl">.7</span>MBH) <span class="sc">%&gt;%</span> </span>
<span id="cb37-63"><a href="#cb37-63" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">1</span>,<span class="dv">12</span>,<span class="dv">9</span>)  <span class="sc">%&gt;%</span> </span>
<span id="cb37-64"><a href="#cb37-64" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Group=</span><span class="fu">factor</span>(Group,<span class="at">levels=</span> <span class="fu">c</span>(<span class="st">'All'</span>,<span class="st">'MBH Cluster'</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb37-65"><a href="#cb37-65" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>Group,<span class="at">y=</span>Log2FC,<span class="at">color=</span>Group))<span class="sc">+</span></span>
<span id="cb37-66"><a href="#cb37-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">outlier.colour =</span> <span class="st">'NA'</span>)<span class="sc">+</span></span>
<span id="cb37-67"><a href="#cb37-67" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylim</span>(<span class="sc">-</span><span class="fl">1.2</span>,<span class="dv">3</span>)<span class="sc">+</span></span>
<span id="cb37-68"><a href="#cb37-68" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>()<span class="sc">+</span></span>
<span id="cb37-69"><a href="#cb37-69" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">labels=</span><span class="fu">c</span>(<span class="fu">paste</span>(<span class="st">'All Genes'</span>),</span>
<span id="cb37-70"><a href="#cb37-70" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">paste</span>(<span class="st">'Genes regulated by</span><span class="sc">\n</span><span class="st">Broad MBH clusters'</span>)))<span class="sc">+</span></span>
<span id="cb37-71"><a href="#cb37-71" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">'grey'</span>,<span class="st">'red'</span>))<span class="sc">+</span></span>
<span id="cb37-72"><a href="#cb37-72" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_test</span>(<span class="dv">20</span>)<span class="sc">+</span></span>
<span id="cb37-73"><a href="#cb37-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"log2 Fold Change</span><span class="sc">\n</span><span class="st">(RTT/WT)"</span>)<span class="sc">+</span></span>
<span id="cb37-74"><a href="#cb37-74" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">'Cortex (Renthal et. al, Nature Neuroscience 2018)'</span>)<span class="sc">+</span></span>
<span id="cb37-75"><a href="#cb37-75" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb37-76"><a href="#cb37-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-77"><a href="#cb37-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-78"><a href="#cb37-78" aria-hidden="true" tabindex="-1"></a>Rtt.plt</span>
<span id="cb37-79"><a href="#cb37-79" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"Plots/RTT_SingleNuclei_geneExpression.pdf"</span>,<span class="at">height =</span> <span class="dv">6</span>,<span class="at">width =</span> <span class="dv">5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></li>
<li><p>Supplementary Figure 4A</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"results/RData/MECP2_regulatedGenes.RData"</span>)</span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a>Pie.chart.list <span class="ot">=</span> <span class="fu">list</span>()</span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a><span class="co"># all MECP2 annotated and unannotated genes</span></span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a>MECP2.pie.all <span class="ot">=</span> mecp2.binding.meth.RNAseqKO_nuclear <span class="sc">%&gt;%</span> </span>
<span id="cb38-5"><a href="#cb38-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(FDR <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb38-6"><a href="#cb38-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(ENSEMBL)) <span class="sc">%&gt;%</span> </span>
<span id="cb38-7"><a href="#cb38-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Meth_CA<span class="sc">==</span><span class="st">'CA'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb38-8"><a href="#cb38-8" aria-hidden="true" tabindex="-1"></a>  <span class="co">#filter(FDR &lt; 0.05) %&gt;% </span></span>
<span id="cb38-9"><a href="#cb38-9" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(ENSEMBL,Binding) <span class="sc">%&gt;%</span> </span>
<span id="cb38-10"><a href="#cb38-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(ENSEMBL,Binding) <span class="sc">%&gt;%</span> </span>
<span id="cb38-11"><a href="#cb38-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">Count=</span>dplyr<span class="sc">::</span><span class="fu">n</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb38-12"><a href="#cb38-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Count =</span> <span class="fu">ifelse</span>(Binding <span class="sc">==</span> <span class="st">'Unbound'</span>,<span class="dv">0</span>,Count)) <span class="sc">%&gt;%</span> </span>
<span id="cb38-13"><a href="#cb38-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Count) <span class="sc">%&gt;%</span> </span>
<span id="cb38-14"><a href="#cb38-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">Count_1=</span>dplyr<span class="sc">::</span><span class="fu">n</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb38-15"><a href="#cb38-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Count_2 =</span> <span class="fu">ifelse</span>(Count <span class="sc">&gt;=</span><span class="dv">10</span>, <span class="fu">sum</span>(Count_1[Count <span class="sc">&gt;=</span><span class="dv">10</span>]),Count_1)) <span class="sc">%&gt;%</span> </span>
<span id="cb38-16"><a href="#cb38-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Peaks =</span> (<span class="fu">ifelse</span>(Count <span class="sc">&gt;=</span> <span class="dv">10</span>, <span class="st">'&gt;=10'</span>, Count))) <span class="sc">%&gt;%</span> </span>
<span id="cb38-17"><a href="#cb38-17" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">3</span>,<span class="dv">4</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb38-18"><a href="#cb38-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span></span>
<span id="cb38-19"><a href="#cb38-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Peaks =</span> <span class="fu">factor</span>(Peaks, <span class="at">levels =</span> <span class="fu">c</span>(<span class="fu">as.character</span>(<span class="dv">0</span><span class="sc">:</span><span class="dv">9</span>),<span class="st">'&gt;=10'</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb38-20"><a href="#cb38-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Percent =</span> <span class="fu">round</span>(Count_2<span class="sc">/</span><span class="fu">sum</span>(Count_2)<span class="sc">*</span><span class="dv">100</span>,<span class="dv">2</span>))</span>
<span id="cb38-21"><a href="#cb38-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-22"><a href="#cb38-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-23"><a href="#cb38-23" aria-hidden="true" tabindex="-1"></a>nb.cols <span class="ot">=</span> <span class="dv">10</span></span>
<span id="cb38-24"><a href="#cb38-24" aria-hidden="true" tabindex="-1"></a>mycolors <span class="ot">=</span> <span class="fu">colorRampPalette</span>(<span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">"Set2"</span>))(nb.cols)</span>
<span id="cb38-25"><a href="#cb38-25" aria-hidden="true" tabindex="-1"></a>mycolors <span class="ot">=</span> <span class="fu">c</span>(<span class="st">'#71CAF7'</span>,mycolors)</span>
<span id="cb38-26"><a href="#cb38-26" aria-hidden="true" tabindex="-1"></a>threshold <span class="ot">=</span> <span class="fl">0.05</span></span>
<span id="cb38-27"><a href="#cb38-27" aria-hidden="true" tabindex="-1"></a>Pie.chart.list[[<span class="dv">1</span>]] <span class="ot">=</span> MECP2.pie.all <span class="sc">%&gt;%</span> </span>
<span id="cb38-28"><a href="#cb38-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="st">""</span>, <span class="at">y =</span> Percent, <span class="at">fill =</span> Peaks)) <span class="sc">+</span></span>
<span id="cb38-29"><a href="#cb38-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">width =</span> <span class="dv">1</span>, <span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">color =</span> <span class="st">"white"</span>) <span class="sc">+</span></span>
<span id="cb38-30"><a href="#cb38-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_polar</span>(<span class="st">"y"</span>, <span class="at">start =</span> <span class="dv">0</span>)<span class="sc">+</span></span>
<span id="cb38-31"><a href="#cb38-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fl">1.55</span>, <span class="at">label =</span> scales<span class="sc">::</span><span class="fu">percent</span>(Percent, <span class="at">accuracy =</span> <span class="fl">0.1</span>,<span class="at">scale =</span> <span class="dv">1</span>)), <span class="at">position =</span> <span class="fu">position_stack</span>(<span class="at">vjust =</span> .<span class="dv">5</span>),<span class="at">size=</span><span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb38-32"><a href="#cb38-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_polar</span>(<span class="st">"y"</span>) <span class="sc">+</span></span>
<span id="cb38-33"><a href="#cb38-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_void</span>()<span class="sc">+</span></span>
<span id="cb38-34"><a href="#cb38-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">10</span>),</span>
<span id="cb38-35"><a href="#cb38-35" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">10</span>))<span class="sc">+</span></span>
<span id="cb38-36"><a href="#cb38-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">name =</span> <span class="st">'All Genes with</span><span class="sc">\n</span><span class="st"> number of MECP2 peaks'</span>,<span class="at">values =</span> mycolors)</span>
<span id="cb38-37"><a href="#cb38-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-38"><a href="#cb38-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-39"><a href="#cb38-39" aria-hidden="true" tabindex="-1"></a><span class="co"># Repressed  MECP2 annotated and unannotated genes</span></span>
<span id="cb38-40"><a href="#cb38-40" aria-hidden="true" tabindex="-1"></a>MECP2.pie.repressed <span class="ot">=</span> mecp2.binding.meth.RNAseqKO_nuclear <span class="sc">%&gt;%</span> </span>
<span id="cb38-41"><a href="#cb38-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(DEG<span class="sc">==</span><span class="st">'Repressed'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb38-42"><a href="#cb38-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(ENSEMBL)) <span class="sc">%&gt;%</span> </span>
<span id="cb38-43"><a href="#cb38-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Meth_CA<span class="sc">==</span><span class="st">'CA'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb38-44"><a href="#cb38-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-45"><a href="#cb38-45" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(ENSEMBL,Binding) <span class="sc">%&gt;%</span> </span>
<span id="cb38-46"><a href="#cb38-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(ENSEMBL,Binding) <span class="sc">%&gt;%</span> </span>
<span id="cb38-47"><a href="#cb38-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">Count=</span>dplyr<span class="sc">::</span><span class="fu">n</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb38-48"><a href="#cb38-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Count =</span> <span class="fu">ifelse</span>(Binding <span class="sc">==</span> <span class="st">'Unbound'</span>,<span class="dv">0</span>,Count)) <span class="sc">%&gt;%</span> </span>
<span id="cb38-49"><a href="#cb38-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Count) <span class="sc">%&gt;%</span> </span>
<span id="cb38-50"><a href="#cb38-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">Count_1=</span>dplyr<span class="sc">::</span><span class="fu">n</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb38-51"><a href="#cb38-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Count_2 =</span> <span class="fu">ifelse</span>(Count <span class="sc">&gt;=</span><span class="dv">10</span>, <span class="fu">sum</span>(Count_1[Count <span class="sc">&gt;=</span><span class="dv">10</span>]),Count_1)) <span class="sc">%&gt;%</span> </span>
<span id="cb38-52"><a href="#cb38-52" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Peaks =</span> (<span class="fu">ifelse</span>(Count <span class="sc">&gt;=</span> <span class="dv">10</span>, <span class="st">'&gt;=10'</span>, Count))) <span class="sc">%&gt;%</span> </span>
<span id="cb38-53"><a href="#cb38-53" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">3</span>,<span class="dv">4</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb38-54"><a href="#cb38-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span></span>
<span id="cb38-55"><a href="#cb38-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Peaks =</span> <span class="fu">factor</span>(Peaks, <span class="at">levels =</span> <span class="fu">c</span>(<span class="fu">as.character</span>(<span class="dv">0</span><span class="sc">:</span><span class="dv">9</span>),<span class="st">'&gt;=10'</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb38-56"><a href="#cb38-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Percent =</span> <span class="fu">round</span>(Count_2<span class="sc">/</span><span class="fu">sum</span>(Count_2)<span class="sc">*</span><span class="dv">100</span>,<span class="dv">2</span>))</span>
<span id="cb38-57"><a href="#cb38-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-58"><a href="#cb38-58" aria-hidden="true" tabindex="-1"></a>nb.cols <span class="ot">=</span> <span class="dv">10</span></span>
<span id="cb38-59"><a href="#cb38-59" aria-hidden="true" tabindex="-1"></a>mycolors <span class="ot">=</span> <span class="fu">colorRampPalette</span>(<span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">"Set2"</span>))(nb.cols)</span>
<span id="cb38-60"><a href="#cb38-60" aria-hidden="true" tabindex="-1"></a>mycolors <span class="ot">=</span> <span class="fu">c</span>(<span class="st">'#71CAF7'</span>,mycolors)</span>
<span id="cb38-61"><a href="#cb38-61" aria-hidden="true" tabindex="-1"></a>threshold <span class="ot">=</span> <span class="fl">0.05</span></span>
<span id="cb38-62"><a href="#cb38-62" aria-hidden="true" tabindex="-1"></a>Pie.chart.list[[<span class="dv">3</span>]] <span class="ot">=</span> MECP2.pie.repressed <span class="sc">%&gt;%</span> </span>
<span id="cb38-63"><a href="#cb38-63" aria-hidden="true" tabindex="-1"></a>  <span class="co">#mutate(Peaks =as.character(Peaks)) %&gt;% </span></span>
<span id="cb38-64"><a href="#cb38-64" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="st">""</span>, <span class="at">y =</span> Percent, <span class="at">fill =</span> Peaks)) <span class="sc">+</span></span>
<span id="cb38-65"><a href="#cb38-65" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">width =</span> <span class="dv">1</span>, <span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">color =</span> <span class="st">"white"</span>) <span class="sc">+</span></span>
<span id="cb38-66"><a href="#cb38-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_polar</span>(<span class="st">"y"</span>, <span class="at">start =</span> <span class="dv">0</span>)<span class="sc">+</span></span>
<span id="cb38-67"><a href="#cb38-67" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fl">1.55</span>, <span class="at">label =</span> scales<span class="sc">::</span><span class="fu">percent</span>(Percent, <span class="at">accuracy =</span> <span class="fl">0.1</span>,<span class="at">scale =</span> <span class="dv">1</span>)), <span class="at">position =</span> <span class="fu">position_stack</span>(<span class="at">vjust =</span> .<span class="dv">5</span>),<span class="at">size=</span><span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb38-68"><a href="#cb38-68" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_polar</span>(<span class="st">"y"</span>) <span class="sc">+</span></span>
<span id="cb38-69"><a href="#cb38-69" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_void</span>()<span class="sc">+</span></span>
<span id="cb38-70"><a href="#cb38-70" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">10</span>),</span>
<span id="cb38-71"><a href="#cb38-71" aria-hidden="true" tabindex="-1"></a>        <span class="co">#legend.key.size = unit(2, 'cm'),</span></span>
<span id="cb38-72"><a href="#cb38-72" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">10</span>))<span class="sc">+</span></span>
<span id="cb38-73"><a href="#cb38-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">name =</span> <span class="st">'Repressed Genes with</span><span class="sc">\n</span><span class="st"> number of MECP2 peaks'</span>,<span class="at">values =</span> mycolors)</span>
<span id="cb38-74"><a href="#cb38-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-75"><a href="#cb38-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-76"><a href="#cb38-76" aria-hidden="true" tabindex="-1"></a><span class="co"># Activated  MECP2 annotated and unannotated genes</span></span>
<span id="cb38-77"><a href="#cb38-77" aria-hidden="true" tabindex="-1"></a>MECP2.pie.activated<span class="ot">=</span> mecp2.binding.meth.RNAseqKO_nuclear <span class="sc">%&gt;%</span> </span>
<span id="cb38-78"><a href="#cb38-78" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(DEG<span class="sc">==</span><span class="st">'Activated'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb38-79"><a href="#cb38-79" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(ENSEMBL)) <span class="sc">%&gt;%</span> </span>
<span id="cb38-80"><a href="#cb38-80" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Meth_CA<span class="sc">==</span><span class="st">'CA'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb38-81"><a href="#cb38-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-82"><a href="#cb38-82" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(ENSEMBL,Binding) <span class="sc">%&gt;%</span> </span>
<span id="cb38-83"><a href="#cb38-83" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(ENSEMBL,Binding) <span class="sc">%&gt;%</span> </span>
<span id="cb38-84"><a href="#cb38-84" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">Count=</span>dplyr<span class="sc">::</span><span class="fu">n</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb38-85"><a href="#cb38-85" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Count =</span> <span class="fu">ifelse</span>(Binding <span class="sc">==</span> <span class="st">'Unbound'</span>,<span class="dv">0</span>,Count)) <span class="sc">%&gt;%</span> </span>
<span id="cb38-86"><a href="#cb38-86" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Count) <span class="sc">%&gt;%</span> </span>
<span id="cb38-87"><a href="#cb38-87" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">Count_1=</span>dplyr<span class="sc">::</span><span class="fu">n</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb38-88"><a href="#cb38-88" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Count_2 =</span> <span class="fu">ifelse</span>(Count <span class="sc">&gt;=</span><span class="dv">10</span>, <span class="fu">sum</span>(Count_1[Count <span class="sc">&gt;=</span><span class="dv">10</span>]),Count_1)) <span class="sc">%&gt;%</span> </span>
<span id="cb38-89"><a href="#cb38-89" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Peaks =</span> (<span class="fu">ifelse</span>(Count <span class="sc">&gt;=</span> <span class="dv">10</span>, <span class="st">'&gt;=10'</span>, Count))) <span class="sc">%&gt;%</span> </span>
<span id="cb38-90"><a href="#cb38-90" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">3</span>,<span class="dv">4</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb38-91"><a href="#cb38-91" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span></span>
<span id="cb38-92"><a href="#cb38-92" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Peaks =</span> <span class="fu">factor</span>(Peaks, <span class="at">levels =</span> <span class="fu">c</span>(<span class="fu">as.character</span>(<span class="dv">0</span><span class="sc">:</span><span class="dv">9</span>),<span class="st">'&gt;=10'</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb38-93"><a href="#cb38-93" aria-hidden="true" tabindex="-1"></a>  <span class="co">#mutate(Peaks = factor(Peaks, levels = c('&lt;5','&gt;=5'))) %&gt;% </span></span>
<span id="cb38-94"><a href="#cb38-94" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Percent =</span> <span class="fu">round</span>(Count_2<span class="sc">/</span><span class="fu">sum</span>(Count_2)<span class="sc">*</span><span class="dv">100</span>,<span class="dv">2</span>))</span>
<span id="cb38-95"><a href="#cb38-95" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-96"><a href="#cb38-96" aria-hidden="true" tabindex="-1"></a>nb.cols <span class="ot">=</span> <span class="dv">10</span></span>
<span id="cb38-97"><a href="#cb38-97" aria-hidden="true" tabindex="-1"></a>mycolors <span class="ot">=</span> <span class="fu">colorRampPalette</span>(<span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">"Set2"</span>))(nb.cols)</span>
<span id="cb38-98"><a href="#cb38-98" aria-hidden="true" tabindex="-1"></a>mycolors <span class="ot">=</span> <span class="fu">c</span>(<span class="st">'#71CAF7'</span>,mycolors)</span>
<span id="cb38-99"><a href="#cb38-99" aria-hidden="true" tabindex="-1"></a>threshold <span class="ot">=</span> <span class="fl">0.05</span></span>
<span id="cb38-100"><a href="#cb38-100" aria-hidden="true" tabindex="-1"></a>Pie.chart.list[[<span class="dv">2</span>]] <span class="ot">=</span> MECP2.pie.activated <span class="sc">%&gt;%</span> </span>
<span id="cb38-101"><a href="#cb38-101" aria-hidden="true" tabindex="-1"></a>  <span class="co">#mutate(Peaks =as.character(Peaks)) %&gt;% </span></span>
<span id="cb38-102"><a href="#cb38-102" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="st">""</span>, <span class="at">y =</span> Percent, <span class="at">fill =</span> Peaks)) <span class="sc">+</span></span>
<span id="cb38-103"><a href="#cb38-103" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">width =</span> <span class="dv">1</span>, <span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">color =</span> <span class="st">"white"</span>) <span class="sc">+</span></span>
<span id="cb38-104"><a href="#cb38-104" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_polar</span>(<span class="st">"y"</span>, <span class="at">start =</span> <span class="dv">0</span>)<span class="sc">+</span></span>
<span id="cb38-105"><a href="#cb38-105" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fl">1.55</span>, <span class="at">label =</span> scales<span class="sc">::</span><span class="fu">percent</span>(Percent, <span class="at">accuracy =</span> <span class="fl">0.1</span>,<span class="at">scale =</span> <span class="dv">1</span>)), <span class="at">position =</span> <span class="fu">position_stack</span>(<span class="at">vjust =</span> .<span class="dv">5</span>),<span class="at">size=</span><span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb38-106"><a href="#cb38-106" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_polar</span>(<span class="st">"y"</span>) <span class="sc">+</span></span>
<span id="cb38-107"><a href="#cb38-107" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_void</span>()<span class="sc">+</span></span>
<span id="cb38-108"><a href="#cb38-108" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">10</span>),</span>
<span id="cb38-109"><a href="#cb38-109" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">10</span>))<span class="sc">+</span></span>
<span id="cb38-110"><a href="#cb38-110" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">name =</span> <span class="st">'Activated Genes with</span><span class="sc">\n</span><span class="st"> number of MECP2 peaks'</span>,<span class="at">values =</span> mycolors)</span>
<span id="cb38-111"><a href="#cb38-111" aria-hidden="true" tabindex="-1"></a>Pie.chart.list[[<span class="dv">2</span>]]</span>
<span id="cb38-112"><a href="#cb38-112" aria-hidden="true" tabindex="-1"></a>position <span class="ot">=</span> <span class="fu">position_stack</span>(<span class="at">vjust =</span> .<span class="dv">5</span>)</span>
<span id="cb38-113"><a href="#cb38-113" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-114"><a href="#cb38-114" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">'Plots/MECP2_regulated_DEGs_peak_distribution.pdf'</span>,<span class="at">width =</span> <span class="dv">15</span>,<span class="at">height =</span> <span class="dv">10</span>)</span>
<span id="cb38-115"><a href="#cb38-115" aria-hidden="true" tabindex="-1"></a><span class="fu">do.call</span>(grid.arrange,<span class="fu">c</span>(Pie.chart.list, <span class="at">ncol=</span><span class="dv">3</span>))</span>
<span id="cb38-116"><a href="#cb38-116" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></li>
<li><p>Supplementary Figure 4B</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">'results/MECP2/MECP2_regulatedGenes.RData'</span>)</span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Correlation of number of MECP2 peaks with Average logFC</span></span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a><span class="co"># WholeCell</span></span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a>Mecp2.Bound.meth.expressed.wholeCell <span class="sc">%&gt;%</span> </span>
<span id="cb39-6"><a href="#cb39-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(DEG<span class="sc">!=</span><span class="st">'Unchanged'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb39-7"><a href="#cb39-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="st">"DEG"</span>,<span class="st">"ENSEMBL"</span>,<span class="st">"logFC"</span>,<span class="st">"geneLength"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb39-8"><a href="#cb39-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(DEG,ENSEMBL,logFC)  <span class="sc">%&gt;%</span> </span>
<span id="cb39-9"><a href="#cb39-9" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">count=</span>dplyr<span class="sc">::</span><span class="fu">n</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb39-10"><a href="#cb39-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">count=</span><span class="fu">as.character</span>(count)) <span class="sc">%&gt;%</span> </span>
<span id="cb39-11"><a href="#cb39-11" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(DEG,count) <span class="sc">%&gt;%</span> </span>
<span id="cb39-12"><a href="#cb39-12" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">AvglogFC =</span> <span class="fu">mean</span>(logFC)) <span class="sc">%&gt;%</span></span>
<span id="cb39-13"><a href="#cb39-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">count=</span><span class="fu">as.integer</span>(count)) <span class="sc">%&gt;%</span></span>
<span id="cb39-14"><a href="#cb39-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>count,<span class="at">y=</span>AvglogFC))<span class="sc">+</span></span>
<span id="cb39-15"><a href="#cb39-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>()<span class="sc">+</span></span>
<span id="cb39-16"><a href="#cb39-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_cor</span>()<span class="sc">+</span></span>
<span id="cb39-17"><a href="#cb39-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">'lm'</span>)<span class="sc">+</span></span>
<span id="cb39-18"><a href="#cb39-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>DEG,<span class="at">scales =</span> <span class="st">"free"</span>)<span class="sc">+</span></span>
<span id="cb39-19"><a href="#cb39-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_test</span>(<span class="dv">20</span>)<span class="sc">+</span></span>
<span id="cb39-20"><a href="#cb39-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">'Number of MECP2 peaks associated with DEGs'</span>)</span>
<span id="cb39-21"><a href="#cb39-21" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"Plots/MECP2_peaks_correlation_with_logFC.WholeCell.pdf"</span>,<span class="at">height =</span> <span class="dv">5</span>,<span class="at">width =</span> <span class="dv">10</span>)</span>
<span id="cb39-22"><a href="#cb39-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-23"><a href="#cb39-23" aria-hidden="true" tabindex="-1"></a><span class="co">#Nuclear</span></span>
<span id="cb39-24"><a href="#cb39-24" aria-hidden="true" tabindex="-1"></a>Mecp2.Bound.meth.expressed.nuclear <span class="sc">%&gt;%</span> </span>
<span id="cb39-25"><a href="#cb39-25" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(DEG<span class="sc">!=</span><span class="st">'Unchanged'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb39-26"><a href="#cb39-26" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="st">"DEG"</span>,<span class="st">"ENSEMBL"</span>,<span class="st">"logFC"</span>,<span class="st">"geneLength"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb39-27"><a href="#cb39-27" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(DEG,ENSEMBL,logFC)  <span class="sc">%&gt;%</span> </span>
<span id="cb39-28"><a href="#cb39-28" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">count=</span>dplyr<span class="sc">::</span><span class="fu">n</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb39-29"><a href="#cb39-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">count=</span><span class="fu">as.character</span>(count)) <span class="sc">%&gt;%</span> </span>
<span id="cb39-30"><a href="#cb39-30" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(DEG,count) <span class="sc">%&gt;%</span> </span>
<span id="cb39-31"><a href="#cb39-31" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">AvglogFC =</span> <span class="fu">mean</span>(logFC)) <span class="sc">%&gt;%</span></span>
<span id="cb39-32"><a href="#cb39-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">count=</span><span class="fu">as.integer</span>(count)) <span class="sc">%&gt;%</span></span>
<span id="cb39-33"><a href="#cb39-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>count,<span class="at">y=</span>AvglogFC))<span class="sc">+</span></span>
<span id="cb39-34"><a href="#cb39-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>()<span class="sc">+</span></span>
<span id="cb39-35"><a href="#cb39-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_cor</span>()<span class="sc">+</span></span>
<span id="cb39-36"><a href="#cb39-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">'lm'</span>)<span class="sc">+</span></span>
<span id="cb39-37"><a href="#cb39-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>DEG,<span class="at">scales =</span> <span class="st">"free"</span>)<span class="sc">+</span></span>
<span id="cb39-38"><a href="#cb39-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_test</span>(<span class="dv">20</span>)<span class="sc">+</span></span>
<span id="cb39-39"><a href="#cb39-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">'Number of MECP2 peaks associated with DEGs'</span>)</span>
<span id="cb39-40"><a href="#cb39-40" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"Plots/MECP2_peaks_correlation_with_logFC.Nuclear.pdf"</span>,<span class="at">height =</span> <span class="dv">5</span>,<span class="at">width =</span> <span class="dv">10</span>)</span>
<span id="cb39-41"><a href="#cb39-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-42"><a href="#cb39-42" aria-hidden="true" tabindex="-1"></a><span class="co">#Chromatin</span></span>
<span id="cb39-43"><a href="#cb39-43" aria-hidden="true" tabindex="-1"></a>Mecp2.Bound.meth.expressed.chromatin <span class="sc">%&gt;%</span> </span>
<span id="cb39-44"><a href="#cb39-44" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(DEG<span class="sc">!=</span><span class="st">'Unchanged'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb39-45"><a href="#cb39-45" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="st">"DEG"</span>,<span class="st">"ENSEMBL"</span>,<span class="st">"logFC"</span>,<span class="st">"geneLength"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb39-46"><a href="#cb39-46" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(DEG,ENSEMBL,logFC)  <span class="sc">%&gt;%</span> </span>
<span id="cb39-47"><a href="#cb39-47" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">count=</span>dplyr<span class="sc">::</span><span class="fu">n</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb39-48"><a href="#cb39-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">count=</span><span class="fu">as.character</span>(count)) <span class="sc">%&gt;%</span> </span>
<span id="cb39-49"><a href="#cb39-49" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(DEG,count) <span class="sc">%&gt;%</span> </span>
<span id="cb39-50"><a href="#cb39-50" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">AvglogFC =</span> <span class="fu">mean</span>(logFC)) <span class="sc">%&gt;%</span></span>
<span id="cb39-51"><a href="#cb39-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">count=</span><span class="fu">as.integer</span>(count)) <span class="sc">%&gt;%</span></span>
<span id="cb39-52"><a href="#cb39-52" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>count,<span class="at">y=</span>AvglogFC))<span class="sc">+</span></span>
<span id="cb39-53"><a href="#cb39-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>()<span class="sc">+</span></span>
<span id="cb39-54"><a href="#cb39-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_cor</span>()<span class="sc">+</span></span>
<span id="cb39-55"><a href="#cb39-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">'lm'</span>)<span class="sc">+</span></span>
<span id="cb39-56"><a href="#cb39-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>DEG,<span class="at">scales =</span> <span class="st">"free"</span>)<span class="sc">+</span></span>
<span id="cb39-57"><a href="#cb39-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_test</span>(<span class="dv">20</span>)<span class="sc">+</span></span>
<span id="cb39-58"><a href="#cb39-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">'Number of MECP2 peaks associated with DEGs'</span>)</span>
<span id="cb39-59"><a href="#cb39-59" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"Plots/MECP2_peaks_correlation_with_logFC.Chromatin.pdf"</span>,<span class="at">height =</span> <span class="dv">5</span>,<span class="at">width =</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></li>
<li><p>Supplementary Figure 4D</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="co"># read SuperEnhacer Table </span></span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a>SE_size.GR <span class="ot">=</span><span class="fu">read.csv</span>(<span class="st">'results/MECP2_SE/MECP2WT_AllEnhancers.table.txt'</span>,<span class="at">sep=</span><span class="st">'</span><span class="sc">\t</span><span class="st">'</span>,</span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a>                     <span class="at">header =</span> T,<span class="at">skip =</span> <span class="dv">5</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb40-4"><a href="#cb40-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">1</span>,<span class="dv">10</span>) <span class="sc">%&gt;%</span> <span class="fu">GRanges</span>()</span>
<span id="cb40-5"><a href="#cb40-5" aria-hidden="true" tabindex="-1"></a>txdb <span class="ot">=</span> TxDb.Mmusculus.UCSC.mm10.knownGene</span>
<span id="cb40-6"><a href="#cb40-6" aria-hidden="true" tabindex="-1"></a>MECP2_SE.all.peakAnno  <span class="ot">=</span> <span class="fu">annotatePeak</span>(SE_size.GR, <span class="at">tssRegion=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">1000</span>, <span class="dv">1000</span>),<span class="at">TxDb=</span>txdb, <span class="at">annoDb=</span><span class="st">"org.Mm.eg.db"</span>)</span>
<span id="cb40-7"><a href="#cb40-7" aria-hidden="true" tabindex="-1"></a>MECP2_SE.all.peakAnno.df <span class="ot">=</span> MECP2_SE.all.peakAnno <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>()</span>
<span id="cb40-8"><a href="#cb40-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-9"><a href="#cb40-9" aria-hidden="true" tabindex="-1"></a>MECP2_SE_genes.wholeCell <span class="ot">=</span> <span class="fu">merge</span>(MECP2_SE.all.peakAnno.df,mecp2.binding.meth.RNAseqKO_wholeCell <span class="sc">%&gt;%</span> <span class="fu">filter</span>(FDR <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="dv">1</span>,<span class="dv">21</span>,<span class="dv">22</span>) <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(),</span>
<span id="cb40-10"><a href="#cb40-10" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">by =</span> <span class="st">'ENSEMBL'</span>,<span class="at">all=</span><span class="cn">TRUE</span>)</span>
<span id="cb40-11"><a href="#cb40-11" aria-hidden="true" tabindex="-1"></a>  <span class="co">#filter(DEG == 'Repressed') </span></span>
<span id="cb40-12"><a href="#cb40-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-13"><a href="#cb40-13" aria-hidden="true" tabindex="-1"></a>MECP2_SE_genes.wholeCell <span class="sc">%&gt;%</span> </span>
<span id="cb40-14"><a href="#cb40-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(logFC)) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="st">'logFC'</span>,<span class="st">'isSuper'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb40-15"><a href="#cb40-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(logFC <span class="sc">&gt;</span> <span class="sc">-</span><span class="dv">2</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb40-16"><a href="#cb40-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">isSuper =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(isSuper),<span class="dv">2</span>,isSuper)) <span class="sc">%&gt;%</span> </span>
<span id="cb40-17"><a href="#cb40-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(isSuper <span class="sc">!=</span><span class="dv">0</span>) <span class="sc">%&gt;%</span> <span class="fu">distinct</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb40-18"><a href="#cb40-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">isSuper =</span> <span class="fu">ifelse</span>(isSuper<span class="sc">==</span><span class="dv">2</span>,<span class="dv">0</span>,isSuper)) <span class="sc">%&gt;%</span> </span>
<span id="cb40-19"><a href="#cb40-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span><span class="fu">as.character</span>(isSuper),<span class="at">y=</span>logFC,<span class="at">color=</span><span class="fu">as.character</span>(isSuper)))<span class="sc">+</span><span class="fu">geom_boxplot</span>()<span class="sc">+</span></span>
<span id="cb40-20"><a href="#cb40-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>(<span class="at">method =</span> <span class="st">'wilcox'</span>,<span class="at">label.y =</span> <span class="fl">1.2</span>)<span class="sc">+</span></span>
<span id="cb40-21"><a href="#cb40-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_test</span>(<span class="dv">20</span>)<span class="sc">+</span></span>
<span id="cb40-22"><a href="#cb40-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">'grey'</span>,<span class="st">'red'</span>))<span class="sc">+</span></span>
<span id="cb40-23"><a href="#cb40-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">labels=</span><span class="fu">c</span>(<span class="fu">paste</span>(<span class="st">'All Genes'</span>),</span>
<span id="cb40-24"><a href="#cb40-24" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">paste</span>(<span class="st">'Genes with</span><span class="sc">\n</span><span class="st">Broad MBH clusters'</span>)))<span class="sc">+</span></span>
<span id="cb40-25"><a href="#cb40-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_test</span>(<span class="dv">20</span>)<span class="sc">+</span></span>
<span id="cb40-26"><a href="#cb40-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"log2 Fold Change</span><span class="sc">\n</span><span class="st">(Mecp2 KO/WT)"</span>)<span class="sc">+</span></span>
<span id="cb40-27"><a href="#cb40-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">'none'</span>,<span class="at">axis.title.x =</span> <span class="fu">element_blank</span>())</span>
<span id="cb40-28"><a href="#cb40-28" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">'Plots/MECP2_SE_genes_logFC.pdf'</span>,<span class="at">height =</span> <span class="dv">7</span>,<span class="at">width =</span> <span class="dv">6</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></li>
</ol>
</section>
<section id="figure-5-mbhs-repress-genes-independently-of-the-mca-pathway." class="level3">
<h3 class="anchored" data-anchor-id="figure-5-mbhs-repress-genes-independently-of-the-mca-pathway."><strong>Figure 5: MBHs repress genes independently of the mCA pathway.</strong></h3>
<ol type="1">
<li><p>Figure 5A</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"results/RData/Boxer_RNAseq_data.RData"</span>)</span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">'results/RData/MECP2_regulatedGenes.RData'</span>,<span class="at">verbose =</span> <span class="cn">TRUE</span>)</span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">'results/RData/MECP2_peakAnnotation.RData'</span>,<span class="at">verbose =</span> <span class="cn">TRUE</span>)</span>
<span id="cb41-4"><a href="#cb41-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-5"><a href="#cb41-5" aria-hidden="true" tabindex="-1"></a>MECP2KO_up_unique <span class="ot">=</span> <span class="fu">setdiff</span>(RNAseq.list[[<span class="dv">1</span>]] <span class="sc">%&gt;%</span> <span class="fu">filter</span>(FDR <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> logFC <span class="sc">&gt;</span><span class="dv">0</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(Gene),</span>
<span id="cb41-6"><a href="#cb41-6" aria-hidden="true" tabindex="-1"></a>                            RNAseq.list[[<span class="dv">7</span>]] <span class="sc">%&gt;%</span> <span class="fu">filter</span>(FDR <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> logFC <span class="sc">&gt;</span><span class="dv">0</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(Gene_name)) </span>
<span id="cb41-7"><a href="#cb41-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-8"><a href="#cb41-8" aria-hidden="true" tabindex="-1"></a>DNMT3AKO_up_unique <span class="ot">=</span> <span class="fu">setdiff</span>(RNAseq.list[[<span class="dv">7</span>]] <span class="sc">%&gt;%</span> <span class="fu">filter</span>(FDR <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> logFC <span class="sc">&gt;</span><span class="dv">0</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(Gene_name),</span>
<span id="cb41-9"><a href="#cb41-9" aria-hidden="true" tabindex="-1"></a>                             RNAseq.list[[<span class="dv">1</span>]] <span class="sc">%&gt;%</span> <span class="fu">filter</span>(FDR <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> logFC <span class="sc">&gt;</span><span class="dv">0</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(Gene))</span>
<span id="cb41-10"><a href="#cb41-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-11"><a href="#cb41-11" aria-hidden="true" tabindex="-1"></a>MECP2KO_up_unique_peaks <span class="ot">=</span> mecp2.binding.meth.RNAseqKO_wholeCell <span class="sc">%&gt;%</span> </span>
<span id="cb41-12"><a href="#cb41-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Gene <span class="sc">%in%</span> MECP2KO_up_unique) <span class="sc">%&gt;%</span> </span>
<span id="cb41-13"><a href="#cb41-13" aria-hidden="true" tabindex="-1"></a>  <span class="co">#filter(!is.na(ENSEMBL)) %&gt;% </span></span>
<span id="cb41-14"><a href="#cb41-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Binding =</span> <span class="fu">ifelse</span>(<span class="sc">!</span><span class="fu">is.na</span>(SYMBOL),<span class="st">'Bound'</span>,<span class="st">'Unbound'</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb41-15"><a href="#cb41-15" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(Gene,Binding) <span class="sc">%&gt;%</span> </span>
<span id="cb41-16"><a href="#cb41-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Gene,Binding) <span class="sc">%&gt;%</span> </span>
<span id="cb41-17"><a href="#cb41-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">MECP2 =</span> dplyr<span class="sc">::</span><span class="fu">n</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb41-18"><a href="#cb41-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">MECP2 =</span> <span class="fu">ifelse</span>(Binding <span class="sc">==</span> <span class="st">'Unbound'</span>,<span class="dv">0</span>,MECP2)) </span>
<span id="cb41-19"><a href="#cb41-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-20"><a href="#cb41-20" aria-hidden="true" tabindex="-1"></a>DNMT3A_up_unique_peaks <span class="ot">=</span> <span class="fu">merge</span>(RNAseq.list[[<span class="dv">7</span>]],mecp2.binding,<span class="at">by=</span><span class="st">'ENSEMBL'</span>,<span class="at">all.x=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb41-21"><a href="#cb41-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Gene_name <span class="sc">%in%</span> DNMT3AKO_up_unique) <span class="sc">%&gt;%</span> </span>
<span id="cb41-22"><a href="#cb41-22" aria-hidden="true" tabindex="-1"></a>  <span class="co">#filter(!is.na(ENSEMBL)) %&gt;%</span></span>
<span id="cb41-23"><a href="#cb41-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Binding =</span> <span class="fu">ifelse</span>(<span class="sc">!</span><span class="fu">is.na</span>(Gene_name),<span class="st">'Bound'</span>,<span class="st">'Unbound'</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb41-24"><a href="#cb41-24" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(Gene_name,Binding) <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(Gene_name,Binding) <span class="sc">%&gt;%</span> </span>
<span id="cb41-25"><a href="#cb41-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">MECP2 =</span> dplyr<span class="sc">::</span><span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb41-26"><a href="#cb41-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">MECP2 =</span> <span class="fu">ifelse</span>(Binding <span class="sc">==</span> <span class="st">'Unbound'</span>,<span class="dv">0</span>,MECP2)) </span>
<span id="cb41-27"><a href="#cb41-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-28"><a href="#cb41-28" aria-hidden="true" tabindex="-1"></a><span class="fu">rbind</span>(DNMT3A_up_unique_peaks[,<span class="dv">2</span><span class="sc">:</span><span class="dv">3</span>] <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">Group=</span><span class="st">'DNMT3A'</span>),</span>
<span id="cb41-29"><a href="#cb41-29" aria-hidden="true" tabindex="-1"></a>        MECP2KO_up_unique_peaks[,<span class="dv">2</span><span class="sc">:</span><span class="dv">3</span>] <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">Group=</span><span class="st">'MECP2'</span>)) <span class="sc">%&gt;%</span>  </span>
<span id="cb41-30"><a href="#cb41-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(MECP2 <span class="sc">&gt;</span><span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb41-31"><a href="#cb41-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>Group,<span class="at">y=</span>MECP2))<span class="sc">+</span></span>
<span id="cb41-32"><a href="#cb41-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>()<span class="sc">+</span></span>
<span id="cb41-33"><a href="#cb41-33" aria-hidden="true" tabindex="-1"></a>  <span class="co">#ylim(0,20)+</span></span>
<span id="cb41-34"><a href="#cb41-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">55</span>))<span class="sc">+</span></span>
<span id="cb41-35"><a href="#cb41-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>(<span class="at">label.y =</span> <span class="dv">45</span>) <span class="sc">+</span></span>
<span id="cb41-36"><a href="#cb41-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_test</span>(<span class="dv">20</span>)<span class="sc">+</span></span>
<span id="cb41-37"><a href="#cb41-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Number of MBHs"</span>)</span>
<span id="cb41-38"><a href="#cb41-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-39"><a href="#cb41-39" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">'Plots/MECP2_peak_distribution_MECP2KO_DNMTKO_genes.pdf'</span>,<span class="at">height =</span> <span class="dv">4</span>,<span class="at">width =</span> <span class="dv">4</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></li>
<li><p>Figure 5B, Supplementary Figure 5A-B.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"scripts/R/logFC_mCA_correlation.R"</span>)</span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">'results/MECP2/MECP2_regulatedGenes.RData'</span>,<span class="at">verbose=</span><span class="cn">TRUE</span>)</span>
<span id="cb42-3"><a href="#cb42-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-4"><a href="#cb42-4" aria-hidden="true" tabindex="-1"></a><span class="co"># WholeCell</span></span>
<span id="cb42-5"><a href="#cb42-5" aria-hidden="true" tabindex="-1"></a>pltt.linePlot <span class="ot">=</span> <span class="fu">logFC_mCA_corr</span>(mecp2.binding.meth.RNAseqKO_wholeCell,<span class="st">"linePlot"</span>)</span>
<span id="cb42-6"><a href="#cb42-6" aria-hidden="true" tabindex="-1"></a>pltt.boxplot <span class="ot">=</span> <span class="fu">logFC_mCA_corr</span>(mecp2.binding.meth.RNAseqKO_wholeCell,<span class="st">"boxplot"</span>)</span>
<span id="cb42-7"><a href="#cb42-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-8"><a href="#cb42-8" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(pltt.linePlot)) {</span>
<span id="cb42-9"><a href="#cb42-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pdf</span>(<span class="fu">paste0</span>(<span class="st">'Plots/logFC_mCA_correlation_WholeCellRNAseq_'</span>,i,<span class="st">'.pdf'</span>),<span class="at">height =</span> <span class="dv">5</span>,<span class="at">width =</span> <span class="dv">10</span>)</span>
<span id="cb42-10"><a href="#cb42-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(pltt.linePlot[[i]])</span>
<span id="cb42-11"><a href="#cb42-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dev.off</span>()</span>
<span id="cb42-12"><a href="#cb42-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-13"><a href="#cb42-13" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb42-14"><a href="#cb42-14" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(pltt.linePlot)) {</span>
<span id="cb42-15"><a href="#cb42-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pdf</span>(<span class="fu">paste0</span>(<span class="st">'Plots/logFC_mCA_correlation_WholeCellRNAseq_boxplot_'</span>,i,<span class="st">'.pdf'</span>),<span class="at">height =</span><span class="dv">8</span>,<span class="at">width =</span> <span class="dv">6</span>)</span>
<span id="cb42-16"><a href="#cb42-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(pltt.boxplot[[i]])</span>
<span id="cb42-17"><a href="#cb42-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dev.off</span>()</span>
<span id="cb42-18"><a href="#cb42-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-19"><a href="#cb42-19" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb42-20"><a href="#cb42-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-21"><a href="#cb42-21" aria-hidden="true" tabindex="-1"></a><span class="co"># Nuclear</span></span>
<span id="cb42-22"><a href="#cb42-22" aria-hidden="true" tabindex="-1"></a>pltt.linePlot <span class="ot">=</span> <span class="fu">logFC_mCA_corr</span>(mecp2.binding.meth.RNAseqKO_nuclear,<span class="st">"linePlot"</span>)</span>
<span id="cb42-23"><a href="#cb42-23" aria-hidden="true" tabindex="-1"></a>pltt.boxplot <span class="ot">=</span> <span class="fu">logFC_mCA_corr</span>(mecp2.binding.meth.RNAseqKO_nuclear,<span class="st">"boxplot"</span>)</span>
<span id="cb42-24"><a href="#cb42-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-25"><a href="#cb42-25" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(pltt.linePlot)) {</span>
<span id="cb42-26"><a href="#cb42-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pdf</span>(<span class="fu">paste0</span>(<span class="st">'Plots/logFC_mCA_correlation_Nuclear_'</span>,i,<span class="st">'.pdf'</span>),<span class="at">height =</span> <span class="dv">5</span>,<span class="at">width =</span> <span class="dv">10</span>)</span>
<span id="cb42-27"><a href="#cb42-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(pltt.linePlot[[i]])</span>
<span id="cb42-28"><a href="#cb42-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dev.off</span>()</span>
<span id="cb42-29"><a href="#cb42-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-30"><a href="#cb42-30" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb42-31"><a href="#cb42-31" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(pltt.linePlot)) {</span>
<span id="cb42-32"><a href="#cb42-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pdf</span>(<span class="fu">paste0</span>(<span class="st">'Plots/logFC_mCA_correlation_Nuclear_boxplot_'</span>,i,<span class="st">'.pdf'</span>),<span class="at">height =</span><span class="dv">8</span>,<span class="at">width =</span> <span class="dv">6</span>)</span>
<span id="cb42-33"><a href="#cb42-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(pltt.boxplot[[i]])</span>
<span id="cb42-34"><a href="#cb42-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dev.off</span>()</span>
<span id="cb42-35"><a href="#cb42-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-36"><a href="#cb42-36" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb42-37"><a href="#cb42-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-38"><a href="#cb42-38" aria-hidden="true" tabindex="-1"></a><span class="co"># Chromatin</span></span>
<span id="cb42-39"><a href="#cb42-39" aria-hidden="true" tabindex="-1"></a>pltt.linePlot <span class="ot">=</span> <span class="fu">logFC_mCA_corr</span>(mecp2.binding.meth.RNAseqKO_chromatin,<span class="st">"linePlot"</span>)</span>
<span id="cb42-40"><a href="#cb42-40" aria-hidden="true" tabindex="-1"></a>pltt.boxplot <span class="ot">=</span> <span class="fu">logFC_mCA_corr</span>(mecp2.binding.meth.RNAseqKO_chromatin,<span class="st">"boxplot"</span>)</span>
<span id="cb42-41"><a href="#cb42-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-42"><a href="#cb42-42" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(pltt.linePlot)) {</span>
<span id="cb42-43"><a href="#cb42-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pdf</span>(<span class="fu">paste0</span>(<span class="st">'Plots/logFC_mCA_correlation_Chromatin_'</span>,i,<span class="st">'.pdf'</span>),<span class="at">height =</span> <span class="dv">5</span>,<span class="at">width =</span> <span class="dv">10</span>)</span>
<span id="cb42-44"><a href="#cb42-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(pltt.linePlot[[i]])</span>
<span id="cb42-45"><a href="#cb42-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dev.off</span>()</span>
<span id="cb42-46"><a href="#cb42-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-47"><a href="#cb42-47" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb42-48"><a href="#cb42-48" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(pltt.linePlot)) {</span>
<span id="cb42-49"><a href="#cb42-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pdf</span>(<span class="fu">paste0</span>(<span class="st">'Plots/logFC_mCA_correlation_Chromatin_boxplot_'</span>,i,<span class="st">'.pdf'</span>),<span class="at">height =</span><span class="dv">8</span>,<span class="at">width =</span> <span class="dv">6</span>)</span>
<span id="cb42-50"><a href="#cb42-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(pltt.boxplot[[i]])</span>
<span id="cb42-51"><a href="#cb42-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dev.off</span>()</span>
<span id="cb42-52"><a href="#cb42-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-53"><a href="#cb42-53" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb42-54"><a href="#cb42-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-55"><a href="#cb42-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-56"><a href="#cb42-56" aria-hidden="true" tabindex="-1"></a>MECP2_regulated_genes_with_mCA <span class="ot">=</span> <span class="fu">list</span>()</span>
<span id="cb42-57"><a href="#cb42-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-58"><a href="#cb42-58" aria-hidden="true" tabindex="-1"></a>MECP2_regulated_genes_with_mCA[[<span class="dv">1</span>]] <span class="ot">=</span> mecp2.binding.meth.RNAseqKO_wholeCell <span class="sc">%&gt;%</span> </span>
<span id="cb42-59"><a href="#cb42-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(DEG <span class="sc">!=</span> <span class="st">'Unchanged'</span>, Meth_CA <span class="sc">==</span> <span class="st">'CA'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb42-60"><a href="#cb42-60" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mCA =</span> <span class="fu">ifelse</span>(CA_meth <span class="sc">&gt;</span> <span class="fu">median</span>(CA_meth),<span class="st">'High_mCA'</span>, <span class="st">'Low_mCA'</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb42-61"><a href="#cb42-61" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">22</span>,<span class="dv">26</span>,<span class="dv">31</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb42-62"><a href="#cb42-62" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(logFC) <span class="sc">%&gt;%</span> </span>
<span id="cb42-63"><a href="#cb42-63" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">MECP2 =</span> dplyr<span class="sc">::</span><span class="fu">n</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb42-64"><a href="#cb42-64" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb42-65"><a href="#cb42-65" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb42-66"><a href="#cb42-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">MECP2 =</span> <span class="fu">ifelse</span>(Binding <span class="sc">==</span> <span class="st">'Unbound'</span> <span class="sc">&amp;</span> MECP2 <span class="sc">==</span><span class="dv">1</span>, <span class="dv">0</span>, MECP2),</span>
<span id="cb42-67"><a href="#cb42-67" aria-hidden="true" tabindex="-1"></a>         <span class="at">Group1 =</span> <span class="st">'All'</span>,</span>
<span id="cb42-68"><a href="#cb42-68" aria-hidden="true" tabindex="-1"></a>         <span class="at">Group2 =</span> <span class="fu">ifelse</span>(MECP2 <span class="sc">&gt;=</span><span class="dv">7</span>, <span class="st">'&gt;=7 MBHs'</span>,<span class="st">'NA'</span>))  <span class="sc">%&gt;%</span> </span>
<span id="cb42-69"><a href="#cb42-69" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">5</span>,<span class="dv">6</span>,<span class="dv">3</span>,<span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb42-70"><a href="#cb42-70" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols=</span><span class="fu">c</span>(<span class="st">'Group1'</span>,<span class="st">'Group2'</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb42-71"><a href="#cb42-71" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(value <span class="sc">!=</span> <span class="st">'NA'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb42-72"><a href="#cb42-72" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">value =</span> <span class="fu">factor</span>(value,<span class="at">levels=</span> <span class="fu">c</span>(<span class="st">'All'</span>,<span class="st">'&gt;=7 MBHs'</span>)),</span>
<span id="cb42-73"><a href="#cb42-73" aria-hidden="true" tabindex="-1"></a>         <span class="at">mCA=</span> <span class="fu">factor</span>(mCA, <span class="at">levels=</span> <span class="fu">c</span>(<span class="st">'Low_mCA'</span>,<span class="st">'High_mCA'</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb42-74"><a href="#cb42-74" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>value,<span class="at">y=</span>logFC,<span class="at">color=</span>value))<span class="sc">+</span></span>
<span id="cb42-75"><a href="#cb42-75" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>()<span class="sc">+</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>mCA)<span class="sc">+</span></span>
<span id="cb42-76"><a href="#cb42-76" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>()<span class="sc">+</span></span>
<span id="cb42-77"><a href="#cb42-77" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">'grey'</span>,<span class="st">'red'</span>))<span class="sc">+</span></span>
<span id="cb42-78"><a href="#cb42-78" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_test</span>(<span class="dv">20</span>)</span>
<span id="cb42-79"><a href="#cb42-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-80"><a href="#cb42-80" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-81"><a href="#cb42-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-82"><a href="#cb42-82" aria-hidden="true" tabindex="-1"></a>MECP2_regulated_genes_with_mCA[[<span class="dv">2</span>]] <span class="ot">=</span> mecp2.binding.meth.RNAseqKO_nuclear <span class="sc">%&gt;%</span> </span>
<span id="cb42-83"><a href="#cb42-83" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(DEG <span class="sc">!=</span> <span class="st">'Unchanged'</span>, Meth_CA <span class="sc">==</span> <span class="st">'CA'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb42-84"><a href="#cb42-84" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mCA =</span> <span class="fu">ifelse</span>(CA_meth <span class="sc">&gt;</span> <span class="fu">median</span>(CA_meth),<span class="st">'High_mCA'</span>, <span class="st">'Low_mCA'</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb42-85"><a href="#cb42-85" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">22</span>,<span class="dv">26</span>,<span class="dv">31</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb42-86"><a href="#cb42-86" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(logFC) <span class="sc">%&gt;%</span> </span>
<span id="cb42-87"><a href="#cb42-87" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">MECP2 =</span> dplyr<span class="sc">::</span><span class="fu">n</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb42-88"><a href="#cb42-88" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb42-89"><a href="#cb42-89" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb42-90"><a href="#cb42-90" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">MECP2 =</span> <span class="fu">ifelse</span>(Binding <span class="sc">==</span> <span class="st">'Unbound'</span> <span class="sc">&amp;</span> MECP2 <span class="sc">==</span><span class="dv">1</span>, <span class="dv">0</span>, MECP2),</span>
<span id="cb42-91"><a href="#cb42-91" aria-hidden="true" tabindex="-1"></a>         <span class="at">Group1 =</span> <span class="st">'All'</span>,</span>
<span id="cb42-92"><a href="#cb42-92" aria-hidden="true" tabindex="-1"></a>         <span class="at">Group2 =</span> <span class="fu">ifelse</span>(MECP2 <span class="sc">&gt;=</span><span class="dv">7</span>, <span class="st">'&gt;=7 MBHs'</span>,<span class="st">'NA'</span>))  <span class="sc">%&gt;%</span> </span>
<span id="cb42-93"><a href="#cb42-93" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">5</span>,<span class="dv">6</span>,<span class="dv">3</span>,<span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb42-94"><a href="#cb42-94" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols=</span><span class="fu">c</span>(<span class="st">'Group1'</span>,<span class="st">'Group2'</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb42-95"><a href="#cb42-95" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(value <span class="sc">!=</span> <span class="st">'NA'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb42-96"><a href="#cb42-96" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">value =</span> <span class="fu">factor</span>(value,<span class="at">levels=</span> <span class="fu">c</span>(<span class="st">'All'</span>,<span class="st">'&gt;=7 MBHs'</span>)),</span>
<span id="cb42-97"><a href="#cb42-97" aria-hidden="true" tabindex="-1"></a>         <span class="at">mCA=</span> <span class="fu">factor</span>(mCA, <span class="at">levels=</span> <span class="fu">c</span>(<span class="st">'Low_mCA'</span>,<span class="st">'High_mCA'</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb42-98"><a href="#cb42-98" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>value,<span class="at">y=</span>logFC,<span class="at">color=</span>value))<span class="sc">+</span></span>
<span id="cb42-99"><a href="#cb42-99" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>()<span class="sc">+</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>mCA)<span class="sc">+</span></span>
<span id="cb42-100"><a href="#cb42-100" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>()<span class="sc">+</span></span>
<span id="cb42-101"><a href="#cb42-101" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">'grey'</span>,<span class="st">'red'</span>))<span class="sc">+</span></span>
<span id="cb42-102"><a href="#cb42-102" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_test</span>(<span class="dv">20</span>)</span>
<span id="cb42-103"><a href="#cb42-103" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-104"><a href="#cb42-104" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-105"><a href="#cb42-105" aria-hidden="true" tabindex="-1"></a>MECP2_regulated_genes_with_mCA[[<span class="dv">3</span>]] <span class="ot">=</span> mecp2.binding.meth.RNAseqKO_chromatin <span class="sc">%&gt;%</span> </span>
<span id="cb42-106"><a href="#cb42-106" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(DEG <span class="sc">!=</span> <span class="st">'Unchanged'</span>, Meth_CA <span class="sc">==</span> <span class="st">'CA'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb42-107"><a href="#cb42-107" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mCA =</span> <span class="fu">ifelse</span>(CA_meth <span class="sc">&gt;</span> <span class="fu">median</span>(CA_meth),<span class="st">'High_mCA'</span>, <span class="st">'Low_mCA'</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb42-108"><a href="#cb42-108" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">22</span>,<span class="dv">26</span>,<span class="dv">31</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb42-109"><a href="#cb42-109" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(logFC) <span class="sc">%&gt;%</span> </span>
<span id="cb42-110"><a href="#cb42-110" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">MECP2 =</span> dplyr<span class="sc">::</span><span class="fu">n</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb42-111"><a href="#cb42-111" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb42-112"><a href="#cb42-112" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb42-113"><a href="#cb42-113" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">MECP2 =</span> <span class="fu">ifelse</span>(Binding <span class="sc">==</span> <span class="st">'Unbound'</span> <span class="sc">&amp;</span> MECP2 <span class="sc">==</span><span class="dv">1</span>, <span class="dv">0</span>, MECP2),</span>
<span id="cb42-114"><a href="#cb42-114" aria-hidden="true" tabindex="-1"></a>         <span class="at">Group1 =</span> <span class="st">'All'</span>,</span>
<span id="cb42-115"><a href="#cb42-115" aria-hidden="true" tabindex="-1"></a>         <span class="at">Group2 =</span> <span class="fu">ifelse</span>(MECP2 <span class="sc">&gt;=</span><span class="dv">7</span>, <span class="st">'&gt;=7 MBHs'</span>,<span class="st">'NA'</span>))  <span class="sc">%&gt;%</span> </span>
<span id="cb42-116"><a href="#cb42-116" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">5</span>,<span class="dv">6</span>,<span class="dv">3</span>,<span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb42-117"><a href="#cb42-117" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols=</span><span class="fu">c</span>(<span class="st">'Group1'</span>,<span class="st">'Group2'</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb42-118"><a href="#cb42-118" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(value <span class="sc">!=</span> <span class="st">'NA'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb42-119"><a href="#cb42-119" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">value =</span> <span class="fu">factor</span>(value,<span class="at">levels=</span> <span class="fu">c</span>(<span class="st">'All'</span>,<span class="st">'&gt;=7 MBHs'</span>)),</span>
<span id="cb42-120"><a href="#cb42-120" aria-hidden="true" tabindex="-1"></a>         <span class="at">mCA=</span> <span class="fu">factor</span>(mCA, <span class="at">levels=</span> <span class="fu">c</span>(<span class="st">'Low_mCA'</span>,<span class="st">'High_mCA'</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb42-121"><a href="#cb42-121" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>value,<span class="at">y=</span>logFC,<span class="at">color=</span>value))<span class="sc">+</span></span>
<span id="cb42-122"><a href="#cb42-122" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>()<span class="sc">+</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>mCA)<span class="sc">+</span></span>
<span id="cb42-123"><a href="#cb42-123" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>()<span class="sc">+</span></span>
<span id="cb42-124"><a href="#cb42-124" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">'grey'</span>,<span class="st">'red'</span>))<span class="sc">+</span></span>
<span id="cb42-125"><a href="#cb42-125" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_test</span>(<span class="dv">20</span>)</span>
<span id="cb42-126"><a href="#cb42-126" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-127"><a href="#cb42-127" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">'Plots/MECP2_regulated_DEGs_with_mCA_boxplot.pdf'</span>,<span class="at">height =</span> <span class="dv">4</span>,<span class="at">width =</span> <span class="dv">15</span>) </span>
<span id="cb42-128"><a href="#cb42-128" aria-hidden="true" tabindex="-1"></a><span class="fu">do.call</span>(grid.arrange,<span class="fu">c</span>(MECP2_regulated_genes_with_mCA,<span class="at">ncol=</span><span class="dv">3</span>))</span>
<span id="cb42-129"><a href="#cb42-129" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>() </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></li>
<li><p>Figure 5C, Supplementary Figure 5C-D.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">'results/MECP2/MECP2_regulatedGenes.RData'</span>,<span class="at">verbose=</span><span class="cn">TRUE</span>)</span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">'results/RData/ENSEMBL.IDs_with_start_end.RData'</span>,<span class="at">verbose =</span> <span class="cn">TRUE</span>)</span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a>df <span class="ot">=</span> <span class="fu">merge</span>(mecp2.binding.meth.RNAseqKO_nuclear,ensenbl_location,</span>
<span id="cb43-4"><a href="#cb43-4" aria-hidden="true" tabindex="-1"></a>           <span class="at">by.x=</span><span class="st">'ENSEMBL'</span>,<span class="at">by.y=</span><span class="st">'ensembl_gene_id'</span>,<span class="at">all.x=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(ENSEMBL))</span>
<span id="cb43-5"><a href="#cb43-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-6"><a href="#cb43-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-7"><a href="#cb43-7" aria-hidden="true" tabindex="-1"></a>logFC_geneLength.nuclear.bound <span class="ot">=</span> df <span class="sc">%&gt;%</span> </span>
<span id="cb43-8"><a href="#cb43-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Binding <span class="sc">==</span> <span class="st">'Bound'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb43-9"><a href="#cb43-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(DEG<span class="sc">!=</span><span class="st">'Unchanged'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb43-10"><a href="#cb43-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Meth_CA<span class="sc">==</span><span class="st">'CA'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb43-11"><a href="#cb43-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">geneLength =</span> end_position<span class="sc">-</span>start_position) <span class="sc">%&gt;%</span> </span>
<span id="cb43-12"><a href="#cb43-12" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(logFC,geneLength) <span class="sc">%&gt;%</span> </span>
<span id="cb43-13"><a href="#cb43-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(geneLength)) <span class="sc">%&gt;%</span></span>
<span id="cb43-14"><a href="#cb43-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>() </span>
<span id="cb43-15"><a href="#cb43-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-16"><a href="#cb43-16" aria-hidden="true" tabindex="-1"></a>logFC_geneLength.nuclear.all <span class="ot">=</span> df <span class="sc">%&gt;%</span> </span>
<span id="cb43-17"><a href="#cb43-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(DEG<span class="sc">!=</span><span class="st">'Unchanged'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb43-18"><a href="#cb43-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Meth_CA<span class="sc">==</span><span class="st">'CA'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb43-19"><a href="#cb43-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">geneLength =</span> end_position<span class="sc">-</span>start_position) <span class="sc">%&gt;%</span> </span>
<span id="cb43-20"><a href="#cb43-20" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(logFC,geneLength) <span class="sc">%&gt;%</span> </span>
<span id="cb43-21"><a href="#cb43-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(geneLength)) <span class="sc">%&gt;%</span></span>
<span id="cb43-22"><a href="#cb43-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>() </span>
<span id="cb43-23"><a href="#cb43-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-24"><a href="#cb43-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-25"><a href="#cb43-25" aria-hidden="true" tabindex="-1"></a>df.bin <span class="ot">=</span><span class="fu">list</span>()</span>
<span id="cb43-26"><a href="#cb43-26" aria-hidden="true" tabindex="-1"></a>summarize_logFC_mCA <span class="ot">=</span> <span class="cf">function</span>(x,<span class="at">binSize =</span> <span class="dv">19</span>,<span class="at">stepSize =</span><span class="dv">4</span>){</span>
<span id="cb43-27"><a href="#cb43-27" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">seq</span>(<span class="at">from=</span><span class="dv">1</span>, <span class="at">to=</span><span class="fu">nrow</span>(x), <span class="at">by=</span>stepSize)){</span>
<span id="cb43-28"><a href="#cb43-28" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(i)</span>
<span id="cb43-29"><a href="#cb43-29" aria-hidden="true" tabindex="-1"></a>    start <span class="ot">=</span> i</span>
<span id="cb43-30"><a href="#cb43-30" aria-hidden="true" tabindex="-1"></a>    stop  <span class="ot">=</span> i<span class="sc">+</span>binSize</span>
<span id="cb43-31"><a href="#cb43-31" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(stop <span class="sc">&lt;=</span> <span class="fu">nrow</span>(x)){</span>
<span id="cb43-32"><a href="#cb43-32" aria-hidden="true" tabindex="-1"></a>      df <span class="ot">=</span> x[start<span class="sc">:</span>stop,]</span>
<span id="cb43-33"><a href="#cb43-33" aria-hidden="true" tabindex="-1"></a>      df.bin[[i]] <span class="ot">=</span> df  <span class="sc">%&gt;%</span> <span class="fu">summarize</span>(<span class="at">logFC.mean=</span><span class="fu">mean</span>(logFC),<span class="at">logFC.sd =</span> <span class="fu">sd</span>(logFC),<span class="at">CA_meth.mean=</span><span class="fu">mean</span>(geneLength))</span>
<span id="cb43-34"><a href="#cb43-34" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb43-35"><a href="#cb43-35" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>{</span>
<span id="cb43-36"><a href="#cb43-36" aria-hidden="true" tabindex="-1"></a>      <span class="fu">return</span>(df.bin)</span>
<span id="cb43-37"><a href="#cb43-37" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb43-38"><a href="#cb43-38" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb43-39"><a href="#cb43-39" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb43-40"><a href="#cb43-40" aria-hidden="true" tabindex="-1"></a>x <span class="ot">=</span> <span class="fu">summarize_logFC_mCA</span>(logFC_geneLength.nuclear.bound, <span class="at">binSize =</span> <span class="dv">39</span>,<span class="at">stepSize =</span><span class="dv">4</span>)</span>
<span id="cb43-41"><a href="#cb43-41" aria-hidden="true" tabindex="-1"></a>y <span class="ot">=</span> <span class="fu">summarize_logFC_mCA</span>(logFC_geneLength.nuclear.all, <span class="at">binSize =</span> <span class="dv">79</span>,<span class="at">stepSize =</span><span class="dv">4</span>)</span>
<span id="cb43-42"><a href="#cb43-42" aria-hidden="true" tabindex="-1"></a>MECP2_bound <span class="ot">=</span> <span class="fu">do.call</span>(rbind,x) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">MECP2 =</span><span class="st">'MECP2 Bound'</span>)</span>
<span id="cb43-43"><a href="#cb43-43" aria-hidden="true" tabindex="-1"></a>MECP2_all   <span class="ot">=</span> <span class="fu">do.call</span>(rbind,y) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">MECP2 =</span><span class="st">'All'</span>)</span>
<span id="cb43-44"><a href="#cb43-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-45"><a href="#cb43-45" aria-hidden="true" tabindex="-1"></a>plt <span class="ot">=</span> <span class="fu">list</span>()</span>
<span id="cb43-46"><a href="#cb43-46" aria-hidden="true" tabindex="-1"></a>plt.boxplot <span class="ot">=</span> <span class="fu">list</span>()</span>
<span id="cb43-47"><a href="#cb43-47" aria-hidden="true" tabindex="-1"></a>plt[[<span class="dv">1</span>]] <span class="ot">=</span> <span class="fu">rbind</span>(MECP2_bound,MECP2_all)  <span class="sc">%&gt;%</span> </span>
<span id="cb43-48"><a href="#cb43-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>CA_meth.mean,<span class="at">y=</span>logFC.mean,<span class="at">color=</span>MECP2))<span class="sc">+</span></span>
<span id="cb43-49"><a href="#cb43-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size=</span><span class="fl">0.5</span>)<span class="sc">+</span></span>
<span id="cb43-50"><a href="#cb43-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>()<span class="sc">+</span></span>
<span id="cb43-51"><a href="#cb43-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_ribbon</span>(<span class="fu">aes</span>(<span class="at">x=</span>CA_meth.mean,<span class="at">y=</span>logFC.mean,</span>
<span id="cb43-52"><a href="#cb43-52" aria-hidden="true" tabindex="-1"></a>                  <span class="at">ymin=</span>logFC.mean<span class="sc">-</span>logFC.sd,</span>
<span id="cb43-53"><a href="#cb43-53" aria-hidden="true" tabindex="-1"></a>                  <span class="at">ymax=</span>logFC.mean<span class="sc">+</span>logFC.sd,<span class="at">fill=</span>MECP2),</span>
<span id="cb43-54"><a href="#cb43-54" aria-hidden="true" tabindex="-1"></a>              <span class="at">alpha=</span><span class="fl">0.1</span>,<span class="at">color=</span><span class="st">'NA'</span>) <span class="sc">+</span></span>
<span id="cb43-55"><a href="#cb43-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">name =</span> <span class="st">"DEGs (MECP2 KO vs WT)"</span>,<span class="at">values =</span> <span class="fu">c</span>(<span class="st">'black'</span>,<span class="st">'red'</span>),<span class="at">aesthetics =</span> <span class="fu">c</span>(<span class="st">"colour"</span>, <span class="st">"fill"</span>))<span class="sc">+</span></span>
<span id="cb43-56"><a href="#cb43-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_log10</span>(<span class="at">breaks =</span> <span class="fu">trans_breaks</span>(<span class="st">"log10"</span>, <span class="cf">function</span>(x) <span class="dv">10</span><span class="sc">^</span>x),</span>
<span id="cb43-57"><a href="#cb43-57" aria-hidden="true" tabindex="-1"></a>                <span class="at">labels =</span> <span class="fu">trans_format</span>(<span class="st">"log10"</span>, <span class="fu">math_format</span>(<span class="dv">10</span><span class="sc">^</span>.x))) <span class="sc">+</span></span>
<span id="cb43-58"><a href="#cb43-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_cor</span>()<span class="sc">+</span></span>
<span id="cb43-59"><a href="#cb43-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_test</span>(<span class="dv">20</span>)<span class="sc">+</span></span>
<span id="cb43-60"><a href="#cb43-60" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">'Gene Length(bp)'</span>)<span class="sc">+</span></span>
<span id="cb43-61"><a href="#cb43-61" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">'Mean log2 Fold Change'</span>)<span class="sc">+</span></span>
<span id="cb43-62"><a href="#cb43-62" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">color =</span> <span class="fu">guide_legend</span>(<span class="at">override.aes =</span> <span class="fu">aes</span>(<span class="at">label =</span> <span class="st">""</span>)))</span>
<span id="cb43-63"><a href="#cb43-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-64"><a href="#cb43-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-65"><a href="#cb43-65" aria-hidden="true" tabindex="-1"></a>plt[[<span class="dv">1</span>]]</span>
<span id="cb43-66"><a href="#cb43-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-67"><a href="#cb43-67" aria-hidden="true" tabindex="-1"></a>logFC_geneLength_corr <span class="ot">=</span> <span class="cf">function</span>(df,plotType){</span>
<span id="cb43-68"><a href="#cb43-68" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">=</span> <span class="fu">merge</span>(df,ensenbl_location,<span class="at">by.x=</span><span class="st">'ENSEMBL'</span>,<span class="at">by.y=</span><span class="st">'ensembl_gene_id'</span>,<span class="at">all.x=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(ENSEMBL))</span>
<span id="cb43-69"><a href="#cb43-69" aria-hidden="true" tabindex="-1"></a>logFC_geneLength.nuclear.bound_with_count <span class="ot">=</span> df <span class="sc">%&gt;%</span> </span>
<span id="cb43-70"><a href="#cb43-70" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Binding <span class="sc">==</span> <span class="st">'Bound'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb43-71"><a href="#cb43-71" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(DEG<span class="sc">!=</span><span class="st">'Unchanged'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb43-72"><a href="#cb43-72" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Meth_CA<span class="sc">==</span><span class="st">'CA'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb43-73"><a href="#cb43-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">geneLength =</span> end_position<span class="sc">-</span>start_position) <span class="sc">%&gt;%</span> </span>
<span id="cb43-74"><a href="#cb43-74" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(logFC,geneLength) <span class="sc">%&gt;%</span> </span>
<span id="cb43-75"><a href="#cb43-75" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(logFC,geneLength) <span class="sc">%&gt;%</span> </span>
<span id="cb43-76"><a href="#cb43-76" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">Count=</span>dplyr<span class="sc">::</span><span class="fu">n</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb43-77"><a href="#cb43-77" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">MECP2 =</span> <span class="fu">case_when</span>(Count <span class="sc">&lt;</span><span class="dv">7</span><span class="sc">~</span><span class="st">'1-6 MBHs'</span>,</span>
<span id="cb43-78"><a href="#cb43-78" aria-hidden="true" tabindex="-1"></a>                           Count <span class="sc">&gt;=</span><span class="dv">7</span><span class="sc">~</span><span class="st">'&gt;=7 MBHs'</span>)) </span>
<span id="cb43-79"><a href="#cb43-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-80"><a href="#cb43-80" aria-hidden="true" tabindex="-1"></a>logFC_geneLength.nuclear.bound_with_count.list <span class="ot">=</span> <span class="fu">split</span>(logFC_geneLength.nuclear.bound_with_count, <span class="at">f =</span>logFC_geneLength.nuclear.bound_with_count<span class="sc">$</span>MECP2)</span>
<span id="cb43-81"><a href="#cb43-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-82"><a href="#cb43-82" aria-hidden="true" tabindex="-1"></a>logFC_geneLength.nuclear.bound_with_count.list <span class="ot">=</span> <span class="fu">lapply</span>(logFC_geneLength.nuclear.bound_with_count.list, <span class="cf">function</span>(x) x <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(geneLength)))</span>
<span id="cb43-83"><a href="#cb43-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-84"><a href="#cb43-84" aria-hidden="true" tabindex="-1"></a>tmp <span class="ot">=</span> <span class="fu">lapply</span>(logFC_geneLength.nuclear.bound_with_count.list, <span class="cf">function</span>(x) <span class="fu">summarize_logFC_mCA</span>(x <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>(),<span class="at">binSize =</span> <span class="dv">19</span>,<span class="at">stepSize =</span> <span class="dv">2</span>))</span>
<span id="cb43-85"><a href="#cb43-85" aria-hidden="true" tabindex="-1"></a>tmp1 <span class="ot">=</span> <span class="fu">lapply</span>(tmp, <span class="cf">function</span>(x) <span class="fu">do.call</span>(rbind,x))</span>
<span id="cb43-86"><a href="#cb43-86" aria-hidden="true" tabindex="-1"></a>tmp2 <span class="ot">=</span> plyr<span class="sc">::</span><span class="fu">ldply</span>(tmp1,data.frame)</span>
<span id="cb43-87"><a href="#cb43-87" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(tmp2)[<span class="dv">1</span>] <span class="ot">=</span> <span class="st">'MECP2'</span></span>
<span id="cb43-88"><a href="#cb43-88" aria-hidden="true" tabindex="-1"></a>logFC_mCA.nuclear.bound_with_count.avg <span class="ot">=</span> tmp2[,<span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">1</span>)]</span>
<span id="cb43-89"><a href="#cb43-89" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-90"><a href="#cb43-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-91"><a href="#cb43-91" aria-hidden="true" tabindex="-1"></a>plt <span class="ot">=</span> <span class="fu">rbind</span>(logFC_mCA.nuclear.bound_with_count.avg,MECP2_all)  <span class="sc">%&gt;%</span> </span>
<span id="cb43-92"><a href="#cb43-92" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(MECP <span class="sc">!=</span> <span class="st">'1-6 MBHs'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb43-93"><a href="#cb43-93" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>CA_meth.mean,<span class="at">y=</span>logFC.mean,<span class="at">color=</span>MECP2))<span class="sc">+</span></span>
<span id="cb43-94"><a href="#cb43-94" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size=</span><span class="fl">0.5</span>)<span class="sc">+</span></span>
<span id="cb43-95"><a href="#cb43-95" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>()<span class="sc">+</span></span>
<span id="cb43-96"><a href="#cb43-96" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_ribbon</span>(<span class="fu">aes</span>(<span class="at">x=</span>CA_meth.mean,<span class="at">y=</span>logFC.mean,</span>
<span id="cb43-97"><a href="#cb43-97" aria-hidden="true" tabindex="-1"></a>                  <span class="at">ymin=</span>logFC.mean<span class="sc">-</span>logFC.sd,</span>
<span id="cb43-98"><a href="#cb43-98" aria-hidden="true" tabindex="-1"></a>                  <span class="at">ymax=</span>logFC.mean<span class="sc">+</span>logFC.sd,<span class="at">fill=</span>MECP2),</span>
<span id="cb43-99"><a href="#cb43-99" aria-hidden="true" tabindex="-1"></a>              <span class="at">alpha=</span><span class="fl">0.1</span>,<span class="at">color=</span><span class="st">'NA'</span>) <span class="sc">+</span></span>
<span id="cb43-100"><a href="#cb43-100" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">name =</span> <span class="st">"DEGs (MECP2 KO vs WT)"</span>,<span class="at">values =</span> <span class="fu">c</span>(<span class="st">'red'</span>,<span class="st">'black'</span>),<span class="at">aesthetics =</span> <span class="fu">c</span>(<span class="st">"colour"</span>, <span class="st">"fill"</span>))<span class="sc">+</span></span>
<span id="cb43-101"><a href="#cb43-101" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_log10</span>(<span class="at">breaks =</span> <span class="fu">trans_breaks</span>(<span class="st">"log10"</span>, <span class="cf">function</span>(x) <span class="dv">10</span><span class="sc">^</span>x),</span>
<span id="cb43-102"><a href="#cb43-102" aria-hidden="true" tabindex="-1"></a>                <span class="at">labels =</span> <span class="fu">trans_format</span>(<span class="st">"log10"</span>, <span class="fu">math_format</span>(<span class="dv">10</span><span class="sc">^</span>.x))) <span class="sc">+</span></span>
<span id="cb43-103"><a href="#cb43-103" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_cor</span>()<span class="sc">+</span></span>
<span id="cb43-104"><a href="#cb43-104" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_test</span>(<span class="dv">20</span>)<span class="sc">+</span></span>
<span id="cb43-105"><a href="#cb43-105" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">'Gene Length(bp)'</span>)<span class="sc">+</span></span>
<span id="cb43-106"><a href="#cb43-106" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">'Mean log2 Fold Change'</span>)<span class="sc">+</span></span>
<span id="cb43-107"><a href="#cb43-107" aria-hidden="true" tabindex="-1"></a>  <span class="co">#ylim(-0.9,0.9)+</span></span>
<span id="cb43-108"><a href="#cb43-108" aria-hidden="true" tabindex="-1"></a>  <span class="co">#scale_y_continuous(limits = c(-0.6, 0.7), breaks = seq(-0.6, 0.7, by = 0.3))+</span></span>
<span id="cb43-109"><a href="#cb43-109" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">color =</span> <span class="fu">guide_legend</span>(<span class="at">override.aes =</span> <span class="fu">aes</span>(<span class="at">label =</span> <span class="st">""</span>)))</span>
<span id="cb43-110"><a href="#cb43-110" aria-hidden="true" tabindex="-1"></a>plt[[<span class="dv">3</span>]]</span>
<span id="cb43-111"><a href="#cb43-111" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-112"><a href="#cb43-112" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-113"><a href="#cb43-113" aria-hidden="true" tabindex="-1"></a>plt.boxplot <span class="ot">=</span> <span class="fu">rbind</span>(logFC_mCA.nuclear.bound_with_count.avg,MECP2_all)  <span class="sc">%&gt;%</span> </span>
<span id="cb43-114"><a href="#cb43-114" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">MECP2 =</span> <span class="fu">factor</span>(MECP2, <span class="at">levels=</span><span class="fu">c</span>(<span class="st">'All'</span>,<span class="st">'1-6 MBHs'</span>,<span class="st">'&gt;=7 MBHs'</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb43-115"><a href="#cb43-115" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>MECP2,<span class="at">y=</span>logFC.mean,<span class="at">color=</span>MECP2))<span class="sc">+</span></span>
<span id="cb43-116"><a href="#cb43-116" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>()<span class="sc">+</span><span class="fu">stat_compare_means</span>(<span class="at">comparisons =</span> <span class="fu">list</span>(<span class="fu">c</span>(<span class="st">'All'</span>,<span class="st">'&gt;=7 MBHs'</span>)))</span>
<span id="cb43-117"><a href="#cb43-117" aria-hidden="true" tabindex="-1"></a>                                                       <span class="co">#c('All','1-6 MBHs'),</span></span>
<span id="cb43-118"><a href="#cb43-118" aria-hidden="true" tabindex="-1"></a>                                                       <span class="co">#c('1-6 MBHs','&gt;=7 MBHs'))) + </span></span>
<span id="cb43-119"><a href="#cb43-119" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">'black'</span>,<span class="st">'red'</span>))<span class="sc">+</span></span>
<span id="cb43-120"><a href="#cb43-120" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_test</span>(<span class="dv">20</span>) <span class="sc">+</span> </span>
<span id="cb43-121"><a href="#cb43-121" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="dv">45</span>,<span class="at">hjust =</span> <span class="dv">1</span>,<span class="at">vjust =</span> <span class="dv">1</span>))<span class="sc">+</span></span>
<span id="cb43-122"><a href="#cb43-122" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">'DEGs (MECP2 KO vs WT)'</span>)</span>
<span id="cb43-123"><a href="#cb43-123" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-124"><a href="#cb43-124" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(plotType <span class="sc">==</span> <span class="st">'linePlot'</span>){</span>
<span id="cb43-125"><a href="#cb43-125" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(plt)</span>
<span id="cb43-126"><a href="#cb43-126" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb43-127"><a href="#cb43-127" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(plotType <span class="sc">==</span> <span class="st">'boxplot'</span>){</span>
<span id="cb43-128"><a href="#cb43-128" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(plt.boxplot)</span>
<span id="cb43-129"><a href="#cb43-129" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb43-130"><a href="#cb43-130" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb43-131"><a href="#cb43-131" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-132"><a href="#cb43-132" aria-hidden="true" tabindex="-1"></a><span class="co"># Nuclear</span></span>
<span id="cb43-133"><a href="#cb43-133" aria-hidden="true" tabindex="-1"></a>plt.linePlot <span class="ot">=</span> <span class="fu">logFC_geneLength_corr</span>(mecp2.binding.meth.RNAseqKO_nuclear,<span class="st">"linePlot"</span>)</span>
<span id="cb43-134"><a href="#cb43-134" aria-hidden="true" tabindex="-1"></a>plt.boxplot <span class="ot">=</span> <span class="fu">logFC_geneLength_corr</span>(mecp2.binding.meth.RNAseqKO_nuclear,<span class="st">"boxplot"</span>)</span>
<span id="cb43-135"><a href="#cb43-135" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-136"><a href="#cb43-136" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">'Plots/logFC_geneLength_correlation_Nuclear.pdf'</span>,<span class="at">height =</span> <span class="dv">5</span>,<span class="at">width =</span> <span class="dv">10</span>)</span>
<span id="cb43-137"><a href="#cb43-137" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(plt.linePlot)</span>
<span id="cb43-138"><a href="#cb43-138" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb43-139"><a href="#cb43-139" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-140"><a href="#cb43-140" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-141"><a href="#cb43-141" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">'Plots/logFC_geneLength_correlation_Nuclear_boxplot.pdf'</span>,<span class="at">height =</span> <span class="dv">8</span>,<span class="at">width =</span> <span class="dv">6</span>)</span>
<span id="cb43-142"><a href="#cb43-142" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(plt.boxplot)</span>
<span id="cb43-143"><a href="#cb43-143" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb43-144"><a href="#cb43-144" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-145"><a href="#cb43-145" aria-hidden="true" tabindex="-1"></a><span class="co"># WholeCell</span></span>
<span id="cb43-146"><a href="#cb43-146" aria-hidden="true" tabindex="-1"></a>plt.linePlot <span class="ot">=</span> <span class="fu">logFC_geneLength_corr</span>(mecp2.binding.meth.RNAseqKO_wholeCell,<span class="st">"linePlot"</span>)</span>
<span id="cb43-147"><a href="#cb43-147" aria-hidden="true" tabindex="-1"></a>plt.boxplot <span class="ot">=</span> <span class="fu">logFC_geneLength_corr</span>(mecp2.binding.meth.RNAseqKO_wholeCell,<span class="st">"boxplot"</span>)</span>
<span id="cb43-148"><a href="#cb43-148" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-149"><a href="#cb43-149" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">'Plots/logFC_geneLength_correlation_WholeCell.pdf'</span>,<span class="at">height =</span> <span class="dv">5</span>,<span class="at">width =</span> <span class="dv">10</span>)</span>
<span id="cb43-150"><a href="#cb43-150" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(plt.linePlot)</span>
<span id="cb43-151"><a href="#cb43-151" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb43-152"><a href="#cb43-152" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-153"><a href="#cb43-153" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-154"><a href="#cb43-154" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">'Plots/logFC_geneLength_correlation_WholeCell_boxplot.pdf'</span>,<span class="at">height =</span> <span class="dv">8</span>,<span class="at">width =</span> <span class="dv">6</span>)</span>
<span id="cb43-155"><a href="#cb43-155" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(plt.boxplot)</span>
<span id="cb43-156"><a href="#cb43-156" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb43-157"><a href="#cb43-157" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-158"><a href="#cb43-158" aria-hidden="true" tabindex="-1"></a><span class="co"># Chromatin</span></span>
<span id="cb43-159"><a href="#cb43-159" aria-hidden="true" tabindex="-1"></a>plt.linePlot <span class="ot">=</span> <span class="fu">logFC_geneLength_corr</span>(mecp2.binding.meth.RNAseqKO_chromatin,<span class="st">"linePlot"</span>)</span>
<span id="cb43-160"><a href="#cb43-160" aria-hidden="true" tabindex="-1"></a>plt.boxplot <span class="ot">=</span> <span class="fu">logFC_geneLength_corr</span>(mecp2.binding.meth.RNAseqKO_chromatin,<span class="st">"boxplot"</span>)</span>
<span id="cb43-161"><a href="#cb43-161" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-162"><a href="#cb43-162" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">'Plots/logFC_geneLength_correlation_Chromatin.pdf'</span>,<span class="at">height =</span> <span class="dv">5</span>,<span class="at">width =</span> <span class="dv">10</span>)</span>
<span id="cb43-163"><a href="#cb43-163" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(plt.linePlot)</span>
<span id="cb43-164"><a href="#cb43-164" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb43-165"><a href="#cb43-165" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-166"><a href="#cb43-166" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-167"><a href="#cb43-167" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">'Plots/logFC_geneLength_correlation_Chromatin_boxplot.pdf'</span>,<span class="at">height =</span> <span class="dv">8</span>,<span class="at">width =</span> <span class="dv">6</span>)</span>
<span id="cb43-168"><a href="#cb43-168" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(plt.boxplot)</span>
<span id="cb43-169"><a href="#cb43-169" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></li>
<li><p>Figure 5D and Supplementary Figure 7</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"results/RData/MECP2_regulatedGenes.RData"</span>,<span class="at">verbose =</span><span class="cn">TRUE</span>)</span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a><span class="co"># All repressed genes (MECP2 KO vs WT)</span></span>
<span id="cb44-4"><a href="#cb44-4" aria-hidden="true" tabindex="-1"></a>MECP2_repressed_gene.wholeCell <span class="ot">=</span> mecp2.binding.meth.RNAseqKO_wholeCell <span class="sc">%&gt;%</span> </span>
<span id="cb44-5"><a href="#cb44-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(FDR <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb44-6"><a href="#cb44-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Meth_CA <span class="sc">==</span> <span class="st">'CA'</span> <span class="sc">&amp;</span> Meth_CG <span class="sc">==</span><span class="st">'CG'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb44-7"><a href="#cb44-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(DEG<span class="sc">==</span><span class="st">'Repressed'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb44-8"><a href="#cb44-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(ENSEMBL) <span class="sc">%&gt;%</span> </span>
<span id="cb44-9"><a href="#cb44-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unique</span>()</span>
<span id="cb44-10"><a href="#cb44-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-11"><a href="#cb44-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert Gene Symbol to ENSEMBL id </span></span>
<span id="cb44-12"><a href="#cb44-12" aria-hidden="true" tabindex="-1"></a>MECP2_repressed_gene.wholeCell.geneId <span class="ot">=</span> <span class="fu">mapIds</span>(org.Mm.eg.db,</span>
<span id="cb44-13"><a href="#cb44-13" aria-hidden="true" tabindex="-1"></a>                                               <span class="at">keys=</span>MECP2_activated_gene.wholeCell,</span>
<span id="cb44-14"><a href="#cb44-14" aria-hidden="true" tabindex="-1"></a>                                               <span class="at">column=</span><span class="st">"ENTREZID"</span>,</span>
<span id="cb44-15"><a href="#cb44-15" aria-hidden="true" tabindex="-1"></a>                                               <span class="at">keytype=</span><span class="st">"ENSEMBL"</span>,<span class="at">multiVals=</span><span class="st">"first"</span>) </span>
<span id="cb44-16"><a href="#cb44-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-17"><a href="#cb44-17" aria-hidden="true" tabindex="-1"></a>formula_res <span class="ot">=</span> <span class="fu">enrichGO</span>(MECP2_repressed_gene.wholeCell.geneId,</span>
<span id="cb44-18"><a href="#cb44-18" aria-hidden="true" tabindex="-1"></a>                        <span class="at">OrgDb=</span><span class="st">'org.Mm.eg.db'</span>,</span>
<span id="cb44-19"><a href="#cb44-19" aria-hidden="true" tabindex="-1"></a>                        <span class="at">ont=</span><span class="st">"BP"</span>,</span>
<span id="cb44-20"><a href="#cb44-20" aria-hidden="true" tabindex="-1"></a>                        <span class="co">#universe =unique(mecp2.binding.meth.RNAseqKO_wholeCell$geneId),</span></span>
<span id="cb44-21"><a href="#cb44-21" aria-hidden="true" tabindex="-1"></a>                        <span class="at">readable=</span><span class="cn">TRUE</span> )</span>
<span id="cb44-22"><a href="#cb44-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-23"><a href="#cb44-23" aria-hidden="true" tabindex="-1"></a>formula_res <span class="sc">%&gt;%</span> </span>
<span id="cb44-24"><a href="#cb44-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write.table</span>(<span class="st">"results/MECP2_RNAseq/All_MECP2_repressed_genes_GOBP_terms.tsv"</span>,</span>
<span id="cb44-25"><a href="#cb44-25" aria-hidden="true" tabindex="-1"></a>              <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>,<span class="at">quote =</span> T)</span>
<span id="cb44-26"><a href="#cb44-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-27"><a href="#cb44-27" aria-hidden="true" tabindex="-1"></a>formula_res<span class="sc">@</span>result <span class="sc">%&gt;%</span> <span class="fu">filter</span>(p.adjust <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> <span class="co">#View() </span></span>
<span id="cb44-28"><a href="#cb44-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(p.adjust) <span class="sc">%&gt;%</span> </span>
<span id="cb44-29"><a href="#cb44-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">18</span>) <span class="sc">%&gt;%</span></span>
<span id="cb44-30"><a href="#cb44-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span><span class="sc">-</span><span class="fu">log10</span>(pvalue),<span class="at">y=</span><span class="fu">reorder</span>(Description,<span class="sc">-</span>pvalue)))<span class="sc">+</span></span>
<span id="cb44-31"><a href="#cb44-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">'identity'</span>)<span class="sc">+</span><span class="fu">theme_test</span>(<span class="dv">20</span>)<span class="sc">+</span></span>
<span id="cb44-32"><a href="#cb44-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">'Enriched GO terms for All repressed genes'</span>)</span>
<span id="cb44-33"><a href="#cb44-33" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">'Plots/All_MECP2_repressed_genes_GO_terms.pdf'</span>,<span class="at">height =</span> <span class="dv">6</span>,<span class="at">width =</span> <span class="fl">12.2</span>)</span>
<span id="cb44-34"><a href="#cb44-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-35"><a href="#cb44-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-36"><a href="#cb44-36" aria-hidden="true" tabindex="-1"></a><span class="co"># MBHs repressed genes</span></span>
<span id="cb44-37"><a href="#cb44-37" aria-hidden="true" tabindex="-1"></a>MECP2_regulated_gene.wholeCell <span class="ot">=</span> Mecp2.Bound.meth.expressed.wholeCell <span class="sc">%&gt;%</span> </span>
<span id="cb44-38"><a href="#cb44-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(FDR <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb44-39"><a href="#cb44-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(DEG<span class="sc">==</span><span class="st">'Repressed'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb44-40"><a href="#cb44-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(geneId) <span class="sc">%&gt;%</span> </span>
<span id="cb44-41"><a href="#cb44-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unique</span>()</span>
<span id="cb44-42"><a href="#cb44-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-43"><a href="#cb44-43" aria-hidden="true" tabindex="-1"></a>formula_res <span class="ot">=</span> <span class="fu">enrichGO</span>(MECP2_regulated_gene.wholeCell,</span>
<span id="cb44-44"><a href="#cb44-44" aria-hidden="true" tabindex="-1"></a>                        <span class="at">OrgDb=</span><span class="st">'org.Mm.eg.db'</span>,</span>
<span id="cb44-45"><a href="#cb44-45" aria-hidden="true" tabindex="-1"></a>                        <span class="at">ont=</span><span class="st">"BP"</span>,</span>
<span id="cb44-46"><a href="#cb44-46" aria-hidden="true" tabindex="-1"></a>                        <span class="co">#universe =unique(mecp2.binding.meth.RNAseqKO_wholeCell$geneId),</span></span>
<span id="cb44-47"><a href="#cb44-47" aria-hidden="true" tabindex="-1"></a>                        <span class="at">readable=</span><span class="cn">TRUE</span> )</span>
<span id="cb44-48"><a href="#cb44-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-49"><a href="#cb44-49" aria-hidden="true" tabindex="-1"></a>formula_res <span class="sc">%&gt;%</span> </span>
<span id="cb44-50"><a href="#cb44-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write.table</span>(<span class="st">"results/MECP2_RNAseq/MBHs_repressed_genes_GOBP_terms.tsv"</span>,</span>
<span id="cb44-51"><a href="#cb44-51" aria-hidden="true" tabindex="-1"></a>              <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>,<span class="at">quote =</span> T)</span>
<span id="cb44-52"><a href="#cb44-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-53"><a href="#cb44-53" aria-hidden="true" tabindex="-1"></a>formula_res<span class="sc">@</span>result <span class="sc">%&gt;%</span> <span class="fu">filter</span>(p.adjust <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> <span class="co">#View() </span></span>
<span id="cb44-54"><a href="#cb44-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(p.adjust) <span class="sc">%&gt;%</span>  <span class="fu">View</span>()</span>
<span id="cb44-55"><a href="#cb44-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">18</span>) <span class="sc">%&gt;%</span></span>
<span id="cb44-56"><a href="#cb44-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span><span class="sc">-</span><span class="fu">log10</span>(pvalue),<span class="at">y=</span><span class="fu">reorder</span>(Description,<span class="sc">-</span>pvalue)))<span class="sc">+</span></span>
<span id="cb44-57"><a href="#cb44-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">'identity'</span>)<span class="sc">+</span><span class="fu">theme_test</span>(<span class="dv">20</span>)<span class="sc">+</span></span>
<span id="cb44-58"><a href="#cb44-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">'Enriched GO terms for &gt;=7 MBHs repressed genes'</span>)</span>
<span id="cb44-59"><a href="#cb44-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-60"><a href="#cb44-60" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">'Plots/MBHs_repressed_genes_GO_terms.pdf'</span>,<span class="at">height =</span> <span class="dv">6</span>,<span class="at">width =</span> <span class="dv">12</span>)</span>
<span id="cb44-61"><a href="#cb44-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-62"><a href="#cb44-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-63"><a href="#cb44-63" aria-hidden="true" tabindex="-1"></a><span class="co"># All activated genes (MECP2 KO vs WT)</span></span>
<span id="cb44-64"><a href="#cb44-64" aria-hidden="true" tabindex="-1"></a>MECP2_activated_gene.wholeCell <span class="ot">=</span> mecp2.binding.meth.RNAseqKO_wholeCell <span class="sc">%&gt;%</span> </span>
<span id="cb44-65"><a href="#cb44-65" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(FDR <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb44-66"><a href="#cb44-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Meth_CA <span class="sc">==</span> <span class="st">'CA'</span> <span class="sc">&amp;</span> Meth_CG <span class="sc">==</span><span class="st">'CG'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb44-67"><a href="#cb44-67" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(DEG<span class="sc">==</span><span class="st">'Activated'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb44-68"><a href="#cb44-68" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(ENSEMBL) <span class="sc">%&gt;%</span> </span>
<span id="cb44-69"><a href="#cb44-69" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unique</span>()</span>
<span id="cb44-70"><a href="#cb44-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-71"><a href="#cb44-71" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert Gene Symbol to ENSEMBL id </span></span>
<span id="cb44-72"><a href="#cb44-72" aria-hidden="true" tabindex="-1"></a>MECP2_activated_gene.wholeCell.geneId <span class="ot">=</span> <span class="fu">mapIds</span>(org.Mm.eg.db,</span>
<span id="cb44-73"><a href="#cb44-73" aria-hidden="true" tabindex="-1"></a>                                               <span class="at">keys=</span>MECP2_activated_gene.wholeCell,</span>
<span id="cb44-74"><a href="#cb44-74" aria-hidden="true" tabindex="-1"></a>                                               <span class="at">column=</span><span class="st">"ENTREZID"</span>,</span>
<span id="cb44-75"><a href="#cb44-75" aria-hidden="true" tabindex="-1"></a>                                               <span class="at">keytype=</span><span class="st">"ENSEMBL"</span>,<span class="at">multiVals=</span><span class="st">"first"</span>) </span>
<span id="cb44-76"><a href="#cb44-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-77"><a href="#cb44-77" aria-hidden="true" tabindex="-1"></a>formula_res <span class="ot">=</span> <span class="fu">enrichGO</span>(MECP2_activated_gene.wholeCell.geneId,</span>
<span id="cb44-78"><a href="#cb44-78" aria-hidden="true" tabindex="-1"></a>                        <span class="at">OrgDb=</span><span class="st">'org.Mm.eg.db'</span>,</span>
<span id="cb44-79"><a href="#cb44-79" aria-hidden="true" tabindex="-1"></a>                        <span class="at">ont=</span><span class="st">"BP"</span>,</span>
<span id="cb44-80"><a href="#cb44-80" aria-hidden="true" tabindex="-1"></a>                        <span class="co">#universe =unique(mecp2.binding.meth.RNAseqKO_wholeCell$geneId),</span></span>
<span id="cb44-81"><a href="#cb44-81" aria-hidden="true" tabindex="-1"></a>                        <span class="at">readable=</span><span class="cn">TRUE</span> )</span>
<span id="cb44-82"><a href="#cb44-82" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-83"><a href="#cb44-83" aria-hidden="true" tabindex="-1"></a>formula_res <span class="sc">%&gt;%</span> </span>
<span id="cb44-84"><a href="#cb44-84" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write.table</span>(<span class="st">"results/MECP2_RNAseq/All_MECP2_activated_genes_GOBP_terms.tsv"</span>,</span>
<span id="cb44-85"><a href="#cb44-85" aria-hidden="true" tabindex="-1"></a>              <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>,<span class="at">quote =</span> T)</span>
<span id="cb44-86"><a href="#cb44-86" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-87"><a href="#cb44-87" aria-hidden="true" tabindex="-1"></a>formula_res<span class="sc">@</span>result <span class="sc">%&gt;%</span> <span class="fu">filter</span>(p.adjust <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> <span class="co">#View() </span></span>
<span id="cb44-88"><a href="#cb44-88" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(p.adjust) <span class="sc">%&gt;%</span> </span>
<span id="cb44-89"><a href="#cb44-89" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">18</span>) <span class="sc">%&gt;%</span></span>
<span id="cb44-90"><a href="#cb44-90" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span><span class="sc">-</span><span class="fu">log10</span>(pvalue),<span class="at">y=</span><span class="fu">reorder</span>(Description,<span class="sc">-</span>pvalue)))<span class="sc">+</span></span>
<span id="cb44-91"><a href="#cb44-91" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">'identity'</span>)<span class="sc">+</span><span class="fu">theme_test</span>(<span class="dv">20</span>)<span class="sc">+</span></span>
<span id="cb44-92"><a href="#cb44-92" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">'Enriched GO terms for All activated genes'</span>)</span>
<span id="cb44-93"><a href="#cb44-93" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">'Plots/All_MECP2_activated_genes_GO_terms.pdf'</span>,<span class="at">height =</span> <span class="dv">6</span>,<span class="at">width =</span> <span class="fl">12.2</span>)</span>
<span id="cb44-94"><a href="#cb44-94" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-95"><a href="#cb44-95" aria-hidden="true" tabindex="-1"></a><span class="co"># MBHS activated genes</span></span>
<span id="cb44-96"><a href="#cb44-96" aria-hidden="true" tabindex="-1"></a>MECP2_regulated_gene.wholeCell <span class="ot">=</span> Mecp2.Bound.meth.expressed.wholeCell <span class="sc">%&gt;%</span> </span>
<span id="cb44-97"><a href="#cb44-97" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(FDR <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb44-98"><a href="#cb44-98" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(DEG<span class="sc">==</span><span class="st">'Activated'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb44-99"><a href="#cb44-99" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(geneId) <span class="sc">%&gt;%</span> </span>
<span id="cb44-100"><a href="#cb44-100" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unique</span>()</span>
<span id="cb44-101"><a href="#cb44-101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-102"><a href="#cb44-102" aria-hidden="true" tabindex="-1"></a>formula_res <span class="ot">=</span> <span class="fu">enrichGO</span>(MECP2_regulated_gene.wholeCell,</span>
<span id="cb44-103"><a href="#cb44-103" aria-hidden="true" tabindex="-1"></a>                        <span class="at">OrgDb=</span><span class="st">'org.Mm.eg.db'</span>,</span>
<span id="cb44-104"><a href="#cb44-104" aria-hidden="true" tabindex="-1"></a>                        <span class="at">ont=</span><span class="st">"BP"</span>,</span>
<span id="cb44-105"><a href="#cb44-105" aria-hidden="true" tabindex="-1"></a>                        <span class="at">universe =</span><span class="fu">unique</span>(mecp2.binding.meth.RNAseqKO_wholeCell<span class="sc">$</span>geneId),</span>
<span id="cb44-106"><a href="#cb44-106" aria-hidden="true" tabindex="-1"></a>                        <span class="at">readable=</span><span class="cn">TRUE</span> )</span>
<span id="cb44-107"><a href="#cb44-107" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-108"><a href="#cb44-108" aria-hidden="true" tabindex="-1"></a>formula_res <span class="sc">%&gt;%</span> </span>
<span id="cb44-109"><a href="#cb44-109" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write.table</span>(<span class="st">"results/MECP2_RNAseq/MBHs_activated_genes_GOBP_terms.tsv"</span>,<span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>,<span class="at">quote =</span> T)</span>
<span id="cb44-110"><a href="#cb44-110" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-111"><a href="#cb44-111" aria-hidden="true" tabindex="-1"></a>formula_res<span class="sc">@</span>result <span class="sc">%&gt;%</span> <span class="fu">filter</span>(p.adjust <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> <span class="co">#View() </span></span>
<span id="cb44-112"><a href="#cb44-112" aria-hidden="true" tabindex="-1"></a><span class="fu">arrange</span>(p.adjust) <span class="sc">%&gt;%</span> </span>
<span id="cb44-113"><a href="#cb44-113" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">18</span>) <span class="sc">%&gt;%</span></span>
<span id="cb44-114"><a href="#cb44-114" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span><span class="sc">-</span><span class="fu">log10</span>(pvalue),<span class="at">y=</span><span class="fu">reorder</span>(Description,<span class="sc">-</span>pvalue)))<span class="sc">+</span></span>
<span id="cb44-115"><a href="#cb44-115" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">'identity'</span>)<span class="sc">+</span><span class="fu">theme_test</span>(<span class="dv">20</span>)<span class="sc">+</span></span>
<span id="cb44-116"><a href="#cb44-116" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">'Enriched GO terms for MBHs activated genes'</span>)</span>
<span id="cb44-117"><a href="#cb44-117" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">'Plots/MBHs_activated_genes_GO_terms.pdf'</span>,<span class="at">height =</span> <span class="dv">6</span>,<span class="at">width =</span> <span class="fl">12.2</span>)</span>
<span id="cb44-118"><a href="#cb44-118" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-119"><a href="#cb44-119" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-120"><a href="#cb44-120" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-121"><a href="#cb44-121" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-122"><a href="#cb44-122" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-123"><a href="#cb44-123" aria-hidden="true" tabindex="-1"></a><span class="co"># 7 or more MBHs regulated genes</span></span>
<span id="cb44-124"><a href="#cb44-124" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(clusterProfiler)</span>
<span id="cb44-125"><a href="#cb44-125" aria-hidden="true" tabindex="-1"></a>MECP2_regulated_gene.wholeCell <span class="ot">=</span> Mecp2.Bound.meth.expressed.wholeCell <span class="sc">%&gt;%</span> </span>
<span id="cb44-126"><a href="#cb44-126" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(FDR <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb44-127"><a href="#cb44-127" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(DEG<span class="sc">==</span><span class="st">'Repressed'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb44-128"><a href="#cb44-128" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(geneId,logFC,FDR) <span class="sc">%&gt;%</span> </span>
<span id="cb44-129"><a href="#cb44-129" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(geneId,logFC,FDR) <span class="sc">%&gt;%</span> </span>
<span id="cb44-130"><a href="#cb44-130" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">Count=</span>dplyr<span class="sc">::</span><span class="fu">n</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb44-131"><a href="#cb44-131" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">MECP2 =</span> <span class="fu">case_when</span>(Count <span class="sc">&lt;</span><span class="dv">7</span><span class="sc">~</span><span class="st">'&lt;7 MECP2 peaks'</span>,</span>
<span id="cb44-132"><a href="#cb44-132" aria-hidden="true" tabindex="-1"></a>                           Count <span class="sc">&gt;=</span><span class="dv">7</span><span class="sc">~</span><span class="st">'&gt;=1 MECP2 peaks'</span>)) </span>
<span id="cb44-133"><a href="#cb44-133" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-134"><a href="#cb44-134" aria-hidden="true" tabindex="-1"></a>Mecp2.Bound.meth.expressed.wholeCell <span class="sc">%&gt;%</span> </span>
<span id="cb44-135"><a href="#cb44-135" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(FDR <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb44-136"><a href="#cb44-136" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(DEG<span class="sc">==</span><span class="st">'Repressed'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb44-137"><a href="#cb44-137" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(SYMBOL,logFC,FDR) <span class="sc">%&gt;%</span> </span>
<span id="cb44-138"><a href="#cb44-138" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(SYMBOL,logFC,FDR) <span class="sc">%&gt;%</span> </span>
<span id="cb44-139"><a href="#cb44-139" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">Count=</span>dplyr<span class="sc">::</span><span class="fu">n</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb44-140"><a href="#cb44-140" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">MECP2 =</span> <span class="fu">case_when</span>(Count <span class="sc">&lt;</span><span class="dv">7</span><span class="sc">~</span><span class="st">'&lt;7 MECP2 peaks'</span>,</span>
<span id="cb44-141"><a href="#cb44-141" aria-hidden="true" tabindex="-1"></a>                           Count <span class="sc">&gt;=</span><span class="dv">7</span><span class="sc">~</span><span class="st">'&gt;=7 MECP2 peaks'</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb44-142"><a href="#cb44-142" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write.table</span>(<span class="st">"results/MECP2_RNAseq/7orMore_MBHs_repressed_genes.tsv"</span>,<span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>,<span class="at">quote =</span> <span class="cn">FALSE</span>)</span>
<span id="cb44-143"><a href="#cb44-143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-144"><a href="#cb44-144" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-145"><a href="#cb44-145" aria-hidden="true" tabindex="-1"></a><span class="fu">rbind</span>(Mecp2.Bound.meth.expressed.nuclear,Mecp2.UnBound.meth.expressed.nuclear) <span class="sc">%&gt;%</span> </span>
<span id="cb44-146"><a href="#cb44-146" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(FDR <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb44-147"><a href="#cb44-147" aria-hidden="true" tabindex="-1"></a>    <span class="co">#filter(!is.na(SYMBOL)) %&gt;% </span></span>
<span id="cb44-148"><a href="#cb44-148" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(Gene,logFC, FDR,Binding) <span class="sc">%&gt;%</span></span>
<span id="cb44-149"><a href="#cb44-149" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(Gene,logFC, FDR,Binding) <span class="sc">%&gt;%</span> </span>
<span id="cb44-150"><a href="#cb44-150" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">MECP2_peaks=</span>dplyr<span class="sc">::</span><span class="fu">n</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb44-151"><a href="#cb44-151" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">MECP2_peaks =</span> <span class="fu">ifelse</span>(Binding <span class="sc">==</span> <span class="st">'Unbound'</span>,<span class="dv">0</span>,MECP2_peaks)) <span class="sc">%&gt;%</span> </span>
<span id="cb44-152"><a href="#cb44-152" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">5</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb44-153"><a href="#cb44-153" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(<span class="st">'results/MECP2/MECP2_repressed_genes_with_numberof_MECP2_peaks.tsv'</span>,<span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>,</span>
<span id="cb44-154"><a href="#cb44-154" aria-hidden="true" tabindex="-1"></a>                <span class="at">row.names =</span> <span class="cn">FALSE</span>,<span class="at">quote =</span> <span class="cn">FALSE</span>)</span>
<span id="cb44-155"><a href="#cb44-155" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-156"><a href="#cb44-156" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-157"><a href="#cb44-157" aria-hidden="true" tabindex="-1"></a>MECP2_regulated_gene.nuclear <span class="sc">%&gt;%</span> </span>
<span id="cb44-158"><a href="#cb44-158" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write.table</span>(<span class="st">'results/MECP2/MECP2_repressed_genes_having_5_or_morePeaks.tsv'</span>,<span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>,</span>
<span id="cb44-159"><a href="#cb44-159" aria-hidden="true" tabindex="-1"></a>              <span class="at">row.names =</span> <span class="cn">FALSE</span>,<span class="at">quote =</span> <span class="cn">FALSE</span>)</span>
<span id="cb44-160"><a href="#cb44-160" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-161"><a href="#cb44-161" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-162"><a href="#cb44-162" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-163"><a href="#cb44-163" aria-hidden="true" tabindex="-1"></a>MECP2_regulated_gene.wholeCell.list <span class="ot">=</span> <span class="fu">split</span>(MECP2_regulated_gene.wholeCell<span class="sc">$</span>geneId,</span>
<span id="cb44-164"><a href="#cb44-164" aria-hidden="true" tabindex="-1"></a>                                            MECP2_regulated_gene.wholeCell<span class="sc">$</span>MECP2)</span>
<span id="cb44-165"><a href="#cb44-165" aria-hidden="true" tabindex="-1"></a>MECP2_regulated_gene.wholeCell.GO  <span class="ot">=</span> <span class="fu">compareCluster</span>(<span class="at">geneClusters =</span> MECP2_regulated_gene.wholeCell.list,</span>
<span id="cb44-166"><a href="#cb44-166" aria-hidden="true" tabindex="-1"></a>                                          <span class="at">fun=</span><span class="st">"enrichGO"</span>,<span class="at">OrgDb=</span><span class="st">'org.Mm.eg.db'</span>,<span class="at">ont=</span><span class="st">"BP"</span>,<span class="at">readable=</span><span class="cn">TRUE</span> )</span>
<span id="cb44-167"><a href="#cb44-167" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-168"><a href="#cb44-168" aria-hidden="true" tabindex="-1"></a>MECP2_regulated_gene.wholeCell.GO <span class="sc">%&gt;%</span> </span>
<span id="cb44-169"><a href="#cb44-169" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write.table</span>(<span class="st">"results/MECP2_RNAseq/7orMore_MBHs_repressed_genes_GOBP_terms.tsv"</span>,</span>
<span id="cb44-170"><a href="#cb44-170" aria-hidden="true" tabindex="-1"></a>              <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>,<span class="at">quote =</span> T)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></li>
<li><p>Supplementary Figure 5E-F.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load MECP2 bound and unbound target genes with expression change of </span></span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a><span class="co"># MECP2 KOvsWT and CA and CG methylation</span></span>
<span id="cb45-3"><a href="#cb45-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-4"><a href="#cb45-4" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">'results/RData/MECP2_regulatedGenes.RData'</span>,<span class="at">verbose =</span> <span class="cn">TRUE</span>)</span>
<span id="cb45-5"><a href="#cb45-5" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">'results/RData/MECP2_peakAnnotation.RData'</span>,<span class="at">verbose =</span> <span class="cn">TRUE</span>)</span>
<span id="cb45-6"><a href="#cb45-6" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">'results/BSMap/8wk_CTX_Stroud2017_mm10_BSmap_CA_within_mm10_ensembl_genes.First3kb_excluded.RData'</span>,<span class="at">verbose =</span> T)</span>
<span id="cb45-7"><a href="#cb45-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-8"><a href="#cb45-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Read RNA-seq data to extract DNMT3A KOvsWT analysis</span></span>
<span id="cb45-9"><a href="#cb45-9" aria-hidden="true" tabindex="-1"></a>path <span class="ot">=</span> <span class="st">"data/Boxer_RNAseq_data.xlsx"</span></span>
<span id="cb45-10"><a href="#cb45-10" aria-hidden="true" tabindex="-1"></a>sheets <span class="ot">=</span> openxlsx<span class="sc">::</span><span class="fu">getSheetNames</span>(path)</span>
<span id="cb45-11"><a href="#cb45-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-12"><a href="#cb45-12" aria-hidden="true" tabindex="-1"></a><span class="co"># print name of all the sheet</span></span>
<span id="cb45-13"><a href="#cb45-13" aria-hidden="true" tabindex="-1"></a>sheets</span>
<span id="cb45-14"><a href="#cb45-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-15"><a href="#cb45-15" aria-hidden="true" tabindex="-1"></a><span class="co"># open each sheet by name ans make a list </span></span>
<span id="cb45-16"><a href="#cb45-16" aria-hidden="true" tabindex="-1"></a>RNAseq.list <span class="ot">=</span> <span class="fu">lapply</span>(sheets, openxlsx<span class="sc">::</span>read.xlsx, <span class="at">xlsxFile=</span>path,<span class="at">startRow=</span><span class="dv">2</span>)</span>
<span id="cb45-17"><a href="#cb45-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-18"><a href="#cb45-18" aria-hidden="true" tabindex="-1"></a><span class="co"># name each element of the list</span></span>
<span id="cb45-19"><a href="#cb45-19" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(RNAseq.list) <span class="ot">=</span> sheets</span>
<span id="cb45-20"><a href="#cb45-20" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(RNAseq.list)</span>
<span id="cb45-21"><a href="#cb45-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-22"><a href="#cb45-22" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert Gene Symbol to ENSEMBL id </span></span>
<span id="cb45-23"><a href="#cb45-23" aria-hidden="true" tabindex="-1"></a>RNAseq.list <span class="ot">=</span> <span class="fu">lapply</span>(RNAseq.list, <span class="cf">function</span>(x) { <span class="fu">mutate</span>(x,<span class="at">ENSEMBL =</span> <span class="fu">mapIds</span>(org.Mm.eg.db,</span>
<span id="cb45-24"><a href="#cb45-24" aria-hidden="true" tabindex="-1"></a>                                                                          <span class="at">keys=</span>x<span class="sc">$</span>Gene, </span>
<span id="cb45-25"><a href="#cb45-25" aria-hidden="true" tabindex="-1"></a>                                                                          <span class="at">column=</span><span class="st">"ENSEMBL"</span>,</span>
<span id="cb45-26"><a href="#cb45-26" aria-hidden="true" tabindex="-1"></a>                                                                          <span class="at">keytype=</span><span class="st">"SYMBOL"</span>,</span>
<span id="cb45-27"><a href="#cb45-27" aria-hidden="true" tabindex="-1"></a>                                                                          <span class="at">multiVals=</span><span class="st">"first"</span> )) })</span>
<span id="cb45-28"><a href="#cb45-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-29"><a href="#cb45-29" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter only significantly changing genes</span></span>
<span id="cb45-30"><a href="#cb45-30" aria-hidden="true" tabindex="-1"></a>mecp2.binding.meth.RNAseqKO_wholeCell.sig <span class="ot">=</span> mecp2.binding.meth.RNAseqKO_wholeCell <span class="sc">%&gt;%</span> </span>
<span id="cb45-31"><a href="#cb45-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(DEG <span class="sc">!=</span> <span class="st">'Unchanged'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-32"><a href="#cb45-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Meth_CA <span class="sc">==</span> <span class="st">'CA'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb45-33"><a href="#cb45-33" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">1</span>,<span class="dv">20</span>,<span class="dv">22</span>,<span class="dv">26</span>,<span class="dv">30</span>)</span>
<span id="cb45-34"><a href="#cb45-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-35"><a href="#cb45-35" aria-hidden="true" tabindex="-1"></a><span class="co"># Merge MECP2 KOvsWT data with DNMT3A KOvsWT </span></span>
<span id="cb45-36"><a href="#cb45-36" aria-hidden="true" tabindex="-1"></a>MECP2_KOvsWT_DNMT3a_KOvsWT <span class="ot">=</span>  <span class="fu">merge</span>(mecp2.binding.meth.RNAseqKO_wholeCell.sig, </span>
<span id="cb45-37"><a href="#cb45-37" aria-hidden="true" tabindex="-1"></a>                                    RNAseq.list[[<span class="dv">7</span>]] <span class="sc">%&gt;%</span> <span class="fu">filter</span>(FDR <span class="sc">&lt;</span> <span class="fl">0.05</span>),<span class="at">by =</span> <span class="st">'ENSEMBL'</span>,<span class="at">all.x=</span><span class="cn">TRUE</span>)</span>
<span id="cb45-38"><a href="#cb45-38" aria-hidden="true" tabindex="-1"></a>MECP2_KOvsWT_DNMT3a_KOvsWT <span class="ot">=</span>  MECP2_KOvsWT_DNMT3a_KOvsWT <span class="sc">%&gt;%</span> </span>
<span id="cb45-39"><a href="#cb45-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Binding =</span> <span class="fu">ifelse</span>(ENSEMBL <span class="sc">%in%</span> mecp2.binding<span class="sc">$</span>ENSEMBL, <span class="st">'Bound'</span> , <span class="st">'Unbound'</span> ))</span>
<span id="cb45-40"><a href="#cb45-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-41"><a href="#cb45-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-42"><a href="#cb45-42" aria-hidden="true" tabindex="-1"></a>MECP2_KOvsWT_DNMT3a_KOvsWT.df <span class="ot">=</span> MECP2_KOvsWT_DNMT3a_KOvsWT <span class="sc">%&gt;%</span> </span>
<span id="cb45-43"><a href="#cb45-43" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">1</span>,<span class="dv">3</span>,<span class="dv">7</span>,<span class="dv">2</span>,<span class="dv">4</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb45-44"><a href="#cb45-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(ENSEMBL,logFC.x,logFC.y,CA_meth,Binding) <span class="sc">%&gt;%</span> </span>
<span id="cb45-45"><a href="#cb45-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">Count =</span>dplyr<span class="sc">::</span><span class="fu">n</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb45-46"><a href="#cb45-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Count=</span> <span class="fu">ifelse</span>(Count<span class="sc">==</span><span class="dv">1</span> <span class="sc">&amp;</span> Binding <span class="sc">==</span><span class="st">'Unbound'</span>,<span class="dv">0</span>, Count)) <span class="sc">%&gt;%</span> </span>
<span id="cb45-47"><a href="#cb45-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(ENSEMBL)) <span class="sc">%&gt;%</span> </span>
<span id="cb45-48"><a href="#cb45-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">duplicated</span>(ENSEMBL)) <span class="sc">%&gt;%</span> </span>
<span id="cb45-49"><a href="#cb45-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="st">'ENSEMBL'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb45-50"><a href="#cb45-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">logFC.y =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(logFC.y),<span class="dv">0</span>,logFC.y),</span>
<span id="cb45-51"><a href="#cb45-51" aria-hidden="true" tabindex="-1"></a>         <span class="at">logFC.x =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(logFC.x),<span class="dv">0</span>,logFC.x)) <span class="sc">%&gt;%</span></span>
<span id="cb45-52"><a href="#cb45-52" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Count =</span> <span class="fu">ifelse</span>(Count <span class="sc">&gt;=</span><span class="dv">10</span>,<span class="dv">10</span>,Count))</span>
<span id="cb45-53"><a href="#cb45-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-54"><a href="#cb45-54" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(ensenbl_location.df_CA_meth)</span>
<span id="cb45-55"><a href="#cb45-55" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(ensenbl_location.df_CA_meth)[<span class="dv">9</span>] <span class="ot">=</span><span class="st">"CA_meth"</span></span>
<span id="cb45-56"><a href="#cb45-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-57"><a href="#cb45-57" aria-hidden="true" tabindex="-1"></a>MECP2_KOvsWT_DNMT3a_KOvsWT.df<span class="ot">=</span><span class="fu">merge</span>(MECP2_KOvsWT_DNMT3a_KOvsWT.df,</span>
<span id="cb45-58"><a href="#cb45-58" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">y =</span> ensenbl_location.df_CA_meth,</span>
<span id="cb45-59"><a href="#cb45-59" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">by.x=</span><span class="dv">0</span>,<span class="at">by.y=</span><span class="st">'ENSEMBL'</span>,<span class="at">all.x =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb45-60"><a href="#cb45-60" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">14</span>,<span class="dv">5</span>,<span class="dv">6</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb45-61"><a href="#cb45-61" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="st">'Row.names'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb45-62"><a href="#cb45-62" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Count =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(Count),<span class="dv">0</span>,Count),</span>
<span id="cb45-63"><a href="#cb45-63" aria-hidden="true" tabindex="-1"></a>         <span class="at">CA_meth.y =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(CA_meth.y),<span class="dv">0</span>,CA_meth.y)) <span class="sc">%&gt;%</span> </span>
<span id="cb45-64"><a href="#cb45-64" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mCA =</span> <span class="fu">ifelse</span>(CA_meth.y <span class="sc">&gt;=</span> <span class="fu">median</span>(CA_meth.y),<span class="st">'high mCA'</span> ,<span class="st">'low mCA'</span>)) </span>
<span id="cb45-65"><a href="#cb45-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-66"><a href="#cb45-66" aria-hidden="true" tabindex="-1"></a>geneClass.list <span class="ot">=</span> <span class="fu">list</span>()</span>
<span id="cb45-67"><a href="#cb45-67" aria-hidden="true" tabindex="-1"></a>geneClass.list[[<span class="st">'MECP2_KO (High_mCA)'</span>]] <span class="ot">=</span> MECP2_KOvsWT_DNMT3a_KOvsWT.df <span class="sc">%&gt;%</span> </span>
<span id="cb45-68"><a href="#cb45-68" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(mCA <span class="sc">==</span> <span class="st">'high mCA'</span>) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(logFC.x <span class="sc">&gt;</span><span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb45-69"><a href="#cb45-69" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>() <span class="sc">%&gt;%</span> <span class="fu">pull</span>(rowname)</span>
<span id="cb45-70"><a href="#cb45-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-71"><a href="#cb45-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-72"><a href="#cb45-72" aria-hidden="true" tabindex="-1"></a>geneClass.list[[<span class="st">'MBHs'</span>]] <span class="ot">=</span> MECP2_KOvsWT_DNMT3a_KOvsWT.df <span class="sc">%&gt;%</span> </span>
<span id="cb45-73"><a href="#cb45-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(logFC.x <span class="sc">&gt;</span><span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb45-74"><a href="#cb45-74" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Binding <span class="sc">==</span> <span class="st">'Bound'</span> <span class="sc">&amp;</span> Count <span class="sc">&gt;=</span><span class="dv">7</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb45-75"><a href="#cb45-75" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>() <span class="sc">%&gt;%</span> <span class="fu">pull</span>(rowname)</span>
<span id="cb45-76"><a href="#cb45-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-77"><a href="#cb45-77" aria-hidden="true" tabindex="-1"></a>geneClass.list[[<span class="st">'DNMT3AKO (High mCA)'</span>]] <span class="ot">=</span> MECP2_KOvsWT_DNMT3a_KOvsWT.df <span class="sc">%&gt;%</span> </span>
<span id="cb45-78"><a href="#cb45-78" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(mCA <span class="sc">==</span> <span class="st">'high mCA'</span>) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(logFC.y <span class="sc">&gt;</span><span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb45-79"><a href="#cb45-79" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>() <span class="sc">%&gt;%</span> <span class="fu">pull</span>(rowname)</span>
<span id="cb45-80"><a href="#cb45-80" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-81"><a href="#cb45-81" aria-hidden="true" tabindex="-1"></a>geneClass.list[[<span class="st">'only MBHs'</span>]] <span class="ot">=</span> <span class="fu">setdiff</span>(geneClass.list[[<span class="dv">2</span>]],</span>
<span id="cb45-82"><a href="#cb45-82" aria-hidden="true" tabindex="-1"></a>        <span class="fu">intersect</span>(geneClass.list[[<span class="dv">1</span>]],geneClass.list[[<span class="dv">2</span>]]))</span>
<span id="cb45-83"><a href="#cb45-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-84"><a href="#cb45-84" aria-hidden="true" tabindex="-1"></a>geneClass.list[[<span class="st">'only MECP2KO'</span>]] <span class="ot">=</span> <span class="fu">setdiff</span>(<span class="fu">setdiff</span>(geneClass.list[[<span class="dv">1</span>]],</span>
<span id="cb45-85"><a href="#cb45-85" aria-hidden="true" tabindex="-1"></a>                                        <span class="fu">intersect</span>(geneClass.list[[<span class="dv">1</span>]],geneClass.list[[<span class="dv">2</span>]])),</span>
<span id="cb45-86"><a href="#cb45-86" aria-hidden="true" tabindex="-1"></a>                                                  geneClass.list[[<span class="dv">3</span>]])</span>
<span id="cb45-87"><a href="#cb45-87" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-88"><a href="#cb45-88" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(geneClass.list)</span>
<span id="cb45-89"><a href="#cb45-89" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-90"><a href="#cb45-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-91"><a href="#cb45-91" aria-hidden="true" tabindex="-1"></a>DEGs_venn <span class="ot">=</span> <span class="fu">ggvenn</span>(</span>
<span id="cb45-92"><a href="#cb45-92" aria-hidden="true" tabindex="-1"></a>  geneClass.list[<span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>)], <span class="at">show_percentage =</span> <span class="cn">FALSE</span>,</span>
<span id="cb45-93"><a href="#cb45-93" aria-hidden="true" tabindex="-1"></a>  <span class="at">fill_color =</span> <span class="fu">c</span>(<span class="st">"#0073C2FF"</span>, <span class="st">"#868686FF"</span>),</span>
<span id="cb45-94"><a href="#cb45-94" aria-hidden="true" tabindex="-1"></a>  <span class="at">stroke_size =</span> <span class="fl">0.5</span>, <span class="at">set_name_size =</span> <span class="dv">5</span>,<span class="at">text_size =</span> <span class="dv">6</span></span>
<span id="cb45-95"><a href="#cb45-95" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb45-96"><a href="#cb45-96" aria-hidden="true" tabindex="-1"></a>DEGs_venn</span>
<span id="cb45-97"><a href="#cb45-97" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">'Plots/MECP2_high_mCA_and_MBHs_gene_overlap.pdf'</span>,<span class="at">height =</span> <span class="dv">5</span>,<span class="at">width =</span> <span class="dv">5</span>)</span>
<span id="cb45-98"><a href="#cb45-98" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-99"><a href="#cb45-99" aria-hidden="true" tabindex="-1"></a>geneClass.list.geneID <span class="ot">=</span> <span class="fu">lapply</span>(geneClass.list, <span class="cf">function</span>(x) {<span class="fu">mapIds</span>(org.Mm.eg.db,<span class="at">keys=</span>x, </span>
<span id="cb45-100"><a href="#cb45-100" aria-hidden="true" tabindex="-1"></a>                                                               <span class="at">column=</span><span class="st">"ENTREZID"</span>,</span>
<span id="cb45-101"><a href="#cb45-101" aria-hidden="true" tabindex="-1"></a>                                                               <span class="at">keytype=</span><span class="st">"ENSEMBL"</span>,</span>
<span id="cb45-102"><a href="#cb45-102" aria-hidden="true" tabindex="-1"></a>                                                               <span class="at">multiVals=</span><span class="st">"first"</span> ) })</span>
<span id="cb45-103"><a href="#cb45-103" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(geneClass.list.geneID) <span class="ot">=</span> <span class="fu">paste0</span>(<span class="fu">names</span>(geneClass.list.geneID),<span class="st">' (n='</span>,</span>
<span id="cb45-104"><a href="#cb45-104" aria-hidden="true" tabindex="-1"></a>                                      <span class="fu">lengths</span>(geneClass.list.geneID),<span class="st">')'</span>)</span>
<span id="cb45-105"><a href="#cb45-105" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(clusterProfiler)</span>
<span id="cb45-106"><a href="#cb45-106" aria-hidden="true" tabindex="-1"></a>formula_res <span class="ot">=</span> <span class="fu">compareCluster</span>( geneClass.list.geneID[<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">4</span>)], </span>
<span id="cb45-107"><a href="#cb45-107" aria-hidden="true" tabindex="-1"></a>                              <span class="at">fun=</span><span class="st">"enrichGO"</span>,</span>
<span id="cb45-108"><a href="#cb45-108" aria-hidden="true" tabindex="-1"></a>                              <span class="at">OrgDb=</span><span class="st">'org.Mm.eg.db'</span>,</span>
<span id="cb45-109"><a href="#cb45-109" aria-hidden="true" tabindex="-1"></a>                              <span class="at">ont=</span><span class="st">"BP"</span>,</span>
<span id="cb45-110"><a href="#cb45-110" aria-hidden="true" tabindex="-1"></a>                              <span class="co">#universe =unique(mecp2.binding.meth.RNAseqKO_wholeCell$geneId),</span></span>
<span id="cb45-111"><a href="#cb45-111" aria-hidden="true" tabindex="-1"></a>                              <span class="at">readable=</span><span class="cn">TRUE</span> )</span>
<span id="cb45-112"><a href="#cb45-112" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-113"><a href="#cb45-113" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidytext)</span>
<span id="cb45-114"><a href="#cb45-114" aria-hidden="true" tabindex="-1"></a>formula_res<span class="sc">@</span>compareClusterResult <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">group_by</span>(Cluster) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Cluster <span class="sc">!=</span> <span class="st">'MBHs (n=232)'</span>) <span class="sc">%&gt;%</span>  <span class="co">#filter(p.adjust &lt; 0.05) %&gt;%</span></span>
<span id="cb45-115"><a href="#cb45-115" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p.value=</span> <span class="sc">-</span><span class="fu">log10</span>(pvalue)) <span class="sc">%&gt;%</span> </span>
<span id="cb45-116"><a href="#cb45-116" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb45-117"><a href="#cb45-117" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>()<span class="sc">+</span></span>
<span id="cb45-118"><a href="#cb45-118" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x=</span>p.value,<span class="at">y=</span><span class="fu">reorder_within</span>(Description,p.value,Cluster)), </span>
<span id="cb45-119"><a href="#cb45-119" aria-hidden="true" tabindex="-1"></a>           <span class="at">stat =</span> <span class="st">"identity"</span>, </span>
<span id="cb45-120"><a href="#cb45-120" aria-hidden="true" tabindex="-1"></a>           <span class="at">show.legend =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb45-121"><a href="#cb45-121" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>Cluster,<span class="at">scales =</span> <span class="st">'free_y'</span>)<span class="sc">+</span></span>
<span id="cb45-122"><a href="#cb45-122" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_reordered</span>()<span class="sc">+</span></span>
<span id="cb45-123"><a href="#cb45-123" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">'-log10(pvalue)'</span>)</span>
<span id="cb45-124"><a href="#cb45-124" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"Plots/GO_terms_MBHs_regulated_genes_high_low_mCA.pdf"</span>,<span class="at">height =</span> <span class="dv">4</span>,<span class="at">width =</span> <span class="dv">12</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></li>
<li><p>Supplementary Figure 8.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dir.create</span>(<span class="fu">file.path</span>(<span class="fu">paste0</span>(path,<span class="st">"/data/Allen_brain_10X"</span>)), <span class="at">showWarnings =</span> <span class="cn">FALSE</span>)</span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"results/RData/MECP2_regulatedGenes.RData"</span>)</span>
<span id="cb46-4"><a href="#cb46-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-5"><a href="#cb46-5" aria-hidden="true" tabindex="-1"></a>MECP2_regulated_gene.wholeCell <span class="ot">=</span> Mecp2.Bound.meth.expressed.wholeCell <span class="sc">%&gt;%</span> </span>
<span id="cb46-6"><a href="#cb46-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(FDR <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb46-7"><a href="#cb46-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(DEG<span class="sc">==</span><span class="st">'Repressed'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb46-8"><a href="#cb46-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(SYMBOL,logFC,FDR) <span class="sc">%&gt;%</span> </span>
<span id="cb46-9"><a href="#cb46-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(SYMBOL,logFC,FDR) <span class="sc">%&gt;%</span> </span>
<span id="cb46-10"><a href="#cb46-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">Count=</span>dplyr<span class="sc">::</span><span class="fu">n</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb46-11"><a href="#cb46-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Count <span class="sc">&gt;=</span> <span class="dv">7</span>)</span>
<span id="cb46-12"><a href="#cb46-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-13"><a href="#cb46-13" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(<span class="st">"https://idk-etl-prod-download-bucket.s3.amazonaws.com/aibs_mouse_ctx-hpf_10x/trimmed_means.csv"</span>, </span>
<span id="cb46-14"><a href="#cb46-14" aria-hidden="true" tabindex="-1"></a>              <span class="fu">paste</span>(<span class="st">"data/Allen_brain_10X/"</span>, <span class="st">"aibs_mouse_ctx-hpf_10x_trimmed_means.csv"</span>, <span class="at">sep =</span> <span class="st">""</span>))</span>
<span id="cb46-15"><a href="#cb46-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-16"><a href="#cb46-16" aria-hidden="true" tabindex="-1"></a>expr <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">"data/Allen_brain_10X/aibs_mouse_ctx-hpf_10x_trimmed_means.csv"</span>,</span>
<span id="cb46-17"><a href="#cb46-17" aria-hidden="true" tabindex="-1"></a>                <span class="at">check.names =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb46-18"><a href="#cb46-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(feature <span class="sc">%in%</span> MECP2_regulated_gene.wholeCell<span class="sc">$</span>SYMBOL) <span class="sc">%&gt;%</span> </span>
<span id="cb46-19"><a href="#cb46-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="st">'feature'</span>)</span>
<span id="cb46-20"><a href="#cb46-20" aria-hidden="true" tabindex="-1"></a>expr <span class="ot">=</span> <span class="fu">log2</span>(expr<span class="sc">+</span><span class="dv">1</span>)</span>
<span id="cb46-21"><a href="#cb46-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-22"><a href="#cb46-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-23"><a href="#cb46-23" aria-hidden="true" tabindex="-1"></a><span class="co">#dend_label = unlist(partition_leaves(updated_dendrogram),use.names = FALSE) </span></span>
<span id="cb46-24"><a href="#cb46-24" aria-hidden="true" tabindex="-1"></a><span class="co">#dend_label[dend_label %in% colnames(expr) ] %&gt;% length()</span></span>
<span id="cb46-25"><a href="#cb46-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-26"><a href="#cb46-26" aria-hidden="true" tabindex="-1"></a>expr_colname <span class="ot">=</span> <span class="fu">colnames</span>(expr) <span class="sc">%&gt;%</span>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb46-27"><a href="#cb46-27" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"group"</span> <span class="ot">=</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb46-28"><a href="#cb46-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">CellType =</span> <span class="fu">gsub</span>(<span class="st">'.*_'</span>,<span class="st">''</span>,group)) </span>
<span id="cb46-29"><a href="#cb46-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-30"><a href="#cb46-30" aria-hidden="true" tabindex="-1"></a>expr_colname <span class="sc">%&gt;%</span> </span>
<span id="cb46-31"><a href="#cb46-31" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">2</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb46-32"><a href="#cb46-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb46-33"><a href="#cb46-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write.table</span>(<span class="st">"data/Allen_brain_10X/Allen_10x_cellType.tsv"</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb46-34"><a href="#cb46-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-35"><a href="#cb46-35" aria-hidden="true" tabindex="-1"></a>celltype_anno_1 <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">"data/Allen_brain_10X/CellType_Class.tsv"</span>,<span class="at">header =</span> <span class="cn">TRUE</span>,<span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>) <span class="co">#%&gt;% </span></span>
<span id="cb46-36"><a href="#cb46-36" aria-hidden="true" tabindex="-1"></a>celltype_anno_1<span class="sc">$</span>anno <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">s+[^ ]+$"</span>, <span class="st">""</span>,celltype_anno_1<span class="sc">$</span>Class.marker)</span>
<span id="cb46-37"><a href="#cb46-37" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(celltype_anno_1)</span>
<span id="cb46-38"><a href="#cb46-38" aria-hidden="true" tabindex="-1"></a>celltype_anno_1<span class="sc">$</span>anno <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">" "</span>, <span class="st">"_"</span>,celltype_anno_1<span class="sc">$</span>anno)</span>
<span id="cb46-39"><a href="#cb46-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-40"><a href="#cb46-40" aria-hidden="true" tabindex="-1"></a>celltype_anno_2 <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">"data/Allen_brain_10X/CellType_Class_metadata.tsv"</span>,<span class="at">header =</span> <span class="cn">TRUE</span>,<span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb46-41"><a href="#cb46-41" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(celltype_anno_2)</span>
<span id="cb46-42"><a href="#cb46-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-43"><a href="#cb46-43" aria-hidden="true" tabindex="-1"></a>metaData1 <span class="ot">=</span> <span class="fu">merge</span>(expr_colname,celltype_anno_2, <span class="at">by.x =</span> <span class="st">"CellType"</span>, <span class="at">by.y=</span><span class="st">"column_in_exp_file"</span>)</span>
<span id="cb46-44"><a href="#cb46-44" aria-hidden="true" tabindex="-1"></a>metaData2 <span class="ot">=</span> <span class="fu">merge</span>(metaData1,celltype_anno_1,<span class="at">by.x=</span><span class="st">"column_in_metadata"</span>,<span class="at">by.y=</span><span class="st">"anno"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb46-45"><a href="#cb46-45" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">5</span>) <span class="sc">%&gt;%</span> <span class="fu">distinct</span>() <span class="sc">%&gt;%</span></span>
<span id="cb46-46"><a href="#cb46-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(Class)</span>
<span id="cb46-47"><a href="#cb46-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-48"><a href="#cb46-48" aria-hidden="true" tabindex="-1"></a>expr <span class="ot">=</span> expr <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(metaData2<span class="sc">$</span>group)</span>
<span id="cb46-49"><a href="#cb46-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-50"><a href="#cb46-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-51"><a href="#cb46-51" aria-hidden="true" tabindex="-1"></a>ha <span class="ot">=</span> <span class="fu">HeatmapAnnotation</span>(<span class="at">Cell_type=</span>metaData2[,<span class="dv">4</span>],</span>
<span id="cb46-52"><a href="#cb46-52" aria-hidden="true" tabindex="-1"></a>                       <span class="co">#col = list(NEURONAL.NON.NEURONAL =c("GABAergic Neuron" = "red", "Glutamatergic Neuron"="blue","Neuronal"="cyan",</span></span>
<span id="cb46-53"><a href="#cb46-53" aria-hidden="true" tabindex="-1"></a>                        <span class="co">#                                   "Non-Neuronal"="green")),</span></span>
<span id="cb46-54"><a href="#cb46-54" aria-hidden="true" tabindex="-1"></a>                       <span class="at">annotation_height =</span> <span class="fu">unit</span>(<span class="fu">c</span>(<span class="fl">0.5</span>, <span class="fl">0.5</span>), <span class="st">"cm"</span>),</span>
<span id="cb46-55"><a href="#cb46-55" aria-hidden="true" tabindex="-1"></a>                       <span class="at">show_annotation_name =</span> <span class="cn">TRUE</span>,</span>
<span id="cb46-56"><a href="#cb46-56" aria-hidden="true" tabindex="-1"></a>                       <span class="at">annotation_name_offset =</span> <span class="fu">unit</span>(<span class="dv">2</span>, <span class="st">"mm"</span>),</span>
<span id="cb46-57"><a href="#cb46-57" aria-hidden="true" tabindex="-1"></a>                       <span class="at">annotation_name_rot =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>),</span>
<span id="cb46-58"><a href="#cb46-58" aria-hidden="true" tabindex="-1"></a>                       <span class="at">annotation_name_align =</span> <span class="cn">TRUE</span>)</span>
<span id="cb46-59"><a href="#cb46-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-60"><a href="#cb46-60" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">"Plots/Allen_10x_mouse_MBH_repressed_genes.pdf"</span>,<span class="at">height =</span> <span class="dv">7</span>,<span class="at">width =</span> <span class="dv">10</span>)</span>
<span id="cb46-61"><a href="#cb46-61" aria-hidden="true" tabindex="-1"></a><span class="fu">Heatmap</span>(expr, <span class="at">name =</span> <span class="st">"log2(CPM +1)"</span>,</span>
<span id="cb46-62"><a href="#cb46-62" aria-hidden="true" tabindex="-1"></a>        <span class="at">cluster_rows =</span> <span class="cn">TRUE</span>, <span class="at">cluster_columns =</span> <span class="cn">FALSE</span>,<span class="at">top_annotation =</span> ha,</span>
<span id="cb46-63"><a href="#cb46-63" aria-hidden="true" tabindex="-1"></a>        <span class="at">show_column_names =</span> <span class="cn">FALSE</span>,<span class="at">show_row_names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb46-64"><a href="#cb46-64" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></li>
<li><p>Supplementary Figure 9A.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a>hic_mecp2_overlap <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">"results/HiC/Jiang_2017.Neuron.mm10.peakachu-merged.loops_closest_MECP2.bed"</span>,<span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>,<span class="at">header =</span> F) <span class="sc">%&gt;%</span> </span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">id =</span> <span class="fu">paste0</span>(V5,<span class="st">'_'</span>,V6,<span class="st">'_'</span>,V7)) <span class="sc">%&gt;%</span></span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(id) <span class="sc">%&gt;%</span> </span>
<span id="cb47-4"><a href="#cb47-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">count=</span> dplyr<span class="sc">::</span><span class="fu">n</span>()) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(count)) <span class="sc">%&gt;%</span> </span>
<span id="cb47-5"><a href="#cb47-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(count) <span class="sc">%&gt;%</span> </span>
<span id="cb47-6"><a href="#cb47-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">n =</span> dplyr<span class="sc">::</span><span class="fu">n</span>()) </span>
<span id="cb47-7"><a href="#cb47-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-8"><a href="#cb47-8" aria-hidden="true" tabindex="-1"></a>hic_mecp2_overlap[<span class="dv">12</span>,<span class="dv">2</span>] <span class="ot">=</span> hic_mecp2_overlap[<span class="dv">12</span>,<span class="dv">1</span>]</span>
<span id="cb47-9"><a href="#cb47-9" aria-hidden="true" tabindex="-1"></a>hic_mecp2_overlap[<span class="dv">12</span>,<span class="dv">1</span>] <span class="ot">=</span> <span class="dv">0</span></span>
<span id="cb47-10"><a href="#cb47-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-11"><a href="#cb47-11" aria-hidden="true" tabindex="-1"></a>hic_mecp2_overlap.pie <span class="ot">=</span> hic_mecp2_overlap <span class="sc">%&gt;%</span> </span>
<span id="cb47-12"><a href="#cb47-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">count_1 =</span> <span class="fu">ifelse</span>(count <span class="sc">&gt;=</span><span class="dv">5</span>, <span class="st">"&gt;=5"</span>,count)) <span class="sc">%&gt;%</span> </span>
<span id="cb47-13"><a href="#cb47-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(count_1) <span class="sc">%&gt;%</span> <span class="fu">summarise</span>(<span class="at">Total =</span> <span class="fu">sum</span>(n)) <span class="sc">%&gt;%</span>   </span>
<span id="cb47-14"><a href="#cb47-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Percent =</span> <span class="fu">round</span>(Total<span class="sc">/</span><span class="fu">sum</span>(Total)<span class="sc">*</span><span class="dv">100</span>,<span class="dv">3</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb47-15"><a href="#cb47-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">count_1 =</span> <span class="fu">factor</span>(count_1,<span class="at">levels=</span> <span class="fu">as.character</span>(<span class="fu">c</span>(<span class="dv">0</span><span class="sc">:</span><span class="dv">4</span>,<span class="st">"&gt;=5"</span>))))</span>
<span id="cb47-16"><a href="#cb47-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-17"><a href="#cb47-17" aria-hidden="true" tabindex="-1"></a>nb.cols <span class="ot">&lt;-</span> <span class="dv">12</span></span>
<span id="cb47-18"><a href="#cb47-18" aria-hidden="true" tabindex="-1"></a>mycolors <span class="ot">&lt;-</span> <span class="fu">colorRampPalette</span>(<span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">"Set2"</span>))(nb.cols)</span>
<span id="cb47-19"><a href="#cb47-19" aria-hidden="true" tabindex="-1"></a>mycolors <span class="ot">=</span> <span class="fu">c</span>(<span class="st">'#71CAF7'</span>,mycolors)</span>
<span id="cb47-20"><a href="#cb47-20" aria-hidden="true" tabindex="-1"></a>threshold <span class="ot">=</span> <span class="fl">0.05</span></span>
<span id="cb47-21"><a href="#cb47-21" aria-hidden="true" tabindex="-1"></a>hic_mecp2_overlap.pie <span class="sc">%&gt;%</span> </span>
<span id="cb47-22"><a href="#cb47-22" aria-hidden="true" tabindex="-1"></a>  <span class="co">#mutate(Peaks =as.character(Peaks)) %&gt;% </span></span>
<span id="cb47-23"><a href="#cb47-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="st">""</span>, <span class="at">y =</span> Percent, <span class="at">fill =</span> count_1)) <span class="sc">+</span></span>
<span id="cb47-24"><a href="#cb47-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">width =</span> <span class="dv">1</span>, <span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">color =</span> <span class="st">"white"</span>) <span class="sc">+</span></span>
<span id="cb47-25"><a href="#cb47-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_polar</span>(<span class="st">"y"</span>, <span class="at">start =</span> <span class="dv">0</span>)<span class="sc">+</span></span>
<span id="cb47-26"><a href="#cb47-26" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_text(aes(y = lab.ypos, label = Percent), color = "white")+</span></span>
<span id="cb47-27"><a href="#cb47-27" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_text(aes(x = 1.6, label = scales::percent(Percent, accuracy = 0.1)), position = position_stack(vjust = .5)) +</span></span>
<span id="cb47-28"><a href="#cb47-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fl">1.55</span>, <span class="at">label =</span> scales<span class="sc">::</span><span class="fu">percent</span>(Percent, <span class="at">accuracy =</span> <span class="fl">0.1</span>,<span class="at">scale =</span> <span class="dv">1</span>)), <span class="at">position =</span> <span class="fu">position_stack</span>(<span class="at">vjust =</span> .<span class="dv">5</span>),<span class="at">size=</span><span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb47-29"><a href="#cb47-29" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_segment(aes(x = ifelse(Percent&lt;threshold,1, xpos), xend = xpos, y = ypos, yend = ypos)) + </span></span>
<span id="cb47-30"><a href="#cb47-30" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_text(aes(x = xpos, y = ypos, label = ifelse(Percent&gt;threshold,percent(Percent, accuracy = 0.1),"")), hjust = "outward", nudge_x  = 0.2, size = 4) + </span></span>
<span id="cb47-31"><a href="#cb47-31" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_text_repel(aes(x = xpos, y = ypos, label = ifelse(Percent&lt;threshold, percent(Percent, accuracy = 0.1), "")), nudge_x  = 0.2, size = 4)+ </span></span>
<span id="cb47-32"><a href="#cb47-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_polar</span>(<span class="st">"y"</span>) <span class="sc">+</span></span>
<span id="cb47-33"><a href="#cb47-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_void</span>()<span class="sc">+</span></span>
<span id="cb47-34"><a href="#cb47-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">10</span>),</span>
<span id="cb47-35"><a href="#cb47-35" aria-hidden="true" tabindex="-1"></a>        <span class="co">#legend.key.size = unit(2, 'cm'),</span></span>
<span id="cb47-36"><a href="#cb47-36" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">10</span>),</span>
<span id="cb47-37"><a href="#cb47-37" aria-hidden="true" tabindex="-1"></a>        <span class="at">strip.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">"#FFF5DE"</span>,  <span class="at">color =</span> <span class="st">"transparent"</span>))<span class="sc">+</span></span>
<span id="cb47-38"><a href="#cb47-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">name =</span> <span class="st">'Number of MBHs</span><span class="sc">\n</span><span class="st"> within chromatin loops'</span>,<span class="at">values =</span> mycolors)</span>
<span id="cb47-39"><a href="#cb47-39" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"Plots/Number_of_MBHs_within_chromatinLoops.pdf"</span>,<span class="at">height =</span> <span class="dv">15</span>, <span class="at">width =</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></li>
<li><p>Supplementary Figure 9B.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a>hic <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"chromatin_loops"</span> <span class="ot">=</span> <span class="st">"data/mm10_hic_northwerstern_YueLab/Jiang_2017.Neuron.mm10.peakachu-merged.loops_pm10kb.sort.bed"</span>,</span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a>    <span class="st">"H3K27ac"</span> <span class="ot">=</span> <span class="st">"results/H3K27ac/H3K27ac_MECP2WT_filtered_peaks.bed"</span>)</span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a>hic.GR <span class="ot">=</span> <span class="fu">GRangesList</span>(<span class="fu">lapply</span>(hic, readPeakFile))</span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-6"><a href="#cb48-6" aria-hidden="true" tabindex="-1"></a>pn.area <span class="ot">=</span> <span class="dv">100</span></span>
<span id="cb48-7"><a href="#cb48-7" aria-hidden="true" tabindex="-1"></a>pn.dist <span class="ot">=</span> <span class="dv">100</span></span>
<span id="cb48-8"><a href="#cb48-8" aria-hidden="true" tabindex="-1"></a>pn.jacc <span class="ot">=</span> <span class="dv">100</span></span>
<span id="cb48-9"><a href="#cb48-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-10"><a href="#cb48-10" aria-hidden="true" tabindex="-1"></a>hic.GR.test <span class="ot">=</span> <span class="fu">mclapply</span>(hic.GR,<span class="at">mc.cores =</span> <span class="dv">45</span>,<span class="cf">function</span>(x){</span>
<span id="cb48-11"><a href="#cb48-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">GenometriCorrelation</span>(x, MECP2_peaks.GR, <span class="at">chromosomes.length =</span> chrLen.vec,</span>
<span id="cb48-12"><a href="#cb48-12" aria-hidden="true" tabindex="-1"></a>                       <span class="at">permut.number =</span> <span class="dv">0</span>, <span class="at">keep.distributions =</span> <span class="cn">TRUE</span>, </span>
<span id="cb48-13"><a href="#cb48-13" aria-hidden="true" tabindex="-1"></a>                       <span class="at">showProgressBar =</span> <span class="cn">TRUE</span>,<span class="at">awhole.only=</span><span class="cn">TRUE</span>,</span>
<span id="cb48-14"><a href="#cb48-14" aria-hidden="true" tabindex="-1"></a>                       <span class="at">ecdf.area.permut.number =</span> pn.area,</span>
<span id="cb48-15"><a href="#cb48-15" aria-hidden="true" tabindex="-1"></a>                       <span class="at">mean.distance.permut.number =</span> pn.dist,</span>
<span id="cb48-16"><a href="#cb48-16" aria-hidden="true" tabindex="-1"></a>                       <span class="at">jaccard.measure.permut.number =</span> pn.jacc)</span>
<span id="cb48-17"><a href="#cb48-17" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb48-18"><a href="#cb48-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-19"><a href="#cb48-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-20"><a href="#cb48-20" aria-hidden="true" tabindex="-1"></a><span class="fu">graphical.report</span>(hic.GR.test[[<span class="dv">1</span>]],</span>
<span id="cb48-21"><a href="#cb48-21" aria-hidden="true" tabindex="-1"></a>                 <span class="at">pdffile =</span> <span class="fu">paste0</span>(<span class="st">"results/ENCODE/ENCODE_MECP2_overlap_GenometricCorr_"</span>,</span>
<span id="cb48-22"><a href="#cb48-22" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">names</span>(hic.GR.test)[<span class="dv">1</span>],<span class="st">".pdf"</span>), </span>
<span id="cb48-23"><a href="#cb48-23" aria-hidden="true" tabindex="-1"></a>                 <span class="at">show.all =</span> <span class="cn">TRUE</span>)</span>
<span id="cb48-24"><a href="#cb48-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-25"><a href="#cb48-25" aria-hidden="true" tabindex="-1"></a><span class="fu">graphical.report</span>(hic.GR.test[[<span class="dv">2</span>]],</span>
<span id="cb48-26"><a href="#cb48-26" aria-hidden="true" tabindex="-1"></a>                 <span class="at">pdffile =</span> <span class="fu">paste0</span>(<span class="st">"results/ENCODE/ENCODE_MECP2_overlap_GenometricCorr_"</span>,</span>
<span id="cb48-27"><a href="#cb48-27" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">names</span>(hic.GR.test)[<span class="dv">2</span>],<span class="st">".pdf"</span>), </span>
<span id="cb48-28"><a href="#cb48-28" aria-hidden="true" tabindex="-1"></a>                 <span class="at">show.all =</span> <span class="cn">TRUE</span>)</span>
<span id="cb48-29"><a href="#cb48-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-30"><a href="#cb48-30" aria-hidden="true" tabindex="-1"></a>hic.GR.mecp2_overlap <span class="ot">=</span> <span class="fu">list</span>()</span>
<span id="cb48-31"><a href="#cb48-31" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(hic.GR.test)){</span>
<span id="cb48-32"><a href="#cb48-32" aria-hidden="true" tabindex="-1"></a>  test1 <span class="ot">=</span> <span class="fu">unlist</span>(hic.GR.test[[i]]<span class="sc">$</span>awhole)</span>
<span id="cb48-33"><a href="#cb48-33" aria-hidden="true" tabindex="-1"></a>  hic.GR.mecp2_overlap[[<span class="fu">names</span>(hic.GR.test)[i]]] <span class="ot">=</span> test1 <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb48-34"><a href="#cb48-34" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rownames_to_column</span>(<span class="st">'param'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb48-35"><a href="#cb48-35" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span>(<span class="fu">grepl</span>(<span class="st">".data|null"</span>,param))) </span>
<span id="cb48-36"><a href="#cb48-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(hic.GR.mecp2_overlap[[i]])[<span class="dv">2</span>] <span class="ot">=</span> <span class="st">"value"</span></span>
<span id="cb48-37"><a href="#cb48-37" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb48-38"><a href="#cb48-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-39"><a href="#cb48-39" aria-hidden="true" tabindex="-1"></a>jaccard <span class="ot">=</span> <span class="fu">rbindlist</span>(hic.GR.mecp2_overlap,<span class="at">idcol =</span> <span class="st">"Group"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb48-40"><a href="#cb48-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dcast</span>(Group<span class="sc">~</span>param,<span class="at">value.var =</span> <span class="st">"value"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb48-41"><a href="#cb48-41" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">1</span>,<span class="dv">3</span>,<span class="dv">5</span>)</span>
<span id="cb48-42"><a href="#cb48-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-43"><a href="#cb48-43" aria-hidden="true" tabindex="-1"></a><span class="co">#jaccard = jaccard %&gt;% column_to_rownames('Group') %&gt;% t()</span></span>
<span id="cb48-44"><a href="#cb48-44" aria-hidden="true" tabindex="-1"></a>jaccard_index <span class="ot">=</span> jaccard <span class="sc">%&gt;%</span> <span class="fu">column_to_rownames</span>(<span class="st">'Group'</span>) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb48-45"><a href="#cb48-45" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">1</span>) <span class="sc">%&gt;%</span> <span class="fu">t</span>()</span>
<span id="cb48-46"><a href="#cb48-46" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(jaccard_index) <span class="ot">=</span> <span class="st">"MBHs"</span></span>
<span id="cb48-47"><a href="#cb48-47" aria-hidden="true" tabindex="-1"></a>jaccard_index <span class="ot">=</span>  <span class="fu">as.data.frame</span>(jaccard_index) <span class="sc">%&gt;%</span> <span class="fu">mutate_if</span>(is.character, as.numeric)</span>
<span id="cb48-48"><a href="#cb48-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-49"><a href="#cb48-49" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">"Plots/ENCODE_dataOverlap_with_MECP2_HiC_JiangLoop.pdf"</span>,<span class="at">height =</span> <span class="dv">2</span>,<span class="at">width =</span> <span class="dv">3</span>)</span>
<span id="cb48-50"><a href="#cb48-50" aria-hidden="true" tabindex="-1"></a><span class="fu">draw</span>(<span class="fu">Heatmap</span>(jaccard_index,<span class="at">name =</span> <span class="st">"Jaccard measure"</span>,</span>
<span id="cb48-51"><a href="#cb48-51" aria-hidden="true" tabindex="-1"></a>             <span class="co">#col= colorRamp2(c(0,0.06),c('blue',"red")),</span></span>
<span id="cb48-52"><a href="#cb48-52" aria-hidden="true" tabindex="-1"></a>             <span class="co">#heatmap_legend_param=list(at=c(0,5,10,15,20),color_bar="continuous", </span></span>
<span id="cb48-53"><a href="#cb48-53" aria-hidden="true" tabindex="-1"></a>             <span class="co">#                           legend_direction="horizontal", legend_width=unit(5,"cm"),</span></span>
<span id="cb48-54"><a href="#cb48-54" aria-hidden="true" tabindex="-1"></a>             <span class="co">#                          title_position="topcenter", title_gp=gpar(fontsize=10, fontface="bold")),</span></span>
<span id="cb48-55"><a href="#cb48-55" aria-hidden="true" tabindex="-1"></a>             <span class="at">cluster_columns =</span> <span class="cn">FALSE</span>,</span>
<span id="cb48-56"><a href="#cb48-56" aria-hidden="true" tabindex="-1"></a>             <span class="at">cluster_rows =</span> <span class="cn">TRUE</span>,</span>
<span id="cb48-57"><a href="#cb48-57" aria-hidden="true" tabindex="-1"></a>             <span class="at">rect_gp =</span> <span class="fu">gpar</span>(<span class="at">col =</span> <span class="st">"black"</span>),</span>
<span id="cb48-58"><a href="#cb48-58" aria-hidden="true" tabindex="-1"></a>             <span class="at">column_names_side =</span> <span class="st">"top"</span>,</span>
<span id="cb48-59"><a href="#cb48-59" aria-hidden="true" tabindex="-1"></a>             <span class="at">column_names_rot =</span> <span class="dv">30</span>,</span>
<span id="cb48-60"><a href="#cb48-60" aria-hidden="true" tabindex="-1"></a>             <span class="at">column_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">8</span>),</span>
<span id="cb48-61"><a href="#cb48-61" aria-hidden="true" tabindex="-1"></a>             <span class="at">row_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">8</span>)),</span>
<span id="cb48-62"><a href="#cb48-62" aria-hidden="true" tabindex="-1"></a>     <span class="co">#cell_fun = function(j, i, x, y, width, height, fill) {</span></span>
<span id="cb48-63"><a href="#cb48-63" aria-hidden="true" tabindex="-1"></a>     <span class="co">#  grid.text(jaccard_direction[i, j], x, y, gp = gpar(fontsize = 12))},</span></span>
<span id="cb48-64"><a href="#cb48-64" aria-hidden="true" tabindex="-1"></a>     <span class="at">heatmap_legend_side=</span><span class="st">"right"</span>)</span>
<span id="cb48-65"><a href="#cb48-65" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></li>
</ol>
</section>
<section id="section" class="level3">
<h3 class="anchored" data-anchor-id="section"></h3>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>